{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "gpuType": "V6E1",
      "runtime_attributes": {
        "runtime_version": "2025.07"
      }
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "TPU"
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "First, we install the necessary software package."
      ],
      "metadata": {
        "id": "5odADDrKBhqK"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KMa_aAt_8nHn",
        "outputId": "a8d5a456-17c7-4158-b1e2-adbc3b05bb0f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ],
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# In Colab:\n",
        "!pip install --index-url https://test.pypi.org/simple/ \\\n",
        "            --extra-index-url https://pypi.org/simple \\\n",
        "            \"scikit-misc==0.2.0rc1\"\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "rPhuoFANFxx8",
        "outputId": "97107a87-4c3e-415b-c265-e67160dbd150"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://test.pypi.org/simple/, https://pypi.org/simple\n",
            "Collecting scikit-misc==0.2.0rc1\n",
            "  Downloading https://test-files.pythonhosted.org/packages/8a/66/594fbbe003d00b2f688c224a78f36efb37d5a279f07623e6d94e8de9a7f4/scikit_misc-0.2.0rc1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.4 kB)\n",
            "Requirement already satisfied: numpy>=1.19.5 in /usr/local/lib/python3.11/dist-packages (from scikit-misc==0.2.0rc1) (2.0.2)\n",
            "Downloading https://test-files.pythonhosted.org/packages/8a/66/594fbbe003d00b2f688c224a78f36efb37d5a279f07623e6d94e8de9a7f4/scikit_misc-0.2.0rc1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (10.7 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m10.7/10.7 MB\u001b[0m \u001b[31m5.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: scikit-misc\n",
            "Successfully installed scikit-misc-0.2.0rc1\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -U https://github.com/PythonOT/POT/archive/master.zip # with --user for user install (no root)\n",
        "!pip install pykeops\n",
        "!pip install geomloss\n",
        "!pip install muon\n",
        "!pip install gprofiler-official\n",
        "\n",
        "!pip install scanpy\n",
        "!pip install muon\n",
        "!pip install pyranges"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VGt1qSJPCLl_",
        "outputId": "62ae5ece-646d-4d48-8304-d3106c475f7e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting https://github.com/PythonOT/POT/archive/master.zip\n",
            "  Downloading https://github.com/PythonOT/POT/archive/master.zip\n",
            "\u001b[2K     \u001b[32m|\u001b[0m \u001b[32m3.1 MB\u001b[0m \u001b[31m16.0 MB/s\u001b[0m \u001b[33m0:00:00\u001b[0m\n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: numpy>=1.16 in /usr/local/lib/python3.11/dist-packages (from POT==0.9.7.dev0) (2.0.2)\n",
            "Requirement already satisfied: scipy>=1.6 in /usr/local/lib/python3.11/dist-packages (from POT==0.9.7.dev0) (1.15.3)\n",
            "Building wheels for collected packages: POT\n",
            "  Building wheel for POT (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for POT: filename=pot-0.9.7.dev0-cp311-cp311-linux_x86_64.whl size=1366862 sha256=5a3eb65f0735a9f6d7f53d31a519fac40ec0496c69b24c821c87f3726c2d978c\n",
            "  Stored in directory: /tmp/pip-ephem-wheel-cache-9mw6bwr5/wheels/fc/b3/69/deb9c12f26c4c311bb43fab28b37c7a1faa3c089b07eea1ceb\n",
            "Successfully built POT\n",
            "Installing collected packages: POT\n",
            "Successfully installed POT-0.9.7.dev0\n",
            "Collecting pykeops\n",
            "  Downloading pykeops-2.3.tar.gz (552 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m552.2/552.2 kB\u001b[0m \u001b[31m22.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from pykeops) (2.0.2)\n",
            "Collecting pybind11 (from pykeops)\n",
            "  Downloading pybind11-3.0.1-py3-none-any.whl.metadata (10.0 kB)\n",
            "Collecting keopscore==2.3 (from pykeops)\n",
            "  Downloading keopscore-2.3.tar.gz (118 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m118.5/118.5 kB\u001b[0m \u001b[31m7.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Downloading pybind11-3.0.1-py3-none-any.whl (293 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m293.6/293.6 kB\u001b[0m \u001b[31m39.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hBuilding wheels for collected packages: pykeops, keopscore\n",
            "  Building wheel for pykeops (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pykeops: filename=pykeops-2.3-py3-none-any.whl size=120570 sha256=f4a4dd7082159f86731b04003f0e0dfdb8b68a077a60e59faebe77b0dddf7d10\n",
            "  Stored in directory: /root/.cache/pip/wheels/a8/91/ea/d9e54997a840e38595b9a2c5b9021f3969173edbda2fc99686\n",
            "  Building wheel for keopscore (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for keopscore: filename=keopscore-2.3-py3-none-any.whl size=185959 sha256=4ba9b073b05b08b3a22d7f9cb3cc9a9aa4ec61bd5c0133ba9c38586eb769f4d5\n",
            "  Stored in directory: /root/.cache/pip/wheels/32/77/92/17707f162cb65cfa8e97f0360cdb30681038f7762cf929b1b4\n",
            "Successfully built pykeops keopscore\n",
            "Installing collected packages: pybind11, keopscore, pykeops\n",
            "Successfully installed keopscore-2.3 pybind11-3.0.1 pykeops-2.3\n",
            "Collecting geomloss\n",
            "  Downloading geomloss-0.2.6.tar.gz (26 kB)\n",
            "  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from geomloss) (2.0.2)\n",
            "Requirement already satisfied: torch in /usr/local/lib/python3.11/dist-packages (from geomloss) (2.6.0+cpu)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.11/dist-packages (from torch->geomloss) (3.18.0)\n",
            "Requirement already satisfied: typing-extensions>=4.10.0 in /usr/local/lib/python3.11/dist-packages (from torch->geomloss) (4.14.1)\n",
            "Requirement already satisfied: networkx in /usr/local/lib/python3.11/dist-packages (from torch->geomloss) (3.5)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from torch->geomloss) (3.1.6)\n",
            "Requirement already satisfied: fsspec in /usr/local/lib/python3.11/dist-packages (from torch->geomloss) (2025.5.1)\n",
            "Requirement already satisfied: sympy==1.13.1 in /usr/local/lib/python3.11/dist-packages (from torch->geomloss) (1.13.1)\n",
            "Requirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.11/dist-packages (from sympy==1.13.1->torch->geomloss) (1.3.0)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from jinja2->torch->geomloss) (3.0.2)\n",
            "Building wheels for collected packages: geomloss\n",
            "  Building wheel for geomloss (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for geomloss: filename=geomloss-0.2.6-py3-none-any.whl size=32341 sha256=54bb5d531c605b9c2ca0e4805087b4f0383999f63ba49f9939b627096e9a8ef7\n",
            "  Stored in directory: /root/.cache/pip/wheels/f5/cf/07/9d1d883feac2951b968fed8ef676842dc90b9860ea49a4dcac\n",
            "Successfully built geomloss\n",
            "Installing collected packages: geomloss\n",
            "Successfully installed geomloss-0.2.6\n",
            "Collecting muon\n",
            "  Downloading muon-0.1.7-py3-none-any.whl.metadata (7.2 kB)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from muon) (2.0.2)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (from muon) (2.2.2)\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.11/dist-packages (from muon) (3.10.0)\n",
            "Requirement already satisfied: seaborn in /usr/local/lib/python3.11/dist-packages (from muon) (0.13.2)\n",
            "Requirement already satisfied: h5py in /usr/local/lib/python3.11/dist-packages (from muon) (3.14.0)\n",
            "Collecting anndata (from muon)\n",
            "  Downloading anndata-0.12.3-py3-none-any.whl.metadata (9.9 kB)\n",
            "Collecting scanpy (from muon)\n",
            "  Downloading scanpy-1.11.5-py3-none-any.whl.metadata (9.3 kB)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.11/dist-packages (from muon) (1.6.1)\n",
            "Collecting umap-learn (from muon)\n",
            "  Downloading umap_learn-0.5.9.post2-py3-none-any.whl.metadata (25 kB)\n",
            "Requirement already satisfied: numba in /usr/local/lib/python3.11/dist-packages (from muon) (0.61.2)\n",
            "Requirement already satisfied: protobuf in /usr/local/lib/python3.11/dist-packages (from muon) (6.31.1)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.11/dist-packages (from muon) (4.67.1)\n",
            "Collecting mudata (from muon)\n",
            "  Downloading mudata-0.3.2-py3-none-any.whl.metadata (8.4 kB)\n",
            "Collecting array-api-compat>=1.7.1 (from anndata->muon)\n",
            "  Downloading array_api_compat-1.12.0-py3-none-any.whl.metadata (2.5 kB)\n",
            "Collecting legacy-api-wrap (from anndata->muon)\n",
            "  Downloading legacy_api_wrap-1.4.1-py3-none-any.whl.metadata (2.1 kB)\n",
            "Collecting natsort (from anndata->muon)\n",
            "  Downloading natsort-8.4.0-py3-none-any.whl.metadata (21 kB)\n",
            "Requirement already satisfied: packaging>=24.2 in /usr/local/lib/python3.11/dist-packages (from anndata->muon) (25.0)\n",
            "Requirement already satisfied: scipy>=1.12 in /usr/local/lib/python3.11/dist-packages (from anndata->muon) (1.15.3)\n",
            "Collecting zarr!=3.0.*,>=2.18.7 (from anndata->muon)\n",
            "  Downloading zarr-3.1.3-py3-none-any.whl.metadata (10 kB)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.11/dist-packages (from pandas->muon) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas->muon) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas->muon) (2025.2)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (1.3.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (0.12.1)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (4.58.5)\n",
            "Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (1.4.8)\n",
            "Requirement already satisfied: pillow>=8 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (11.3.0)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (3.2.3)\n",
            "Requirement already satisfied: llvmlite<0.45,>=0.44.0dev0 in /usr/local/lib/python3.11/dist-packages (from numba->muon) (0.44.0)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.11/dist-packages (from scanpy->muon) (1.5.1)\n",
            "Requirement already satisfied: networkx>=2.7.1 in /usr/local/lib/python3.11/dist-packages (from scanpy->muon) (3.5)\n",
            "Collecting patsy!=1.0.0 (from scanpy->muon)\n",
            "  Downloading patsy-1.0.2-py2.py3-none-any.whl.metadata (3.6 kB)\n",
            "Collecting pynndescent>=0.5.13 (from scanpy->muon)\n",
            "  Downloading pynndescent-0.5.13-py3-none-any.whl.metadata (6.8 kB)\n",
            "Collecting session-info2 (from scanpy->muon)\n",
            "  Downloading session_info2-0.2.3-py3-none-any.whl.metadata (3.4 kB)\n",
            "Collecting statsmodels>=0.14.5 (from scanpy->muon)\n",
            "  Downloading statsmodels-0.14.5-cp311-cp311-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl.metadata (9.5 kB)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.11/dist-packages (from scanpy->muon) (4.14.1)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.11/dist-packages (from scikit-learn->muon) (3.6.0)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.8.2->pandas->muon) (1.17.0)\n",
            "Collecting donfig>=0.8 (from zarr!=3.0.*,>=2.18.7->anndata->muon)\n",
            "  Downloading donfig-0.8.1.post1-py3-none-any.whl.metadata (5.0 kB)\n",
            "Collecting numcodecs>=0.14 (from numcodecs[crc32c]>=0.14->zarr!=3.0.*,>=2.18.7->anndata->muon)\n",
            "  Downloading numcodecs-0.16.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.3 kB)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.11/dist-packages (from donfig>=0.8->zarr!=3.0.*,>=2.18.7->anndata->muon) (6.0.2)\n",
            "Collecting crc32c>=2.7 (from numcodecs[crc32c]>=0.14->zarr!=3.0.*,>=2.18.7->anndata->muon)\n",
            "  Downloading crc32c-2.8-cp311-cp311-manylinux1_x86_64.manylinux_2_28_x86_64.manylinux_2_5_x86_64.whl.metadata (7.8 kB)\n",
            "Downloading muon-0.1.7-py3-none-any.whl (293 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m293.7/293.7 kB\u001b[0m \u001b[31m19.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading anndata-0.12.3-py3-none-any.whl (169 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m169.3/169.3 kB\u001b[0m \u001b[31m32.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading mudata-0.3.2-py3-none-any.whl (41 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m41.9/41.9 kB\u001b[0m \u001b[31m6.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading scanpy-1.11.5-py3-none-any.whl (2.1 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m2.1/2.1 MB\u001b[0m \u001b[31m104.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading umap_learn-0.5.9.post2-py3-none-any.whl (90 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m90.1/90.1 kB\u001b[0m \u001b[31m13.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading array_api_compat-1.12.0-py3-none-any.whl (58 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m58.2/58.2 kB\u001b[0m \u001b[31m8.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading legacy_api_wrap-1.4.1-py3-none-any.whl (10.0 kB)\n",
            "Downloading patsy-1.0.2-py2.py3-none-any.whl (233 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m233.3/233.3 kB\u001b[0m \u001b[31m37.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pynndescent-0.5.13-py3-none-any.whl (56 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m56.9/56.9 kB\u001b[0m \u001b[31m10.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading statsmodels-0.14.5-cp311-cp311-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl (10.5 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m10.5/10.5 MB\u001b[0m \u001b[31m111.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading zarr-3.1.3-py3-none-any.whl (276 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m276.4/276.4 kB\u001b[0m \u001b[31m39.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading natsort-8.4.0-py3-none-any.whl (38 kB)\n",
            "Downloading session_info2-0.2.3-py3-none-any.whl (16 kB)\n",
            "Downloading donfig-0.8.1.post1-py3-none-any.whl (21 kB)\n",
            "Downloading numcodecs-0.16.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (8.9 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m8.9/8.9 MB\u001b[0m \u001b[31m146.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading crc32c-2.8-cp311-cp311-manylinux1_x86_64.manylinux_2_28_x86_64.manylinux_2_5_x86_64.whl (80 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m80.2/80.2 kB\u001b[0m \u001b[31m14.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: session-info2, patsy, numcodecs, natsort, legacy-api-wrap, donfig, crc32c, array-api-compat, zarr, statsmodels, pynndescent, umap-learn, anndata, scanpy, mudata, muon\n",
            "Successfully installed anndata-0.12.3 array-api-compat-1.12.0 crc32c-2.8 donfig-0.8.1.post1 legacy-api-wrap-1.4.1 mudata-0.3.2 muon-0.1.7 natsort-8.4.0 numcodecs-0.16.3 patsy-1.0.2 pynndescent-0.5.13 scanpy-1.11.5 session-info2-0.2.3 statsmodels-0.14.5 umap-learn-0.5.9.post2 zarr-3.1.3\n",
            "Collecting gprofiler-official\n",
            "  Downloading gprofiler_official-1.0.0-py3-none-any.whl.metadata (11 kB)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from gprofiler-official) (2.32.3)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->gprofiler-official) (3.4.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->gprofiler-official) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->gprofiler-official) (2.4.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->gprofiler-official) (2025.7.14)\n",
            "Downloading gprofiler_official-1.0.0-py3-none-any.whl (9.3 kB)\n",
            "Installing collected packages: gprofiler-official\n",
            "Successfully installed gprofiler-official-1.0.0\n",
            "Requirement already satisfied: scanpy in /usr/local/lib/python3.11/dist-packages (1.11.5)\n",
            "Requirement already satisfied: anndata>=0.8 in /usr/local/lib/python3.11/dist-packages (from scanpy) (0.12.3)\n",
            "Requirement already satisfied: h5py>=3.7.0 in /usr/local/lib/python3.11/dist-packages (from scanpy) (3.14.0)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.11/dist-packages (from scanpy) (1.5.1)\n",
            "Requirement already satisfied: legacy-api-wrap>=1.4.1 in /usr/local/lib/python3.11/dist-packages (from scanpy) (1.4.1)\n",
            "Requirement already satisfied: matplotlib>=3.7.5 in /usr/local/lib/python3.11/dist-packages (from scanpy) (3.10.0)\n",
            "Requirement already satisfied: natsort in /usr/local/lib/python3.11/dist-packages (from scanpy) (8.4.0)\n",
            "Requirement already satisfied: networkx>=2.7.1 in /usr/local/lib/python3.11/dist-packages (from scanpy) (3.5)\n",
            "Requirement already satisfied: numba!=0.62.0rc1,>=0.57.1 in /usr/local/lib/python3.11/dist-packages (from scanpy) (0.61.2)\n",
            "Requirement already satisfied: numpy>=1.24.1 in /usr/local/lib/python3.11/dist-packages (from scanpy) (2.0.2)\n",
            "Requirement already satisfied: packaging>=21.3 in /usr/local/lib/python3.11/dist-packages (from scanpy) (25.0)\n",
            "Requirement already satisfied: pandas>=1.5.3 in /usr/local/lib/python3.11/dist-packages (from scanpy) (2.2.2)\n",
            "Requirement already satisfied: patsy!=1.0.0 in /usr/local/lib/python3.11/dist-packages (from scanpy) (1.0.2)\n",
            "Requirement already satisfied: pynndescent>=0.5.13 in /usr/local/lib/python3.11/dist-packages (from scanpy) (0.5.13)\n",
            "Requirement already satisfied: scikit-learn>=1.1.3 in /usr/local/lib/python3.11/dist-packages (from scanpy) (1.6.1)\n",
            "Requirement already satisfied: scipy>=1.8.1 in /usr/local/lib/python3.11/dist-packages (from scanpy) (1.15.3)\n",
            "Requirement already satisfied: seaborn>=0.13.2 in /usr/local/lib/python3.11/dist-packages (from scanpy) (0.13.2)\n",
            "Requirement already satisfied: session-info2 in /usr/local/lib/python3.11/dist-packages (from scanpy) (0.2.3)\n",
            "Requirement already satisfied: statsmodels>=0.14.5 in /usr/local/lib/python3.11/dist-packages (from scanpy) (0.14.5)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.11/dist-packages (from scanpy) (4.67.1)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.11/dist-packages (from scanpy) (4.14.1)\n",
            "Requirement already satisfied: umap-learn>=0.5.6 in /usr/local/lib/python3.11/dist-packages (from scanpy) (0.5.9.post2)\n",
            "Requirement already satisfied: array-api-compat>=1.7.1 in /usr/local/lib/python3.11/dist-packages (from anndata>=0.8->scanpy) (1.12.0)\n",
            "Requirement already satisfied: zarr!=3.0.*,>=2.18.7 in /usr/local/lib/python3.11/dist-packages (from anndata>=0.8->scanpy) (3.1.3)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.7.5->scanpy) (1.3.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.7.5->scanpy) (0.12.1)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.7.5->scanpy) (4.58.5)\n",
            "Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.7.5->scanpy) (1.4.8)\n",
            "Requirement already satisfied: pillow>=8 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.7.5->scanpy) (11.3.0)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.7.5->scanpy) (3.2.3)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.7.5->scanpy) (2.9.0.post0)\n",
            "Requirement already satisfied: llvmlite<0.45,>=0.44.0dev0 in /usr/local/lib/python3.11/dist-packages (from numba!=0.62.0rc1,>=0.57.1->scanpy) (0.44.0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas>=1.5.3->scanpy) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas>=1.5.3->scanpy) (2025.2)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.11/dist-packages (from scikit-learn>=1.1.3->scanpy) (3.6.0)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.7->matplotlib>=3.7.5->scanpy) (1.17.0)\n",
            "Requirement already satisfied: donfig>=0.8 in /usr/local/lib/python3.11/dist-packages (from zarr!=3.0.*,>=2.18.7->anndata>=0.8->scanpy) (0.8.1.post1)\n",
            "Requirement already satisfied: numcodecs>=0.14 in /usr/local/lib/python3.11/dist-packages (from numcodecs[crc32c]>=0.14->zarr!=3.0.*,>=2.18.7->anndata>=0.8->scanpy) (0.16.3)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.11/dist-packages (from donfig>=0.8->zarr!=3.0.*,>=2.18.7->anndata>=0.8->scanpy) (6.0.2)\n",
            "Requirement already satisfied: crc32c>=2.7 in /usr/local/lib/python3.11/dist-packages (from numcodecs[crc32c]>=0.14->zarr!=3.0.*,>=2.18.7->anndata>=0.8->scanpy) (2.8)\n",
            "Requirement already satisfied: muon in /usr/local/lib/python3.11/dist-packages (0.1.7)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from muon) (2.0.2)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (from muon) (2.2.2)\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.11/dist-packages (from muon) (3.10.0)\n",
            "Requirement already satisfied: seaborn in /usr/local/lib/python3.11/dist-packages (from muon) (0.13.2)\n",
            "Requirement already satisfied: h5py in /usr/local/lib/python3.11/dist-packages (from muon) (3.14.0)\n",
            "Requirement already satisfied: anndata in /usr/local/lib/python3.11/dist-packages (from muon) (0.12.3)\n",
            "Requirement already satisfied: scanpy in /usr/local/lib/python3.11/dist-packages (from muon) (1.11.5)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.11/dist-packages (from muon) (1.6.1)\n",
            "Requirement already satisfied: umap-learn in /usr/local/lib/python3.11/dist-packages (from muon) (0.5.9.post2)\n",
            "Requirement already satisfied: numba in /usr/local/lib/python3.11/dist-packages (from muon) (0.61.2)\n",
            "Requirement already satisfied: protobuf in /usr/local/lib/python3.11/dist-packages (from muon) (6.31.1)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.11/dist-packages (from muon) (4.67.1)\n",
            "Requirement already satisfied: mudata in /usr/local/lib/python3.11/dist-packages (from muon) (0.3.2)\n",
            "Requirement already satisfied: array-api-compat>=1.7.1 in /usr/local/lib/python3.11/dist-packages (from anndata->muon) (1.12.0)\n",
            "Requirement already satisfied: legacy-api-wrap in /usr/local/lib/python3.11/dist-packages (from anndata->muon) (1.4.1)\n",
            "Requirement already satisfied: natsort in /usr/local/lib/python3.11/dist-packages (from anndata->muon) (8.4.0)\n",
            "Requirement already satisfied: packaging>=24.2 in /usr/local/lib/python3.11/dist-packages (from anndata->muon) (25.0)\n",
            "Requirement already satisfied: scipy>=1.12 in /usr/local/lib/python3.11/dist-packages (from anndata->muon) (1.15.3)\n",
            "Requirement already satisfied: zarr!=3.0.*,>=2.18.7 in /usr/local/lib/python3.11/dist-packages (from anndata->muon) (3.1.3)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.11/dist-packages (from pandas->muon) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas->muon) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas->muon) (2025.2)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (1.3.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (0.12.1)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (4.58.5)\n",
            "Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (1.4.8)\n",
            "Requirement already satisfied: pillow>=8 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (11.3.0)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->muon) (3.2.3)\n",
            "Requirement already satisfied: llvmlite<0.45,>=0.44.0dev0 in /usr/local/lib/python3.11/dist-packages (from numba->muon) (0.44.0)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.11/dist-packages (from scanpy->muon) (1.5.1)\n",
            "Requirement already satisfied: networkx>=2.7.1 in /usr/local/lib/python3.11/dist-packages (from scanpy->muon) (3.5)\n",
            "Requirement already satisfied: patsy!=1.0.0 in /usr/local/lib/python3.11/dist-packages (from scanpy->muon) (1.0.2)\n",
            "Requirement already satisfied: pynndescent>=0.5.13 in /usr/local/lib/python3.11/dist-packages (from scanpy->muon) (0.5.13)\n",
            "Requirement already satisfied: session-info2 in /usr/local/lib/python3.11/dist-packages (from scanpy->muon) (0.2.3)\n",
            "Requirement already satisfied: statsmodels>=0.14.5 in /usr/local/lib/python3.11/dist-packages (from scanpy->muon) (0.14.5)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.11/dist-packages (from scanpy->muon) (4.14.1)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.11/dist-packages (from scikit-learn->muon) (3.6.0)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.8.2->pandas->muon) (1.17.0)\n",
            "Requirement already satisfied: donfig>=0.8 in /usr/local/lib/python3.11/dist-packages (from zarr!=3.0.*,>=2.18.7->anndata->muon) (0.8.1.post1)\n",
            "Requirement already satisfied: numcodecs>=0.14 in /usr/local/lib/python3.11/dist-packages (from numcodecs[crc32c]>=0.14->zarr!=3.0.*,>=2.18.7->anndata->muon) (0.16.3)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.11/dist-packages (from donfig>=0.8->zarr!=3.0.*,>=2.18.7->anndata->muon) (6.0.2)\n",
            "Requirement already satisfied: crc32c>=2.7 in /usr/local/lib/python3.11/dist-packages (from numcodecs[crc32c]>=0.14->zarr!=3.0.*,>=2.18.7->anndata->muon) (2.8)\n",
            "Collecting pyranges\n",
            "  Downloading pyranges-0.1.4-py3-none-any.whl.metadata (3.7 kB)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (from pyranges) (2.2.2)\n",
            "Collecting ncls>=0.0.63 (from pyranges)\n",
            "  Downloading ncls-0.0.70-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.9 kB)\n",
            "Collecting tabulate (from pyranges)\n",
            "  Downloading tabulate-0.9.0-py3-none-any.whl.metadata (34 kB)\n",
            "Collecting sorted_nearest>=0.0.33 (from pyranges)\n",
            "  Downloading sorted_nearest-0.0.41.tar.gz (1.6 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m65.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: natsort in /usr/local/lib/python3.11/dist-packages (from pyranges) (8.4.0)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from ncls>=0.0.63->pyranges) (2.0.2)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.11/dist-packages (from pandas->pyranges) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas->pyranges) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas->pyranges) (2025.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.8.2->pandas->pyranges) (1.17.0)\n",
            "Downloading pyranges-0.1.4-py3-none-any.whl (1.5 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m1.5/1.5 MB\u001b[0m \u001b[31m136.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading ncls-0.0.70-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.7 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m2.7/2.7 MB\u001b[0m \u001b[31m13.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tabulate-0.9.0-py3-none-any.whl (35 kB)\n",
            "Building wheels for collected packages: sorted_nearest\n",
            "  Building wheel for sorted_nearest (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for sorted_nearest: filename=sorted_nearest-0.0.41-cp311-cp311-linux_x86_64.whl size=6485577 sha256=49699bfa2ab93b3efe07ee9e97e891d169278b2582642a408517f3bb0d0e9ead\n",
            "  Stored in directory: /root/.cache/pip/wheels/33/6e/5e/190e13df4a337b51046ef3a9e55b07542e97987b384529066c\n",
            "Successfully built sorted_nearest\n",
            "Installing collected packages: tabulate, sorted_nearest, ncls, pyranges\n",
            "Successfully installed ncls-0.0.70 pyranges-0.1.4 sorted_nearest-0.0.41 tabulate-0.9.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "pip install numpy==1.26.4"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 367
        },
        "id": "09FmRGv7BHHj",
        "outputId": "1fe2da96-07c7-444e-8d2e-268ebb729381"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting numpy==1.26.4\n",
            "  Downloading numpy-1.26.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)\n",
            "\u001b[?25l     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m0.0/61.0 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m61.0/61.0 kB\u001b[0m \u001b[31m5.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading numpy-1.26.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.3 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m18.3/18.3 MB\u001b[0m \u001b[31m245.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: numpy\n",
            "  Attempting uninstall: numpy\n",
            "    Found existing installation: numpy 2.0.2\n",
            "    Uninstalling numpy-2.0.2:\n",
            "      Successfully uninstalled numpy-2.0.2\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "thinc 8.3.6 requires numpy<3.0.0,>=2.0.0, but you have numpy 1.26.4 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed numpy-1.26.4\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "numpy"
                ]
              },
              "id": "43ac5c0f7f954d95b3503d743cab2d21"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import Grotia_utils\n",
        "importlib.reload(Grotia_utils)\n",
        "from Grotia_utils import (\n",
        "   load_results,interpret_embedding_top_genes,plot_top_genes_by_dimension_sequential,\n",
        "   write_all_peaks_to_bed,bed_to_fasta,r\n",
        "   un_gimme_motifs_with_genome,\n",
        "   run_pathway_analysis_top_genes\n",
        ")\n"
      ],
      "metadata": {
        "id": "i7WeT8qpo1_N"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "As our interpretability framework relies on the learned projection matrix and the corresponding kernel matrix, we first load or compute both from the data."
      ],
      "metadata": {
        "id": "-2J0CXeKGO1b"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import muon as mu\n",
        "mdata = mu.read_h5mu(\"/content/drive/MyDrive/pbmc10X.h5mu.gz\")\n",
        "\n",
        "sc.pp.filter_cells(mdata[\"rna\"], min_genes=100)\n",
        "sc.pp.filter_genes(mdata[\"rna\"], min_cells=20)\n",
        "\n",
        "sc.pp.filter_cells(mdata[\"atac\"], min_genes=100)\n",
        "# we're actually filtering peaks here\n",
        "sc.pp.filter_genes(mdata[\"atac\"], min_cells=20)\n",
        "mdata.update()\n",
        "\n",
        "\n",
        "mdata[\"rna\"].layers[\"counts\"] = mdata[\"rna\"].X.copy()\n",
        "\n",
        "# Log-normalize the scRNA counts\n",
        "sc.pp.normalize_total(mdata[\"rna\"], target_sum=10000.)\n",
        "sc.pp.log1p(mdata[\"rna\"])\n",
        "\n",
        "# Since we use both raw and normalized gene counts in our autoencoder it makes sense to select highly variable genes based on both criteria\n",
        "raw_hvg = sc.pp.highly_variable_genes(mdata[\"rna\"], layer=\"counts\", n_top_genes=3000, subset=False, inplace=False,\n",
        "                                     flavor=\"seurat_v3\")[\"highly_variable\"].values\n",
        "norm_hvg = sc.pp.highly_variable_genes(mdata[\"rna\"], n_top_genes=3000, subset=False,\n",
        "                                       inplace=False)[\"highly_variable\"].values\n",
        "mdata.mod[\"rna\"] = mdata[\"rna\"][:, np.logical_or(raw_hvg, norm_hvg)].copy()\n",
        "\n",
        "# Perform PCA on the selected genes\n",
        "sc.tl.pca(mdata[\"rna\"], n_comps=50, zero_center=None)\n",
        "mu.atac.pp.tfidf(mdata[\"atac\"], log_tf=True, log_idf=True)\n",
        "sc.tl.pca(mdata[\"atac\"], n_comps=50, zero_center=None)\n",
        "\n",
        "X_normalized = (mdata[\"rna\"].obsm['X_pca'] )\n",
        "y_normalized = (mdata[\"atac\"].obsm['X_pca'] )\n",
        "gamma_x = calculate_gamma(X_normalized)\n",
        "gamma_y = calculate_gamma(y_normalized)\n",
        "# celltype_array = np.array(mdata.obs['celltype'][:4000])\n",
        "\n",
        "\n",
        "kernel_X_original= rbf_kernel(X_normalized,gamma_x)\n",
        "transformer = KernelCenterer().fit(kernel_X_original)\n",
        "kernel_X_center = transformer.transform(kernel_X_original)\n",
        "\n",
        "kernel_y_original= rbf_kernel(y_normalized,gamma_y)\n",
        "transformer = KernelCenterer().fit(kernel_y_original)\n",
        "kernel_y_center = transformer.transform(kernel_y_original)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WUKt7FjjBKRA",
        "outputId": "8eeef698-9d85-4007-92a0-b64e40027578"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.11/dist-packages/numba/np/ufunc/parallel.py:371: NumbaWarning: \u001b[1mThe TBB threading layer requires TBB version 2021 update 6 or later i.e., TBB_INTERFACE_VERSION >= 12060. Found TBB_INTERFACE_VERSION = 12050. The TBB threading layer is disabled.\u001b[0m\n",
            "  warnings.warn(problem)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We load the projection matrix from both domains (RNA and ATAC), together with the corresponding optimal transport (OT) map."
      ],
      "metadata": {
        "id": "CVwWPWwhCqW2"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import numpy as np\n",
        "\n",
        "# Example usage:\n",
        "mapped_K1, mapped_K2, ot_plan, alpha, beta = load_results(\n",
        "    path_mappedK1=\"/content/drive/MyDrive/mappedK1.npy\",\n",
        "    path_mappedK2=\"/content/drive/MyDrive/mappedK2.npy\",\n",
        "    path_otplan=\"/content/drive/MyDrive/ot_plan.npy\",\n",
        "    path_alpha=\"/content/drive/MyDrive/alpha.npy\",\n",
        "    path_beta=\"/content/drive/MyDrive/beta.npy\"\n",
        ")\n",
        "\n",
        "print(\"Loaded shapes:\")\n",
        "print(\"  mapped_K1:\", mapped_K1.shape)\n",
        "print(\"  mapped_K2:\", mapped_K2.shape)\n",
        "print(\"  ot_plan:\", ot_plan.shape)\n",
        "print(\"  alpha:\", alpha.shape)\n",
        "print(\"  beta:\", beta.shape)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pznsHPFz9_3G",
        "outputId": "721b366f-0bb8-4c4f-d04a-27a765dc2fac"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Loaded shapes:\n",
            "  mapped_K1: (9378, 8)\n",
            "  mapped_K2: (9378, 8)\n",
            "  ot_plan: (9378, 9378)\n",
            "  alpha: (9378, 8)\n",
            "  beta: (9378, 8)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "To interpret each latent embedding dimension in terms of genes, we apply interpret_embedding_top_genes using the learned per-cell coefficients (alpha), the centered cellâ€“cell kernel (K_centered), and the kernel scale parameter (gamma). The function computes sensitivities in PCA space, propagates them back to gene space via the PCA loadings (adata.varm[\"PCs\"]) using the chain rule, and ranks genes by the mean absolute sensitivity per dimension. We report the top-\n",
        "ğ‘˜\n",
        "k genes for each latent dimension as putative drivers of the corresponding embedding axis."
      ],
      "metadata": {
        "id": "ErzEfEjdFY1C"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "avg_abs_sens_genes, top_genes_by_dim = interpret_embedding_top_genes(\n",
        "    adata=mdata[\"rna\"],\n",
        "    alpha=alpha,\n",
        "    K_centered=kernel_X_center,\n",
        "    gamma=gamma_x,\n",
        "    pca_key=\"X_pca\",\n",
        "    pcs_key=\"PCs\",\n",
        "    top_k=100,\n",
        "    print_top=False  # set True to print\n",
        ")\n",
        "\n",
        "print(\"Top genes per dimension (names only):\")\n",
        "for d, genes in top_genes_by_dim.items():\n",
        "    print(f\"Dim {d}: {genes[:10]} ...\")  # preview\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GSm3kgPjB6X5",
        "outputId": "e9e19070-03f5-4459-ed64-dc9837952538"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Top genes per dimension (names only):\n",
            "Dim 0: ['LYZ', 'ZEB2', 'PLXDC2', 'SLC8A1', 'LYN', 'VCAN', 'CD74', 'SAT1', 'AOAH', 'HLA-DRA'] ...\n",
            "Dim 1: ['ZEB2', 'LYZ', 'AOAH', 'PLXDC2', 'CD74', 'SLC8A1', 'LEF1', 'HLA-DRA', 'BCL11B', 'LRMDA'] ...\n",
            "Dim 2: ['GNLY', 'CCL5', 'LEF1', 'BACH2', 'NKG7', 'MAML2', 'PDE3B', 'CAMK4', 'ZEB2', 'SERINC5'] ...\n",
            "Dim 3: ['VCAN', 'S100A4', 'MTSS1', 'HLA-DPA1', 'IFITM3', 'ITGB1', 'COTL1', 'BACH2', 'S100A11', 'MAML2'] ...\n",
            "Dim 4: ['VCAN', 'ANXA1', 'PLXDC2', 'BACH2', 'LRMDA', 'PLCB1', 'NEAT1', 'LEF1', 'DPYD', 'CD36'] ...\n",
            "Dim 5: ['TCF7L2', 'FCGR3A', 'MTSS1', 'CDKN1C', 'IFITM3', 'AFF3', 'AIF1', 'SERPINA1', 'WARS', 'MS4A7'] ...\n",
            "Dim 6: ['THEMIS', 'TSHZ2', 'FHIT', 'CD8B', 'ANK3', 'CD8A', 'NELL2', 'AC139720.1', 'LYZ', 'IL32'] ...\n",
            "Dim 7: ['SLC2A3', 'S100A9', 'S100A8', 'CAMK2D', 'HLA-DRA', 'ANK3', 'PDE4D', 'HLA-DQA1', 'CD83', 'AOAH'] ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We then visualize the per-dimension gene importance scores using plot_top_genes_by_dimension_sequential, which produces a heatmap of the top-\n",
        "ğ‘˜\n",
        "k genes for each latent dimension and highlights each dimensionâ€™s top genes with a red bounding box."
      ],
      "metadata": {
        "id": "lHPE_JtFHWxA"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "# Example usage:\n",
        "plot_top_genes_by_dimension_sequential(avg_abs_sens_genes, mdata[\"rna\"].var_names, top_k=5)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "kEOgjFtDHBYZ",
        "outputId": "3fd3e6a0-2c6e-45fc-ce9c-41c0e9df5ca6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 900x1200 with 2 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAA4IAAASmCAYAAAB7kxXPAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjAsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvlHJYcgAAAAlwSFlzAAAPYQAAD2EBqD+naQABAABJREFUeJzs3Xtczuf/B/DXXan7TimpVKKSnEZDzvMlOeSQNCNyGJmxzakZtkYOOcSo0cxhbDJ105zLZDUJ3zEzp82UQ1sRiqSSQ8fP7w/f7p97910q1efmfj0fj8/D7utzfa7P+3Mt6d11fa5LIgiCACIiIiIiItIaOmIHQERERERERLWLiSAREREREZGWYSJIRERERESkZZgIEhERERERaRkmgkRERERERFqGiSAREREREZGWYSJIRERERESkZZgIEhERERERaRkmgkRERERERFqGiSAREb20hIQESCQSLFq0SOxQatyiRYsgkUiQkJAgdihERERVxkSQiIgUUlJSIJFIlA5DQ0PY2NigT58+WLBgAZKTk8UOk4iIiF6SRBAEQewgiIhIM6SkpMDBwQGOjo4YO3YsACA/Px93797Fb7/9hkuXLkFXVxdz587FsmXLIJFIAACPHz/GjRs3YG5uDnNzczEfocZlZmYiMzMTTZo0gaGhodjhEBERVQkTQSIiUihNBN3d3XH48GGV8//9738xbtw4pKSkYP78+ViyZIkIURIREdHL4tRQIiKqsB49euDw4cMwMDDAF198gZs3bwIo+x1Be3t72NvbIycnBx9++CGsra1Rt25d9OzZE+fOnQMA3L59G2PHjoWlpSVkMhn69++Pa9euqb3/P//8g0mTJqFJkyYwMDCAtbU1JkyYgNTUVJW6EokErq6uyMjIwPjx42Fubg6ZTIauXbuqfb/vzp07mDlzJpycnCCTyWBqaopWrVrhgw8+QE5OjqJeee8IRkdHo3fv3jAxMYFMJsObb76JkJAQFBUVKdUrnYI7YcIEXL9+HW+//Tbq16+PunXrom/fvrh48WJ5/xuIiIheGhNBIiKqlBYtWsDb2xsFBQXYv3//C+sXFBSgX79++OWXXzBy5Eh4enril19+Qd++fZGUlIRu3brh+vXrGDt2LPr164e4uDgMHjwYxcXFSu2cPn0a7du3x7Zt2+Di4oKZM2fiP//5DyIiItC5c2f8/fffKvfOzs5Gjx498Ndff2HcuHEYNmwYfv/9d7i7u+PSpUuKeo8fP8Zbb72Fr776Co6Ojpg+fTomTJiA5s2bY/v27bh3794LnzMkJASenp74448/MHr0aEydOhVPnjzBJ598ghEjRkDdBJyUlBR07doVWVlZmDhxIvr164cjR46gd+/eyMjIeOE9iYiIqkwgIiL6n3/++UcAILi7u5db79tvvxUACOPGjRMEQRCOHj0qABAWLlyoVM/Ozk4AIIwYMUIoLCxUlK9cuVIAIJiamgoff/yxUFJSojj34YcfCgCEPXv2KMoKCgoEe3t7wdjYWDh37pzSPU6cOCHo6uoKHh4eSuUABADCRx99JBQXFyvKt2zZIgAQpkyZoiiLiooSAAh+fn4qz/rw4UPh6dOnis8LFy4UAAhHjx5VlF2/fl3Q09MTLC0thRs3bijKnz59KvTo0UMAIHz//feK8tJ+BiCsWLFC6X7z588XAAhBQUEqsRAREVUXjggSEVGl2djYAHi2cEpFrF69Gnp6eorPPj4+AICioiIsXbpUsejM8+eenx558OBBpKSkYM6cOWjfvr1S2z169MDQoUNx6NAh5ObmKp2rW7cuVq5cCR2d///nbvz48dDT08OZM2dU4pTJZCplRkZGMDAwKPf55HI5ioqK8Mknn6Bx48aKcgMDA6xcuRIAEBYWpnKdg4MD5syZo1T23nvvAYDa+IiIiKqL3ourEBERVV39+vXRpEkTpTJra2sAgJOTk8rKm6Xnbt++rSj79ddfAQBXrlxRu1dheno6SkpKcPXqVXTs2FFR3rx5cxgZGSnV1dPTQ8OGDZGdna0o69mzJ6ytrbFixQpcvHgRHh4e6NWrF1q1aqWUpJbl/PnzAABXV1eVc926dYNUKsWFCxdUzrVr104pSQUAW1tbAFCKj4iIqLoxESQiokorTdIsLCxeWLdevXoqZaWjg+WdKywsVJRlZWUBACIiIsq916NHj15479J7PP8OoomJCX799VcsWLAA0dHROHToEACgcePG+Oyzz/DRRx+Ve9/SkciGDRuqnJNIJGjYsCFu3bqlcq685//3O5JERETViVNDiYio0kpXzOzUqVOt3K80YYqOjoYgCGUevXr1qvI9mjRpgrCwMNy7dw/nz5/HypUrUVJSgqlTp2LHjh0Vik/dAi+CICAjI6PMpJSIiEgMTASJiKhSrl69ih9++AEGBgZ4++23a+WeXbp0AQCcOnWqxu+lo6ODdu3aYe7cuYoEMCoqqtxrSt9bVLelxOnTp/H06VO0a9euukMlIiKqMiaCRERUYb/88gvc3d2Rn5+Pzz77DI0aNaqV+w4dOhRNmjRBSEgIjh8/rnK+sLAQ//3vf6vc/l9//aV2NK+0TCqVlnv96NGjoaenh5CQEKV3GwsKCvDpp58CACZMmFDl+IiIiKob3xEkIiIV169fVyzKUlBQgLt37+K3337Dn3/+CV1dXcyfPx8LFy6stXgMDAywe/duDBw4EL169YKbmxvatm0LiUSC1NRUnDhxAg0aNEBSUlKV2o+Li8OcOXPw1ltvoXnz5mjQoAH+/vtvREVFQSqVYurUqeVe7+joiJUrV+KTTz6Bs7MzvL29UbduXURHR+PKlSsYOnQoxo4dW6XYiIiIagITQSIiUpGcnIzFixcDeLalgqmpKVq2bImAgACMHz8ejo6OtR5Tp06dcPHiRaxatQqHDh3CL7/8AgMDAzRq1AheXl6KbSeqwt3dHSkpKTh+/Dj27t2LvLw8NGrUCCNHjsTcuXPRunXrF7Yxa9YsNGvWDCEhIQgPD0dBQQGaN2+O4OBgzJgxo0KrjxIREdUWiSAIgthBEBERERERUe3hO4JERERERERahokgERERERGRlmEiSEREREREpGWYCBIREREREWkZJoJERERERERahokgERERERGRluE+gkTPuXnzJv755x/07NlT7FA0ypMnT3Dv3j00adJE7FBqjSAIKCgogIGBgaIsPz8fx48fR05ODjp06ICmTZuKGKFmuXfvHqZMmQJ/f3906tRJ7HA0wu3bt5GVlQULCws0bNhQ7HA0Qnx8PE6ePIkHDx7AwsIC/fr1g4uLi9hh1ZpTp06hU6dO0NPjj1/qXLx4ETKZDM2bN1eUJSUl4eTJkygqKkL79u35/YWoOglEpLB06VJBR0dH7DBqzZEjR4T//Oc/grm5udCyZUth0aJFwqNHj1TqhYeHa1W/zJs3TzA2NhZ0dXWF//znP0JaWppw9epVoUmTJoKOjo4gkUgEXV1dwc/PT+xQNUZKSoogkUiEqKgosUOpVZGRkcKNGzeUyg4dOiS0atVK0NHRURzt27cXjh8/LlKUtW/+/PnCxIkTFZ+zsrKEHj16KP7+lB46OjrC2LFjhaKiIhGjrT0SiURo0KCB8MEHHwgnTpwQOxyNcf/+faFDhw6Kvy/vvPOOUFxcLMyePVvQ1dVV+noZMGCA8OTJE7FDJnotcGookZY6e/Ys3N3dceXKFfTq1QsNGjRAYGAg2rVrh8TERLHDE83333+P5cuXw9XVFR9++CHOnz8PX19fTJ06FQMGDMDhw4exf/9+DBo0CKGhodi+fbvYIdeKevXqlXu0adMGADBq1CjUq1cPJiYmIkdcO3x8fHDixAnF57i4OAwZMgQPHjzAp59+inXr1uHjjz9Gamoq+vfvjwsXLogXbC2KjIxE48aNFZ8/+ugjnD59GsuXL0dqaiqePHmC5ORkfPrpp5DL5ViyZImI0dauunXr4ptvvkGvXr1gb2+Pzz//HJcuXRI7LFEtW7YMly5dQkBAAL7++mv89ttvmDRpEtauXYsFCxbg3LlzOH36NGbMmIGffvoJixYtEjtkoteCRBAEQewgiGpSYGBgheseO3YMCQkJKC4ursGINIOHhweuXr2KX3/9FWZmZgCAEydOYPTo0Xj06BGioqLQo0cPAEBERATeffddreiXLl26wMbGBvv27QMAbNu2Db6+vhgzZoxK0telSxcYGBjg+PHjYoRaq3R0dGBkZIShQ4eqndaWl5eHPXv2oHfv3oopxFu3bq3tMGudjo4OwsPDMXr0aABA586dce/ePZw7dw7169dX1Lt16xbatWsHV1dX7Nq1S6xwa42hoSHWrVuHiRMnoqSkBIaGhvjss8/U/gA/ZcoU/PTTT0hJSan1OGtb6ddLz549IZfLIZfL8ccff0AikaBNmzYYO3YsfHx8YGtrK3aotapZs2YYPHgw1q5dCwA4dOgQPDw88Mknn2DVqlVKdUePHo1ff/0Vf//9txihEr1exB6SJKpppdNJnp+OVN6hLVMgra2thdWrV6uU3717V+jUqZMgk8mE/fv3C4KgXVNDTU1Nha+//lrxuXTK486dO1Xqrly5UjAxManF6MQTHh4uNGrUSGjVqpUQExOjcv6ff/4RJBKJcODAARGiE49EIhEiIiIEQRCEgoICQUdHRwgNDVVb9/PPPxcsLS1rMzzRWFpaCl988YUgCILw+PFjQSKRCHv37lVbd/369YKBgUFthiea579eSv3111+Cv7+/YG9vr5h27urqKmzevFl48OCBOIHWMplMJnz77beKz2lpaWV+P9m0aZPWfL0Q1TRODaXXnqWlJdzd3XHv3r0XHp999pnY4daavLw8tdP3LCwskJCQgP/85z8YMWIEvv32WxGiE09xcTHq1Kmj+GxkZATgWb/8m5mZGZ4+fVprsYlpzJgxuHLlCjw9PeHl5YXBgwfjypUrivMSiUTE6DRDYWEhBEEocxEhBwcHZGdn125QIunTpw++//57FBQUQCaToUOHDopR9n/bv38/HBwcajlCzdG6dWssX74c//zzD44fP473338fly5dwuTJk2FtbS12eLXCysoKN2/eVHxOS0sDANy4cUOlbmpqKqysrGotNqLXGRNBeu116dIFFy9eRIMGDV541K1bV+xwa42joyN+++03tecMDQ1x8OBBDBs2DJMnT8aGDRtqOTrx2NjYKP3wUbduXQQFBcHJyUml7o0bN9QmiK+runXrYsWKFfjzzz8BAM7OzvDz89Oa5KYshw4dQkhICDZu3AhjY2PcunVLbb3bt28rTRd9nS1duhRpaWno3bs3YmNjsWrVKsTExMDDwwPh4eGIi4tDWFgYevXqhZ9//hnTp08XO2SN0KNHD2zYsAF37txBVFQU3n77bbFDqhV9+/bF+vXrcfjwYZw7dw6zZ8+GqakpfvzxR5w/f15R7/jx41i3bh26dOkiYrRErxGxhySJatrSpUsFiUQipKamvrDu9u3bBVdX11qISnxz5swRzMzM1K4SWqqkpET46KOPtGrK7JgxY4TevXtXqG6nTp0ET0/PGo5Icx08eFBo3ry50KBBA+Gzzz4TdHR0tHJq6L+PAQMGqK3bs2dPrfn+IgiCcP78eaFNmzYq0/NLV4aUSCSCVCoVlixZInaotUbd1FAShDt37gh2dnaKrw1dXV3h22+/FQ4fPizo6ekJtra2grW1taCjoyMYGRkJV69eFTtkotcCF4sh0lJXrlzBli1bMHr0aLRv377cumvWrMHFixe1YvGPO3fuIDMzE23bti23XmZmJpYtWwYPDw/06dOnlqLTPIWFhQgJCcGyZcuQl5eH/fv3w9PTU+ywak1qaqpKmY6OjtKKmcCzr5e5c+fC3d0dI0eOrK3wRCcIAn766SccOXIE165dQ15eHmQyGWxsbODi4oKhQ4dq1R6LixcvxjvvvKNYZZf+X+kiZbm5uejZsydatWoFAPjll1/wzTffICMjA82bN8e0adOU9hkkoqpjIkhERC8tNzcXDx48gKWlJWQymdjhEBER0QswESQiIiIiItIyXCyGiIiIiIhIyzARJCIiIiIi0jJMBImIiIiIiLQME0EiIiIiIiItw0SQiIiIiIhIy+iJHQBRbcnPz0d4eDhiY2ORnJyMhw8fwtjYGM2aNcOAAQMwevRo6Ovrix1mrWO/qMd+UY/9oh77RT32i3rsF/XYL0S1TMTN7IlqzR9//CE4ODgIOjo6gkQiEUxNTQVbW1vB1NRUkEgkgo6OjtCsWTPh8uXLYodaq9gv6rFf1GO/qMd+UY/9oh77RT32C1HtYyJIr72HDx8K9vb2gqGhoRAUFCSkpaUpnU9LSxOWL18uGBoaCo6OjkJeXp5IkdYu9ot67Bf12C/qsV/UY7+ox35Rj/1CJA4mgvTaCw0NFXR0dISjR4+WW+/IkSOCjo6OsG7dutoJTGTsF/XYL+qxX9Rjv6jHflGP/aIe+4VIHBJBEASxp6cS1aQBAwZAIpEgJiamQnUB4PDhwzUdlujYL+qxX9Rjv6jHflGP/aIe+0U99guROLhqKL32/vzzT7i6ulaorpubG/7888+aDUhDsF/UY7+ox35Rj/2iHvtFPfaLeuwXInEwEaTXXlZWFqysrCpUt2HDhsjKyqrhiDQD+0U99ot67Bf12C/qsV/UY7+ox34hEgcTQXrt5efno06dOhWqq6enh4KCghqOSDOwX9Rjv6jHflGP/aIe+0U99ot67BcicXAfQdIKKSkpOHfu3Avr/fPPP7UQjeZgv6jHflGP/aIe+0U99ot67Bf12C9EtY+LxdBrT0dHBxKJpEJ1BUGARCJBcXFxDUclPvaLeuwX9dgv6rFf1GO/qMd+UY/9QiQOjgjSa2/r1q1ih6CR2C/qsV/UY7+ox35Rj/2iHvtFPfYLkTg4IkhERERERKRluFgMERERERGRlmEiSEREREREpGWYCBIREREREWkZJoJERERERERahokgab38/HwsWrQI+fn5YoeiUdgv6rFf1GO/qMd+UY/9oh77RT32C1HN4KqhpPVyc3NhYmKCnJwc1KtXT+xwNAb7RT32i3rsF/XYL+qxX9Rjv6jHfiGqGRwRJCIiIiIi0jJMBImIiIiIiLQME0EiIiIiIiItoyd2AKS9Dv91T+wQAAAFBfkY8+FsJCTnQl9f/BfRpXq6YocA4Fm/jJ86F7+mPoK+fpHY4cCpoZHYIQAA8vMl+PjT+ch8IsHDkgKxw0Hy3TyxQwDw7OtlwrS5+O2GZny9GBloxj9vBfn5mDzzM1xOz4f+g1yxw4GFsYHYIQAA8vMBv7nzcPcRkFMk/vfdi7ezxQ4BAFBYkA+fDz/BkevZqKP/RNRY+vRpD9ndO6LGUMoAwEIABiYmYofy/zRoiQ1Z+2lih1BjnpxfJ3YIrzUuFkOi0ZREUNNoSiKoaTQlEdQ0mpIIahpNSQQ1jaYkgppGUxJBTVKaCJbo6OBhfQuxw9Eoxg/uQae4WOwwFJgIUlXxX0oiIiIiUuthfQt8sf242GFolLnjekKDxiaJqozvCBIREREREWkZjggSEREREb2qJBzXoarhVw4REREREZGWeSUTwbCwMEgkEsUhlUphY2MDd3d3hIaG4uHDhyrXnD17Fh4eHrCysoKRkRGcnZ0RGhqK4kq+7DthwgQYGZW9aIVEIsG0af//0m5KSgokEglWr15d4XvMnTsXEokEI0eOrHRsz/eLkZERmjZtiuHDh2PPnj0oKSlRucbV1VXpGplMBmdnZ6xZs0Zt/VKdO3eGRCLBhg0bKhUjERERERGJ75WeGhoYGAgHBwcUFhYiPT0dCQkJ8PPzQ0hICKKiouDs7AzgWRLYvXt3ODk54dNPP4WhoSFiYmIwc+ZMJCcnY+3atSI/yf8TBAE7duyAvb09oqOj8fDhQxgbG1f4egMDA2zZsgUA8OTJE6SmpiI6OhrDhw+Hq6srDhw4gHr16ildY2tri6CgIABAZmYm5HI5Pv74Y9y7dw/Lli1Tuce1a9dw5swZ2NvbIyIiAh9++OFLPDEREREREdW2VzoRHDhwIDp27Kj47O/vj/j4eHh4eMDT0xOJiYmQyWTYtGkTAOD48eMwMzMDAEyZMgW9evVCWFiYRiWCCQkJSEtLQ3x8PNzd3bF3716MHz++wtfr6elh7NixSmVLly7FihUr4O/vj/fffx+RkZFK501MTJSu+eCDD9CyZUt89dVXCAwMhK6u8nYG4eHhsLS0RHBwMIYPH46UlBTY29tX/mGJiIiIiEgUr+TU0PK4ubkhICAAqampCA8PBwDk5uZCKpXC1NRUqa61tTVkMpkIUZYtIiICrVu3Ru/evdG3b19ERERUS7ufffYZ+vfvj127duHq1avl1pVKpejUqRMePnyIu3fvqpyXy+UYPnw4PDw8YGJiArlcXi0xEhEREVElSSSv70E16rVLBAFg3LhxAIDY2FgAz96Dy83NxZQpU5CYmIjU1FRs3LgRe/fuhb+/f5XukZmZqfZ4Gfn5+dizZw98fHwAAD4+PoiPj0d6evpLtVtq3LhxEAQBcXFxL6xb+m7jv5Pn06dP4/r16/Dx8YG+vj6GDRtWbckqERERERHVjtcyEbS1tYWJiQmSk5MBAO+//z6mTZuGbdu2oXXr1rC3t8e0adMQGhqKmTNnVrr9R48ewcLCQu3xMg4ePIjs7GyMGjUKAODl5YU6depg586dL9VuqTZt2gCAol9KFRcXKxLZK1euYO7cufj9998xaNAglRHT8PBwNG7cGG+99RYAYNSoUbh8+TIuXLhQLTESEREREVHNe6XfESyPkZGRYvVQXV1dODo6wt3dHSNGjIBUKsWOHTswffp0WFlZwcvLq1JtS6VSREdHqz3Xr1+/KsccERGBjh07olmzZgAAY2NjDB48GBEREfDz86tyu6VKVzv996qqSUlJKkmsp6cnvv32W6WyoqIiREZGYvz48ZD8b7jezc0NlpaWiIiIQLt27V46RiIiIiIiqnmvbSKYl5cHS0tLAMCKFSuwdu1aXLt2TZEMeXt7o3fv3pg6dSo8PDygp1fxrtDV1UXfvn2rNd7s7GwcOnQI06ZNw/Xr1xXlb731Fvbs2YOrV6+iefPmePLkCXJycpSutbKyqtA98vLyAEBlFVJ7e3ts3rwZJSUlSE5OxrJly3Dv3j1IpVKlerGxsbh37x46d+6sFGPv3r2xY8cOrFy5Ejo66geZ8/PzkZ+fr1RWUJAPfX2DCsVORERERGpwQ3mqotfyKyctLQ05OTmKkbX169fDzc1NZf8/T09P3L59GykpKSJEqWzXrl3Iz89HcHAwnJycFMesWbMAQPEeXmRkJKytrZWOirp06RIAKPqlVN26ddG3b1/0798fH374IQ4dOoTffvsNn3/+uVK90hi8vb2VYoyMjMStW7dw7NixMu8dFBQEExMTpeOHzZqzWisRERERkTZ5LUcEt2/fDgBwd3cHAGRkZKjdOL6wsBDAsymPYouIiECbNm2wcOFClXObNm2CXC7H4sWL4e7uXqHFXtTZvn07JBLJC6evOjs7Y+zYsdi0aRNmz56NJk2a4NGjRzhw4ABGjhyJ4cOHq1wzY8YMREREoHfv3mrb9Pf3VyS1pRKSc6v0HERERERE9HJeu0QwPj4eS5YsgYODA8aMGQMAaN68OeLi4nD//n00aNAAwLMFUn744QcYGxvD0dFRzJBx8+ZNHD9+HIsXL1abZBUUFGDMmDE4ffo0unTpUqlRwFIrVqxAbGwsRo0aBScnpxfWnzt3Lr7//nuEhIRgzZo12LdvHx49eoSpU6fiP//5j0r92NhY7Nq1C19//TUMDFSnexoYGKiU6+vnq9QjIiIiIqKa90ongjExMUhKSkJRUREyMjIQHx+PuLg42NnZISoqSvGO22effYaxY8eiS5cumDx5MmQyGXbs2IGzZ89i6dKlqFOnTo3HeuTIETx9+lSl3MvLCz/++CMEQYCnp6faawcNGgQ9PT1ERESgS5cu5d6nqKhIsX/i06dPkZqaiqioKPzxxx/o3bs3vvnmmwrF27p1awwaNAhbtmxBQEAAIiIi0KBBA3Tv3l1tfU9PT2zevBk//vgjhg0bVqF7EBEREdFL4n57VEWvdCK4YMECAIC+vj7MzMzQtm1brFmzBr6+vkoLoowZMwbm5uYICgrCqlWrkJubixYtWmDjxo2YMmVKrcR6+PBhHD58WKXc3t4eERERaNKkCd58802115qamqJHjx6IjIxESEhIuQvb5OfnK/ZRNDQ0hKWlJVxcXLBgwQK8/fbbZS7mos6cOXPw448/IiAgAD///DN8fHygq6urtm6fPn1gaGiI8PBwJoJERERERBpOIgiCIHYQpJ0O/3VP7BA0klRPfbKt7ZwaGr24khZKvpsndggaycjglf49Z42xMOZKzepcvJ0tdggap0+f9pDdvYOcBg3xxfbjYoejUeaO6wmTzHSxw1CQdZr14kqvqCdnQsQO4bX2Wq4aSkRERERERGXjr0z/JysrCwUFBWWe19XVVdl0nYiIiIiI6FXERPB/hg0bVu4+eHZ2dhqx3yARERERkQI3lKcqYiL4P8HBwXjw4EGZ52UyWS1GQ0REREREVHOYCP6Pi4uL2CEQERERERHVCo4lExERERERaRmOCBIRERERvaq4oTxVEUcEiYiIiIjotfD111/D3t4eUqkUXbp0wW+//VZu/V27dqFly5aQSqVo27YtDh06pHReEAQsWLAA1tbWkMlk6Nu3L65du6ZUZ9myZejevTsMDQ1hampa7v3u378PW1tbSCQSZGdnV+URqw1HBEk0lkbc2Fgdy3rsF3V0+AtPtZpbGYsdgkbKe1okdgga6UlBsdghaCSZnq7YIWic0kEmAcDTIkHUWDQNe0NzRUZGYtasWdi4cSO6dOmCNWvWwN3dHVeuXIGlpaVK/ZMnT8LHxwdBQUHw8PCAXC6Hl5cXzp07hzZt2gAAvvjiC4SGhmLbtm1wcHBAQEAA3N3dcfnyZUilUgBAQUEBRowYgW7duuHbb78tN8b33nsPzs7OuHXrVvV3QCVxRJCIiIiIiF55ISEheP/99+Hr64vWrVtj48aNMDQ0xHfffae2/tq1azFgwADMmTMHrVq1wpIlS9ChQwesW7cOwLPRwDVr1mD+/PkYOnQonJ2d8f333+P27dvYv3+/op3Fixfj448/Rtu2bcuNb8OGDcjOzsbs2bOr7ZlfBhNBIiIiIqJXlUTn9T0qoaCgAGfPnkXfvn0VZTo6Oujbty9OnTql9ppTp04p1QcAd3d3Rf1//vkH6enpSnVMTEzQpUuXMtssy+XLlxEYGIjvv/8eOjqakYJpRhRERERERETPyc/PR25urtKRn5+vtm5mZiaKi4vRsGFDpfKGDRsiPT1d7TXp6enl1i/9szJtlvUcPj4+WLVqFZo0aVLh62oaE0EiIiIiItI4QUFBMDExUTqCgoLEDqvS/P390apVK4wdO1bsUJQwESQiIiIiIo3j7++PnJwcpcPf319tXXNzc+jq6iIjI0OpPCMjA1ZWVmqvsbKyKrd+6Z+VaVOd+Ph47Nq1C3p6etDT00OfPn0UMS9cuLDC7VS3Vz4RDAsLg0QiURxSqRQ2NjZwd3dHaGgoHj58qHLN2bNn4eHhASsrKxgZGcHZ2RmhoaEoLq7aamoJCQkYNmwYrKysoK+vD0tLSwwZMgR79+5VWz8xMVERa1nLxrq6ukIikcDJyUnt+bi4OMUz7969W1Fe2h+///67Uv2cnBx07twZUqkUhw8fBgAcOXIEEydORPPmzWFoaIimTZti0qRJuHPnjsr97O3tFffT0dGBqakp2rZti8mTJ+P06dMV6SYiIiIiogozMDBAvXr1lA4DA/Wrq+vr68PFxQVHjhxRlJWUlODIkSPo1q2b2mu6deumVB949jN2aX0HBwdYWVkp1cnNzcXp06fLbFOdPXv24OLFi7hw4QIuXLiALVu2AABOnDiBqVOnVrid6vbabB8RGBgIBwcHFBYWIj09HQkJCfDz80NISAiioqLg7OwM4FkS2L17dzg5OeHTTz+FoaEhYmJiMHPmTCQnJ2Pt2rWVuu/ChQsRGBgIJycnTJkyBXZ2drh//z4OHTqEd955BxERERg9erTSNeHh4bCyssKDBw+we/duTJo0SW3bUqkU169fx2+//YbOnTsrnYuIiIBUKsXTp09fGGNubi769++PP/74A/v27cOAAQMAAJ9++imysrIwYsQIODk54e+//8a6detw8OBBXLhwQeU3He3atcMnn3wCAHj48CESExOxa9cubN68GR9//DFCQkIq3G9EREREVA24obzCrFmzMH78eHTs2BGdO3fGmjVr8OjRI/j6+gIA3n33XTRq1EgxvXTmzJno1asXgoODMXjwYOzcuRO///47vvnmGwCARCKBn58fli5dCicnJ8X2ETY2NvDy8lLc98aNG8jKysKNGzdQXFyMCxcuAACaNWsGIyMjODo6KsWZmZkJAGjVqtUL9x2sSa9NIjhw4EB07NhR8dnf3x/x8fHw8PCAp6cnEhMTIZPJsGnTJgDA8ePHYWZmBgCYMmUKevXqhbCwsEolgrt370ZgYCCGDx8OuVyOOnXqKM7NmTMHP/30EwoLC5WuEQQBcrkco0ePxj///IOIiIgyE0FHR0cUFRVhx44dSong06dPsW/fPgwePBh79uwpN8aHDx/C3d0dFy5cwN69ezFw4EDFuZCQEPTo0UNp5aIBAwagV69eWLduHZYuXarUVqNGjVTmNq9cuRKjR4/Gl19+CScnJ3z44YflxkNEREREVBNGjhyJe/fuYcGCBUhPT0e7du1w+PBhxWIvN27cUPq5t3v37pDL5Zg/fz4+//xzODk5Yf/+/Yo9BAFg7ty5ePToESZPnozs7Gz06NEDhw8fVuwhCAALFizAtm3bFJ/bt28PADh69ChcXV1r+Kmr7pWfGloeNzc3BAQEIDU1FeHh4QCejY5JpVKV7Nva2hoymaxS7QcEBMDMzAzfffedUhJYyt3dHR4eHkplv/zyC1JSUjBq1CiMGjUKx48fR1paWpn38PHxQWRkJEpKShRl0dHRePz4Mby9vcuNLy8vDwMGDMC5c+ewZ88eDB48WOl8z549VZav7dmzJ8zMzJCYmFhu26VkMhm2b98OMzMzLFu2DILAbVaJiIiISBzTpk1Damoq8vPzcfr0aXTp0kVxLiEhAWFhYUr1R4wYgStXriA/Px+XLl3CoEGDlM5LJBIEBgYiPT0dT58+xc8//4zmzZsr1QkLC4MgCCpHWUmgq6srBEEQdTQQeM0TQQAYN24cACA2NhbAs47Pzc3FlClTkJiYiNTUVGzcuBF79+4t8+VTda5du4akpCR4eXnB2Ni4wtdFRETA0dERnTp1wpAhQ2BoaIgdO3aUWX/06NG4c+cOEhISFGVyuRx9+vSBpaVlmdc9evQIAwcOxJkzZ7Br1y6VhLQseXl5yMvLg7m5eYWfycjICG+//TZu3bqFy5cvV/g6IiIiIiISx2ufCNra2sLExATJyckAgPfffx/Tpk3Dtm3b0Lp1a9jb22PatGkIDQ3FzJkzK9xu6YhZ27ZtK3xNYWEhdu3ahVGjRgF4Nprm6emJiIiIMq9xcnJCx44dIZfLAQDZ2dk4dOiQynuH/zZ+/HicPn0au3btgqenZ4VjXLNmDQoKCjBy5MgKXwNAMYRe2s9EREREVAvE3vRdQzaUp8rTih42MjJSrB6qq6sLR0dHuLu7Y9u2bYiMjMSQIUMwffp07N+/v8Jt5ubmAkClRgNjYmJw//59+Pj4KMp8fHxw8eJF/PXXX2VeN3r0aOzduxcFBQXYvXs3dHV18fbbb5d7r4yMDEilUjRu3LjC8R0/fhyLFy+Gt7c33NzcKnwd8KyPAahdpZWIiIiIiDSLViSCeXl5ioRtxYoVWLlyJXbs2IF3330X3t7e2LdvH3r06IGpU6eiqKioQm3Wq1cPQOUSn/DwcDg4OMDAwADXr1/H9evX4ejoCENDw3JHBUeNGoWcnBzExMQgIiICHh4eL0xAN23aBH19fQwYMABXrlx5YWxJSUl4++230aZNG8WStpWRl5cHoOzEOD8/H7m5uUpHQX5+pe9DREREREQv77VPBNPS0pCTk4NmzZoBANavXw83NzfFCFYpT09P3L59GykpKRVqt2XLlgCAP//8s0L1c3NzER0djX/++QdOTk6Ko3Xr1nj8+DHkcnmZC61YW1vD1dUVwcHBOH78+AunhQJA69atcejQITx58gT9+vXDzZs3y6x78+ZN9O/fHyYmJjh06FClRjlLXbp0CQAU/fxvQUFBMDExUTq2rud2E0REREREYnhtto8oy/bt2wE8W8ETeDZlUt3G8aXbPFR0RLB58+Zo0aIFDhw4gLVr16oklv+2d+9ePH36FBs2bFBZiOXKlSuYP38+fvnlF/To0UPt9aNHj8akSZNgamqqsppRWTp37oz9+/dj8ODB6NevH06cOAELCwulOvfv30f//v2Rn5+PI0eOwNraukJtPy8vLw/79u1D48aN0apVK7V1/P39MWvWLKWyy+kcESQiIiJ6KdxHkKrotU4E4+PjsWTJEjg4OGDMmDEAniVwcXFxuH//Pho0aAAAKC4uxg8//ABjY2OVDR/Ls3jxYowaNQqTJk1CeHg49PSUuzM2NhYFBQXw8PBAeHg4mjZtig8++EClnfz8fKxYsQIRERFlJoLDhw/HzZs30aJFC+jr61c4xj59+mDHjh0YMWIEBgwYgKNHjyqmtT569AiDBg3CrVu3cPToUTg5OVW43VJPnjzBuHHjkJWVheXLl0NSxjcjAwMDGBgYKJXpP8it9P2IiIiIiOjlvTaJYExMDJKSklBUVISMjAzEx8cjLi4OdnZ2iIqKUmz6+Nlnn2Hs2LHo0qULJk+eDJlMhh07duDs2bNYunSp2v0AyzJy5Ej8+eefWLZsGc6fPw8fHx/Y2dnh/v37OHz4MI4cOQK5XI7bt2/j6NGjmDFjhtp2DAwM4O7ujl27diE0NFRtDCYmJli0aFGV+ubtt9/G5s2bMXHiRHh6eio2wRwzZgx+++03TJw4EYmJiUp7BxoZGcHLy0upnVu3bin2Y8zLy8Ply5exa9cupKen45NPPsGUKVOqFB8REREREdWu1yYRXLBgAQBAX18fZmZmaNu2LdasWQNfX1+ld97GjBkDc3NzBAUFYdWqVcjNzUWLFi2wcePGKiUyS5cuhZubG0JDQ7FhwwZkZWWhfv366Nq1Kw4cOABPT0+EhISgpKQEQ4YMKbOdIUOGYM+ePYiJianUdg8V5evri6ysLMyePRsjRozAvn37cOHCBQDAd999h++++06pvp2dnUoieOHCBYwbNw4SiQTGxsZo3LgxhgwZgkmTJqFz587VHjMREREREdUMiVDWCiVENexcKqeGqmNZz+DFlbSQrg7fgVCnrOnY2i7vacXe99Y2JSX8J1+dm9mPxQ5B47zl6gxpxh1kN2iIJVuPiR2ORgnw7QXTzHSxw1CQvTVP7BBqzJNflokdwmvttRkRJCIiIiLSOtx4naqIiaAaWVlZKCgoKPO8rq6uyuqbRERERERErwomgmoMGzYMx46VPQ3Czs6uwvsNEhERERERaRomgmoEBwfjwYMHZZ6XyWS1GA0REREREVH1YiKohouLi9ghEBERERG9GBcNoyri26VERERERERahokgERERERGRlmEiSEREREREpGX4jiCJRodz2tXKLywROwSN1MBYX+wQNNJ/kzPFDkEjRSexX9RZPrCF2CFopPpG/P7yb3V0n40VCAKQl18kcjSaRRDEjuBfuI8gVRG/coiIiIiIiLQME0EiIiIiIiItw0SQiIiIiIhIyzARJCIiIiIi0jKvdSIYFhYGiUSiOKRSKWxsbODu7o7Q0FA8fPhQ5ZqzZ8/Cw8MDVlZWMDIygrOzM0JDQ1FcXFzp+0dHR6NXr16wtLSEoaEhmjZtCm9vbxw+fFip3oYNGzBixAg0adIEEokEEyZMKLPN7OxsTJ48GRYWFqhbty569+6Nc+fOqa0bFRWFDh06QCqVokmTJli4cCGKisp+4Xvu3LmQSCQYOXJkmXVycnIwd+5cODk5QSaTwc7ODu+99x5u3LhRfmcQERERUfWT6Ly+B9UorVg1NDAwEA4ODigsLER6ejoSEhLg5+eHkJAQREVFwdnZGcCzJLB79+5wcnLCp59+CkNDQ8TExGDmzJlITk7G2rVrK3zP1atXY86cOejVqxf8/f1haGiI69ev4+eff8bOnTsxYMAARd2VK1fi4cOH6Ny5M+7cuVNmmyUlJRg8eDAuXryIOXPmwNzcHOvXr4erqyvOnj0LJycnRd2YmBh4eXnB1dUVX331Ff78808sXboUd+/exYYNG1TaFgQBO3bsgL29PaKjo/Hw4UMYGxur3L9fv364fPkyPvroIzRv3hzXr1/H+vXr8dNPPyExMVHlGiIiIiIi0jxakQgOHDgQHTt2VHz29/dHfHw8PDw84OnpicTERMhkMmzatAkAcPz4cZiZmQEApkyZgl69eiEsLKzCiWBRURGWLFmCfv36ITY2VuX83bt3lT4fO3ZMMRpoZGRUZru7d+/GyZMnsWvXLgwfPhwA4O3tjebNm2PhwoWQy+WKurNnz4azszNiY2Ohp/fsf3O9evWwfPlyzJw5Ey1btlRqOyEhAWlpaYiPj4e7uzv27t2L8ePHK9X59ddfcebMGaxbtw5Tp05VlLdo0QITJ07Ezz//jLfffrtCfUREREREROLR2jFXNzc3BAQEIDU1FeHh4QCA3NxcSKVSmJqaKtW1traGTCarcNuZmZnIzc3FW2+9pfa8paWl0mc7OztIKrCn3u7du9GwYUMMGzZMUWZhYQFvb28cOHAA+fn5AIDLly/j8uXLmDx5siIJBICPPvoIgiBg9+7dKm1HRESgdevW6N27N/r27YuIiAiVOrm5uQCAhg0bKpVbW1sDQKX6iIiIiIiIxKO1iSAAjBs3DgAUo3aurq7Izc3FlClTkJiYiNTUVGzcuBF79+6Fv79/hdu1tLSETCZDdHQ0srKyqi3e8+fPo0OHDtDRUf7f1rlzZzx+/BhXr15V1AOgNAoKADY2NrC1tVWcL5Wfn489e/bAx8cHAODj44P4+Hikp6cr1evYsSPq1q2LgIAAxMfH49atWzh27Bjmzp2LTp06oW/fvtX2rERERERUATqS1/egGqXViaCtrS1MTEyQnJwMAHj//fcxbdo0bNu2Da1bt4a9vT2mTZuG0NBQzJw5s8Lt6ujoYM6cOTh79iyaNGmCQYMGYfny5WUu6lJRd+7cUYy+Pa+07Pbt24p6z5f/u25pvVIHDx5EdnY2Ro0aBQDw8vJCnTp1sHPnTqV65ubmiIyMRE5ODvr06QNbW1u4urrCxsYG8fHxSqOPRERERESkubQ6EQQAIyMjxeqhurq6cHR0hLu7O7Zt24bIyEgMGTIE06dPx/79+yvV7uLFiyGXy9G+fXv89NNPmDdvHlxcXNChQwckJiZWKdYnT57AwMBApVwqlSrOP/9nWXVLz5eKiIhAx44d0axZMwCAsbExBg8erHZ6qIWFBdq3b49ly5Zh//79WLRoEU6cOAFfX98qPRMREREREdU+rR/CycvLU7yzt2LFCqxduxbXrl1TLNri7e2N3r17Y+rUqfDw8KjUqJePjw98fHyQm5uL06dPIywsDHK5HEOGDMGlS5cUCVxFyWQyxXuAz3v69Kni/PN/llX3+Xf5srOzcejQIUybNg3Xr19XlL/11lvYs2cPrl69iubNmwMA/v77b/Tu3Rvff/893nnnHQDA0KFDYW9vjwkTJiAmJgYDBw5UG3t+fr5KPAX5BdBXk6wSEREREVHN0uoRwbS0NOTk5ChGwtavXw83NzeVlTs9PT1x+/ZtpKSkVOk+9erVQ79+/RAREYHx48cjOTkZp0+frnQ71tbWareXKC2zsbFR1Hu+/N91S+sBwK5du5Cfn4/g4GA4OTkpjlmzZgGA0qhgWFgYnj59Cg8PD6U2PT09AQC//PJLmbEHBQXBxMRE6fhufXCFnpuIiIiIyiD2Xn/cR/CVpdU9vH37dgCAu7s7ACAjI0PtxvGFhYUAUO5m7BVVuoBLefsFlqVdu3Y4d+4cSkpKlMpPnz4NQ0NDxchdu3btAAC///67Ur3bt28jLS1NcR54lui1adMGu3btUjn69u2rtCVFRkYGBEFQ6aOK9I+/vz9ycnKUjokffVLpPiAiIiIiopentYlgfHw8lixZAgcHB4wZMwYA0Lx5c8TFxeH+/fuKesXFxfjhhx9gbGwMR0fHCrX9+PFjnDp1Su25mJgYAM/23qus4cOHIyMjA3v37lWUZWZmYteuXRgyZIjincA33ngDLVu2xDfffKOUtG3YsAESiUSxB+HNmzdx/PhxeHt7Y/jw4SqHr68vrl+/rhi9bN68OQRBwA8//KAU144dOwAA7du3LzN2AwMD1KtXT+ngtFAiIiIiInFoxTuCMTExSEpKQlFRETIyMhAfH4+4uDjY2dkhKipK8a7eZ599hrFjx6JLly6YPHkyZDIZduzYgbNnz2Lp0qWoU6dOhe73+PFjdO/eHV27dsWAAQPQuHFjZGdnY//+/Thx4gS8vLyUkqbo6GhcvHgRwLPRtT/++ANLly4F8GzapbOzM4BniWDXrl3h6+uLy5cvw9zcHOvXr0dxcTEWL16sFMOqVavg6emJ/v37Y9SoUbh06RLWrVuHSZMmoVWrVgAAuVwOQRAUUzv/bdCgQdDT00NERAS6dOmCCRMmYPXq1ZgyZQrOnz+PN954A+fOncOWLVvwxhtvcDN5IiIiIqJXhEQQBEHsIGpKWFiY0mqW+vr6MDMzQ9u2beHh4QFfX18YGxsrXfPTTz8hKCgIf/31F3Jzc9GiRQtMnToVU6ZMqfB9i4qKEBYWhh9//BHnz59Heno6dHV10aJFC4wePRozZsyAvr6+ov6ECROwbds2tW1t3boVEyZMUHx+8OAB5syZg/379+PJkyfo1KkTVq9erbJnIADs378fixcvRmJiIiwsLDBhwgQsWLBAkdA6OzsjJycHqampZT5L7969cfnyZdy6dQt6enq4desWFixYgKNHj+LWrVto0KABPDw8sHz5cpibm1e4jwDgwo2HlaqvLYylWvH7mUprYKz/4kpa6L/JmWKHoJGik9gv6iwfWPnZKNqg5LX9SajqzFrYQ/f2LTwwa4jPNh0ROxyNsmJKH9S/n/7iirVE5rZM7BBqzJP4eWKH8Fp7rRNB0mxMBNVjIqgeE0H1mAiqx0RQPSaC6jERVMVEsGwalwj2WS52CDXmyZHPxQ7htaa17wgSERERERFpKw49VFJWVhYKCgrKPK+rqwsLC4tajIiIiIiIiKhymAhW0rBhw3Ds2LEyz9vZ2VV5v0EiIiIiIqLawESwkoKDg/HgwYMyz8tkslqMhoiIiIi0GjdepypiIlhJLi4uYodARERERET0UvgrBCIiIiIiIi3DRJCIiIiIiEjLcGooEREREdGrSiIROwJ6RTERJNEUFpWIHYJGKmC/qCVww2e1zGUGYoegkZ4UFIsdAr1CMnPzxQ5B45iWCND933/z+y/R64lTQ4mIiIiIiLQME0EiIiIiIiItw0SQiIiIiIhIy/AdQSIiIiKiVxU3lKcq4lcOERERERGRlnmtE8GwsDBIJBLFIZVKYWNjA3d3d4SGhuLhw4cq15w9exYeHh6wsrKCkZERnJ2dERoaiuLiyq9AFx0djV69esHS0hKGhoZo2rQpvL29cfjwYaV6GzZswIgRI9CkSRNIJBJMmDChzDazs7MxefJkWFhYoG7duujduzfOnTuntm5UVBQ6dOgAqVSKJk2aYOHChSgqKiqz7blz50IikWDkyJFl1snIyICvry8sLS0hk8nQoUMH7Nq1q/yOICIiIiIijfJaJ4KlAgMDsX37dmzYsAHTp08HAPj5+aFt27b4448/FPXOnj2L7t27IyUlBZ9++imCg4PRtGlTzJw5E7NmzarUPVevXg1PT09IJBL4+/vjyy+/xDvvvINr165h586dSnVXrlyJ+Ph4vPHGG9DTK3u2bklJCQYPHgy5XI5p06bhiy++wN27d+Hq6opr164p1Y2JiYGXlxdMTU3x1VdfwcvLC0uXLlU8/78JgoAdO3bA3t4e0dHRapPk3Nxc9OjRA3v27MGUKVOwevVqGBsbw9vbG3K5vFL9Q0RERERE4tGKdwQHDhyIjh07Kj77+/sjPj4eHh4e8PT0RGJiImQyGTZt2gQAOH78OMzMzAAAU6ZMQa9evRAWFoa1a9dW6H5FRUVYsmQJ+vXrh9jYWJXzd+/eVfp87NgxxWigkZFRme3u3r0bJ0+exK5duzB8+HAAgLe3N5o3b46FCxcqJWOzZ8+Gs7MzYmNjFcllvXr1sHz5csycORMtW7ZUajshIQFpaWmIj4+Hu7s79u7di/HjxyvV2bRpE65fv44jR47Azc0NAPDhhx+ia9eu+OSTTzB8+HDo6+tXqI+IiIiIqBpwQ3mqIq0YEVTHzc0NAQEBSE1NRXh4OIBnI15SqRSmpqZKda2trSGTySrcdmZmJnJzc/HWW2+pPW9paan02c7ODpIK/CXevXs3GjZsiGHDhinKLCws4O3tjQMHDiA//9mGuJcvX8bly5cxefJkpRHGjz76CIIgYPfu3SptR0REoHXr1ujduzf69u2LiIgIlTonTpyAhYWFIgkEAB0dHXh7eyM9PR3Hjh174TMQEREREZH4tDYRBIBx48YBgGLUztXVFbm5uZgyZQoSExORmpqKjRs3Yu/evfD3969wu6Xvz0VHRyMrK6va4j1//jw6dOgAHR3l/22dO3fG48ePcfXqVUU9AEqjoABgY2MDW1tbxflS+fn52LNnD3x8fAAAPj4+iI+PR3p6uko9dQmxoaEhgGdTa4mIiIiISPNpdSJoa2sLExMTJCcnAwDef/99TJs2Ddu2bUPr1q1hb2+PadOmITQ0FDNnzqxwuzo6OpgzZw7Onj2LJk2aYNCgQVi+fHmZi7pU1J07d2Btba1SXlp2+/ZtRb3ny/9dt7ReqYMHDyI7OxujRo0CAHh5eaFOnToq7zK2aNECaWlpSE1NVSo/ceIEAODWrVtVeSwiIiIiIqplWp0IAoCRkZFiYRRdXV04OjrC3d0d27ZtQ2RkJIYMGYLp06dj//79lWp38eLFkMvlaN++PX766SfMmzcPLi4u6NChAxITE6sU65MnT2BgYKBSLpVKFeef/7OsuqXnS0VERKBjx45o1qwZAMDY2BiDBw9WmR46adIk6OrqwtvbGydPnkRycjKCgoKwb98+pfsSEREREZFm04rFYsqTl5eneGdvxYoVWLt2La5du6ZYtMXb2xu9e/fG1KlT4eHhUe6qnv/m4+MDHx8f5Obm4vTp0wgLC4NcLseQIUNw6dIlRQJXUTKZTPEe4POePn2qOP/8n2XVfX56Z3Z2Ng4dOoRp06bh+vXrivK33noLe/bswdWrV9G8eXMAgLOzM+RyOT744APF+49WVlZYs2YNPvzww3IXusnPz1eJpyA/H/pqklUiIiIiqiBuKE9VpNVfOWlpacjJyVGMhK1fvx5ubm4qCY2npydu376NlJSUKt2nXr166NevHyIiIjB+/HgkJyfj9OnTlW7H2tpaMe3zeaVlNjY2inrPl/+7bmk9ANi1axfy8/MRHBwMJycnxVG6Xca/RwWHDx+O27dv47fffsOpU6eQmpqKpk2bAoAiYVQnKCgIJiYmSkfYxpDKPD4REREREVUTrR4R3L59OwDA3d0dwLPN0tVtHF9YWAgA5W7GXlEdO3bEtm3b1CZpL9KuXTucOHECJSUlSgvGnD59GoaGhopErF27dgCA33//HZ07d1bUu337NtLS0jB58mRFWUREBNq0aYOFCxeq3G/Tpk2Qy+VYvHixUrm+vj46deqk+Pzzzz8DAPr27Vtm7P7+/ip7Mf556+mLHpmIiIiIiGqA1iaC8fHxWLJkCRwcHDBmzBgAz0a04uLicP/+fTRo0AAAUFxcjB9++AHGxsZwdHSsUNuPHz/GxYsX0a1bN5VzMTExAJ4tvFJZw4cPx+7du7F3717FPoKZmZnYtWsXhgwZongn8I033kDLli3xzTffYMqUKdDV1QUAbNiwARKJRHHtzZs3cfz4cSxevFhR9ryCggKMGTMGp0+fRpcuXdTGdO3aNWzcuBEeHh7ljggaGBiovLOonylUug+IiIiIiOjlaUUiGBMTg6SkJBQVFSEjIwPx8fGIi4uDnZ0doqKiFO/qffbZZxg7diy6dOmCyZMnQyaTYceOHTh79iyWLl2KOnXqVOh+jx8/Rvfu3dG1a1cMGDAAjRs3RnZ2Nvbv348TJ07Ay8sL7du3V9SPjo7GxYsXATwbffzjjz+wdOlSAM+mpTo7OwN4lgh27doVvr6+uHz5MszNzbF+/XoUFxerjNqtWrUKnp6e6N+/P0aNGoVLly5h3bp1mDRpElq1agUAkMvlEAQBnp6eap9j0KBB0NPTQ0REhCIRbN26NUaMGIEmTZrgn3/+wYYNG2BmZoaNGzdW9H8HEREREVUXbihPVaQVieCCBQsAPJvSaGZmhrZt22LNmjXw9fWFsbGxot6YMWNgbm6OoKAgrFq1Crm5uWjRogU2btyIKVOmVPh+pqam2Lx5M3788Uds3boV6enp0NXVRYsWLbBq1SrMmDFDqf6ePXuwbds2xefz588r9vqztbVVJIK6uro4dOgQ5syZg9DQUDx58gSdOnVCWFiYygijh4cH9u7di8WLF2P69OmwsLDA559/rugL4Nm00CZNmuDNN98s8zl69OiByMhIhISEQE9PD2+++Sa2bt2KjIwMmJubw9vbG4sXL1YsuENERERERJpPIggC5+eRKM78nSN2CBrJSKoVv5+pNCvTyq2yqy2u3Hkodggaaf3pG2KHoJG+HNpa7BA00t0c1VW2tV0zl+aok34bD8wa4tONR8QOR6Os/KAP6t9PFzsMBdnAL8UOocY8iflY7BBea1q9aigREREREZE24tBDJWVlZaGgoKDM87q6urCwsKjFiIiIiIhIa3EfQaoiJoKVNGzYMBw7dqzM83Z2dlXeb5CIiIiIiKg2MBGspODgYDx48KDM8zKZrBajISIiIiIiqjwmgpXk4uIidghEREREREQvhZOKiYiIiIiItAxHBImIiIiIXlXcUJ6qiCOCREREREREWoYjgiQamb6u2CFoJDMjfbFD0Eh5T4vEDkEjZT8tFDsEjfRRlyZih6CRnhSUiB2CRuK/R6pKB5kEQcAjfp9RIgiC2CEQVQuOCBIREREREWkZjggSEREREb2quKE8VRG/coiIiIiIiLQME0EiIiIiIiItw0SQiIiIiIhIy7zWiWBYWBgkEonikEqlsLGxgbu7O0JDQ/Hw4UOVa86ePQsPDw9YWVnByMgIzs7OCA0NRXFxcaXvHx0djV69esHS0hKGhoZo2rQpvL29cfjwYUWdlJQUSCQSrF69+oXtFRcXY+vWrXB1dYWZmRkMDAxgb28PX19f/P7774p6Z86cwbRp0/DGG2+gbt26aNKkCby9vXH16tVy2+/cuTMkEgk2bNig9vxff/2FESNGoGnTpjA0NIS5uTl69uyJ6OjoCvYIEREREVUric7re1CN0ooeDgwMxPbt27FhwwZMnz4dAODn54e2bdvijz/+UNQ7e/YsunfvjpSUFHz66acIDg5G06ZNMXPmTMyaNatS91y9ejU8PT0hkUjg7++PL7/8Eu+88w6uXbuGnTt3VvoZnjx5Ag8PD0ycOBGCIODzzz/Hhg0b8O677+LUqVPo3Lkz0tLSAAArV67Enj170KdPH6xduxaTJ0/G8ePH0aFDB1y6dElt+9euXcOZM2dgb2+PiIgItXVSU1Px8OFDjB8/HmvXrkVAQAAAwNPTE998802ln4mIiIiIiMShFauGDhw4EB07dlR89vf3R3x8PDw8PODp6YnExETIZDJs2rQJAHD8+HGYmZkBAKZMmYJevXohLCwMa9eurdD9ioqKsGTJEvTr1w+xsbEq5+/evVvpZ5gzZw4OHz6ML7/8En5+fkrnFi5ciC+//FLxedasWZDL5dDX///96EaOHIm2bdtixYoVCA8PV2k/PDwclpaWCA4OxvDhw5GSkgJ7e3ulOoMGDcKgQYOUyqZNmwYXFxeEhIRg8uTJlX4uIiIiIiKqfVoxIqiOm5sbAgICkJqaqkiMcnNzIZVKYWpqqlTX2toaMpmswm1nZmYiNzcXb731ltrzlpaWlYo1LS0NmzZtQr9+/VSSQADQ1dXF7NmzYWtrCwDo3r27UhIIAE5OTnjjjTeQmJio9h5yuRzDhw+Hh4cHTExMIJfLKxSbrq4uGjdujOzs7Eo9ExERERERiUdrE0EAGDduHAAoRu1cXV2Rm5uLKVOmIDExEampqdi4cSP27t0Lf3//CrdraWkJmUyG6OhoZGVlvXScMTExKCoqUsRbFYIgICMjA+bm5irnTp8+jevXr8PHxwf6+voYNmxYmdNDAeDRo0fIzMxEcnIyvvzyS8TExKBPnz5Vjo2IiIiIiGqXVkwNLYutrS1MTEyQnJwMAHj//ffx119/YdOmTdiyZQuAZyNe69atwwcffFDhdnV0dDBnzhwEBgaiSZMm6NmzJ3r06IEBAwagQ4cOlY6zdBSvbdu2lb62VEREBG7duoXAwECVc+Hh4WjcuLFiBHPUqFH47rvvcOHCBbRr106l/ieffKKYRqujo4Nhw4Zh3bp1VY6NiIiIiKpIIhE7AnpFafWIIAAYGRkpVg/V1dWFo6Mj3N3dsW3bNkRGRmLIkCGYPn069u/fX6l2Fy9eDLlcjvbt2+Onn37CvHnz4OLigg4dOpQ5PbMsubm5AABjY+NKXVcqKSkJU6dORbdu3TB+/Hilc0VFRYiMjMTIkSMh+d83Ejc3N1haWpY5Kujn54e4uDhs27YNAwcORHFxMQoKCqoUGxERERER1T6tHhEEgLy8PMU7eytWrMDatWtx7do1GBkZAQC8vb3Ru3dvTJ06FR4eHtDTq3iX+fj4wMfHB7m5uTh9+jTCwsIgl8sxZMgQXLp0CVKptELt1KtXDwDUbnfxIunp6Rg8eDBMTEywe/du6OrqKp2PjY3FvXv30LlzZ1y/fl1R3rt3b+zYsQMrV66Ejo7y7wtatmyJli1bAgDeffdd9O/fH0OGDMHp06cVyeS/5efnIz8/X6msIL8Q+gYGlX4mIiIiIiJ6OVo9IpiWloacnBw0a9YMALB+/Xq4ubkpksBSnp6euH37NlJSUqp0n3r16qFfv36IiIjA+PHjkZycjNOnT1f4+tKk688//6zUfXNycjBw4EBkZ2fj8OHDsLGxUalTOurn7e0NJycnxREZGYlbt27h2LFjL7zP8OHDcebMmXL3KQwKCoKJiYnSseXr4Eo9DxERERERVQ+tHhHcvn07AMDd3R0AkJGRoXbj+MLCQgDPplG+rI4dO2Lbtm24c+dOha8ZOHAgdHV1ER4eXuEFY54+fYohQ4bg6tWr+Pnnn9G6dWuVOo8ePcKBAwcwcuRIDB8+XOX8jBkzEBERgd69e5d7rydPngB4lniWxd/fX2Uvxuv3CivyKERERERUFm68TlWktV858fHxWLJkCRwcHDBmzBgAQPPmzREXF4f79+8r6hUXF+OHH36AsbExHB0dK9T248ePcerUKbXnYmJiAAAtWrSocKyNGzfG+++/j9jYWHz11Vcq50tKShAcHKzYUL64uBgjR47EqVOnsGvXLnTr1k1tu/v27cOjR48wdepUDB8+XOXw8PDAnj17FFM61e1/WFhYiO+//x4ymUxtslnKwMAA9erVUzo4LZSIiIiISBxaMSIYExODpKQkFBUVISMjA/Hx8YiLi4OdnR2ioqIU7+p99tlnGDt2LLp06YLJkydDJpNhx44dOHv2LJYuXYo6depU6H6PHz9G9+7d0bVrVwwYMECxz97+/ftx4sQJeHl5oX379krXHDlyBE+fPlVpy8vLC23atEFwcDCSk5MxY8YM7N27Fx4eHqhfvz5u3LiBXbt2ISkpCaNGjQLwbFXPqKgoDBkyBFlZWSobyI8dOxbAs2mhDRo0QPfu3dU+h6enJzZv3owff/wRw4YNw5QpU5Cbm4uePXuiUaNGSE9PR0REBJKSkhAcHKwypZaIiIiIiDSTViSCCxYsAADo6+vDzMwMbdu2xZo1a+Dr66u0EueYMWNgbm6OoKAgrFq1Crm5uWjRogU2btyIKVOmVPh+pqamigRq69atSE9Ph66uLlq0aIFVq1ZhxowZKtccPnwYhw8fVim3t7dHmzZtYGhoiJiYGISFhWHbtm1YsmQJHj9+DBsbG7i5uSEiIgKNGjUCAFy4cAEAEB0djejoaJU2x44di7t37+Lnn3+Gj4+PygIypfr06QNDQ0OEh4dj2LBhGDlyJL799lts2LAB9+/fh7GxMVxcXLBy5Up4enpWuH+IiIiIiEhcEkEQBLGDIO10KS1P7BA0kkU9TplVp6CoROwQNNJfd3LFDkEjmUorNoND29iaGYodgkYqKub3l3+zdXaE3p3byKpvielrfxI7HI3y1Ux3mGVliB2GgszrG7FDqDFP9k8WO4TXmta+I0hERERERKSttGJqaHXKysoqd/N0XV1dWFhY1GJERERERERElcNEsJKGDRtW7t56dnZ2Vd5vkIiIiIiIqDYwEayk4OBgPHjwoMzzMpmsFqMhIiIiIiKqPCaCleTi4iJ2CEREREREz3BDeaoifuUQERERERFpGSaCREREREREWoZTQ4mIiEhrWbp1g05GuthhaBzd//VJ/Qd38dVMd5Gj0Sym2Zlih0BULZgIkmgKuIGvWhk5+WKHoJHMjfXFDkEjmRpw43R1bOpz4S6qGJ2MDOjduS12GBpLAsDswV2xw6DySCRiR0CvKCaCREREpPVKdHSQW5/7AJcyuZ8BCZ71S54Z++Xf6okdAFE1YCJIREREWi+3vgWWhZW9T7C2mTehF0zvZyDPzALrdv4idjga53OxAyCqBlwshoiIiIiISMtwRJCIiIiI6BUl4TuCVEUcESQiIiIiItIyTAQrKCwsDBKJRHFIpVLY2NjA3d0doaGhePjwoVJ9V1dXpfrPH3XqVH2VP29vb0gkEnz66adqzyckJCjuc/bsWZXzEyZMgJGRkUqsbdq0Ual75MgRGBoaokOHDsjKygIALF++HF27doWFhQWkUimcnJzg5+eHe/fuVfmZiIiIiIiodnFqaCUFBgbCwcEBhYWFSE9PR0JCAvz8/BASEoKoqCg4OzsDAObNm4dJkyYpXfvo0SN88MEH6N+/f5XunZubi+joaNjb22PHjh1YsWJFudMBFi1ahOjo6CrdKz4+HkOGDEGLFi3w888/w8zMDABw9uxZtGvXDqNGjYKxsTESExOxefNm/Pjjj7hw4QLq1q1bpfsREREREVHtYSJYSQMHDkTHjh0Vn/39/REfHw8PDw94enoiMTERMpkM/fr1U7k2PDwcADBmzJgq3XvPnj0oLi7Gd999Bzc3Nxw/fhy9evVSW7ddu3Y4ePAgzp07hw4dOlTqPseOHcOQIUPQvHlzpSSwNIZ/69atG4YPH47o6GiMGjWqcg9FRERERES1jlNDq4GbmxsCAgKQmpqqSPbUkcvlqFu3LoYOHVql+0RERKBfv37o3bs3WrVqhYiIiDLrTp8+HfXr18eiRYsqdY8TJ05g8ODBaNasGX7++Wc0aNDghdfY29sDALKzsyt1LyIiIiJ6OWW9ivQ6HFSzmAhWk3HjxgEAYmNj1Z6/d+8e4uLi4OXlVaXpk7dv38bRo0fh4+MDAPDx8cHu3btRUFCgtn69evXw8ccfIzo6GufOnavQPX755RcMGjQIDg4OOHLkCMzNzdXWEwQBmZmZSE9Px4kTJzBjxgzo6urC1dW10s9FRERERES1j4lgNbG1tYWJiQmSk5PVno+MjERRUVGVp4Xu2LEDBgYGitHEUaNG4cGDBzh06FCZ18yYMQP169fH4sWLX9j+nTt3MHDgQNjZ2SE+Pr7MJBAAMjIyYGFhAWtra/Ts2RM3btyAXC5Hy5YtK/9gRERERERU65gIViMjIyOV1UNLyeVyWFhYqH13sCIiIiIwePBgGBsbAwCcnJzg4uJS7vRQExMT+Pn5ISoqCufPny+3/UePHuHhw4do2LAh6tWrV25dMzMzxMXFITo6GoGBgTA3N0deXl7lH4qIiIiIiETBRLAa5eXlKRK15/399984deoURo4cCT29yq/Pk5iYiPPnz+Ott97C9evXFYerqysOHjyI3NzcMq+dOXMmTE1NX/iuYLNmzbBy5UrEx8fDx8cHxcXFZdbV19dH37594eHhgYCAAHz99dd47733cPDgwTKvyc/PR25urtJRkJ//wmcnIiIionJIXuODahQTwWqSlpaGnJwcNGvWTOWcXC4HUPXVQksXoPn444/h5OSkOIKDg/H06VO1K3mWqsyo4Ny5czF37lzs27cP77//PgRBqFB83bt3h7W1dbmjk0FBQTAxMVE6tq4PqVD7RERERERUvbh9RDXZvn07AMDd3V3lnFwuh6OjI7p27VrpdgVBgFwuR+/evfHRRx+pnF+yZAkiIiLg6+tbZht+fn5Ys2YNFi9eDFNT03Lvt3LlSmRlZWHLli2oX78+goODKxTn06dPkZOTU+Z5f39/zJo1S6nscjpHBImIiIiIxMBEsBrEx8djyZIlcHBwUBn1O3/+PBITExEQEFCltn/55RekpKQgMDAQw4cPVzl/9epVBAQE4Pbt27CxsVHbRumo4KJFi/Dmm2++8J6bNm1CdnY2QkJCUL9+fcyfPx/As/cIJRIJDA0Nlerv2bMHDx48UNpf8d8MDAxgYGCgVKb/oOwprUREREREVHOYCFZSTEwMkpKSUFRUhIyMDMTHxyMuLg52dnaIioqCVCpVql86XbKq00IjIiKgq6uLwYMHqz3v6emJefPmYefOnSojbs+bOXMmvvzyS1y8ePGF21fo6OggIiICOTk5CAgIgJmZGT766CNcu3YNffv2xciRI9GyZUvo6Ojg999/R3h4OOzt7TFz5swqPSMRERERVQ3326OqYiJYSQsWLADwbMEUMzMztG3bFmvWrIGvr6/KQjElJSXYuXMnOnTogBYtWlT6XoWFhdi1axe6d+8OMzMztXXatGkDBwcHhIeHl5sImpqaws/Pr0JbSQDPnm/fvn3o27cvpk+fDlNTU/Tv3x/vvPMO4uPjsW3bNhQWFsLOzg7Tpk3DvHnzKrT5PBERERERiU8iVHRFEKJqdi6VU0PV0dPhGk7qmBvrix2CRrqR+VjsEDSSjZlM7BA0kg5HDlRYvdEUenduIbtBQywLOyZ2OBpj3oReML2fgVzzhli38xexw9E4n/dxFDsEBSPvMLFDqDF5P0wQO4TXGn/iJCIiIiIi0jKcGiqirKwsFBQUlHleV1cXFhYWtRgRERERERFpAyaCIho2bBiOHSt7GoqdnR1SUlJqLyAiIiIieqVwsRiqKiaCIgoODsaDBw/KPC+T8R0XIiIiIiKqfkwEReTi4iJ2CEREREREpIW4WAwREREREZGW4YggEREREdEriu8IUlVxRJCIiIiIiEjLcESQRKOvy99DqGNnYSh2CBopv7BE7BA0Uh09/j1S53bWE7FD0EhtGpuIHYLG0Xnur1CJIF4cmkqqp4sBzbiVFdHriD9BEBERERERaRmOCBIRERERvaL4jiBVFUcEiYiIiIiItAwTQSIiIiIiIi3DRJCIiIiIiEjLMBGsgLCwMEgkEsUhlUphY2MDd3d3hIaG4uHDh0r1XV1dleo/f9SpU6dKMSQkJGDYsGGwsrKCvr4+LC0tMWTIEOzdu1dRJyUlReVe5ubm6N69Oz7//HPcuHFDbdt37tzB5MmT4eDgAJlMBkdHR8yaNQv3798v97n09fXh4OCAyZMn4+bNm1V6LiIiIiIiqn1cLKYSAgMD4eDggMLCQqSnpyMhIQF+fn4ICQlBVFQUnJ2dAQDz5s3DpEmTlK599OgRPvjgA/Tv37/S9124cCECAwPh5OSEKVOmwM7ODvfv38ehQ4fwzjvvICIiAqNHj1bU9/HxwaBBg1BSUoIHDx7gzJkzWLNmDdauXYtvv/0Wo0aNUtTNy8tDt27d8OjRI3z00Udo3LgxLl68iHXr1uHo0aM4e/YsdJ5bW9vW1hZBQUEAgIKCAly+fBkbN27ETz/9hMTERBgacusDIiIiolrDtWKoipgIVsLAgQPRsWNHxWd/f3/Ex8fDw8MDnp6eSExMhEwmQ79+/VSuDQ8PBwCMGTOmUvfcvXs3AgMDMXz4cMjlcqURxTlz5uCnn35CYWGh0jUdOnTA2LFjlcpSU1PRv39/jB8/Hq1atcKbb74JAIiKikJqaioOHjyIwYMHK+qbmZkhMDAQFy9eRPv27RXlJiYmKm07ODhg2rRp+OWXX9Q+OxERERERaRZODX1Jbm5uCAgIQGpqqiLZU0cul6Nu3boYOnRopdoPCAiAmZkZvvvuO7XTSt3d3eHh4fHCduzs7BAWFoaCggJ88cUXivLc3FwAQMOGDZXqW1tbAwBkMtkL27aysgIA6Onx9wpERERERK8CJoLVYNy4cQCA2NhYtefv3buHuLg4eHl5oW7duhVu99q1a0hKSoKXlxeMjY1fOs5u3brB0dERcXFxirKePXtCR0cHM2fOxK+//oq0tDQcOnQIy5Ytg5eXF1q2bKnURnFxMTIzM5GZmYk7d+4gPj4eCxcuRLNmzfDWW2+9dIxERERERFTzOIRTDWxtbWFiYoLk5GS15yMjI1FUVFTpaaGJiYkAgLZt2750jKXatGmDAwcOIDc3F/Xq1UPr1q3xzTffYPbs2ejWrZui3vjx47FlyxaV65OSkmBhYaFU1qpVK8TGxkJfX7/a4iQiIiKiF+OG8lRVTASriZGRkcrqoaXkcjksLCwq/f5c6bTN6hgNLGVkZAQAePjwIerVqwcAaNSoETp37oxBgwbBzs4OJ06cQGhoKMzNzbF69Wql6+3t7bF582YAQFFREa5cuYIvvvgCAwcOxIkTJ1SSRCIiIiIi0jxMBKtJXl4eLC0tVcr//vtvnDp1CtOmTav0O3SliVpZCWZV5OXlAfj/5PKXX36Bh4cHfv31V8VCOF5eXqhXrx4WL16MiRMnonXr1orr69ati759+yo+DxgwAD169EDHjh2xYsUKBAcHq71vfn4+8vPzlcoK8guhb2BQbc9GREREREQVw3cEq0FaWhpycnLQrFkzlXNyuRxA5VcLBaB4P+/PP/98uQCfc+nSJVhaWiqSzE2bNqFhw4ZKq6ECgKenJwRBwMmTJ1/YpouLC0xMTHD8+PEy6wQFBcHExETp2PK1+qSRiIiIiIhqFhPBarB9+3YAz1bw/De5XA5HR0d07dq10u02b94cLVq0wIEDBxQjeS/j1KlTSE5OVtrLMCMjA8XFxSp1S7ekKCoqqlDbxcXF5cbo7++PnJwcpWPS1E8q+QRERERE9DyJRPLaHlSzmAi+pPj4eCxZsgQODg4qo37nz59HYmKi0mbvlbV48WLcv38fkyZNUpuUxcbG4uDBgy9sJzU1FRMmTIC+vj7mzJmjKG/evDkyMjKQkJCgVH/Hjh0AoLSHYFmOHj2KvLw8xd6E6hgYGKBevXpKB6eFEhERERGJg+8IVkJMTAySkpJQVFSEjIwMxMfHIy4uDnZ2doiKioJUKlWqHxERAaBq00JLjRw5En/++SeWLVuG8+fPw8fHB3Z2drh//z4OHz6MI0eOKKafljp37hzCw8NRUlKC7OxsnDlzBnv27IFEIsH27dvh7OysqDtt2jRs3boVQ4YMwfTp02FnZ4djx45hx44d6NevH7p06aLUdk5OjmK/xNLFYjZs2ACZTIbPPvusys9JRERERES1hyOClbBgwQKMGzcOU6ZMwZo1ayAIAtasWYM//vgDbdq0UapbUlKCnTt3okOHDmjRosVL3Xfp0qU4cuQIWrVqhQ0bNmDy5Mn44osvYGhoiAMHDsDHx0ep/o4dOzBu3Di89957WLx4Ma5duwY/Pz8kJibC29tbqW6LFi1w9uxZDBgwAOHh4Zg+fTpOnjyJ2bNnY//+/SqxpKWlYdy4cRg3bhwmTpyIb775Br169cJ///tftGvX7qWek4iIiIjoZXz99dewt7eHVCpFly5d8Ntvv5Vbf9euXWjZsiWkUinatm2LQ4cOKZ0XBAELFiyAtbU1ZDIZ+vbti2vXrinVWbZsGbp37w5DQ0OYmpqq3OPixYvw8fFB48aNIZPJ0KpVK6xdu/aln/VlcUSwAiZMmIAJEyZU6hodHR2kpaVVWwxubm5wc3Mrt469vT0EQah02y1atMCuXbteWO/f00eJiIiIiDRFZGQkZs2ahY0bN6JLly5Ys2YN3N3dceXKFbWr+588eRI+Pj4ICgqCh4cH5HI5vLy8cO7cOcUgzxdffIHQ0FBs27YNDg4OCAgIgLu7Oy5fvqyYDVhQUIARI0agW7du+Pbbb1Xuc/bsWVhaWiI8PByNGzfGyZMnMXnyZOjq6mLatGk12ynlkAhVyRyIqsGltJdfAOd1ZGdhKHYIGim/sETsEDRSauZjsUPQSIVF/HpRp01jE7FD0DiGjo2hc+sWshs0xJKtx8QOR2ME+PaC6f0MFFjZ4NKviWKHo3E62NUTOwQFs3HyF1d6RWVtr9w6G126dEGnTp2wbt06AM9m6DVu3BjTp09X+wrTyJEj8ejRI6X1Nrp27Yp27dph48aNEAQBNjY2+OSTTzB79mwAz16TatiwIcLCwjBq1Cil9sLCwuDn54fs7OwXxjp16lQkJiYiPj6+Us9YnTg1VCRZWVlIT08v87h3757YIRIRERERvRIKCgpw9uxZpf2udXR00LdvX5w6dUrtNadOnVKqDzzbBaC0/j///IP09HSlOiYmJujSpUuZbVZUTk4OzMzMXqqNl8WpoSIZNmwYjh0r+zePdnZ2SElJqb2AiIiIiIg0SH5+PvLz85XKDAwMYKBm5fnMzEwUFxejYcOGSuUNGzZEUlKS2vbT09PV1k9PT1ecLy0rq05VnDx5EpGRkfjxxx+r3EZ1YCIokuDgYDx48KDM8zKZrBajISIiIiLSLEFBQVi8eLFS2cKFC7Fo0SJxAqoGly5dwtChQ7Fw4UKlvb3FwERQJC4uLmKHQERERESvuNd543V/f3/MmjVLqUzdaCAAmJubQ1dXFxkZGUrlGRkZsLKyUnuNlZVVufVL/8zIyIC1tbVSnaqsln/58mX06dMHkydPxvz58yt9fXXjO4JERERERKRxDAwMUK9ePaWjrERQX18fLi4uOHLkiKKspKQER44cQbdu3dRe061bN6X6ABAXF6eo7+DgACsrK6U6ubm5OH36dJltluWvv/5C7969MX78eCxbtqxS19YUjggSEREREdErb9asWRg/fjw6duyIzp07Y82aNXj06BF8fX0BAO+++y4aNWqEoKAgAMDMmTPRq1cvBAcHY/Dgwdi5cyd+//13fPPNNwCejbb6+flh6dKlcHJyUmwfYWNjAy8vL8V9b9y4gaysLNy4cQPFxcW4cOECAKBZs2YwMjLCpUuX4ObmBnd3d8yaNUvxfqGuri4sLCxqr4P+hYkgiYb7lqh3J/up2CFopCf5xWKHoJGMpPw2rk5UUtVf4n+d2Tbg++f/Ji15Nj1KEICn3HZEoXRzMQkAAz1dUWMhqqiRI0fi3r17WLBgAdLT09GuXTscPnxYsdjLjRs3oKPz/xMiu3fvDrlcjvnz5+Pzzz+Hk5MT9u/fr9hDEADmzp2LR48eYfLkycjOzkaPHj1w+PBhxR6CALBgwQJs27ZN8bl9+/YAgKNHj8LV1RW7d+/GvXv3EB4ejvDwcEU9sReH5D6CJJo/uY+gWgZ1OGNbHSaC6kn1+QOaOkwE1fN5s5HYIWgcqzccoXfnFh6YNcT8LUfFDkdjLJ3UG/WzMlBoZYOrv18VOxyN80ajumKHoNBg/A6xQ6gx97f5iB3Ca40/cRIREREREWkZJoJERERERERahokgERERERGRlmEiSEREREREpGWYCFaTsLAwSCQSxSGVSmFjYwN3d3eEhobi4cOHSvVdXV2V6j9/1KlTp1L3njBhQpltHT58GACQkpICiUSC1atXq21j0aJFkEgkyMzMVGrXyMhI7fOVddjb21cqdiIiIiKquor8fPaqHlSzuO54NQsMDISDgwMKCwuRnp6OhIQE+Pn5ISQkBFFRUXB2dgYAzJs3D5MmTVK69tGjR/jggw/Qv3//St/XwMAAW7ZsUSl/8803q/Yg/9KzZ09s375dqWzSpEno3LkzJk+erCgrTRyJiIiIiEhzMRGsZgMHDkTHjh0Vn/39/REfHw8PDw94enoiMTERMpkM/fr1U7m2dF+RMWPGVPq+enp6GDt2bNUDf4GmTZuiadOmSmUffPABmjZtWqP3JSIiIiKi6sepobXAzc0NAQEBSE1NVdpE8t/kcjnq1q2LoUOH1mJ0RERERESkbZgI1pJx48YBAGJjY9Wev3fvHuLi4uDl5YW6dau2SWlmZqbSkZOTo1Ln8ePHKvUyMzPx+PHjKt2TiIiIiMQj9nt8fEfw1cWpobXE1tYWJiYmSE5OVns+MjISRUVFVZoWCjx7v9DCwkKprFevXkhISFAqW7hwIRYuXFilexARERER0euBiWAtMjIyUlk9tJRcLoeFhYXadwcrQiqVIjo6Wqmsfv36KvUmT56MESNGqJR///33KovBEBERERHR64mJYC3Ky8uDpaWlSvnff/+NU6dOYdq0adDTq9r/El1dXfTt2/eF9ZycnNTW++9//1ul+1ZUfn4+8vPzlcoK8guhb2BQo/clIiIiIiJVfEewlqSlpSEnJwfNmjVTOSeXywFUbbXQV0VQUBBMTEyUji1fB4sdFhEREdErTez3+PiO4KuLI4K1pHTapbu7u8o5uVwOR0dHdO3atbbDqjX+/v6YNWuWUtm1e4UiRUNEREREpN2YCNaC+Ph4LFmyBA4ODiqjfufPn0diYiICAgJEiq52GBgYwOBf00D1c/NEioaIiIiISLsxEaxmMTExSEpKQlFRETIyMhAfH4+4uDjY2dkhKioKUqlUqX5ERASA13taKBERERERaRYmgtVswYIFAAB9fX2YmZmhbdu2WLNmDXx9fWFsbKxUt6SkBDt37kSHDh3QokULMcIlIiIiIiItJBEEQRA7CNJOf6Zxaqg6BnW4hpM6T/KLxQ5BI0n1dcUOQSNFJaWLHYJG8nmzkdghaByrNxyhd+cWHpg1xPwtR8UOR2MsndQb9bMyUGhlg6u/XxU7HI3zRqO6YoegYPneD2KHUGPufustdgivNf7ESUREREREpGU4NVSDZWVloaCgoMzzurq6sLCwqMWIiIiIiIjodcBEUIMNGzYMx44dK/O8nZ0dUlJSai8gIiIiIiJ6LTAR1GDBwcF48OBBmedlMlktRkNEREREmoYbr1NVMRHUYC4uLmKHQEREREREryEuFkNERERERKRlmAgSERERERFpGU4NJSIiIiJ6RfEdQaoqJoIkGgM9DkirYyzlX0t1LIwNxA5BI2U/LhQ7BI30bofGYoegkfT5fVeFznM/QxcVl4gXiIYSwH4hel3xXwQiIiIiIiItw0SQiIiIiIhIyzARJCIiIiIi0jJ8GYmIiIiI6BXFxWKoqjgiSEREREREpGW0PhEMCwuDRCJRHFKpFDY2NnB3d0doaCgePnyoVN/V1VWp/vNHnTp1KnTPhISEMtv491Hqzz//xPDhw2FnZwepVIpGjRqhX79++Oqrr5Tatre3h4eHR7n33b17t6KsIjEsWrQIAHD//n2sWrUKPXv2hIWFBUxNTdG1a1dERkZW6LmJiIiIiEgzcGro/wQGBsLBwQGFhYVIT09HQkIC/Pz8EBISgqioKDg7OwMA5s2bh0mTJild++jRI3zwwQfo379/he7VqlUrbN++XanM398fRkZGmDdvnkr9kydPonfv3mjSpAnef/99WFlZ4ebNm/j111+xdu1aTJ8+vYpPDZU4nrdo0SIkJyejS5cuAIBTp05h3rx5GDRoEObPnw89PT3s2bMHo0aNwuXLl7F48eIqx0FERERERLWHieD/DBw4EB07dlR89vf3R3x8PDw8PODp6YnExETIZDL069dP5drw8HAAwJgxYyp0r4YNG2Ls2LFKZStWrIC5ublKOQAsW7YMJiYmOHPmDExNTZXO3b17t0L3LIu6+wHAli1bkJycjOnTp2PgwIEAgDfeeAPXrl2DnZ2dot5HH32Evn37YuXKlZg7dy7q1q37UvEQERERUcXxHUGqKq2fGloeNzc3BAQEIDU1VZHsqSOXy1G3bl0MHTq0RuJITk7GG2+8oZIEAoClpWW13++vv/7CjBkz0L59e6xatUpR7uDgoJQEAs+++Xh5eSE/Px9///13tcdCRERERETVj4ngC4wbNw4AEBsbq/b8vXv3EBcXBy8vrxobDbOzs8PZs2dx6dKlCtUvLCxEZmamypGTk/PCax8/fgxvb2/o6upi586dMDAweOE16enpAABzc/MKxUdEREREROLi1NAXsLW1hYmJCZKTk9Wej4yMRFFRUYWnhVbF7NmzMXDgQLRr1w6dO3fGf/7zH/Tp0we9e/dWu0BNbGwsLCwsqnSv6dOn4/Lly9i2bRuaN2/+wvpZWVnYsmUL/vOf/8Da2rpK9yQiIiIiotrFRLACjIyMVFYPLSWXy2FhYaH23cHq0q9fP5w6dQpBQUH46aefcOrUKXzxxRewsLDAli1b4OnpqVS/S5cuWLp0qUo7Fy9exOzZs8u8j1wux3fffYdx48bh3XfffWFcJSUlGDNmDLKzs1VWLyUiIiKiWsBXBKmKmAhWQF5entp38f7++2+cOnUK06ZNg55ezXZlp06dsHfvXhQUFODixYvYt28fvvzySwwfPhwXLlxA69atFXXNzc3Rt29flTbKi/HatWv44IMP0Lx5c6xfv75CMU2fPh2HDx/G999/jzfffLPcuvn5+cjPz1cqK8gvhn4Fpp4SEREREVH14juCL5CWloacnBw0a9ZM5ZxcLgdQ8dVCq4O+vj46deqE5cuXY8OGDSgsLMSuXbteqs38/HyMHDkSBQUF2LlzJ4yMjF54zeLFi7F+/XqsWLFC8R5leYKCgmBiYqJ0bPpq9UvFTUREREREVcMRwRco3WfP3d1d5ZxcLoejoyO6du1a22EBgGK7izt37rxUO7Nnz8b58+exdu1atG/f/oX1v/76ayxatAh+fn749NNPK3QPf39/zJo1S6nsxoPiKsVLREREREQvhyOC5YiPj8eSJUvg4OCgMup3/vx5JCYmYvTo0TUex9GjRyEIgkr5oUOHAAAtWrSoctv79u3DunXr4OnpiRkzZrywfmRkJGbMmIExY8YgJCSkwvcxMDBAvXr1lA5OCyUiIiIiEgdHBP8nJiYGSUlJKCoqQkZGBuLj4xEXFwc7OztERUVBKpUq1Y+IiABQO9NCp0+fjsePH+Ptt99Gy5YtUVBQgJMnTyIyMhL29vbw9fWtUrt37tzBe++9B11dXfTp06fMvRIdHR3RrVs3/Pbbb3j33XfRoEED9OnTR9EHpbp3746mTZtWKRYiIiIiqjxuKE9VxUTwfxYsWADg2Tt4ZmZmaNu2LdasWQNfX18YGxsr1S0pKcHOnTvRoUOHlxqNq6jVq1dj165dOHToEL755hsUFBSgSZMm+OijjzB//ny1G81XxJUrV/DgwQMAwMyZM8usN378eHTr1g2XL19GQUEB7t27h4kTJ6rU27p1KxNBIiIiIqJXgERQN+eQqBZcTX8sdggayVjG38+oI62jK3YIGin7caHYIWgkQ31+vaijr8c3Qv7NxMkOOrdv4YFZQ3y26YjY4WiMFVP6oH5WBgqsbJB4OknscDTOm02MX1ypljT6cJ/YIdSYWxveFjuE1xr/RSAiIiIiItIyHHqoAVlZWSgoKCjzvK6uLiwsLGoxIiIiIiJ6HfEdQaoqJoI1YNiwYTh27FiZ5+3s7JCSklJ7ARERERERET2HiWANCA4OVizCoo5MJqvFaIiIiIiIiJQxEawBLi4uYodARERERERUJiaCRERERESvKL4jSFXFVUOJiIiIiIi0DBNBIiIiIiIiLcOpoSSauw/zxQ5BI+nqcIqHOg+fFIkdgkYqKC4ROwSNFP/3XbFD0EjD2jQSOwSNVSIIyH1cKHYYGqNEEAAAOhIJzIz0RY6GiGoCRwSJiIiIiIi0DEcEiYiIiIheVZxIRFXEEUEiIiIiIiItw0SQiIiIiIhIyzARJCIiIiIi0jKvfSIYFhYGiUSiOKRSKWxsbODu7o7Q0FA8fPhQqb6rq6tS/eePOnXqVOieCQkJZbbx7+P5GH///Xe17bm6uqJNmzZKZfb29mW2OWDAAEW9RYsWQSKRQEdHBzdv3lRpOzc3FzKZDBKJBNOmTVOUp6SkQCKRYPXq1Ur1U1JS4OvrC0dHR0ilUlhZWaFnz55YuHBhhfqGiIiIiKpPRX/mfBUPqllas1hMYGAgHBwcUFhYiPT0dCQkJMDPzw8hISGIioqCs7MzAGDevHmYNGmS0rWPHj3CBx98gP79+1foXq1atcL27duVyvz9/WFkZIR58+ZVzwMBaNeuHT755BOVchsbG5UyAwMD7NixA3PnzlUq37t3b4Xvd/36dXTq1AkymQwTJ06Evb097ty5g3PnzmHlypVYvHhx5R+CiIiIiIhqndYkggMHDkTHjh0Vn/39/REfHw8PDw94enoiMTERMpkM/fr1U7k2PDwcADBmzJgK3athw4YYO3asUtmKFStgbm6uUv4yGjVqVOH2Bg0apDYRlMvlGDx4MPbs2fPCNr788kvk5eXhwoULsLOzUzp39y737CIiIiIielW89lNDy+Pm5oaAgACkpqYqkj115HI56tati6FDh9ZidNVr9OjRuHDhApKSkhRl6enpiI+Px+jRoyvURnJyMmxtbVWSQACwtLSstliJiIiIiKhmaXUiCADjxo0DAMTGxqo9f+/ePcTFxcHLywt169at0VhycnKQmZmpchQWFqqtX1hYqLb+kydPVOr27NkTtra2kMvlirLIyEgYGRlh8ODBFYrPzs4ON2/eRHx8fNUekIiIiIiqldjv8fEdwVeX1ieCtra2MDExQXJystrzkZGRKCoqqvC00JfRt29fWFhYqBwnT55UWz82NlZt/bVr16rUlUgkGDVqFHbs2KEoi4iIwLBhw2BgYFCh+GbMmAF9fX306dMH7du3h5+fHw4cOIDHjx9X7YGJiIiIiEgUWvOOYHmMjIxUVg8tJZfLYWFhofbdwer29ddfo3nz5irln3zyCYqLi1XKu3TpgqVLl6qUOzk5qW1/9OjRWL16Nc6cOYP69evjzJkzWL58eYXje+ONN3DhwgUsWbIEBw8exIULF7B27VoYGRkhJCQE77//foXbIiIiIiIi8TARBJCXl6f2Hbe///4bp06dwrRp06CnV/Nd1blzZ6UFbUrVr18fmZmZKuXm5ubo27dvhdtv3749WrZsCblcDlNTU1hZWcHNza1SMTZv3hzbt29HcXExLl++jIMHD+KLL77A5MmT4eDgUGY8+fn5yM/PVyorKMiHvn7FRiOJiIiIiKj6aP3U0LS0NOTk5KBZs2Yq50rfp6uNaaG1ZfTo0YiMjIRcLsfIkSOho1O1LwFdXV20bdsW/v7+2LdvH4BnU03LEhQUBBMTE6UjfOOXVbo3ERERERG9HK1PBEv3+3N3d1c5J5fL4ejoiK5du9Z2WDVm9OjRuHPnDq5evVrh1UJfpHQU886dO2XW8ff3R05OjtIx9oOPq+X+RERERNpK7AVduFjMq0urp4bGx8djyZIlcHBwUBn1O3/+PBITExEQECBSdDXD0dERa9aswZMnT9C5c+dKXXvixAl07doVderUUSo/dOgQAKBFixZlXmtgYKCyKI2+vup7j0REREREVPO0JhGMiYlBUlISioqKkJGRgfj4eMTFxcHOzg5RUVGQSqVK9UunOWrytNBbt26p3f/QyMgIXl5eZV43c+bMKt1v5cqVOHv2LIYNGwZnZ2cAwLlz5/D999/DzMwMfn5+VWqXiIiIiIhql9YkggsWLAAA6Ovrw8zMDG3btsWaNWvg6+sLY2NjpbolJSXYuXMnOnToUO4ol9guXLig2AfxeXZ2duUmglX1+eefQy6X49ixY4iIiMDjx49hbW2NUaNGISAgAA4ODtV+TyIiIiIiqn4SQRAEsYMg7fTfaw/EDkEjWZtIX1xJCxWX8FuVOgXFJWKHoJFO38oSOwSNNKxNI7FD0DgmTnbQuX0L9+tbYtqan8QOR2Os83NHgwd3UWTdCHcuqd9rWZs1NtOcVc8d/H4UO4Qa88+awWKH8FrT+sViiIiIiIiItI3WTA2tTllZWSgoKCjzvK6uLiwsLGoxIiIiIiIioopjIlgFw4YNw7Fjx8o8b2dnh5SUlNoLiIiIiIiIqBKYCFZBcHAwHjwo+/02mUxWi9EQERERkdbidntURUwEq8DFxUXsEIiIiIiIiKqMi8UQERERERFpGSaCREREREREWoZTQ4mIiLSA53gPmORyf8V/k6TfAQCYPbiLdX7uIkejOepnZ4odAhHVMCaCJJomDQzFDkEjSetwoF4dXR2+Da/O/r9uix2CRupm20DsEDSOafZ96KTz66UsEgANHtwVOwyNoyMB6snqiB0GlUMi4b+PVDVMBImIiLSIoKODJ+YNxQ5DY8ju3oEEz/pFsLIWOxyNU9LQSuwQiKiGMBEkIiLSIk/MG+JQ7Fmxw9AYg/q7wPDuHQhW1shLviF2OBpFEMSOgIhqEuegERERERERaRmOCBIRERERvaL4jiBVFUcEiYiIiIiItAwTQQBhYWGQSCSKQyqVwsbGBu7u7ggNDcXDhw/VXvfzzz/Dzc0NJiYmMDY2houLCyIjIyt17wkTJsDIyEipzNXVVSme54+kpCQAQEJCAiQSCXbv3g0AZdb/95GQkICUlBTF56VLl6qNa8yYMZBIJCqxbd68Gb169ULDhg1hYGAABwcH+Pr6IiUlpVLPTURERERE4uHU0OcEBgbCwcEBhYWFSE9PR0JCAvz8/BASEoKoqCg4Ozsr6m7duhXvvfce+vXrh+XLl0NXVxdXrlzBzZs3qyUWW1tbBAUFqZTb2Niorb99+3alz99//z3i4uJUylu1aoUnT54AAKRSKXbs2IH58+cr1Xn06BEOHDgAqVSqcp/z58/DwcEBnp6eqF+/Pv755x9s3rwZBw8exMWLF8uMj4iIiIiINAcTwecMHDgQHTt2VHz29/dHfHw8PDw84OnpicTERMhkMqSkpGDq1KmYPn061q5dWyOxmJiYYOzYsRWu/++6v/76K+Li4tS2UTp6N2jQIOzduxcXL17Em2++qTh/4MABFBQUYMCAAYiPj1e6dv369SrteXl5oWPHjvj+++/x2WefVThmIiIiIno5fEWQqopTQ1/Azc0NAQEBSE1NRXh4OABg48aNKC4uRmBgIAAgLy8Pwiu4xnK3bt3g4OAAuVyuVB4REYEBAwbAzMysQu3Y29sDALKzs6s5QiIiIiIiqglMBCtg3LhxAIDY2FgAz94NbNmyJQ4dOgRbW1sYGxujQYMGCAgIQElJSbXcs7i4GJmZmUpHXl5etbT9PB8fH+zcuVORyGZmZiI2NhajR48u97r79+/j7t27+P333+Hr6wsA6NOnT7XHR0RERERE1Y+JYAXY2trCxMQEycnJAIBr167h5s2b8PX1xcSJE7F7924MHDgQS5cuxbx586rlnklJSbCwsFA6pk2bVi1tP2/06NG4ceMGfvnlFwDADz/8AKlUCk9Pz3Kva9SoERo2bIhOnTrh5MmTCA0NRb9+/ao9PiIiIiIiqn58R7CCjIyMFKuH5uXloaSkBCtWrMCnn34KAHjnnXeQlZWFtWvX4vPPP4exsfFL3c/e3h6bN29WKquJhVjeeOMNODs7Y8eOHejRowfkcjmGDh0KQ0PDcq+LiYnB06dPkZiYiPDwcDx69KjaYyMiIiIioprBRLCC8vLyYGlpCQCQyWR49OgRfHx8lOr4+Pjg8OHDOH/+PHr27PlS96tbty769u37Um1U1OjRoxEcHIyPP/4YJ0+exOeff/7Ca3r37g3g2QI7Q4cORZs2bWBkZFTmqGV+fj7y8/P/VQYYGBi8/AMQERERaSluKE9VxamhFZCWloacnBw0a9YMwP+PzDVs2FCpXmmi+ODBg9oN8CX5+PggMzMT77//Pho0aID+/ftX6npHR0e0b98eERERZdYJCgqCiYmJ0rF+zRcvGzoREREREVUBE8EKKN2Lz93dHQDg4uICALh165ZSvdu3bwMALCwsajG6l9ekSRO89dZbSEhIwIgRI6CnV/mB4idPniAnJ6fM8/7+/sjJyVE6PvKb+zJhExERERFRFTERfIH4+HgsWbIEDg4OGDNmDABg5MiRAIBvv/1WUa+kpARbt26FmZmZIlF8lSxduhQLFy7E9OnTy6xTVFSkdrTzt99+w59//qm0B+O/GRgYoF69ekoHp4USEREREYmD7wg+JyYmBklJSSgqKkJGRgbi4+MRFxcHOzs7REVFQSqVAgCGDh2KPn36ICgoCJmZmXjzzTexf/9+/Pe//8WmTZteyQSnV69e6NWrV7l18vLy0LhxY4wcORJvvPEG6tatiz///BNbt26FiYkJAgICailaIiIiIgK4oTxVHRPB5yxYsAAAoK+vDzMzM7Rt2xZr1qyBr6+v0iqgEokE+/fvx/z58xEZGYmwsDC0aNEC4eHhilHD15GhoSEmTZqEo0ePYvfu3Xjy5AlsbGzg4+OD+fPnKzaWJyIiIiIizSYRSncSJ6plN7LyX1xJC0nrcMa2Oro6/JWnOvv/ui12CBqpm20DsUPQOE4uzVEn/TYeW1rjUOxZscPRGIP6u8Dw7h2U2DRCXvINscPRKPwJsWwmMs35t7r53MNih1Bjrn4xQOwQXmua81VMREREREREtYJTQ2tIVlYWCgoKyjyvq6v7yq0uSkRERESahfsIUlUxEawhw4YNw7Fjx8o8b2dnh5SUlNoLiIiIiIiI6H+YCNaQ4ODgcjeWl8lktRgNERERERHR/2MiWENexb0EiYiIiIhIO3CxGCIiIiIiIi3DEUEiIiIiolcU14qhquKIIBERERERkZbhiCCJxkRWR+wQNJIBN5RXq7C4ROwQNJKlzEDsEDSSrg5/Ra6CXVIuAcCj/CKxw9Ao+UX8vlsWE5lU7BCIXhp/4iQiIiIiItIyHBEkIiIiInpF6XAGBFURRwSJiIiIiIi0DBNBIiIiIiIiLfNaJIJhYWGQSCSKQyqVwsbGBu7u7ggNDcXDhw/VXvfzzz/Dzc0NJiYmMDY2houLCyIjI6sUQ0JCAoYNGwYrKyvo6+vD0tISQ4YMwd69exV1UlJSlOKsU6cOzM3N0b17d3z++ee4ceOG2nYlEgl2796tVF5QUAAPDw/o6Ojgu+++U+oHqVSKW7duqbTl6uqKNm3aKJXFxsbivffeQ5s2baCrqwt7e/tynzMjIwOzZ89Gy5YtYWhoiLp168LFxQVLly5FdnZ2BXuLiIiIiIjE9Fq9IxgYGAgHBwcUFhYiPT0dCQkJ8PPzQ0hICKKiouDs7Kyou3XrVrz33nvo168fli9fDl1dXVy5cgU3b96s9H0XLlyIwMBAODk5YcqUKbCzs8P9+/dx6NAhvPPOO4iIiMDo0aMV9X18fDBo0CCUlJTgwYMHOHPmDNasWYO1a9fi22+/xahRo8q9X2FhIYYPH45Dhw5h8+bNmDhxotL5/Px8rFixAl999dULY5fL5YiMjESHDh1gY2NTbt0zZ85g0KBByMvLw9ixY+Hi4gIA+P3337FixQocP34csbGxL7wnEREREVUP7iNIVfVaJYIDBw5Ex44dFZ/9/f0RHx8PDw8PeHp6IjExETKZDCkpKZg6dSqmT5+OtWvXvtQ9d+/ejcDAQAwfPhxyuRx16vz/lghz5szBTz/9hMLCQqVrOnTogLFjxyqVpaamon///hg/fjxatWqFN998U+39CgsL4e3tjYMHD2LTpk147733VOq0a9cOmzdvhr+//wuTu+XLl2Pz5s2oU6cOPDw8cOnSJbX1srOz8fbbb0NXVxfnz59Hy5Ytlc4vW7YMmzdvLvdeRERERESkGV6LqaHlcXNzQ0BAAFJTUxEeHg4A2LhxI4qLixEYGAgAyMvLgyAIVWo/ICAAZmZm+O6775SSwFLu7u7w8PB4YTt2dnYICwtDQUEBvvjiC7V1ioqKMGrUKBw4cAAbNmzA+++/r7be559/juLiYqxYseKF97WxsVEb979t2rQJt27dQkhIiEoSCAANGzbE/PnzX9gOERERERGJ77VPBAFg3LhxAKCYtvjzzz+jZcuWOHToEGxtbWFsbIwGDRogICAAJSUV3zz12rVrSEpKgpeXF4yNjV86zm7dusHR0RFxcXEq54qKiuDj44N9+/bh66+/xpQpU8psx8HBAe+++y42b96M27dvv3RcABAVFQWZTIbhw4dXS3tERERERCQerUgEbW1tYWJiguTkZADPEribN2/C19cXEydOxO7d/8fencfVnP79A3+dok6pDqEUqRDGVHZG1rJkSRJK9sY2ZixhjEmyFJMxyjLMGGumTsvYM7biiLF8jd0YGllCqCTapP33h59zO84pp1Oc6PW8H597nOu6Ptf1PkffmfPuuj7XtQP9+vXDkiVL4OPjo3S/N27cAADY2NhUWKzW1tZ48uQJMjIyZMq///577NixA2vXrsWUKVPe2Y+Pjw8KCgrw448/VkhcN27cQNOmTaGlpVUh/RERERERkfpUiUQQAPT09KS7h2ZlZeHZs2dYvHgx/Pz8pBu69O3bF6tXry5xl9G3vU7WKmI28M04AcjFkJycjGrVqsHS0lKpfho1aoTRo0djw4YNePz4cbnjysjIqND3SURERETl9+aO9J/aRe9XlUkEs7KypImMjo4OgFe7d77Jw8MDOTk5uHTpklJ9GhgYAJBP2sobJyCfXC5fvhwNGzbE0KFDcerUKaX6mj9/PgoKCpR6VvBdDAwMyvU+c3NzkZGRIXPl5uaWOy4iIiIiIiq7KpEIJiYmIj09HU2aNAEA6U6axsbGMu2MjIwAAM+ePVOq39ebpvzzzz8VFSquXbsGIyMjaZL5momJCWJiYiASiTBgwABcuXLlnX01atQIo0aNqpBZwebNm+PmzZvIy8tT6f6AgACIRCKZK+in8ieoRERERERUdlUiEQwJCQHwagdPANLz794+dP31xip169ZVqt+mTZuiWbNm2Lt3r3QmrzzOnDmD27dvo0+fPgrrGzVqhMOHD0NDQwOOjo6Ij49/Z5+vZwXL+6zgwIEDkZOTg507d6p0v7e3N9LT02WuWXO+L1dMRERERESkmk8+EZRIJPD394elpSVGjhwJAHB3dwcAbN68WdquqKgIW7duhaGhoTRRVMbixYvx9OlTTJgwAQUFBXL10dHR+PPPP9/Zz7179zBu3DhoaWlhzpw5JbazsbHB/v37kZWVhd69e8sls29r3LgxRo0ahd9++w1JSUnvfkMl+Oqrr2BiYoLZs2fj5s2bcvUpKSlYsmRJifdra2vDwMBA5tLW1lY5HiIiIiJ6daD8p3rR+/VJHSh/8OBBxMXFoaCgAMnJyZBIJIiJiYG5uTmioqIgFAoBAIMGDULPnj0REBCA1NRUtGzZEnv27MHJkyfx22+/lSlBcXd3xz///IOlS5fi0qVL8PDwgLm5OZ4+fYpDhw7h6NGjCAsLk7nn4sWLCA0NRVFREZ4/f45z585h586dEAgECAkJga2tbaljdurUCbt27cLAgQPRu3dv/PXXX6hdu3aJ7X18fBASEoL//vsPn3/+uUzd1atXERUVBQC4desW0tPTpQldy5YtMXDgQABArVq1sHv3bvTv3x+tWrXCqFGjpAnzxYsXER4ejk6dOin9uRERERERkfp8UongggULAABaWlowNDSEjY0NVq1aBU9PT5nNVwQCAfbs2YP58+cjMjISwcHBaNasGUJDQ6WzhmWxZMkSODg4YM2aNfj111+RlpaGWrVq4YsvvsDevXvh7Ows0z48PBzh4eGoVq0aDAwMYGVlBS8vL3z11Vdo2LChUmP26dMHISEh8PDwQL9+/XD06NES2zZp0gSjRo3Ctm3b5OouXrwIX19fmbLXr8eOHStNBAGgY8eOuHbtGn766Sfs378fISEh0NDQwGeffYbvv/8eU6dOVSp2IiIiIiJSL0FxcXGxuoOgqik9p0jdIVRK2tU/+RXbKskv5M+LIrE3n6g7hEqpaV0ed/O2Rm2sUP3xI7wwMsGB6AvqDqfS6N+nLXRTHqPQtD5Srt9RdziVSm4B/71bEovaQnWHIGXjG6PuEN6bf/x7qzuETxq/cRIREREREVUxn9TS0IqUlpZW6lEJmpqaSu8uSkRERET0PvDgdVIVE8ESuLq64vjx4yXWm5ubIyEh4cMFREREREREVEGYCJYgMDCw1IPldXR0PmA0REREREREFYeJYAnKcpYgERERERHRx4SJIBERERHRR4rPCJKquGsoERERERFRFcNEkIiIiIiIqIrh0lBSm8fPX6o7hEpJV1tT3SHQR+Ru+gt1h1ApdTA3VHcIlY6GdPlYMfIKC9UaS+VS/Or/FwOFPD9dRlpmycdoVXWV6UB5IlUxESQiIiIi+kjxEUFSFZeGEhERERERVTFMBImIiIiIiKoYJoJERERERERVDBNBIiIiIiKiKqZKJ4LBwcEQCATSSygUwtTUFI6OjlizZg0yMzMV3nfkyBE4ODhAJBJBX18fbdu2RWRkZJnGHjduHAQCAQwMDJCTkyNXHx8fL41rxYoVAAALCwuZeEu6goODAQBZWVlYuHAhrK2tUaNGDdSuXRutWrXCjBkz8OjRI5nxTp48iX79+qF+/foQCoVo2LAhBg4ciLCwMJl2kZGRGDVqFKysrCAQCNCjR48yvW8iIiIiqjjKfDf8WC96v7hrKAA/Pz9YWloiPz8fSUlJiI2NhZeXF4KCghAVFQVbW1tp261bt2L8+PHo3bs3fvjhB2hqauK///7DgwcPyjxutWrV8OLFC+zbtw9ubm4ydWKxGEKhEC9f/t8RC6tWrUJWVpb09YEDBxAeHo6VK1eiTp060nI7Ozvk5+ejW7duiIuLw9ixYzFt2jRkZWXh33//RVhYGAYPHgxTU1MAwPbt2+Hu7i5NEmvVqoW7d+/ixIkT2LhxI0aMGCHt+9dff8WFCxfQvn17PH36tMzvmYiIiIiI1I+JIIB+/fqhXbt20tfe3t6QSCRwcnKCs7Mzbty4AR0dHSQkJOCbb77BtGnTsHr16nKPq62tjc6dOyM8PFwuEQwLC8OAAQOwc+dOaZmLi4tMm6SkJISHh8PFxQUWFhYyddu3b8elS5cgFotlEjkAePnyJfLy/u9soEWLFqFFixb43//+By0tLZm2KSkpMq9DQkJQv359aGhowNrauqxvmYiIiIiIKoEqvTS0NA4ODvD19cW9e/cQGhoKAFi/fj0KCwvh5+cH4NXSy+Li4nKNM2LECBw8eBDPnz+Xlp07dw7x8fFyCVxZ3L59GwDQuXNnuTqhUAgDAwOZtu3bt5dLAgHAyMhI5rWZmRk0NPhjQ0RERET0MeM3+lKMHj0aABAdHQ3g1bOBzZs3x4EDB9CgQQPo6+ujdu3a8PX1RVFRkUpjuLq6QiAQYNeuXdKysLAwNG/eHG3atFE5dnNzcwDA77///s5k1dzcHEePHkViYqLK4xERERHRhycQfLqXKtatWwcLCwsIhUJ07NgRf//9d6ntt2/fjubNm0MoFMLGxgYHDhyQqS8uLsaCBQtgYmICHR0d9OrVC/Hx8TJtli5dCjs7O+jq6qJmzZoKx7l//z4GDBgAXV1dGBkZYc6cOSgoKFDtTVYQJoKlaNCgAUQikXR2LT4+Hg8ePICnpye+/PJL7NixA/369cOSJUvg4+Oj0hj6+vpwcnKSbspSVFSEiIgIeHh4lCt2FxcXNGvWDAsWLIClpSU8PT2xZcsWuaWeADB37lw8ePAAjRs3hoODAxYsWICTJ0+qnNwSEREREX1okZGRmDVrFhYuXIiLFy+iZcuWcHR0VPj9FwBOnz4NDw8PjB8/HpcuXYKLiwtcXFxw7do1aZvly5djzZo1WL9+Pc6ePYsaNWrA0dFRZh+PvLw8DBs2DFOmTFE4TmFhIQYMGIC8vDycPn0a27ZtQ3BwMBYsWFCxH0AZMRF8Bz09PenuoVlZWXj27BkWL14MPz8/DBkyBGKxGH379sXq1atL3GX0XUaMGIHY2FgkJSVBIpEgKSmpXMtCAUBHRwdnz57FnDlzALzaIXX8+PEwMTHBtGnTkJubK2375Zdf4tChQ+jRowdOnjwJf39/dO3aFVZWVjh9+nS54iAiIiIi+hCCgoIwceJEeHp6okWLFli/fj10dXWxZcsWhe1Xr16Nvn37Ys6cOfjss8/g7++PNm3aYO3atQBezQauWrUK8+fPx6BBg2Bra4vff/8djx49wp49e6T9LF68GDNnzoSNjY3CcaKjo3H9+nWEhoaiVatW6NevH/z9/bFu3TqZfTs+NCaC75CVlQV9fX0Ar5IrAHKzdR4eHsjJycGlS5dUGqN///7Q19dHZGQkxGIx2rdvjyZNmpQvcAAikQjLly9HQkICEhISsHnzZjRr1gxr166Fv7+/TFtHR0ccPnwYz58/x4kTJ/DNN9/g3r17cHJyKvG3KGWRm5uLjIwMmSvvjWSUiIiIiOhNir4/5pbw/TEvLw8XLlxAr169pGUaGhro1asXzpw5o/CeM2fOyLQHXn0nft3+7t27SEpKkmkjEonQsWPHEvssaRwbGxsYGxvLjJORkYF///1X6X4qGhPBUiQmJiI9PV2alL0+buHNv0Tg/zZUefbsmUrjaGtrw9XVFdu2bcPu3bvLPRuoiLm5Ob788kucOnUKNWvWhFgsVthOV1cXXbt2xdq1azF//nw8e/YMBw8eLPf4AQEBEIlEMteGn1eUu18iIiKiqkzdZ/29z0vR98eAgACFn0NqaioKCwvlvqcbGxsjKSlJ4T1JSUmltn/9z7L0WZZx3hxDHZgIliIkJATAq4wdANq2bQsAePjwoUy714ez161bV+WxRowYgUuXLiEzMxPDhw9XuZ93qVWrFho3bozHjx+/s+3rIzWUafsu3t7eSE9Pl7kmTfu23P0SERER0adJ0fdHb29vdYf1yWAiWAKJRAJ/f39YWlpi5MiRAAB3d3cAwObNm6XtioqKsHXrVhgaGkoTRVXY29vD398fa9euRb169coXPIArV64gNTVVrvzevXu4fv06mjVrJi07evSowj5e75r0ZltVaWtrw8DAQObS0tYud79ERERE9GlS9P1Ru4Tvj3Xq1IGmpiaSk5NlypOTk0v8bl2vXr1S27/+Z1n6LMs4b46hDjxQHsDBgwcRFxeHgoICJCcnQyKRICYmBubm5oiKioJQKAQADBo0CD179kRAQABSU1PRsmVL7NmzBydPnsRvv/1W4g+mMjQ0NDB//vyKekuIiYnBwoUL4ezsjC+++AJ6enq4c+cOtmzZgtzcXCxatEjadtCgQbC0tMTAgQPRuHFjZGdn48iRI9i3bx/at2+PgQMHStueOHECJ06cAAA8efIE2dnZWLJkCQCgW7du6NatW4W9ByIiIiIiZWhpaaFt27Y4evQoXFxcALyasDl69CimTp2q8J5OnTrh6NGj8PLykpbFxMSgU6dOAABLS0vUq1cPR48eRatWrQAAGRkZOHv2bIk7hJY0ztKlS5GSkiJ9pCwmJgYGBgZo0aJF2d9sBWEiCEi3btXS0oKhoSFsbGywatUqeHp6SjeKAV6twd6zZw/mz5+PyMhIBAcHo1mzZggNDZXOGlYWQ4YMQWZmJqKjoyGRSJCWloZatWqhQ4cOmD17Nuzt7aVtN23ahL179+KPP/7Ao0ePUFxcjEaNGsHHxwdz585FtWr/92MikUiwePFimbF8fX0BAAsXLmQiSERERERqMWvWLIwdOxbt2rVDhw4dsGrVKmRnZ8PT0xMAMGbMGNSvX1/6nOGMGTPQvXt3BAYGYsCAAYiIiMD58+exYcMGAK+++3t5eWHJkiWwsrKCpaUlfH19YWpqKk02gVdnBKalpeH+/fsoLCzE5cuXAQBNmjSBnp4e+vTpgxYtWmD06NFYvnw5kpKSMH/+fHzzzTflmkgqL0Hxu04bJ3pP4h6/UHcIlZKutqa6Q6CPyJ7rj9QdQqXkbttA3SFUOnU+s4Tmo4d4YVQPew6eU3c4lYZLv/bQTUlCgUl9JP17R93hVCop6S/f3aiKamNhoO4QpNotOabuEN6b8/Pt393oLWvXrsVPP/2EpKQktGrVCmvWrEHHjh0BAD169ICFhQWCg4Ol7bdv34758+cjISEBVlZWWL58Ofr37y+tLy4uxsKFC7FhwwY8f/4cXbp0wS+//IKmTZtK24wbNw7btm2Ti+XYsWPo0aMHgFePZ02ZMgWxsbGoUaMGxo4di2XLlslMuHxoTARJbZgIKsZEkMqCiaBiTATlMRFUjIlgyZgIloyJ4IehSiJIyuPS0AqWlpZW6sGQmpqa5dpdlIiIiIiIqLyYCFYwV1dXHD9+vMR6c3NzJCQkfLiAiIiIiIiI3sJEsIIFBgaWerC8jo7OB4yGiIiIiD5lAoFA3SHQR4qJYAUrz1mCREREREREHwIPlCciIiIiIqpimAgSERERERFVMVwaSkRERET0keIjgqQqJoKkNjV1q6s7hEpJqMWJekWeZ+erO4RKqZ6etrpDqJSuPUpXdwiVTufCImgCKAaQV1ik7nAqjdeHKQsEQDVNfqN+k6G+lrpDIKL3iN84iYiIiIiIqhgmgkRERERERFUME0EiIiIiIqIqhs8IEhERERF9pHigPKmKM4JERERERERVzCeXCAYHB0MgEEgvoVAIU1NTODo6Ys2aNcjMzFR435EjR+Dg4ACRSAR9fX20bdsWkZGRZRp73LhxMmO/eR06dEimbUZGBhYvXoyWLVtCT08POjo6sLa2xty5c/Ho0SO5vv/66y+4ubmhfv360NLSgkgkQseOHeHn54fk5GSZtj169JAZW0dHB7a2tli1ahWKimR3isvJycH48eNhbW0NkUgEPT09tGzZEqtXr0Z+fsm7NH733XcQCARwd3cv02dERERERETq98kuDfXz84OlpSXy8/ORlJSE2NhYeHl5ISgoCFFRUbC1tZW23bp1K8aPH4/evXvjhx9+gKamJv777z88ePCgzONqa2tj06ZNcuUtW7aU/vnOnTvo1asX7t+/j2HDhmHSpEnQ0tLC1atXsXnzZuzevRs3b96Utl+wYAH8/f3RqFEjjBs3Do0aNcLLly9x4cIFBAYGYtu2bbh9+7bMeA0aNEBAQAAAIDU1FWFhYZg5cyaePHmCpUuXStvl5OTg33//Rf/+/WFhYQENDQ2cPn0aM2fOxNmzZxEWFib3XoqLixEeHg4LCwvs27cPmZmZ0NfXL/NnRURERERE6vHJJoL9+vVDu3btpK+9vb0hkUjg5OQEZ2dn3LhxAzo6OkhISMA333yDadOmYfXq1eUet1q1ahg1alSJ9QUFBXB1dUVycjJiY2PRpUsXmfqlS5fixx9/lL6OjIyEv78/3NzcEBISAi0t2TN9Vq5ciZUrV8qNIxKJZOL46quv0Lx5c/z888/w8/ODpqYmAMDQ0BD/+9//ZO796quvIBKJsHbtWgQFBaFevXoy9bGxsUhMTIREIoGjoyN27dqFsWPHvuOTISIiIqKKxkcESVWf3NLQ0jg4OMDX1xf37t1DaGgoAGD9+vUoLCyEn58fACArKwvFxcWldVMuO3fuxJUrV+Dj4yOXBAKAgYGBzIzdggULUKdOHWzevFkuCQReJXyLFi1657hCoRDt27dHZmYmUlJS3tnewsICAPD8+XO5OrFYjBYtWsDe3h69evWCWCx+Z39ERERERFR5VKlEEABGjx4NAIiOjgbw6tnA5s2b48CBA2jQoAH09fVRu3Zt+Pr6yj1Pp6zU1FSZKz09XVoXFRUlE0dpbt68iZs3b8LFxQV6enoqxfKmhIQECAQC1KxZU64uLy8PqampePDgAXbv3o0VK1bA3NwcTZo0kWmXm5uLnTt3wsPDAwDg4eEBiUSCpKSkcsdHREREREQfRpVLBBs0aACRSCR9pi4+Ph4PHjyAp6cnvvzyS+zYsQP9+vXDkiVL4OPjU+b+s7OzUbduXZlr0KBB0vobN25AJBLBzMzsnX3FxcUBAKytrWXKi4uL5ZLNgoICmTaFhYXSuv/++w/fffcdzp8/j/79+0NHR0durF27dqFu3bpo2LAhXF1d0aBBA+zbtw/VqsmuHv7zzz/x/PlzDB8+HADg4uKC6tWrIyIiQrkPiIiIiIiI1O6TfUawNHp6etLdQ7OyslBUVIRly5Zh7ty5AIAhQ4YgLS0Nq1evxrx588q0EYpQKMS+fftkymrVqiX9c0ZGhtL9ZWRkSON9U3p6OurWrStTdu7cOZlnIuPi4uTaODs7Y/PmzQrHsre3R0xMDJ4/f46jR4/iypUryM7OlmsnFovRrl076Uyhvr4+BgwYALFYDC8vL6XeFxERERFVDJ4jSKqqkolgVlYWjIyMAAA6OjrIzs6WLnV8zcPDA4cOHcKlS5fQrVs3pfvW1NREr169Sqw3MDDAnTt3lOrrdcKYlZUlU66np4eYmBgAr5a4/vTTT3L3WlhYYOPGjSgqKsLt27exdOlSPHnyBEKhUOFYxsbGMDY2BgAMHToUP/zwA3r37o34+HjpZjHPnz/HgQMHMHXqVNy6dUt6b+fOnbFz507cvHkTTZs2Vdh/bm4ucnNz3yrTgLa2tjIfBRERERERVaAqtzQ0MTER6enp0hktU1NTAJAmQa+9ThSfPXtWoeM3b94c6enpSh1N0bx5cwDAtWvXZMqrVauGXr16oVevXmjRooXCe2vUqIFevXqhT58+mDJlCg4cOIC///4b8+bNUyrOoUOHIisrC3v37pWWbd++Hbm5uQgMDISVlZX0mjVrFgCUumlMQEAARCKRzPVz0I8lticiIiIiovenyiWCISEhAABHR0cAQNu2bQEADx8+lGn3+lD3t5dXltfAgQMBQLpraWmaNWsGKysr7NmzR+EyzbKwtbXFqFGj8Ntvv+H+/fvvbJ+TkwMAMhvdiMViWFtbY/v27XJXr169FJ45+Jq3tzfS09Nlrmmz5pbrPRERERERkWqqVCIokUjg7+8PS0tLjBw5EgDg7u4OADLPzhUVFWHr1q0wNDSUJooVZejQobCxscHSpUtx5swZufrMzEyZTWoWLVqE1NRUTJw4Efn5+XLty3LUxXfffYf8/HwEBQVJy1JTUxX2sWnTJgCQPnf44MEDnDhxAm5ubhg6dKjc5enpiVu3buHs2bMKx9bW1oaBgYHMxWWhRERERETq8ck+I3jw4EHExcWhoKAAycnJkEgkiImJgbm5OaKioqTPyg0aNAg9e/ZEQEAAUlNT0bJlS+zZswcnT57Eb7/9VuHJSvXq1bFr1y706tUL3bp1g5ubGzp37ozq1avj33//RVhYGGrVqiU9S3DEiBG4du0aAgIC8Pfff2P48OGwtLREdnY2rl27hvDwcOjr68tsSFOSFi1aoH///ti0aRN8fX1Ru3ZthIaGYv369XBxcUGjRo2QmZmJw4cPIyYmBgMHDoSDgwMAICwsDMXFxXB2dlbYd//+/VGtWjWIxWJ07Nix4j4wIiIiIioR94ohVX2yieCCBQsAAFpaWjA0NISNjQ1WrVoFT09PmV07BQIB9uzZg/nz5yMyMhLBwcFo1qwZQkNDpbOGFa1Jkya4fPkyVq5cid27d2PPnj0oKipCkyZNMGHCBEyfPl2m/Q8//ABHR0esXbsWW7ZsQWpqKnR0dNC0aVPMnj0bX331lXRDl3eZM2cO9u/fj59//hmLFi1Cly5dcPr0aYSHhyM5ORnVqlVDs2bNEBQUhGnTpknvE4vFaNiwIVq2bKmw35o1a6JLly6IjIxEUFCQ3LETRERERERUeQiKy7K2kKgCJaXLL3UlQKhVpVZsK+15Nn9eFPk7MU3dIVRKtYVa6g6h0uncwxbC5MfINqqH7X8qXsZfFQ1z6ogaKUkoNK2PJzfuqjucSuVlfqG6Q6i0LGor3oVdHeyWn1B3CO/N6e+U37mfyo7fOImIiIiIiKoYrt9TQlpaGvLy8kqs19TUrPDdRYmIiIiI3oUHypOqmAgqwdXVFcePHy+x3tzcHAkJCR8uICIiIiIionJgIqiEwMDAUg+W19HR+YDREBERERERlQ8TQSVU9FmCRERERERE6sREkIiIiIjoI8VHBElV3DWUiIiIiIioimEiSEREREREVMVwaSipzdOsko/kqMpqaGuqO4RKSV+H/7oi5VmbitQdQqVTXfP//+63GCgsVm8slcr//ywEAkBfyH/PvOluSra6Q6i0KtOB8kSq4owgERERERFRFcNffRERERERfaR4oDypijOCREREREREVQwTQSIiIiIioiqGiSAREREREVEVU6USweDgYAgEAuklFAphamoKR0dHrFmzBpmZmQrvO3LkCBwcHCASiaCvr4+2bdsiMjJSqTFjY2Nlxizteu327duYPHkyGjVqBKFQCAMDA3Tu3BmrV69GTk6OtJ2FhUWJfb18+RIAMG7cuFLHfPjwoUx/Tk5Opb6fXbt2wd3dHY0aNYKuri6aNWuG2bNn4/nz50p9HkRERERUcZT9nvkxXvR+VcnNYvz8/GBpaYn8/HwkJSUhNjYWXl5eCAoKQlRUFGxtbaVtt27divHjx6N379744YcfoKmpif/++w8PHjxQaqzPPvsMISEhMmXe3t7Q09ODj4+PXPv9+/dj2LBh0NbWxpgxY2BtbY28vDycPHkSc+bMwb///osNGzZI27dq1QqzZ8+W60dLSwsAMHnyZPTq1Uumrri4GF999RUsLCxQv359pd7Ha5MmTYKpqSlGjRqFhg0b4p9//sHatWtx4MABXLx4ETo6OmXqj4iIiIiIPrwqmQj269cP7dq1k7729vaGRCKBk5MTnJ2dcePGDejo6CAhIQHffPMNpk2bhtWrV6s0lrGxMUaNGiVTtmzZMtSpU0eu/O7duxg+fDjMzc0hkUhgYmIirfvmm29w69Yt7N+/X+ae+vXry/Xzpk6dOqFTp04yZSdPnsSLFy8wcuTIMr+fHTt2oEePHjJlbdu2xdixYyEWizFhwoQy90lERERERB9WlVoaWhoHBwf4+vri3r17CA0NBQCsX78ehYWF8PPzAwBkZWWhuPj9ncK7fPlyZGVlYfPmzTJJ4GtNmjTBjBkzyj1OWFgYBAIBRowYUeZ7304CAWDw4MEAgBs3bpQ3NCIiIiIi+gCYCL5h9OjRAIDo6GgAr54NbN68OQ4cOIAGDRpAX18ftWvXhq+vL4qKiip8/H379qFRo0aws7NT+p78/HykpqbKXC9evCi1/R9//AE7OztYWFhUQNRAUlISAKBOnToV0h8RERERKUcg+HQver+YCL6hQYMGEIlEuH37NgAgPj4eDx48gKenJ7788kvs2LED/fr1w5IlSxQ+31ceGRkZePjwIWxsbMp0X3R0NOrWrStzLV++vMT2hw8fxtOnT1VaFlqSH3/8EZqamhg6dGiF9UlERERERO9PlXxGsDR6enrS3UOzsrJQVFSEZcuWYe7cuQCAIUOGIC0tDatXr8a8efOgr69fIeNmZGQAQJn769ixI5YsWSJT1qhRoxLbh4WFoXr16nBzcyt7kCX0t3nzZnz33XewsrKqkD6JiIiIiOj9YiL4lqysLBgZGQEAdHR0kJ2dDQ8PD5k2Hh4eOHToEC5duoRu3bpVyLgGBgYAUOIRFiWpU6eO3K6gJcnKysLevXvh6OiI2rVrlznGt/31118YP348HB0dsXTp0lLb5ubmIjc3V6YsL7cAWtra5Y6DiIiIiIjKhktD35CYmIj09HQ0adIEAGBqagrg1c6fb3qdKD579qzCxjYwMICpqSmuXbtWYX2+bc+ePSrvFvq2K1euwNnZGdbW1tixYweqVSv9dwoBAQEQiUQy18a1K8odBxERERERlR0TwTe8Pu/P0dERwKtjEQDIHLoOAI8ePQIA1K1bt0LHd3Jywu3bt3HmzJkK7fc1sVgMPT09ODs7l6uf27dvo2/fvjAyMsKBAwegp6f3znu8vb2Rnp4uc02c+m254iAiIiKq6tR96DsPlP94MRH8/yQSCfz9/WFpaSmdMXN3dwcAbN68WdquqKgIW7duhaGhoTRRrCjfffcdatSogQkTJiA5OVmu/vbt2yqfZ/jkyRMcOXIEgwcPhq6ursoxJiUloU+fPtDQ0MDhw4eVToa1tbVhYGAgc3FZKBERERGRelTJZwQPHjyIuLg4FBQUIDk5GRKJBDExMTA3N0dUVBSEQiEAYNCgQejZsycCAgKQmpqKli1bYs+ePTh58iR+++03aFdwItO4cWOEhYXB3d0dn332GcaMGQNra2vk5eXh9OnT2L59O8aNG6dS35GRkSgoKHjnstBbt27JbT4DAK1bt8aAAQPQt29f3LlzB9999x1OnjyJkydPStsYGxujd+/eKsVHREREREQfTpVMBBcsWAAA0NLSgqGhIWxsbLBq1Sp4enrK7NopEAiwZ88ezJ8/H5GRkQgODkazZs0QGhpaoccvvMnZ2RlXr17FTz/9hL179+LXX3+FtrY2bG1tERgYiIkTJ6rUr1gshpGR0Ts3lvnvv//g6+srVz5+/HgMGDAAV65cAQCFR1R0796diSARERER0UdAUFxcXKzuIKhq+vdhtrpDqJRqaGuqO4RKSV+nSv7e6p2OxqeoO4RKqXujin2G+1NQ5zNLaD56iOy69RDx51l1h1NpDHfqiBpPklBUvz5y7jxQdziVytX76eoOodLq1KSmukOQsl99Wt0hvDfHZtipO4RPGp8RJCIiIiIiqmL4K/ZySktLQ15eXon1mpqaFb67KBERERERUXkwESwnV1dXHD9+vMR6c3NzJCQkfLiAiIiIiIiI3oGJYDkFBgaWerC8jo7OB4yGiIiIiKoSnrdHqmIiWE4VfZYgERERERHR+8bNYoiIiIiIiKoYJoJERERERERVDJeGEhHRJ6XXiH4QpT9VdxiVjkbSYwCA7pMkDHfqqOZoKg+dpzyLk4iqJiaCpDZJmTnqDqFS0srmgfKK2JqJ1B1CpWRjxM/lbfrPUqGZ9EjdYVRaAgA1niSpO4xKp7gYSH+Rr+4wKhXzOrrqDoGUwL1iSFVMBImI6JNUKNBASo1a6g6j0qiX9RQCAMUaGiiqZ6LucCqdQiNjdYdARPRBMREkIqJPUkqNWrAbv03dYVQapzePhUnWUxTVM0HK9TvqDqdSKS5WdwRERB8eN4shIiIiIiKqYjgjSERERET0kdLgQ4KkIs4IEhERERERVTFVLhEMDg6GQCCQXkKhEKampnB0dMSaNWuQmZlZavs3r6Sksu26Nm7cOOjp6ZUpvmrVqqF+/foYN24cHj58KNe+R48eEAgEsLKyUthfTEyMtK8dO3ao/Dm8rUOHDhAIBPj111+VeOdERERERFSZVNmloX5+frC0tER+fj6SkpIQGxsLLy8vBAUFISoqCra2tgrbv6lmzZrvPb6XL1/if//7H4KDg3Hy5Elcu3YNQqFQpq1QKMStW7fw999/o0OHDjJ1YrEYQqEQL1++LHUcZT8HAIiPj8e5c+dgYWEBsViMKVOmVNwbJyIiIiKi967KJoL9+vVDu3btpK+9vb0hkUjg5OQEZ2dn3LhxAzo6OiW2/5DxTZgwAXXq1MGPP/6IqKgouLm5ybRt3LgxCgoKEB4eLpMIvnz5Ert378aAAQOwc+fOd44DvPtzAIDQ0FAYGRkhMDAQQ4cORUJCAiwsLCronRMRERGRsviIIKmqyi0NLY2DgwN8fX1x7949hIaGytVnZmaisLBQDZEBXbt2BQDcvn1bYb2HhwciIyNRVFQkLdu3bx9evHghlzi+y7s+h7CwMAwdOhROTk4QiUQICwsrU/9ERERERKReTATfMnr0aABAdHS0TLm9vT0MDAygq6sLZ2dnxMfHf9C4EhISAAC1aik+HHnEiBF4/PgxYmNjpWVhYWHo2bMnjIyMyjxeSZ/D2bNncevWLXh4eEBLSwuurq4Qi8Vl7p+IiIiIiNSHieBbGjRoAJFIJJ1509XVxbhx47Bu3Trs3r0b3333HY4ePQo7Ozs8ePDgvcWRnp6O1NRUJCYmYufOnVi8eDG0tbXh5OSksL2VlRXatWsnnZ17/vw5Dhw4gBEjRqg0/tufw2uhoaEwMzND586dAQDDhw/H9evXcfnyZZXGISIiIiKiD4+JoAJ6enrSXTPd3NywdetWjBkzBi4uLvD398fhw4fx9OlTLF269L3F0KtXL9StWxdmZmYYOnQoatSogaioKDRo0KDEe0aMGIFdu3YhLy8PO3bsgKamJgYPHqxyDG9+DgBQUFCAyMhIuLu7Q/D/F6Q7ODjAyMiIs4JERERERB8RJoIKZGVlQV9fv8T6Ll26oGPHjjhy5Mh7i2HdunWIiYnBjh070L9/f6SmpkJbW7vUe4YPH4709HQcPHgQYrEYTk5Opb6Pd3n7c4iOjsaTJ0/QoUMH3Lp1C7du3cLdu3dhb2+P8PBwmecT35abm4uMjAyZKy8vV+XYiIiIiAglHnP2KVz0flXZXUNLkpiYiPT0dDRp0qTUdmZmZvjvv//eWxwdOnSQ7ubp4uKCLl26YMSIEfjvv/9KPIvQxMQEPXr0QGBgIE6dOlXiTqHKUPQ5vJ71K2nzmePHj8Pe3l5hXUBAABYvXixTNvqbORg79TuVYyQiIiIiItUwEXxLSEgIAMDR0bHUdnfu3EHdunU/REjQ1NREQEAA7O3tsXbtWnz//fclth0xYgQmTJiAmjVron///iqP+fbnkJ2djb1798Ld3R1Dhw6Vaz99+nSIxeISE0Fvb2/MmjVLpuxUQumH1hMRERER0fvBRPANEokE/v7+sLS0xMiRIwEAT548kUv4Dhw4gAsXLmD69OkfLLYePXqgQ4cOWLVqFby8vOQOlX9t6NChePDgAZo1awYtLS2VxlL0OezevRvZ2dn45ptvpEdZvCk6Ohrbt2/HunXrFC5h1dbWlivX0spTKT4iIiIiIiqfKpsIHjx4EHFxcSgoKEBycjIkEgliYmJgbm6OqKgoaaJlZ2eH1q1bo127dhCJRLh48SK2bNkCMzMzzJs3r8zj5ufnY8mSJXLlhoaG+Prrr0u9d86cORg2bBiCg4Px1VdfKWwjEomwaNEipeNR9nMQi8WoXbs27OzsFPbj7OyMjRs3Yv/+/XB1dVV6fCIiIiJSnQYfpSMVVdlEcMGCBQAALS0tGBoawsbGBqtWrYKnp6fMBinu7u7Yv38/oqOj8eLFC5iYmGDixIlYuHAhjI2NyzxuXl4efH195cobN278zkTQ1dUVjRs3xooVKzBx4kRoamqWefy3KfM5pKSk4MiRI/Dw8ChxzJ49e0JXVxehoaFMBImIiIiIKjlBcXFxsbqDoKrpaFyqukOolLQ0yp/gf4pszUTqDqFSevQsR90hVDpN2jZF9aRHeKxXG3bjt6k7nErj9OaxMMl6ikLT+ki5fkfd4VQq/CZEZWVaU7XHb96Hfr+eVXcI783BKR3VHcInjcdHEBERERERVTFVdmloRUpLS0NeXskbn2hqan6wHUaJiIiIqOrgeXukKiaCFcDV1RXHjx8vsd7c3BwJCQkfLiAiIiIiIqJSqJwI5uXlYc+ePTh37hyeP3+OwsJCuTYCgQCbN28uV4Afg8DAQDx79qzEeh0dnQ8YDRERERERUelUSgTv3buH3r174/bt2yhtr5mqkgi2bdtW3SEQEREREREpTaVEcObMmbh16xZGjx6NL7/8Eg0aNEC1alxlSkRERERE9DFQKXuTSCTo2bMntm3jttxEREREROrCvWJIVSolgkVFRWjdunVFx0JVjEi78pzBU5kUFvFAK0VO3eG5k4qkvSx5x+KqqmFREaoDKC4uRm5OrrrDqTReP8pRXFyM3PwiNUdTueQW8PNQhD8nJatM5wgSqUqlcwQ7duyIGzduVHQsRERERERE9AGolAguW7YMEokEO3bsqOh4iIiIiIiI6D1TaWno/v37YW9vD3d3d3Tv3h1t2rSBgYGBXDuBQABfX99yB0lERERERPIE4EOCpBqVEsFFixZJ/xwbG4vY2FiF7ZgIEhERERERVT4qJYLHjh2r6DiIiIiIiIjoA1HpGcHu3bsrfVVWwcHBEAgE0ksoFMLU1BSOjo5Ys2YNMjMzS23/5pWUlFSmsceNGydzf7Vq1WBmZobhw4fj+vXrJd534MABCAQCmJqaoqio5J28MjIysHjxYrRs2RJ6enrQ0dGBtbU15s6di0ePHsnEoaenV2I/AoEAU6dOlb5+8OABFi9ejA4dOqBWrVqoU6cOevTogSNHjpTp/RMRERERkXpV+VPg/fz8YGlpifz8fCQlJSE2NhZeXl4ICgpCVFQUbG1tFbZ/U82aNcs8rra2NjZt2gQAKCgowO3bt7F+/XocOnQI169fh6mpqdw9YrEYFhYWSEhIgEQiQa9eveTa3LlzB7169cL9+/cxbNgwTJo0CVpaWrh69So2b96M3bt34+bNm2WOFwD27t2LH3/8ES4uLhg7diwKCgrw+++/o3fv3tiyZQs8PT1V6peIiIiIVKPBRwRJReVKBE+dOoXg4GBcvnwZGRkZMDAwQOvWrTFmzBh06dKlomJ8r/r164d27dpJX3t7e0MikcDJyQnOzs64ceMGdHR0SmyvqmrVqmHUqFEyZV988QWcnJywf/9+TJw4UaYuOzsbe/fuRUBAALZu3QqxWCyXCBYUFMDV1RXJycmIjY2V+ztYunQpfvzxR5Vjtre3x/3791GnTh1p2VdffYVWrVphwYIFTASJiIiIiD4SKi0NBYCZM2eiW7du2Lx5My5cuIBbt27hwoUL2LRpE7p3745Zs2ZVZJwflIODA3x9fXHv3j2EhobK1WdmZqKwsLDCx61Xrx6AV0ni23bv3o2cnBwMGzYMw4cPx65du/Dy5UuZNjt37sSVK1fg4+OjMBE3MDDA0qVLVY7v888/l0kCgVczm/3790diYqLccloiIiIiIqqcVEoEt23bhtWrV8PKygpisRiPHj1CQUEBHj9+jLCwMDRt2hSrV6/G77//XtHxfjCjR48GAERHR8uU29vbw8DAALq6unB2dkZ8fLzKY6SmpiI1NRXJyck4c+YMZs6cidq1a8PJyUmurVgshr29PerVq4fhw4cjMzMT+/btk2kTFRUlE3tZ43j7UlZSUhJ0dXWhq6tbpnGJiIiIiEg9VFoa+uuvv6JBgwY4e/YsRCKRtNzY2BjDhw9Hv379YGNjg19++QVjxoypsGA/pAYNGkAkEuH27dsAAF1dXYwbN06aCF64cAFBQUGws7PDxYsXYWZmVqb+s7OzUbduXZmy+vXrIzo6Wq48JSUFR44cwa+//goAaNiwITp16gSxWIxhw4ZJ2924cQMikahMsSiKoyxu3bqFXbt2YdiwYdDU1FS5HyIiIiIi+nBUSgT//fdfTJgwQSYJfJNIJMKQIUOkm6F8rPT09KTLHd3c3ODm5iatc3FxgaOjI7p164alS5di/fr1ZepbKBRKZ/SKioqQkJCAoKAg9O/fHydOnEDTpk2lbSMiIqChoYEhQ4ZIyzw8PDB79mw8e/YMtWrVAvBqt1B9fX2V43hb7969S733xYsXGDZsGHR0dLBs2bIyjUtERERE5ScQcLcYUs172zX0U/ihzMrKgpGRUYn1Xbp0QceOHVU6PkFTU1Nus5f+/fvDysoK3t7e2Llzp7Q8NDQUHTp0wNOnT/H06VMAQOvWrZGXl4ft27dj0qRJAF49A3jnzp1yx6GMwsJC6XEXBw8eVLjL6Ztyc3ORm5srU5aXmwstbe0yj01EREREROWj0jOCn3/+OXbu3ImsrCyF9ZmZmdi5cyc+//zzcgWnTomJiUhPT0eTJk1KbWdmZoa0tLQKGbNBgwZo1qwZTpw4IS2Lj4/HuXPncPLkSVhZWUmv15vBiMViadvmzZsjPT0dDx48qJB4SjNx4kT8+eefCA4OhoODwzvbBwQEQCQSyVzBvwa99ziJiIiIiEieSong5MmTkZiYiE6dOmHnzp3SjUVSU1OxY8cO2NnZITExEVOmTKnQYD+kkJAQAICjo2Op7e7cuVOuZ+zeVlBQIJNgi8ViVK9eHREREdi+fbvMNWPGDPz111+4f/8+AGDgwIEAoHCn04o0Z84cbN26FStXroSHh4dS93h7eyM9PV3mGjfl491ZloiIiIjoY6bS0lBPT09cunQJa9eulT43p6GhgaKiIgBAcXExpk2bhrFjx1ZcpB+QRCKBv78/LC0tMXLkSADAkydP5BK+AwcO4MKFC5g+fXqFjHvz5k38999/aNu2rbRMLBaja9eucHd3l2vfqVMnrFmzBuHh4Zg7dy6GDh2KgIAALF26FD169ECnTp1k2mdmZmLZsmXlOkLip59+wooVKzBv3jzMmDFD6fu0tbWh/dYyUK2nGSrHQURERETAJ/A0FqmJys8IrlmzBsOGDVN4oPzYsWPRtWvXiozzvTl48CDi4uJQUFCA5ORkSCQSxMTEwNzcHFFRURAKhQAAOzs7tG7dGu3atYNIJMLFixexZcsWmJmZYd68eWUet6CgQDpz93qzmPXr16OoqAgLFy4EAJw9exa3bt3C1KlTFfZRv359tGnTBmKxGHPnzkX16tWxa9cu9OrVC926dYObmxs6d+6M6tWr499//0VYWBhq1aqlciK4e/dufPfdd7CyssJnn30mN/PYu3dvGBsbq9Q3ERERERF9OOXaLKZr164fTcJXkgULFgAAtLS0YGhoCBsbG6xatQqenp4yO3C6u7tj//79iI6OxosXL2BiYoKJEydi4cKFKiU/ubm5Muf9GRgYoH379ggJCUHPnj0B/N/zf6+XfCoycOBALFq0CFevXoWtrS2aNGmCy5cvY+XKldi9ezf27NmDoqIiNGnSBBMmTCjX7OWVK1cAvHpuUdFZhceOHWMiSERERET0ERAUFxcXqzsIqprO3+XSUEUKi/g/SUWevsx9d6MqKO1lnrpDqHQG92+PGilJeFTDEO1GblR3OJXGefFEmGanocDEFIlXb6s7nEolt6BI3SFUSrn5/FxKYmump+4QpFw2nVd3CO/Nngnt1B3CJ02pGcHXm5HUr18fmpqa0tfKaNiwoWqRERERERFRqTT4kCCpSKlE0MLCAgKBADdu3EDTpk2lr99FIBCgoKCg3EF+LNLS0pCXV/Jv5zU1NSt0h1EiIiIiIiJVKJUIjhkzBgKBACKRSOY1yXJ1dcXx48dLrDc3N0dCQsKHC4iIiIiIiEgBpRLB4ODgUl/TK4GBgXj27FmJ9To6Oh8wGiIiIiIiIsXKtWsoyXrz/D8iIiIiIqLKSuVEsKioCBoaGjJlZ86cwZ9//gmhUAhPT080aNCg3AESEREREZFifFqLVKXx7ibyZs6cCV1dXTx//lxatmPHDnTt2hUBAQFYuHAh2rRpg8TExIqKk4iIiIiIiCqISongsWPH4ODggJo1a0rLFixYAJFIhN9//x3Lly/Hs2fPsGLFioqKk4iIiIiIiCqISktDHzx4gO7du0tf3717F3FxcVi4cCFGjRoFAPjrr79w6NChiomSPkkPMl+oO4RKSbeaprpDqJQ6WdZWdwiVUmomD5R/m5bm//8dZ3ExXr54qd5gKpPi4v//BwGqaar0e+BPllCL/95VJL+AB8oTfcpUSgSzs7NRo0YN6evjx49DIBCgX79+0rIWLVrg6NGj5Y+QiIiIiIgU4pFupCqVfiVoamqK//77T/r60KFD0NPTk9k1MyMjA9ra2uWPkIiIiIiIiCqUSjOC3bt3R3h4ONauXQuhUIhdu3bBxcUFmpr/t7Ti9u3b3DWUiIiIiIioElJpRtDHxwc6OjqYMWMGJk2aBG1tbSxatEhan5mZiRMnTqBz584VFScRERERERFVEJVmBJs0aYLr169j586dAICBAwfC3NxcWh8fH4/JkydjxIgRFRMlERERERHJ4SOCpCqVtw0zMTHB1KlTMXXqVJkkEADatGmDlStXon379uUOUBXBwcEQCATSSygUwtTUFI6OjlizZg0yMzNLbf/mlZSUVKaxx40bJ3O/gYEBWrZsicDAQOTm5krbLVq0CAKBAKmpqSX2NX/+fAgEAsTGxsrVRUREQCAQYO3atdIyCwsL6bgaGhqoWbMmbGxsMGnSJJw9e7bEcV6+fImVK1eiY8eOEIlEEAqFaNq0KaZOnYqbN29K2x09ehRffvklmjZtCl1dXTRq1AgTJkzA48ePy/QZERERERGReqk0I/ix8PPzg6WlJfLz85GUlITY2Fh4eXkhKCgIUVFRsLW1Vdj+TW+elagsbW1tbNq0CQDw/Plz7Ny5E99++y3OnTuHiIgIpfuZP38+IiIi8NVXX+Hq1avQ0tKS9jlz5ky0b98eX3/9tcw9rVq1wuzZswG8WqJ748YNbN++HRs3bsTMmTMRFBQk0z41NRV9+/bFhQsX4OTkhBEjRkBPTw///fcfIiIisGHDBuTlvdqefu7cuUhLS8OwYcNgZWWFO3fuYO3atfjzzz9x+fJl1KtXr8yfFRERERERfXgqJ4JPnjzB1q1bce7cOTx//hyFhYVybQQCgVqPkOjXrx/atWsnfe3t7Q2JRAInJyc4Ozvjxo0b0NHRKbG9qqpVqyY9TxEAvv76a3Ts2BGRkZEICgqCqampUv0IhUL8+uuv6NOnDwICArBw4UIAwPfff48nT57g4MGD0NCQndStX7++zNgA8OOPP2LEiBFYuXIlrKysMGXKFGnduHHjcOnSJezYsQNDhgyRuc/f3x8+Pj7S10FBQejSpYvMmH379kX37t2xdu1aLFmyRKn3RURERERE6qVSInj16lU4ODjg2bNnKJYeUCuvMp5r4uDgAF9fX8ybNw+hoaGYOHGiTH1mZiZ0dXVldkAtLw0NDfTo0QPnz59HQkKC0okgAPTu3RsjRoxAQEAAPDw88PTpU2zYsAEzZ85Eq1atlOpDR0cHISEhMDc3x9KlS/HVV19BIBDg7Nmz2L9/PyZOnCiXBAKvZjZXrFghfd2tWze5Nt26dYOhoSFu3Lih9HsiIiIiIiL1UukZwdmzZyMtLQ0+Pj64e/cu8vPzUVRUJHcpmiWsDEaPHg0AiI6Olim3t7eHgYEBdHV14ezsjPj4+Aob8/bt2wCA2rVrl/neoKAg6OrqYvLkyZg8eTIaNGiAxYsXl6kPPT09DB48GA8fPsT169cBAFFRUQD+7/NQRVZWFrKyslCnTh2V+yAiIiIi1WgIBJ/sRe+XSjOCZ86cgYuLC/z8/Co6ng+iQYMGEIlE0uRMV1cX48aNkyaCFy5cQFBQEOzs7HDx4kWYmZmVeYzXm8Ckp6fjjz/+wJ49e2Bra4tmzZqVuS9jY2MsW7YMkydPBgDs2bMHenp6Ze7H2toawKuk9PPPP5fO4tnY2JS5r9dWrVqFvLw8uLu7q9wHERERERF9WColglpaWmjcuHFFx/JB6enpSXcPdXNzg5ubm7TOxcUFjo6O6NatG5YuXYr169eXqe/s7GzUrVtXpszOzg4hISEqx/t6xk1XVxddunRRqY/XyePr952RkQEA0NfXV6m/EydOYPHixXBzc4ODg4NKfRARERER0YenUiLYvXt3nD9/vqJj+aCysrJgZGRUYn2XLl3QsWNHHDlypMx9C4VC7Nu3D8Cr5+wsLS3RoEEDlWPNzMzE9OnT0axZM9y+fRtz586V7kpaFllZWQD+L/EzMDCQ9l/W3VHj4uIwePBgWFtbKxVLbm6uzPEZAJCfl4vqWtplGpeIiIiIiMpPpWcEV6xYgWvXrslsJPIxSUxMRHp6Opo0aVJqOzMzM6SlpZW5f01NTfTq1Qu9evVC165dy5UEAoCPjw+SkpIQFhaGmTNnYsuWLTh16lSZ+7l27RoASN938+bNAQD//PNPmfp58OAB+vTpA5FIhAMHDig1oxgQEACRSCRz7dz8cxnfARERERG9SfAJX/R+qTQjuHTpUlhbW2Pu3LlYv349WrVqJZ1depNAIMDmzZvLHWRFe71E09HRsdR2d+7ckVvi+aGdP38e69atw7Rp09CmTRs0a9YMkZGR+Oqrr3Dp0iVUq6bcX2FWVhZ2794NMzMzfPbZZwCAgQMHIiAgAKGhoejatatS/Tx9+hR9+vRBbm4ujh49ChMTE6Xu8/b2xqxZs2TKDt18ptS9RERERERUsVRKBIODg6V/vnPnDu7cuaOwXWVMBCUSCfz9/WFpaYmRI0cCeHUm4tsJ34EDB3DhwgVMnz5dHWECAAoLCzF58mSYmJjA398fAFCjRg38/PPPGDRoEFauXIk5c+a8s5+cnByMHj0aaWlp+OGHH6THenTq1Al9+/bFpk2b0K9fP7i4uMjcl5eXh3nz5klnfrOzs9G/f388fPgQx44dg5WVldLvRVtbG9rasstAq2u9UPp+IiIiIiKqOColgnfv3q3oON6LgwcPIi4uDgUFBUhOToZEIkFMTAzMzc0RFRUFoVAI4NVGLq1bt0a7du0gEolw8eJFbNmyBWZmZpg3b957jfH10RBv0tDQwLx587BmzRpcvHgRO3fulFl+6ezsDGdnZyxevBju7u5o2LChtO7hw4cIDQ0F8GoW8Pr169i+fTuSkpIwe/Zs6c6jr/3+++/o06cPXF1dMXDgQPTs2RM1atRAfHw8IiIi8PjxY2kiOHLkSPz999/48ssvcePGDZmzA/X09OQSSSIiIiIiqpxUSgTNzc0rOo73YsGCBQBe7XJqaGgIGxsbrFq1Cp6enjKJlbu7O/bv34/o6Gi8ePECJiYmmDhxIhYuXAhjY+P3GmNAQIBcmaamJsaMGYMFCxbAyckJrq6ucm1+/vlntGjRAtOmTcPevXul5ZcvX8bo0aMhEAigr68PMzMzDBw4EBMmTECHDh3k+qlbty5Onz6NX375BZGRkfDx8UFeXh7Mzc3h7OyMGTNmyPQNAFu2bMGWLVtk+jE3N2ciSERERPSBCXjeHqlIUFxcXFzeTtLS0pCdna3SeXtUde2+mqTuECol3Wqa6g6hUupgYajuECql1Mw8dYdQ6Vi0boLqjx/hkW4tWLuuVXc4lca1XVNh+uIZCkzqI+lfxY90VFXVNPlFWpH8giJ1h1BpmRlWnl3PPX6/rO4Q3pvwMa3UHcInTaVdQ4FXB6XPmDEDxsbGqFu3LiwtLaV1Z8+eRf/+/XHhwoUKCZKIiIiIiIgqjkpLQ9PS0mBnZ4ebN2+iTZs2qFu3rszzYra2tjh16hTEYjHatm1bYcGqU1paGvLySv7Nu6amptp3GCUiIiIiIlKGSjOCixYtws2bNxEREYHz589j2LBhMvU6Ojro3r07JBJJhQRZGbi6usLExKTEq3379uoOkYiIiIiISCkqzQhGRUXByckJbm5uJbaxsLDA6dOnVQ6ssgkMDMSzZyWfe6ejo/MBoyEiIiIiAjT4iCupSKVE8PHjxxg+fHipbbS1tZGdna1SUJXRp7LElYiIiIiISKWlobVr18aDBw9KbRMXFwcTExOVgiIiIiIiIqL3R6VEsFu3bti7dy8SExMV1l+/fh2HDh1Cr169yhUcERERERERVTyVEkEfHx8UFhaic+fOEIvFSE1NBQDcuHEDmzdvhoODA7S1tTFnzpwKDZaIiIiIiP6PQCD4ZC96v1R6RtDGxgaRkZEYPXo0xowZAwAoLi6GtbU1iouLoa+vjz/++ANWVlYVGix9WurqVJ7DWCsTTf6LT6H0F/nqDqFS0q6u8nGwnywBXv1vqLgYKMgvUHM0lUdx8et/FiMjh/97elNdA/73SJHnuYXqDoGI3iOVv0E4Ozvj7t27WLFiBYYNG4ZevXph8ODB+PHHH3H79m3079+/IuMkIiIiIiIq1bp162BhYQGhUIiOHTvi77//LrX99u3b0bx5cwiFQtjY2ODAgQMy9cXFxViwYAFMTEygo6ODXr16IT4+XqZNWloaRo4cCQMDA9SsWRPjx49HVlaWTJvDhw/jiy++gL6+PurWrYshQ4YgISGhQt6zqsr1q2RDQ0PMnDkTERERiI6Oxo4dOzBnzhzUqVOnouIjIiIiIiJ6p8jISMyaNQsLFy7ExYsX0bJlSzg6OiIlJUVh+9OnT8PDwwPjx4/HpUuX4OLiAhcXF1y7dk3aZvny5VizZg3Wr1+Ps2fPokaNGnB0dMTLly+lbUaOHIl///0XMTEx+PPPP3HixAlMmjRJWn/37l0MGjQIDg4OuHz5Mg4fPozU1FS4urq+vw9DCYLi4teLRYg+rJPxJZ/LWJVxaahiJjWF6g6hUtLgAVJyTK0bo9rjh3ioUwufOa9SdziVxo0oL9TPeYb8eqaIv3BT3eFUKlwaqtjTzDx1h1BpNTfRVXcIUqPFV9QdwnsTMrJlmdp37NgR7du3x9q1awEARUVFMDMzw7Rp0/D999/LtXd3d0d2djb+/PNPadkXX3yBVq1aYf369SguLoapqSlmz56Nb7/9FgCQnp4OY2NjBAcHY/jw4bhx4wZatGiBc+fOoV27dgCAQ4cOoX///khMTISpqSl27NgBDw8P5ObmQkPj1Tzcvn37MGjQIOTm5qJ69eoqfT7lpdIzgq9dvnwZV65cwaNHj5CfL/+8gUAggK+vb3mGICIiIiKiKig3Nxe5ubkyZdra2tDWlv/lTV5eHi5cuABvb29pmYaGBnr16oUzZ84o7P/MmTOYNWuWTJmjoyP27NkD4NVMXlJSksxJCCKRCB07dsSZM2cwfPhwnDlzBjVr1pQmgQDQq1cvaGho4OzZsxg8eDDatm0LDQ0NbN26FePGjUNWVhZCQkLQq1cvtSWBgIqJYEpKCkaMGIFjx44BeLV2VhEmgkREREREpIqAgAAsXrxYpmzhwoVYtGiRXNvU1FQUFhbC2NhYptzY2BhxcXEK+09KSlLYPikpSVr/uqy0NkZGRjL11apVg6GhobSNpaUloqOj4ebmhsmTJ6OwsBCdOnWSex7xQ1PpGcFvvvkGEokE/fr1w7Zt2xATE4Njx47JXRKJpKLjrTDBwcEy29MKhUKYmprC0dERa9asQWZmZqnt37xe/yWrws3NDQKBAHPnzlVYHxsbKx3nwoULcvXjxo2Dnp6e9HVKSgoMDQ3h4OAg1zY/Px82NjawsLBAdna2XP0vv/wCgUCAjh07lhhvZGQkRo0aBSsrKwgEAvTo0UOJd0lEREREVDbe3t5IT0+Xud6c8ftYJCUlYeLEiRg7dizOnTuH48ePQ0tLC0OHDi1xQu1DUGlG8PDhw7C3t5dZT/ux8vPzg6WlJfLz85GUlITY2Fh4eXkhKCgIUVFRsLW1Vdj+TTVr1lRp7IyMDOzbtw8WFhYIDw/HsmXLSj0zZdGiRdi3b1+pfRoZGeHHH3/EpEmTsG3bNowdO1ZaFxgYiGvXrmHfvn2oUaOG3L1isRgWFhb4+++/cevWLTRp0kSuza+//ooLFy6gffv2ePr0aRneLRERERGR8kpaBqpInTp1oKmpieTkZJny5ORk1KtXT+E99erVK7X9638mJyfDxMREpk2rVq2kbd7ejKagoABpaWnS+9etWweRSITly5dL24SGhsLMzAxnz57FF198odR7rGgqzQhWr14dbdu2rehY1KJfv34YNWoUPD094e3tjcOHD+PIkSNISUmBs7MzcnJyFLZ/8xIKVdvEYufOnSgsLMSWLVvw4MEDnDhxosS2rVq1wp9//omLFy++s98JEyagS5cu+Pbbb6XJ2t27d+Hn5wdXV1c4OTnJ3XP37l2cPn0aQUFBqFu3LsRiscK+Q0JCkJ6eDolEAlNTUyXfKRERERG9D+o+9L2yHCivpaWFtm3b4ujRo9KyoqIiHD16FJ06dVJ4T6dOnWTaA0BMTIy0vaWlJerVqyfTJiMjA2fPnpW26dSpE54/fy6zck8ikaCoqEi6yu7FixfSTWJe09TUlMaoLiolgl27dsXly5crOJTKw8HBAb6+vrh37x5CQ0Pl6jMzM1FYWP5DVsViMXr37g17e3t89tlnJSZfADBt2jTUqlVL4ZrotwkEAqxfvx7p6enSHY6+/vprVKtWDWvWrCkxllq1amHAgAEYOnRoibGYmZnJ/SATEREREanbrFmzsHHjRmzbtg03btzAlClTkJ2dDU9PTwDAmDFjZJaWzpgxA4cOHUJgYCDi4uKwaNEinD9/HlOnTgXw6ju1l5cXlixZgqioKPzzzz8YM2YMTE1N4eLiAgD47LPP0LdvX0ycOBF///03Tp06halTp2L48OHSSZMBAwbg3Llz8PPzQ3x8PC5evAhPT0+Ym5ujdevWH/ZDeoNK3+gDAgJw/vx56dasn6LRo0cDAKKjo2XK7e3tYWBgAF1dXTg7O8sdKKmsR48e4dixY/Dw8AAAeHh4YMeOHcjLU7xVs4GBAWbOnIl9+/YpNSv4+eef49tvv0VwcDCmT5+OQ4cOYcmSJahfv77C9mKxGK6urtDS0oKHhwfi4+Nx7tw5ld4bEREREdGH5u7ujhUrVmDBggVo1aoVLl++jEOHDkk3e7l//z4eP34sbW9nZ4ewsDBs2LABLVu2xI4dO7Bnzx5YW1tL23z33XeYNm0aJk2ahPbt2yMrKwuHDh2SWREoFovRvHlz9OzZE/3790eXLl2wYcMGab2DgwPCwsKwZ88etG7dGn379oW2tjYOHToEHR2dD/DJKKbyOYL//vsvunbtijp16sDW1hYGBgbynQsE2Lx5c7mDfB+Cg4Ph6ekpc+bH22rWrIlGjRrh4sWL+OOPP3Dw4EFpInjhwgUEBQVBV1cXFy9ehJmZWZnGDwwMhK+vL5KTk6Gvr4/4+Hg0bdoUu3fvlv6GAXi1WYy9vT22b9+O3r17w9LSEl27dsXevXsBvNosZseOHcjKypIbIycnB9bW1rhz5w7atm2Ls2fPSqeh33ThwgW0a9cOMTEx6NWrF4qLi9GwYUMMGTIEq1atKvE9WFtbo06dOoiNjS3Te3+N5wgqxnMEFeM5gorxHEF5PEdQMZ4jWDKeI6gYzxEsWWU6R3BM2FV1h/De/D7C9t2NSGUqbRZz9+5duLi44Pnz53j+/Dlu3bqlsF1lTgSVoaenJ9091M3NDW5ubtI6FxcXODo6olu3bli6dCnWr19fpr7FYjEGDBgAfX19AICVlRXatm0LsVgskwi+SSQSwcvLCwsXLsSlS5feOZWspaUFkUgEAOjZs6fCJPB1LMbGxrC3twfw6u/N3d0doaGhCAwMLPE+IiIiIlIv/j6QVKXS0tBp06bh9u3bmDJlCk6cOIH4+HjcvXtX7rpz505Fx/tBZWVlSRM1Rbp06YKOHTviyJEjZer3xo0buHTpEjp37oxbt25Jrx49euDPP/9ERkZGiffOmDEDNWvWVOpZwdWrV+PSpUuwtrbGmjVrFCbshYWFiIiIgL29Pe7evSuNpWPHjkhOTpZ7gFZVubm5yMjIkLny8nLffSMREREREVU4lRLBEydOYODAgVi3bh26dOmCxo0bw9zcXOH1sUpMTER6errCIxTeZGZmhrS0tDL1/XoDmpkzZ8LKykp6BQYG4uXLl9i5c2eJ976eFYyKisKlS5dKbPfgwQMsXLgQLi4uiI6OhpaWFr755hu5dhKJBI8fP0ZERIRMLK9nP0vbwKYsAgICIBKJZK7Q9SsrpG8iIiIiIioblZaGamtro2nTphUdS6USEhICAHB0dCy13Z07d1C3bl2l+y0uLkZYWBjs7e3x9ddfy9X7+/tDLBZLdzdSxMvLC6tWrcLixYtLPMPw9W5Ha9asgYmJCZYuXYpp06YhIiICw4cPl7YTi8UwMjLCunXr5PrYtWsXdu/ejfXr15f7QVZvb2/MmjVLpuz8gxfl6pOIiIiIiFSjUiLYu3dvnD59uqJjqTQkEgn8/f1haWmJkSNHAgCePHkil/AdOHAAFy5cwPTp05Xu+9SpU0hISICfnx+GDh0qV3/z5k34+vri0aNHJZ7T93pWcNGiRWjZsqVc/e7duxEVFYWgoCDpJjZff/01goODMWvWLPTv3x8GBgbIycnBrl27MGzYMIWxmJqaIjw8HFFRUXB3d1f6PSqi6EBQLa3yH8FBREREVJWV9bw9otdUSgRXrFiBLl26YM6cOfD391f5QPXK4ODBg4iLi0NBQQGSk5MhkUgQExMDc3NzREVFSd+bnZ0dWrdujXbt2kEkEuHixYvYsmULzMzMMG/ePKXHE4vF0NTUxIABAxTWOzs7w8fHBxEREXIzaG+aMWMGVq5ciStXrqBGjRrS8szMTEyfPh2tW7eWSVA1NDSwfv16dOzYET4+Pvj5558RFRWFzMxMODs7Kxzjiy++kB4u/zoRPHHihPTg+ydPniA7OxtLliwBAHTr1g3dunVT+rMgIiIiIqKyKW3l4Ju2bt1aar1KieCoUaMgEokQFBSEDRs2wMrKqsTjIypqs5H3ZcGCBQBe7bBpaGgIGxsbrFq1Cp6enjIbxbi7u2P//v2Ijo7GixcvYGJigokTJ2LhwoXSs0neJT8/H9u3b4ednR0MDQ0VtrG2toalpSVCQ0NLTQRr1qwJLy8vLF68WKb89Wzirl275Hb7bNeuHb7++mv88ssvGDduHMRiMYRCIXr37q1wDA0NDQwYMABisRhPnz5F7dq1IZFIFI4JAAsXLmQiSERERET0HmVkZEDFEwBlqHSOoIaGcnvMCAQCFBZy+R8pxnMEFeM5gorxHEHFeI6gPJ4jqBjPESwZzxFUjOcIlqwynSPoGfGPukN4b7YOt1F3CJ80lWYEi4qKKjoOIiIiIiIieofXZ3+X5NixY0r1o1IiSIqlpaUhL6/k355pamqWaYdRIiIiIqLScF1I1dO6dWuZ1/n5+bh69SquXLmCMWPGKN0PE8EK5OrqiuPHj5dYb25ujoSEhA8XEBERERERfVKCgoIUlv/www948uSJ0v0olQj+/vvvAIDBgwdDX19f+loZZclKP3aBgYF49qzk597KexYfERERERGRIu7u7mjfvj1WrlypVHulEsFx48ZBIBDgiy++gL6+vvR1aYqLiyEQCKpUIti2bVt1h0BERERERFVMUVERIiMjoaWlpfQ9SiWCW7ZsgUAggImJCYB3n0lBRERERETvnwZ3G69yGjVqJHN8RHFxMZ4+fYqXL19i/fr1Svej9Izgm8aOHav0AERERERERFQxvLy8ZF7n5+fjn3/+wZMnT+Dh4aF0P9wshoiIiIiI6CMxffp0heWbN2/G1KlTsWXLFqX6YSJIapOem6/uEColoxo82FgRQz3l17xXJbP2Xld3CJXO8tx8GAJAQR7y//tb3eFUHgX//3gjAVBNk0vJ3qTJpXUKHYhPUncIlVZzk0bqDoFITo8ePTBz5syKTQQ1NDTeuTmMIgKBAAUFBWW+j4iIiIiI3o2/xyAASE5ORlBQEIyNjZW+R6lEsFu3bnKJ4LNnz3D16lVoamrCzMwMxsbGSE5OxoMHD1BYWAhbW1vUqlWrbO+AiIiIiIiISqSpqSmzWcxrderUKdMxf0olgrGxsTKvExMT0blzZ4wYMQI//PADGjZsKK27f/8+vL29cerUKfz5559KB0JERERERESl2717t8xrDQ0NGBkZwdbWFkKhUOl+VHpG8Ntvv4WJiQlCQ0Pl6ho2bAixWIwvvvgCc+bMQXh4uCpDEBERERER0VucnZ0rpB+VEsEjR45g8uTJpbZxcHDAxo0bVQqKiIiIiIiIXjl+/LjSbbt3765UOw1VAnn58iUeP35captHjx4hJydHle4rleDgYAgEAuklFAphamoKR0dHrFmzBpmZmaW2f/NKSlJu9638/HzUqVMHXbp0KbFNcXExzMzM0KZNG5ny5ORkfPvtt2jevDl0dXVRo0YNtG3bFkuWLMHz588V9tWhQwcIBAL8+uuvpX4GQqEQDx8+lKvv0aMHrK2tlXpvRERERFRxSvre+Slc9H8cHBxgb28PBwcH6Z8Vvba3t1e6T5USwbZt2yIiIgJnzpxRWH/69GlERkaiffv2qnRfKfn5+SEkJAS//vorpk2bBuDVYY42Nja4evVqie3fvGrWrKnUWNWrV8ewYcNw+vRp3Lt3T2GbEydOIDExEaNGjZKWnTt3DtbW1li3bh26du2KoKAgBAYGonXr1li2bBnc3Nzk+omPj8e5c+dgYWEBsVhcaly5ublYtmyZUu+BiIiIiIgqxrNnz/D8+XM8e/YM+/btg7W1Nf744w/cu3cP9+7dwx9//IEWLVqUaY8WlZaGLl26FD179kTXrl0xcOBAdOnSBUZGRkhJScFff/2FP//8E9WqVcOSJUtU6b5S6tevH9q1ayd97e3tDYlEAicnJzg7O+PGjRvQ0dEpsX1ZjRw5EuvXr0d4eDi+//57ufqwsDBoaGhg+PDhAIDnz59j8ODB0NTUxKVLl9C8eXOZ9kuXLlW4VDc0NBRGRkYIDAzE0KFDkZCQAAsLC4UxtWrVChs3boS3tzdMTU1Vfm9ERERERKQ8AwMD6Z/nzp2L1atXw8HBQVo2dOhQGBoaYubMmejfv79Sfao0I9ilSxccOHAA5ubm2Lt3L+bMmYNx48Zhzpw5iIqKgrm5Ofbv34/OnTur0v1Hw8HBAb6+vrh3757CjXMyMzNRWFioUt+dO3eGhYUFwsLC5Ory8/OxY8cO2NvbSxOy3377DQ8fPkRQUJBcEggAxsbGmD9/vlx5WFgYhg4dCicnJ4hEIoXjvTZv3jwUFhZyVpCIiIiISE1u3boFIyMjuXJjY2PcvHlT6X5USgQBoGfPnrh16xaOHz+ONWvWwM/PD2vWrMHx48dx69Yt9OzZU9WuPyqjR48GAERHR8uU29vbw8DAALq6unB2dkZ8fHyZ+hUIBBgxYgT++ecf/PvvvzJ1hw4dQlpaGkaOHCkti4qKgo6ODoYOHar0GGfPnsWtW7fg4eEBLS0tuLq6lro81NLSEmPGjMHGjRvx6NGjMr0fIiIiIqp4AsGne5FirVu3ho+PD549eyYte/bsGby9vdGqVSul+1E5EQReJStdu3bF1KlT4ePjg6lTp6Jr165V6uHOBg0aQCQS4fbt2wAAXV1djBs3DuvWrcPu3bvx3Xff4ejRo7Czs8ODBw/K1PfrRO/t5CwsLAxCoRBDhgyRlt24cQNNmzaFlpaW0v2HhobCzMxMOnM7fPhwXL9+HZcvXy7xHh8fHxQUFODHH38swzshIiIiIqKKsGHDBly9elW6cWSbNm3QoEED/Pvvv9i0aZPS/ZQrEaRX9PT0pLuHurm5YevWrRgzZgxcXFzg7++Pw4cP4+nTp1i6dGmZ+m3RogVat26NiIgIaVl2djaioqLg5OQks1Y4IyMD+vr6SvddUFCAyMhIuLu7SxN3BwcHGBkZlTor2KhRI4wePRobNmx4586xRERERERUsaytrREfH49t27Zh8ODBGDx4MLZt24abN2/i888/V7ofJoIVICsrq9QkrEuXLujYsSOOHDlS5r5HjhyJu3fv4vTp0wCAPXv24MWLFzLLQoFXD5C+fZRFaaKjo/HkyRN06NABt27dwq1bt3D37l3Y29sjPDwcRUVFJd47f/58FBQUlOlZwdzcXGRkZMhc+Xm5St9PRERERESvVKtWDUOGDIGvry98fX0xdOhQaGpqlqkPJoLllJiYiPT0dDRp0qTUdmZmZkhLSytz/x4eHtDQ0JBu4hIWFoZatWrJ7QbUvHlz3Lx5E3l5eUr1+3rWz83NDVZWVtIrMjISDx8+LPXQykaNGmHUqFFlmhUMCAiASCSSuf7YtEape4mIiIhIMQ2B4JO9qGSXLl3CqFGj0LZtW7Rt2xajRo3ChQsXytQHE8FyCgkJAQA4OjqW2u7OnTuoW7dumfs3NTWFvb09tm/fjuTkZMTExGDo0KFyzwIOHDgQOTk52Llz5zv7zM7Oxt69e+Hu7o7t27fLXSYmJu88U/D1rKCyzwp6e3sjPT1d5nKbMF2pe4mIiIiI6JXdu3ejQ4cOSE1NhYuLC1xcXJCamoqOHTsqlQu8ptI5gvSKRCKBv78/LC0tpUs1nzx5IpfwHThwABcuXMD06aolPiNHjsSXX36JyZMnIz8/X25ZKAB89dVX+PnnnzF79my0bdsWTZs2lalPSUnBhg0bMH/+fOzevRvZ2dn45ptv0LVrV7m+oqOjsX37dqxbtw7a2toKY2rcuDFGjRqF3377Debm5qhWrfQfJW1tbbm+qmu9fNdbJyIiIiKiNyxYsAALFy6UOxruhx9+wMKFC2U2lCwNE0ElHTx4EHFxcSgoKEBycjIkEgliYmJgbm6OqKgoCIVCAICdnR1at26Ndu3aQSQS4eLFi9iyZQvMzMwwb948lcYeMmQIvv76a+zduxdmZmbo1q2bXJtatWph9+7d6N+/P1q1aiWdKgaAixcvIjw8HJ06dQLwallo7dq1YWdnp3A8Z2dnbNy4Efv374erq2uJcfn4+CAkJAT//fdfmR5MJSIiIiIi1cTHx8Pd3V2u3M3NDX5+fkr3o1IiGBkZCVdXV1SvXl2V2z9KCxYsAABoaWnB0NAQNjY2WLVqFTw9PWU2inF3d8f+/fsRHR2NFy9ewMTEBBMnTsTChQthbGys0tgGBgYYOHAgtm/fDg8PjxKP5+jYsSOuXbuGn376Cfv370dISAg0NDTw2Wef4fvvv8fUqVORkpKCI0eOwMPDo8QHSnv27AldXV2EhoaWmgg2adIEo0aNwrZt21R6X0REREREVDYmJiY4ffo0rKysZMpPnz4NExMTpfsRFBcXF5d1cA0NDdSpUwdjxozBhAkT0Lx587J2QYT911LUHUKlZFRD8XLcqq6ZifLHo1Qls/ZeV3cIlc7yr3vCMC0FD6vXQJPPx6k7nErj1r/BqJ+fjXwTU9y9FK/ucCqVOnr8964iwRfuqTuESmtWt0bqDkHq612f7n8HfnFtoe4QKqWVK1di/vz5+Oabb9ClSxcAwKlTp7B27Vr4+/tj1qxZSvWj0mYx8+fPh1AoRFBQED7//HN069YNISEhePmSz3wRERERERG9LzNnzsT69etx6NAhuLm5wc3NDQcOHMCvv/6qdBIIqLg01M/PD4sWLcLBgwexadMm7N+/H6dOncL06dMxatQoTJgwAS1btlSl6yohLS2t1GMeNDU1VdphlIiIiIiIPn2jR4/G6NGj8XpxZ0mPjpVG5eMjNDQ0MGDAAOzevRuJiYn44YcfULduXaxbtw5t2rRBhw4dsGnTJmRlZak6xCfL1dUVJiYmJV7t27dXd4hERERERFTJCQQClZJAoIJ2DTUyMsLcuXMxd+5cxMTEwNPTExcuXMDkyZMxa9YsjBo1CnPnzoW5uXlFDPfRCwwMxLNnz0qs19HR+YDREBEREdHHStUkgD5ejRo1QmnbvNy9e1epfirs+Ijr169j48aNCA0NxdOnT1GjRg0MHjwYly9fxvr167Ft2zbs2LED/fr1q6ghP1qvj3UgIiIiIiIqCy8vL5nX2dnZuHDhAo4ePYrZs2cr3U+5EsGcnBxERERg48aNOHv2LIqLi2Fraws/Pz+MGjVKeqzCwYMHMW7cOMydO5eJIBERERERkYqmT5+usHzlypW4fPmy0v2olAieP38emzZtQkREBDIzMyEUCjFmzBh89dVX6Nixo1z7fv36Yfz48VixYoUqwxEREREREVEpXFxcsHjxYqXbq5QIdujQAQDQokULTJ48GWPGjIFIJCr1noYNG6J+/fqqDEdERArU6NwByxMeqDuMSqfms1QAgGl+Nm79G6zeYCqRevkv1B0CERG9Z4MHD0Z+fj6qV6/+zrYqHSg/evRoTJ48WXqAIZEq7qflqjuESklYXeXNfD9pmhp8GP5tNZtaQPPRQ3WHQR+ZF0YmOBB9Qd1hVCp9mhmrO4RKqezfEKsOkU7l+W/1tN031B3Ce/Pz4M/UHUKlVVhYiJSUFJkj6Z48eYIOHTrgzp07EAgE79yoU6UZwZCQEFVuIyKi96BIQwM5tY3UHUalofskCQIAxRoaKDCqp+5wKpX8omK8rMNzaomIPmZbtmzBtGnTkJOTI1cnEAjQuHFjFBcXo6ioqNR+yrVZTG5uLg4cOIBLly4hPT0dIpEIrVu3Rv/+/aGtrV2eromISEk5tY0Q8edZdYdRaQx36ogaT5JQYFQP8RduqjucSuX6kwx1h0BEROXk5+eH+fPno2/fvtDU1JSWp6WlwcHBQekNY1ROBKOiojBp0iQ8efJE5hwLgUAAIyMjbNiwAQMHDlS1eyIiIiIiInrLw4cPMX78eBgZya4GSklJAQDY2toq1Y9KieDRo0cxZMgQaGpq4ssvv0TXrl1hbGyM5ORknDhxAqGhoXB1dcXhw4fh4OCgyhBERERERPQOPFC+6unatSuEQqFcuZaWFnr06KF0Pyo96bpw4ULo6Ojg/Pnz2LhxI8aMGQNHR0eMGTMGmzZtwt9//w2hUIiFCxeq0v0HFxwcDIFAIL2EQiFMTU3h6OiINWvWIDMzs9T7J06cCIFAACcnpzKPPW7cOOjp6cmU9ejRQyaeN6+4uDgAQGxsLAQCAXbs2AEAJbZ/+4qNjUVCQoL09ZIlSxTGNXLkSAgEArnYNm7ciO7du8PY2Bja2tqwtLSEp6cnEhISyvzeiYiIiIiobCQSCQwMDOTKa9asCYlEonQ/Ks0IXrp0CSNGjIC1tbXCeltbW7i5uSEiIkKV7tXGz88PlpaWyM/PR1JSEmJjY+Hl5YWgoCBERUUpnGY9f/48goODFWbl5dGgQQMEBATIlZuamips//YGPr///jtiYmLkyj/77DPpg6VCoRDh4eGYP3++TJvs7Gzs3btX4Xu6dOkSLC0t4ezsjFq1auHu3bvYuHEj/vzzT1y5cqXE+IiIiIiIqPwuX76M8ePHIz4+HnZ2dvj9999hZGSEY8eOQV9fH+3atVOqH5USQV1dXdStW/quY0ZGRtDV1VWle7Xp16+fzAfn7e0NiUQCJycnODs748aNG9DR0ZHWFxcXY/r06RgzZgyOHj1aobGIRCKMGjVK6fZvt/3f//6HmJgYhX28nr3r378/du3ahStXrqBly5bS+r179yIvLw99+/aV+63CL7/8Itefi4sL2rVrh99//x3ff/+90jETEREREVHZTJ48GXXr1sWsWbOwZs0a+Pj4YOPGjXj8+DECAgIQHR2tVD8qLQ3t1asXjhw5UmqbI0eOoHfv3qp0X6k4ODjA19cX9+7dQ2hoqExdSEgIrl27hqVLl6opuvLp1KkTLC0tERYWJlMuFovRt29fGBoaKtWPhYUFAOD58+cVHCERERERlUZD8OlepNi1a9ewatUqjBw5En5+fjh+/DgAoH379rh06ZLS/aiUCK5YsQIpKSkYM2YMHjx4IFP34MEDjB49GqmpqVixYoUq3Vc6o0ePBgCZ7DozMxNz587FvHnzUK9exZ9TVVhYiNTUVJkrKyurwsfx8PBARESEdOfX1NRUREdHY8SIEaXe9/TpU6SkpOD8+fPw9PQEAPTs2bPC4yMiIiIiov9jZmaG9PR0AEDDhg2lu4UWFRUhPz9f6X6UWhqqaOfPWrVqQSwWIyIiAg0bNpTuGnr//n0UFhbC1tYWo0ePrvAlk+rQoEEDiEQi3L59W1rm5+cHHR0dzJw5872MGRcXJ7f8duzYsQgODq7QcUaMGIEffvgBp06dQpcuXfDHH39AKBTC2dkZhw4dKvG++vXrIzc3FwBQu3ZtrFmz5pOYASYiIiIiqsx++OEHfPfddwgNDYW2tjYKCwtRVFSEwMBAtGrVSul+lEoEY2NjS6wrKCjAnTt3cOfOHZnyK1eufFLb2erp6Ul3D7158yZWr16N8PBwaGtrv5fxLCwssHHjRpmy97ERy+effw5bW1uEh4ejS5cuCAsLw6BBg975fOfBgwfx8uVL3LhxA6GhocjOzq7w2IiIiIiISNa3336LlJQUmJubo06dOsjJyUGtWrWgq6uLAwcOKN2PUolgUVGRyoF+KrKysqSHNs6YMQN2dnYYMmTIexuvRo0a6NWr13vr/00jRoxAYGAgZs6cidOnT2PevHnvvMfe3h7Aqw12Bg0aBGtra+jp6WHq1KkK2+fm5kpnEP+vDO8tkSYiIiIi+hR5eXnJvNbS0kLDhg3RvXt31KhRQ+l+VNo1tKpJTExEeno6mjRpAolEgkOHDmHXrl0yZ+cVFBQgJycHCQkJMDQ0VHi2R2Xl4eEBb29vTJw4EbVr10afPn3KdH/jxo3RunVriMXiEhPBgIAALF68WKbM6zsfzJzrq3LcRERERFUdN1WpeqZPn14h/TARVMLrs/gcHR1x//59AICrq6tcu4cPH8LS0hIrV66Uy9Qrs4YNG6Jz586IjY3FlClTUK1a2X8scnJy5Gb83uTt7Y1Zs2bJlCVzNSkRERERkVowEXwHiUQCf39/WFpaYuTIkUhJScHu3bvl2k2aNAnm5ubw8fGBjY2NGiItnyVLluDYsWNwd3cvsU1BQQEyMzNRq1YtmfK///4b//zzT6k7jWpra8stA31eUHLiSERERERE8jQ1NaU7/iui7GN9TATfcPDgQcTFxaGgoADJycmQSCSIiYmBubk5oqKiIBQK0bBhQzRs2FDuXi8vLxgbG8PFxeXDB14Bunfvju7du5faJisrC2ZmZnB3d8fnn3+OGjVq4J9//sHWrVshEong68tlnkRERERE79Pbk1L5+fnS7+QLFixQuh8mgm94/cFpaWnB0NAQNjY2WLVqFTw9PaGvr6/m6NRPV1cXEyZMwLFjx7Bjxw7k5OTA1NQUHh4emD9/vvRgeSIiIiL6MD6lXfpJOc7OznJlQ4YMQYsWLRAREYHx48cr1Y+guLR5RaL36H4al4YqIqyuoe4QKiVNPg0vp2ZTC2g+eojsuvUQ8edZdYdTaQx36ogaT5KQX88U8RduqjucSuX6kwx1h1Ap9WlmrO4QKiV+QyyZSKfy/Ld69r7/1B3CexM4sJm6Q/io3LlzBzY2Nkof61Z5foqJiIiIiIiozF68eIE1a9agfv36St9T4UtDc3NzoaGhgerVq1d01x+VtLQ05OXllVivqamJunXrfsCIiIiIiIjoY2doaCizWUxxcTEyMjKgp6cHsVisdD8qJYInTpzAkSNHMGvWLNSsWRMA8PTpU4waNQpHjhxB9erVMX36dCxbtkyV7j8Jrq6uOH78eIn15ubmMucQEhERERGVFZ+cqHpWrVol81pDQwNGRkbo0KGDNDdThkqJ4IoVK3D9+nX4+flJy2bPno3Dhw+jSZMmyMrKwk8//YQ2bdrAzc1NlSE+eoGBgXj27FmJ9To6Oh8wGiIiIiIi+hSMGTOmQvpRKRG8dOkSevbsKX398uVL/PHHH+jTpw8OHTqEzMxM2Nra4tdff62yiWDbtm3VHQIREREREX2C0tLSsHbtWly9ehU5OTlo2bIlpk6dClNTU6X7UGmzmKdPn8o8iHjmzBm8fPkSnp6eAAB9fX04OTnhv/8+3V2MiIiIiIiIPrRr166hefPm+P3336Gvrw8jIyP88ccfsLW1xfXr15XuR6UZQR0dHWRmZkpfHzt2DAKBQOZAcj09vVKXRhIREREREVHZzJkzB926dUNkZCQ0NTUBAIWFhfDw8MDcuXOxb98+pfpRKRFs0qQJDh06hNzcXAgEAkRERKBFixaoV6+etM39+/dhZGSkSvdERERERKQEnidf9Zw8eRKxsbHSJBB4dSKBt7c3evTooXQ/KiWCEydOxKRJk9CkSRNoaWkhISEBP/30k0ybCxcuoEWLFqp0T1VESjoPlFdEqMXjPRWpo6et7hAqrWIARTz5Wer1JyEQCGCgU+GnJH3UeHC6Yo+fv1R3CJVS3JMMdYdQaQ2yqffuRkTviaamJkQikVy5vr6+zLES76LSN87x48djzpw5yMnJQXp6OqZMmQIvLy9p/ZkzZ3Dz5k2ZDWWIiIiIiIiofFq1aoX//e9/cuWnTp1C69atle5HpV+VCgQC/Pjjj/jxxx8V1rdt2xbPnj1DjRo1VOmeiIiIiIiIFAgODkZ+fr5cuZ2dHbp166Z0P+9lzYyWlha0tLTeR9dERERERPT/afAhwSrHwsJC+ueUlBTo6elBV1cXVlZWZeqnXA8j7d69G25ubrC1tUWTJk2k5XFxcVi+fDkePnxYnu6JiIiIiIjoLZs3b0aDBg1Qr1496OnpwcrKCmKxuEx9qDQjWFRUBA8PD+zYsQPAq+MkcnJypPW1atWCj48PCgsL4e3trcoQRERERERE9JaIiAjMmDED3t7eaNy4MSZMmIAlS5Zg9uzZyMnJwYQJE5TqR6UZwZUrV2L79u2YPHkynj17hm+//Vam3tjYGF27dsX+/fvL3HdwcDAEAoH0EgqFMDU1haOjI9asWSNzfqEiEydOhEAggJOTU5nHHjduHPT09EqsFwgEmDp1qvR1QkICBAIBVqxYofQY3333HQQCAdzd3csc25ufi56eHho1aoShQ4di586dKCoqkrunR48eMvcYGhqiffv22LJli8L2bm5uEAgEmDt3bolxLF26FM7OzjA2NoZAIMCiRYvK9D6IiIiIiEh1y5cvR0BAAHx8fNChQwdpbrFu3ToEBQUp3Y9KM4LBwcFo3749fvnlFwCvEqS3NWnSRKVE8DU/Pz9YWloiPz8fSUlJiI2NhZeXF4KCghAVFQVbW1u5e86fP4/g4GAIhUKVx32fiouLER4eDgsLC+zbtw+ZmZnQ19dX+n5tbW1s2rQJAJCTk4N79+5h3759GDp0KHr06IG9e/fCwMBA5p4GDRogICAAAPDkyRP8/vvvGD9+PG7evIlly5ZJ22VkZGDfvn2wsLBAeHg4li1bpvDvdf78+ahXrx5at26Nw4cPq/IxEBEREVEF4aFTVc+NGzfQt29fufJWrVrh7t27Svej0s/OrVu30LVr11Lb1K5dG0+fPlWlewBAv379MGrUKHh6esLb2xuHDx/GkSNHkJKSAmdnZ5mlqMCrJGv69OkYM2YMjI0r5zlJsbGxSExMxJYtW1BQUIBdu3aV6f5q1aph1KhRGDVqFCZOnIglS5bgypUrCAgIQGxsLCZOnCh3j0gkkt4zc+ZMnDp1Cg0aNMDatWtldhvauXMnCgsLsWXLFjx48AAnTpxQGMPdu3fx+PFjhIaGlu3NExERERFRudWoUQO5ufLncV+6dAmWlpZK96NSIqijo4P09PRS29y7dw81a9ZUpfsSOTg4wNfXF/fu3ZNLREJCQnDt2jUsXbq0QsesSGKxGC1atIC9vT169epV5gc6S/L999+jT58+2L59O27evFlqW11dXXzxxRfIzs7GkydPZGLr3bs37O3t8dlnn5UY25u7FBERERER0YdlY2OD8+fPS18XFhZi6dKlmDhxIqZNm6Z0Pyolgq+XBb58+VJhfVpaGg4dOoQvvvhCle5LNXr0aABAdHS0tCwzMxNz587FvHnzUK9evXKPkZqaqvAqj9zcXOzcuRMeHh4AAA8PD0gkEiQlJZU7XuDV51JcXIyYmJh3tr1z5w40NTWlifqjR49w7Ngxmdh27NiBvLy8ComNiIiIiIgqhpeXF54/fw4A0u/0Bw4cQFBQEKZMmaJ0PyolgtOnT0diYiKGDBmCxMREmbrbt29j8ODBSE9Px/Tp01XpvlQNGjSASCTC7du3pWV+fn7Q0dHBzJkzy91/dnY26tatq/Aqjz///BPPnz/H8OHDAQAuLi6oXr06IiIiyh0zAFhbWwOAzOcCvPoNwetENi4uDjNmzMDFixfRv39/6OrqAgDCw8Ohra2NQYMGAQCGDx+OZ8+e4cCBAxUSGxERERERVYxBgwbBy8sLAGBubo5Hjx7h1KlT0gkzZam0WcygQYMwd+5c/PjjjzA3N0eNGjUAAEZGRnj69CmKi4vh6+sLBwcHVbp/Jz09PenuoTdv3sTq1aulyUx5CYVC7Nu3T2Fd7969Ve5XLBajXbt20vMW9fX1MWDAAIjFYulfZHm83u307V1V4+LiZJJYgUCAAQMGYMuWLTKxDRgwQLpxjZWVFdq2bQuxWAwXF5dyx0ZERERE7wfPkydVqZQIAkBAQAAcHBywdu1anD17Fi9fvkRRURH69u2L6dOnw9HRsSLjlJGVlQUjIyMAwIwZM2BnZ4chQ4ZUSN+ampro1atXhfT12vPnz3HgwAFMnToVt27dkpZ37twZO3fuxM2bN9G0aVPk5OTIPXup7FLXrKwsAJDbhdTCwgIbN26UHsVhZWUl/eyAV7sOXbp0CWPGjJGJrUePHli3bh0yMjLkdiJVRW5urtxDrXm5udCqgOSdiIiIiIjKRuVEEHg1Q1aeWTJVJCYmIj09HU2aNIFEIsGhQ4ewa9cuJCQkSNsUFBQgJycHCQkJMDQ0rJBEpjy2b9+O3NxcBAYGIjAwUK5eLBZj8eLFiIyMhKenp0xdcXGxUmNcu3YNAKQzjq/VqFGj1MT29aY7M2fOVLi0dufOnXIxqSIgIACLFy+WKZs4/XtM8vIud99ERERERFQ25UoE1SEkJAQA4OjoiPv37wMAXF1d5do9fPgQlpaWWLlyZYUsvSwPsVgMa2trLFy4UK7ut99+Q1hYGBYvXgxHR0elNntRJCQkBAKBoEyJeXFxMcLCwmBvb4+vv/5art7f3x9isbhCEkFvb2/MmjVLpuzaI/ltb4mIiIiI6P0rVyJYUFCA//77D8+fP0dhYaHCNt26dSvPEDIkEgn8/f1haWmJkSNHIiUlBbt375ZrN2nSJJibm8PHxwc2NjYVNr4qXp/Jt3jxYgwdOlSuPi8vDyNHjsTZs2fRsWNHmJiYlHmMZcuWITo6GsOHD4eVlZXS9506dQoJCQnw8/NTGNvNmzfh6+uLR48ewdTUtMxxvUlbW1vuGU6tpxnl6pOIiIioqtPgQ4KkIpUSweLiYixYsAA///yz3OYkbyspQXyXgwcPIi4uDgUFBUhOToZEIkFMTAzMzc0RFRUFoVCIhg0bomHDhnL3enl5wdjY+INtdHL06FGFR2m4uLhg//79KC4uhrOzs8J7+/fvj2rVqkEsFqNjx46ljlNQUCBdyvny5Uvcu3cPUVFRuHr1Kuzt7bFhw4YyxS0Wi6GpqYkBAwYorHd2doaPjw8iIiKks3khISG4d+8eXrx4AQA4ceIElixZAuDVERbm5uZlioGIiIiIiD48lRJBf39/LF26FDVr1sSYMWPQoEEDVKtWsatMFyxYAADQ0tKCoaEhbGxssGrVKnh6esptiKJuhw4dwqFDh+TKLSwsIBaL0bBhQ7Rs2VLhvTVr1kSXLl0QGRmJoKCgUj/H3Nxc6bawurq6MDIyQtu2bbFgwQIMHjwYGhrKnwaSn5+P7du3w87ODoaGhgrbWFtbw9LSEqGhodJEcPPmzTh+/Li0zbFjx3Ds2DEAQJcuXZgIEhERERF9BATFyu5G8gYLCwsIBAKcP38etWvXfh9xURVw/i6Xhioi1FLpeM9PXh097jD7trotLKH56CGy6tZD+L7/qTucSsNj4BfQe5KEApP6SPr39rtvqEIMdKqrO4RK6fFz+VU9BMQ94X+nSzLIRrld3T8E30Px6g7hvfHvq/wjT1R2Kk3jJSUlYcqUKUwCiYiIiIjUiI8IkqpUSgQtLS2RkfHx/JYoLS0NeXl5JdZramrKHLpORERERET0KVMpEZwyZQqWLl2KlJQUmcPJKytXV1eZ59reZm5uLnMOIRERERER0adMpURw0KBB+Ouvv2BnZ4cFCxagTZs2JR7armhXzw8tMDAQz549K7FeR0fnA0ZDRERERESkXiovDRUIBCguLi71sHGBQICCggKVg6sobdu2VXcIRERERERElYZKieCYMWMg4JOpRERERERqpcGv5KQilRLB4ODgCg6DiIiIiIiIPhQeWEZERERERFTFqDQj+FpSUhJ27dqFuLg4ZGdnY/PmzQCAJ0+e4O7du7CxseFGLFQiPWG5fvw+WbramuoOoVLKKyxSdwiVTnHx//05J5+fz9sEAqC6Jn/f+aY/rz9SdwiVUjfLyr8DujqY6euqOwQieo9U/ib+yy+/YPbs2cjNzQXwamOY14lgSkoKOnXqhPXr12PixIkVEykREREREcnQ4L4dpCKVflW6b98+TJ06FTY2NoiKisKUKVNk6j///HPY2tpiz549FREjERERERERVSCVZgR/+uknNGzYEMeOHUONGjVw4cIFuTY2Njb466+/yh0gERERERERVSyVZgQvX76MAQMGoEaNGiW2qV+/PpKTk1UOjIiIiIiIiN4PlWYEi4qKUL169VLbpKSkQFtbW6WgiIiIiIjo3fiIIKlKpRnBZs2albrss6CgACdOnICNjY3KgX0IwcHBEAgE0ksoFMLU1BSOjo5Ys2YNMjMzS71/4sSJEAgEcHJyKvPY48aNg0AggIGBAXJycuTq4+PjpXGtWLECAGBhYSETb0nX63Mes7KysHDhQlhbW6NGjRqoXbs2WrVqhRkzZuDRI9md406ePIl+/fqhfv36EAqFaNiwIQYOHIiwsDCZdpGRkRg1ahSsrKwgEAjQo0ePMr93IiIiIiJSL5VmBEeOHIlvv/0WixcvxsKFC2XqCgsL8e233+LOnTuYO3duhQT5vvn5+cHS0hL5+flISkpCbGwsvLy8EBQUhKioKNja2srdc/78eQQHB0MoFKo8brVq1fDixQvs27cPbm5uMnVisRhCoRAvX76Ulq1atQpZWVnS1wcOHEB4eDhWrlyJOnXqSMvt7OyQn5+Pbt26IS4uDmPHjsW0adOQlZWFf//9F2FhYRg8eDBMTU0BANu3b4e7u7s0SaxVqxbu3r2LEydOYOPGjRgxYoS0719//RUXLlxA+/bt8fTpU5XfOxERERERqY9KieC0adOwb98++Pn5SRMWAHBzc8P58+eRkJCAPn36YPz48RUa7PvSr18/tGvXTvra29sbEokETk5OcHZ2xo0bN2TOQywuLsb06dMxZswYHD16VOVxtbW10blzZ4SHh8slgmFhYRgwYAB27twpLXNxcZFpk5SUhPDwcLi4uMDCwkKmbvv27bh06RLEYrFMIgcAL1++RF5envT1okWL0KJFC/zvf/+DlpaWTNuUlBSZ1yEhIahfvz40NDRgbW1d1rdMRERERESVgEpLQ6tXr47Dhw/j+++/xwELAcEAAJxKSURBVNOnT3Ht2jUUFxdjx44dSEtLw9y5cxEVFQXBR7xo2cHBAb6+vrh37x5CQ0Nl6kJCQnDt2jUsXbq03OOMGDECBw8exPPnz6Vl586dQ3x8vFwCVxa3b98GAHTu3FmuTigUwsDAQKZt+/bt5ZJAADAykj1k18zMDBoaPKCZiIiIiOhjpvI3ei0tLSxduhSpqam4fv06Tp48iatXr+Lp06cICAhQmFR8bEaPHg0AiI6OlpZlZmZi7ty5mDdvHurVq1fuMVxdXSEQCLBr1y5pWVhYGJo3b442bdqo3K+5uTkA4Pfff0dxcfE72x49ehSJiYkqj0dEREREH56G4NO96P0q99SOQCBA8+bNYWdnB2tra2hqalZEXJVCgwYNIBKJpLNrwKvnCXV0dDBz5swKGUNfXx9OTk7STVmKiooQEREBDw+PcvXr4uKCZs2aYcGCBbC0tISnpye2bNkit9QTAObOnYsHDx6gcePGcHBwwIIFC3Dy5EkUFRWVKwYiIiIiIqqcuMbvHfT09KS7h968eROrV6/GTz/9VKFHY4wYMQKxsbFISkqCRCJBUlJSuZaFAoCOjg7Onj2LOXPmAHi1Q+r48eNhYmKCadOmITc3V9r2yy+/xKFDh9CjRw+cPHkS/v7+6Nq1K6ysrHD69OlyxUFERERERJWPSpvFNGrU6J1tNDQ0YGBggGbNmmHw4MFym6F8LLKysqTPyc2YMQN2dnYYMmRIhY7Rv39/6OvrIzIyEpcvX0b79u3RpEkTJCQklKtfkUiE5cuXY/ny5bh37x6OHj2KFStWYO3atRCJRFiyZIm0raOjIxwdHfHixQtcuHABkZGRWL9+PZycnBAXFyf3rGBZ5ebmyiSfAJCXWwgtnjVJRERERPTBqTQjWFRUhLy8PCQkJCAhIQGJiYl4+fIlEhMTpWUvX77ErVu3EBkZCQ8PD/Tp00dmp8qPQWJiItLT09GkSRNIJBIcOnQIM2bMkL7HhIQEFBQUICcnBwkJCcjIyFBpHG1tbbi6umLbtm3YvXt3uWcDFTE3N8eXX36JU6dOoWbNmhCLxQrb6erqomvXrli7di3mz5+PZ8+e4eDBg+UePyAgACKRSOba8POKcvdLREREVJUJPuH/o/dLpUTw8uXLMDExgYODA06fPo3c3Fw8evQIubm5OH36NHr27AlTU1Pcv38fN2/eRP/+/XH06FEEBgZWdPzvVUhICIBXs2X3798H8GpzF0tLS+n18OFDSCQSWFpaYsuWLSqPNWLECFy6dAmZmZkYPnx4hcSvSK1atdC4cWM8fvz4nW1fH6mhTNt38fb2Rnp6usw1adq35e6XiIiIiIjKTqWloXPnzkVubi6io6NljhIQCAT44osvcOjQIbRu3Rrff/891q9fj+3bt6NFixaIiIiAt7d3hQX/PkkkEvj7+8PS0hIjR45ESkoKdu/eLddu0qRJMDc3h4+PD2xsbFQez97eHv7+/qhdu3aF7EZ65coV1K9fX+ageQC4d+8erl+/jmbNmknLjh49ip49e8r1ceDAAQCQaasqbW1tuecqtbJflLtfIiIiIiIqO5USwb1792LcuHElnienqamJ/v37Y9u2bVi/fj2EQiEcHBwQHh5ermDfl4MHDyIuLg4FBQVITk6GRCJBTEwMzM3NERUVBaFQiIYNG6Jhw4Zy93p5ecHY2FjusPey0tDQwPz588vVx5tiYmKwcOFCODs744svvoCenh7u3LmDLVu2IDc3F4sWLZK2HTRoECwtLTFw4EA0btwY2dnZOHLkCPbt24f27dtj4MCB0rYnTpzAiRMnAABPnjxBdna29FnDbt26oVu3bhX2HoiIiIiI6P1QKRHMyMh45/Nwr5f/vfb2zFRlsmDBAgCvzkY0NDSEjY0NVq1aBU9PT+jr66s5OtUMGTIEmZmZiI6OhkQiQVpaGmrVqoUOHTpg9uzZsLe3l7bdtGkT9u7diz/++AOPHj1CcXExGjVqBB8fH8ydOxfVqv3fj4lEIsHixYtlxvL19QUALFy4kIkgERER0QfE8/ZIVYLid502rkC7du1w+/ZtXLhwQeEOonfu3EHbtm3RuHFjnD9/HsCrZ+BOnz5d7p0w6dMR95hLQxXR1f50zuKk98vUujGqPX6IrLr1sGUXj3p57UtXO+g9SUKhaX2k3rir7nAqlaO3ktUdQqXUzbJ8O2N/qlLSX6o7hEqrjYWBukOQWia5/e5GH6nvHRqrO4RPmkozgvPmzcPQoUPRqlUrTJgwAZ07d4aRkRFSUlJw6tQpbN68GVlZWZg3bx4AIC8vD9HR0ejTp0+FBk9ERERERERlp1Ii6Orqik2bNsHLywurVq3C6tWrpXXFxcXQ09PDb7/9BldXVwDAixcvsHnzZnz++ecVE3UllpaWVuoxGZqamqhbt+4HjIiIiIiI/h97dx4XVfX+AfwzoDAgMkayCiIoZiiKK+GCiiSgSOSCGqCipmYufF1DxQUttBKJNNevggJqboUbQYxm7mWaUaBpuaAsIrIpss7vD3/Mt2kGhGGZET7vXvdVc+5zz33uZObDOfccIpKlVCEIAJMnT8aoUaPw7bff4tdff0VeXh709fXRrVs3vPPOOxCJRNLYVq1a4Z133qmThNXdyJEj8cMPP1R63tLSktNjiYiIiIhIpZQuBAFAJBJhwoQJdZVLo7B+/Xo8efKk0vM6OjoNmA0RERERNWZcLIaUVatCkOT17NlT1SkQERERERFVSfFGgERERERERNRosRAkIiIiIiJqYjg1lIiIiIjoFSUQ8CVBUg4LQVKZ4tJyVaeglkrL+L0oUlYuUXUKasdE8uI7kQDg1/M/FV9FaZkEf2c+VWku6qaP+euqTkEtFTwvVXUKakmTq5AQNWqcGkpERERERNTEsBAkIiIiIiJqYmpVCB45cgTvvPMO2rZtC5FIhLZt28LLywvffPNNHaVHRERERESV0RA03oPql1LvCJaWluK9997DoUOHIJFI0KxZM7z++utIT09HbGwsjh49ilGjRiEmJgbNmvE1RCIiIiIiInWi1IhgSEgIDh48iAEDBuDHH3/E8+fPkZaWhufPn+PMmTPo378/Dh06hLVr19Z1vkRERERERFRLShWCu3btQqdOnfD999+jX79+0NB40Y2Ghgb69++P77//Hh07dsTOnTvrNNmaioiIgEAgkB5CoRBmZmZwdXVFeHg48vPzq7z+/fffh0AggIeHh1L3P3r0KAYOHAgjIyPo6urC2toa3t7eiIuLk4nbvHkzxowZg7Zt20IgEGDSpEmV9pmTk4Np06bB0NAQLVq0wODBg/HLL78ojI2NjUWPHj0gFArRtm1brFixAqWlla+MtmjRIggEAowdO7Zaz3f27Fnpd5uVlVWta4iIiIiISPWUKgTT0tIwYsSISqd9Nm/eHCNGjEBaWlqtkqsrwcHB2LNnDzZv3ozZs2cDAAICAmBnZ4fr168rvObnn39GREQEhEKhUvf8/PPP4enpCYFAgMDAQGzYsAGjRo3Cn3/+iX379snErlu3DmKxGJ07d65yKm15eTmGDx+OmJgYzJo1C59++ikyMzMxaNAg/PnnnzKxJ0+ehJeXF1q1aoUvv/wSXl5eWLNmjfT5/00ikWDv3r1o164djh49+tIiuby8HLNnz0aLFi2q+Y0QEREREZG6UOoFPgsLCxQUFFQZ8/TpU7Rt21appOqau7s7evXqJf0cGBgIsVgMDw8PeHp6Ijk5GTo6OtLzEokEc+bMwYQJE5CYmFjj+5WWlmL16tV4++23ER8fL3c+MzNT5vMPP/wgHQ3U09OrtN+DBw/i/PnzOHDgAEaPHg0A8Pb2RseOHbFixQrExMRIYxcsWICuXbsiPj5eWlzq6+vjk08+wdy5c9GpUyeZvk+fPo3U1FSIxWK4urri8OHDmDhxYqW5bNu2Dffv38fUqVPxxRdfvPxLISIiIqI6x/3kSVlKjQhOnToVX3/9daUjfg8ePMD+/fsxderUWiVXn5ydnREUFIS7d+8iKipK5tyePXuQlJSEjz/+WKm+s7KykJeXh379+ik8b2RkJPPZ0tISgmr8V3zw4EEYGxtj5MiR0jZDQ0N4e3vj22+/RVFREQDgjz/+wB9//IFp06bJjDDOnDkTEokEBw8elOs7Ojoatra2GDx4MFxcXBAdHV1pHtnZ2Vi2bBmCg4PRqlWrl+ZNRERERETqRalC0NvbG3379kX37t2xdu1anD17Fn/++SfOnj2LkJAQ9OzZE/3798eYMWNw7949mUOd+Pn5AYDMqF1+fj4WL16MJUuWwMTERKl+jYyMoKOjg6NHjyI7O7tOcgWAq1evokePHtJ3Miv06dMHz549w82bN6VxAGRGQQHAzMwM5ubm0vMVioqKcOjQIYwfPx4AMH78eIjFYqSnpyvMIygoCCYmJpg+fXqdPBcRERERETUspaaGWltbQyAQQCKRYOnSpXLnJRIJjh49iqNHj8q0CwSCKhcraWjm5uYQiUS4ffu2tC04OBg6Ojr4z3/+o3S/GhoaWLhwIYKDg9G2bVs4OTmhf//+cHNzQ48ePZTuNy0tDU5OTnLtpqamAICHDx/Czs5OOlJb0f7v2IcPH8q0HTt2DDk5ORg3bhwAwMvLC9OmTcO+ffsQEBAgE3v9+nVs3boVJ06cgKamptLPQkREREREqqNUIThhwoRqTWV8Fejp6UkXRrl58ya++OIL7N27F9ra2rXqd9WqVejUqRO++uorfPfddzh58iSWLl2K7t27Izo6Gm+++WaN+ywsLFSYV8WCNoWFhTJ/ryw2Ly9Ppi06Ohq9evVChw4dAAAtW7bE8OHDER0dLVcIzpkzB+7u7hg6dGiN8yciIiKiuqXRSP5MTg1PqUIwIiKijtNQnYKCAuk7e3PnzkXfvn0xatSoOul7/PjxGD9+PPLy8nDp0iVEREQgJiYGI0aMQFJSUo1XJNXR0ZG+B/hPz58/l57/598ri/3nwjg5OTk4ceIEZs2ahVu3bknb+/Xrh0OHDuHmzZvo2LEjAGD//v04f/48kpKSapR3RS7/zqe4qARatSy4iYiIiIio5pR6R7CxSE1NRW5uLjp06ACxWIy4uDjMnTsXd+7ckR6lpaUoLCzEnTt35EbSqktfXx9vv/02oqOjMXHiRNy+fRuXLl2qcT+mpqYKF+ipaDMzM5PG/bP937EVcQBw4MABFBUVYf369bCxsZEe8+bNAwCZRWMWLlyIMWPGQEtLS/r95OTkAADu378vN+X0n0JCQiASiWSO/25aX8NvgIiIiIiI6oJSI4L/dO7cOVy7dg15eXnQ19eHvb19patlqps9e/YAAFxdXaUL2fxzRc4KDx48gJWVFTZs2CA3VbKmevXqhcjISKX2WLS3t8ePP/6I8vJymQVjLl26BF1dXenInb29PYAXeyH26dNHGvfw4UOkpqZi2rRp0rbo6Gh06dIFK1askLvf1q1bERMTg1WrVgF4UezFxMTIbFNRoUePHujWrRuuXbumMPfAwEBpcVnhZmZJ9R6ciIiIiIjqlNKF4Pnz5+Hv7y+dTiiRSKTvDdrY2GDXrl1wdHSsmyzrgVgsxurVq2FlZQUfHx9kZmbiyJEjcnHTpk2DpaUlli5dCjs7u2r1/ezZM/z6668Kn//kyZMAgDfeeKPGOY8ePRoHDx7E4cOHpfsIZmVl4cCBAxgxYoT0ncDOnTujU6dO2LZtG6ZPny5d1GXz5s0QCATSa+/fv48zZ85g1apV0rZ/Ki4uho+PDy5dugQHBweF38++ffuwf/9+7N69G+bm5pXmrq2tLffOolZu1XtREhEREVHVNPiKIClJqULw999/x9ChQ/Hs2TO8/fbbGDx4MExNTZGeno5Tp04hPj4erq6uuHjxImxtbes65xo7efIkUlJSUFpaioyMDIjFYiQkJMDS0hKxsbEQCoVo27Yt2rZtK3dtQEAAjI2N4eXlVe37PXv2DH379sVbb70FNzc3WFhYICcnB9988w1+/PFHeHl5oXv37tL4o0eP4tdffwUAlJSU4Pr161izZg0AwNPTE127dgXwohB866234O/vjz/++AOtW7fGV199hbKyMumoXYXPPvsMnp6eGDp0KMaNG4ekpCRs3LgRU6dOlS5UExMTA4lEAk9PT4XPMWzYMDRr1gzR0dFwcHBQ+B1UjAC6u7ujdevW1f6OiIiIiIhIdZQqBIODg1FcXIwTJ07Azc1N5tzixYsRFxcHT09PBAcHY9++fXWSaG0sX74cAKClpQUDAwPY2dkhLCwM/v7+aNmyZZ3fr1WrVti+fTuOHz+OXbt2IT09HZqamnjjjTfw2WefYc6cOTLxhw4dQmRkpPTz1atXpXv9mZubSwtBTU1NnDhxAgsXLkR4eDgKCwvRu3dvREREyI0wenh44PDhw1i1ahVmz54NQ0NDLFmyRPpdAC+mhbZt2xbdunWr9Dn69++P/fv3IzQ0VGZzeiIiIiIienUJJBKJpKYXGRsbY8iQIQrfFavw3nvvITExERkZGbVKkBqv6/c5NVQRTvFQrKy8xr9VNXpvOnSCVvpD5Bua4L+Hzqs6HbUxZVRftHyUjiJjM1w9+7uq01ErrfW5UrMipWX8/UWRopIyVaegtrq1rfuBBGWFn/1b1SnUmzn9rVSdQqOm1Kqhubm5sLKq+l+MlZUVcnNzlUqKiIiIiIiopjZt2oR27dpBKBTCwcEBly9frjL+wIED6NSpE4RCIezs7HDixAmZ8xKJBMuXL4epqSl0dHTg4uKCP//8UyYmOzsbPj4+0NfXR6tWrTBlyhQUFBTI9fP555+jY8eO0NbWRps2bfDxxx/XzUMrSalC0MzMDBcvXqwy5tKlSzLbFDQW2dnZSE9Pr/R49OiRqlMkIiIioiZCIGi8R03t378f8+bNw4oVK/DLL7+gW7ducHV1RWZmpsL48+fPY/z48ZgyZQquXr0KLy8veHl5yeyZ/emnnyI8PBxbtmzBpUuX0KJFC7i6ukr38QYAHx8f/P7770hISMCxY8dw5swZmVX6gRf7le/YsQOff/45UlJSEBsbK7O6vyooNTU0ICAAX375JZYsWYKlS5fKbIz+/PlzhISEYM2aNZgzZw42bNhQpwmr2qBBg/DDDz9Uet7S0hJ37txpuIReYZwaqhinhirGqaHyODVUMU4NrRynhirGqaGKcWpo5dRpauiX5xrv1NDZ/Wo2NdTBwQG9e/fGxo0bAQDl5eWwsLDA7Nmz8dFHH8nFjx07Fk+fPsWxY8ekbW+99Rbs7e2xZcsWSCQSmJmZYf78+ViwYAGAFzMjjY2NERERgXHjxiE5ORm2trb46aef0KtXLwBAXFwchg0bhtTUVJiZmSE5ORldu3ZFUlKSUjsH1Jdqr/6hqamJlStXIigoCEFBQTh27Bg++eQTbN26FX369IGxsTEyMjLw008/4dGjR7C2tkZQUFB95q4S69evx5MnTyo9r6Oj04DZEBERERE1TkVFRSgqKpJpU7QlGfBi27MrV64gMDBQ2qahoQEXFxdcuHBBYf8XLlyQ2+fa1dUV33zzDQDg77//Rnp6OlxcXKTnRSIRHBwccOHCBYwbNw4XLlxAq1atpEUgALi4uEBDQwOXLl3Cu+++i6NHj8La2hrHjh2Dm5sbJBIJXFxc8Omnn8LAwKDG30tdqXYhKJFIUDF4+Prrr+PixYtYtGgR9u3bJzOXVigUwt/fH+vWrVPpg9WXnj17qjoFIiIiIqJGLyQkRG6LtBUrVmDlypVysVlZWSgrK4OxsbFMu7GxMVJSUhT2n56erjA+PT1der6iraoYIyMjmfPNmjWDgYGBNOavv/7C3bt3ceDAAezevRtlZWX4z3/+g9GjR0MsFlf1FdQrpfcDaN26NXbu3ImtW7ciJSUFeXl50NfXR6dOndC8efO6zJGIiIiIiBTQQON9pyQwMFBuxE7RaKC6Ky8vR1FREXbv3o2OHTsCAP773/+iZ8+euHHjhsqmi9Z6Y7jmzZvDzs6uLnIhIiIiIiICUPk0UEVat24NTU1Nua3rMjIyYGJiovAaExOTKuMr/p6RkQFTU1OZGHt7e2nMvxejKS0tRXZ2tvR6U1NTNGvWTFoEAsCbb74JALh3757KCsEarRoqUGb5HiIiIiIionqkpaWFnj17IjExUdpWXl6OxMREODo6KrzG0dFRJh4AEhISpPFWVlYwMTGRicnLy8OlS5ekMY6OjsjJycGVK1ekMWKxGOXl5XBwcAAA9OvXD6Wlpbh9+7Y05ubNmwBeLDSpKjUaEVy5cqXCObmVEQgEKC0trWlO1ETcz32m6hTUklGLV2/KQ0N4w1R9VmhTF83+f4nZkrJyJD18quJs1EdJWTkAICs7F4PHLFNxNurl17hPVZ2CWjLiaqoKRVy5q+oU1JY6rRpK/zNv3jxMnDgRvXr1Qp8+fRAWFoanT5/C398fADBhwgS0adMGISEhAF5s6TBw4ECsX78ew4cPx759+/Dzzz9j27ZtAF7UMgEBAVizZg1sbGxgZWWFoKAgmJmZwcvLC8CLkT03Nze8//772LJlC0pKSjBr1iyMGzdOupWei4sLevTogcmTJyMsLAzl5eX48MMP8fbbb8uMEja0GhWCFZskEhERERGR6nHC3v+MHTsWjx49wvLly5Geng57e3vExcVJF3u5d+8eNDT+NyGyb9++iImJwbJly7BkyRLY2Njgm2++QZcuXaQxixYtwtOnTzFt2jTk5OSgf//+iIuLk9k+Lzo6GrNmzcKQIUOgoaGBUaNGITw8XHpeQ0MDR48exezZs+Hk5IQWLVrA3d0d69evb4BvpXLV3kdQQ0MDK1euxPLly+s7J2oijicp3tyzqeOIoGIcEZSn174tNB4+QLaBERZ9lfjyC5qIT2cOgUF2Jh40b4EOnSepOh21whFBxVrr8fddRTgiWLl5TtaqTkHqq/N3VJ1CvZnZt52qU2jUavSOIBEREREREb36WAgSERERERE1MSwEXyIiIgICgUB6CIVCmJmZwdXVFeHh4cjPz6/y+vfffx8CgQAeHh7VvmdJSQlat26N/v37VxojkUhgYWGBHj16yLRnZGRgwYIF6NSpE3R1ddGiRQv07NkTa9asQU5OjsK++vTpA4FAgM2bNys8X/EdCIVCPHjwQO78oEGDZOZSExERERGRemMhWE3BwcHYs2cPNm/ejNmzZwMAAgICYGdnh+vXryu85ueff0ZERITMy6TV0bx5c4wZMwbnz5/H3buK5+efOXMGqamp8PX1lbb99NNP6NKlCzZt2oQBAwYgNDQU69evR/fu3bF27Vp4e3vL9fPnn3/ip59+Qrt27RAdHV1lXkVFRVi7dm2NnoWIiIiI6o+GoPEeVL+qvWpoeXl5feah9tzd3dGrVy/p58DAQIjFYnh4eMDT0xPJycnQ0dGRnpdIJJgzZw4mTJggtz9Jdfj4+GDLli3Yu3cvPvroI7nzMTEx0NDQwLhx4wAAOTk5ePfdd6GpqYmrV6+iU6dOMvEff/wxtm/fLtdPVFQUjIyMsH79eowePRp37txBu3btFOZkb2+P7du3IzAwULocLhERERERvXo4IlgLzs7OCAoKwt27dxEVFSVzbs+ePUhKSsLHH3+sVN/9+vVDu3btEBMTI3eupKQEBw8exODBg6UF2datW/HgwQOEhobKFYEAYGxsjGXL5PfTiomJwejRo+Hh4QGRSKTwfhWWLFmCsrIyjgoSEREREb3iWAjWkp+fHwAgPj5e2pafn4/FixdjyZIlMDExUapfgUCA9957D7/99ht+//13mXNxcXHIzs6Gj4+PtC02NhY6OjoYPXp0te9x6dIl3Lp1C+PHj4eWlhZGjhxZ5fRQKysrTJgwAdu3b8fDhw9r/lBERERERKQWWAjWkrm5OUQiEW7fvi1tCw4Oho6ODv7zn//Uqu+KQu/fxVlMTAyEQiFGjRolbUtOTkbHjh2hpaVV7f6joqJgYWGBfv36AQDGjRuHP/74A9euXav0mqVLl6K0tBTr1q2rwZMQERERUX3QEAga7UH1i4VgHdDT05OuHnrz5k188cUX+Oyzz6CtXbsNam1tbdG9e3fs27dP2vb06VPExsbCw8MD+vr60va8vDy0bFn9DbdLS0uxf/9+jB07FoL//w/N2dkZRkZGVY4KWltbw8/PD9u2bUNaWpoST0VERERERKrGQrAOFBQUSIuwuXPnom/fvjKjdbXh4+ODv//+G+fPnwcAfPPNN3j27JnMtFAA0NfXf+lWFv8UHx+PR48eoU+fPrh16xZu3bqFv//+G4MHD8bevXurXBxo2bJlKC0trdG7gkVFRcjLy5M5SoqLqn09ERERERHVHRaCtZSamorc3Fx06NABYrEYcXFxmDt3Lu7cuSM9SktLUVhYiDt37iAvL69G/Y8fPx4aGhrSRVxiYmLw2muvYdiwYTJxnTp1ws2bN1FcXFytfitG/by9vWFjYyM99u/fjwcPHuCHH36o9Fpra2v4+vrWaFQwJCQEIpFI5vh6R3i1riUiIiIiorrFQrCW9uzZAwBwdXXFvXv3AAAjR46ElZWV9Hjw4AHEYjGsrKywc+fOGvVvZmaGwYMH48CBA8jIyEBCQgJGjx4t9y7giBEjUFhYiEOHDr20z6dPn+Lbb7/F2LFjceDAAbnD1NT0pXsKVowKVvddwcDAQOTm5soc3lPnVOtaIiIiIlJMIGi8B9Wvau8jSPLEYjFWr14NKysr+Pj4IDMzE0eOHJGLmzZtGiwtLbF06VLY2dnV+D4+Pj6YPHkypk+fjpKSErlpoQAwY8YMfPnll5g/fz569uyJjh07ypzPzMzEtm3bsGzZMhw5cgRPnz7Fhx9+iAEDBsj1FR8fjwMHDmDTpk2VvufYvn17+Pr6YuvWrbC0tESzZlX/UtLW1pbrq7nW85c9OhERERER1QMWgtV08uRJpKSkoLS0FBkZGRCLxUhISIClpSViY2MhFArRtm1btG3bVu7agIAAGBsbw8vLS6l7jxo1CjNnzsS3334LCwsLODk5ycW89tprOHLkCIYNGwZ7e3v4+vqiZ8+eAIBffvkFe/fuhaOjI4AX00Jff/119O3bV+H9PD09sX37dhw/fhwjR46sNK+lS5diz549uHHjBjp37qzUsxERERERUcNjIVhNy5cvBwBoaWnBwMAAdnZ2CAsLg7+/f41W61SGvr4+RowYgQMHDmD8+PHSVT7/zcHBAUlJSfjss89w/Phx7NmzBxoaGnjzzTfx0UcfYdasWcjMzMT333+P8ePHQ1NTU2E/Q4YMga6uLqKioqosBDt06ABfX19ERkbWyXMSEREREVHDEEgkEomqk6Cm6XhSpqpTUEtGLWq37Uhj9YZp/f7A5VWk174tNB4+QLaBERZ9lajqdNTGpzOHwCA7Ew+at0CHzpNUnY5a+TXuU1WnoJZa6/H3XUUirtxVdQpqa56TtapTkNp+qfH+e3rfwVLVKTRqHBEkIiIiInpFceN1UhYLQRXIzs6ucpsHTU1NGBoaNmBGRERERETUlLAQVIGRI0dWuU+fpaUl7ty503AJERERERFRk8JCUAXWr1+PJ0+eVHpeR0enAbMhIiIiIqKmhoWgClRs60BEREREVBt8RZCUpaHqBIiIiIiIiKhhsRAkIiIiIiJqYjg1lIjUnq3nIOhkP1J1GmpHkJ4GAHgtOxOfzhyi4mzUR6snWapOgYiISO2xECSV6W7eStUpqKWycomqU1A72o8fQSPtgarTUFsCAAbZmapOQ+0Yvi7ChW9DVJ2GWmn7uq6qU1BLd7OeqToFteTb3ULVKVA1cHofKYuFIBG9MiQaGigyNFZ1GmpDOyMNArz4XspNTFWdjlopK5eglL9WiIiIKsVCkIheGUWGxjid+Kuq01Abg4Z0gzAjDeUmpsi5eUfV6aiV+48LVZ0CERGRWuNoMhERERERURPDQpCIiIiIiKiJ4dRQIiIiIqJXlIA7ypOSOCL4EhERERAIBNJDKBTCzMwMrq6uCA8PR35+vkz8mTNn4OnpCQsLCwiFQpiYmMDNzQ3nzp2r8b1XrlwJgUAAY2NjPHsmv6JZu3bt4OHhIdMmEAgwa9YsudhPPvkEAoEAkydPRnl5ubQ9Ly8PH3/8MXr16gWRSARtbW1YWlpi7NixOH78uMJ8KjuUeUYiIiIiImp4HBGspuDgYFhZWaGkpATp6ek4ffo0AgICEBoaitjYWHTt2hUAcPPmTWhoaGDGjBkwMTHBkydPEBUVBScnJxw/fhxubm41vndmZiY2b96M+fPnK5X72rVrsXTpUkycOBE7duyAhsaL+v/WrVtwdXXF3bt38e6772LChAnQ09PD/fv3ceLECXh4eGD37t3w8/MDAIwcORIdOnSQ63/JkiUoKChA7969lcqPiIiIiIgaFgvBanJ3d0evXr2knwMDAyEWi+Hh4QFPT08kJydDR0cHU6dOxdSpU2WunTlzJqytrREWFqZUIWhvb4/PPvsMM2fOhI6OTo2u/eyzzxAYGIgJEyZg586d0iKwtLQU7777LjIyMvDDDz+gX79+MtetWLEC8fHxKCsrk7Z17dpVWvBWuH//PlJTUzF16lRoaWnV+NmIiIiIiKjhcWpoLTg7OyMoKAh3795FVFRUpXG6urowNDRETk6OUvdZvnw5MjIysHnz5hpdFxoaikWLFsHX1xe7du2SFoEAcODAASQlJSEoKEiuCKwwdOhQuLu7V3mPvXv3QiKRwMfHp0a5EREREVHtCRrxQfWLhWAtVUybjI+Pl2nPy8tDVlYWUlJSsGTJEiQlJWHIkCFK3WPAgAFwdnbGp59+isLC6u2N9cUXX2D+/Pl47733EBERIVMEAsDRo0cBAL6+vkrlVCE6OhoWFhZwcnKqVT9ERERERNRwODW0lszNzSESiXD79m2Zdm9vb3z33XcAAC0tLUyfPh1BQUFK32fFihUYOHAgtmzZgv/85z9Vxh47dgx3797F+PHjsXv3bmhqasrFpKSkoFWrVmjTpo1M+9OnT2WKTS0tLejr6yu8z++//47r169j0aJFXLGKiIiIiOgVwhHBOqCnpye3eujatWsRHx+P//73v3jrrbdQXFyM0tJSpe/h5OSEwYMHV2tUMCMjAwBgZWWlsAgEXoxY6unpybUvXboUhoaG0uO9996r9D7R0dEAwGmhRERERESvGI4I1oGCggIYGRnJtNnb20v/2dfXFz169MCkSZNw8OBBpe+zcuXKao0KTpw4EQ8fPsQnn3yC1q1bK4xt2bIlHj9+LNc+c+ZM6ZYUVU0blUgkiImJQZcuXeQWkFGkqKgIRUVF/2oTQFtb+6XXEhEREZFiGpyVRUriiGAtpaamIjc3V+G2ChW0tLTg6emJw4cPV/sdP0WcnJwwaNCgl44KNmvWDF9//TUGDhyI+fPnY9euXXIxnTp1Qk5ODh48eCDT3rFjR7i4uMDFxQVCobDSe5w7dw53796t9mhgSEgIRCKRzLFxw6fVupaIiIiIiOoWC8Fa2rNnDwDA1dW1yrjCwkJIJBK5KaQ1tXLlSqSnp2Pr1q1VxgmFQsTGxqJ79+54//33ceTIEZnzFaN+FdM7ayo6OhoCgaDKqaP/FBgYiNzcXJlj1n8WKXVvIiIiIiKqHRaCtSAWi7F69WpYWVlJR8YyMzPl4nJycnDo0CFYWFjITSGtqYEDB2LQoEFYt24dnj9/XmWsvr4+4uLi0KFDB4wfPx6JiYnSc97e3rC1tcXq1atx8eJFhddLJBKF7SUlJThw4AD69++Ptm3bVitvbW1t6OvryxycFkpEREREpBp8R7CaTp48iZSUFJSWliIjIwNisRgJCQmwtLREbGysdBqlu7s7zM3N4eDgACMjI9y7dw+7du3Cw4cPsX///jrJZcWKFRg8eHC1Yg0NDZGQkIB+/frBy8sLiYmJ6NOnD5o3b44jR47A1dUV/fv3x8iRIzFgwAC0aNECDx48QGxsLO7du4fhw4fL9fndd9/h8ePHXCSGiIiIiOgVxUKwmpYvXw7gxft+BgYGsLOzQ1hYGPz9/dGyZUtp3OTJk7Fv3z5s2LABOTk5eO211/DWW28hJiYGAwYMqJNcBg0ahIEDB+KHH36oVryFhQXi4+MxYMAAuLu748yZM+jcuTM6duyIa9euITw8HEeOHMHJkydRXFwMY2NjODg4YMWKFdIppP8UHR2N5s2bY8yYMXXyPERERESkHC4VQ8oSSCqb/0dUzx7mFKs6BbVUVs7/JP/NtEt7NEt7gOfGpjid+Kuq01Ebg4Z0gzAjDWVmbZBz846q01Er9x8rvzBXY9bJrOXLg5qgu1nPVJ2CWnqtRXNVp6C2jFqqz3cTfSVV1SnUG5+e5qpOoVHjO4JERERERERNDKeGqkBubu5Lt5EwMTFpoGyIiIiIiKipYSGoAnPnzkVkZGSVMZyxS0REREQvw/3kSVksBFVg0aJF8PX1VXUaRERERETURLEQVAFbW1vY2tqqOg0iIiIiImqiuFgMERERERFRE8MRQSIiIiKiV5SALwmSkjgiSERERERE1MRwRJBU5uGT56pOQS1xQ3l5xv//nUgAlIPfT4WKb+J5aRlOpqSrNBd1s2jbZVWnoJbOfzxM1SnQKyQ9p0jVKagtddpQnkhZHBEkIiIiIiJqYlgIEhERERERNTGcGkpERERE9IriqA4pi792iIiIiIiImhgWgkRERERERE1Mky0EIyIiIBAIpIdQKISZmRlcXV0RHh6O/Px8mfgzZ87A09MTFhYWEAqFMDExgZubG86dO1fje69cuVLm3hoaGjA1NYWHhwcuXrxY6XXJycnSXHNyciqNe/78OTZs2AAHBweIRCIIhUJ07NgRs2bNws2bN+XyyMrKUthPu3bt4OHhIf38+PFjfPbZZ3BycoKhoSFatWqFt956C/v376/xd0BERERERKrT5N8RDA4OhpWVFUpKSpCeno7Tp08jICAAoaGhiI2NRdeuXQEAN2/ehIaGBmbMmAETExM8efIEUVFRcHJywvHjx+Hm5lbje2/evBl6enooLy/H/fv3sX37djg5OeHy5cuwt7eXi4+KipLe++DBg5g6dapcTFZWFtzc3HDlyhV4eHjgvffeg56eHm7cuIF9+/Zh27ZtKC4urnGuAHDhwgUsXboUw4YNw7Jly9CsWTMcOnQI48aNwx9//IFVq1Yp1S8RERERKYcbypOymnwh6O7ujl69ekk/BwYGQiwWw8PDA56enkhOToaOjg6mTp0qV3jNnDkT1tbWCAsLU6oQHD16NFq3bi397OXlhS5duuDAgQNyhaBEIkFMTAzee+89/P3334iOjlZYCE6aNAlXr17FwYMHMWrUKJlzq1evxtKlS2ucZ4XOnTvjzz//hKWlpbRt5syZcHFxwbp167Bo0SK0aNFC6f6JiIiIiKhhNNmpoVVxdnZGUFAQ7t69i6ioqErjdHV1YWhoWOU0zZowMTEBADRrJl+fnzt3Dnfu3MG4ceMwbtw4nDlzBqmpqTIxly5dwvHjxzFlyhS5IhAAtLW18fnnnyudn5WVlUwRCLz4KZSXlxeKiorw119/Kd03ERERERE1HBaClfDz8wMAxMfHy7Tn5eUhKysLKSkpWLJkCZKSkjBkyBCl7pGdnY2srCxkZmbi6tWreP/99yEUCuHt7S0XGx0djfbt26N3794YMWIEdHV1sXfvXpmY2NhYmdxrmse/j/Ly8mpdn56eDgAyo5tERERERKS+mvzU0MqYm5tDJBLh9u3bMu3e3t747rvvAABaWlqYPn06goKClLrHG2+8IfO5VatW+Oabb9C5c2eZ9pKSEhw4cAAzZswAAOjo6MDT0xPR0dFYuHChNC45ORkAYGdnV6s8/qniHcnKZGdnY8eOHRgwYABMTU1rdF8iIiIiqh2+IUjKYiFYBT09PbnVQ9euXYv58+fj/v37iIyMRHFxMUpLS5Xq/9ChQ9DX14dEIsGDBw+wefNmjBo1CvHx8ejbt6807uTJk3j8+DHGjx8vbRs/fjxGjBiB33//XVo45uXlAQBatmypVB7/5uvrW+V15eXl8PHxQU5ODr788ssa3ZOIiIiIiFSHhWAVCgoKYGRkJNP2z0VcfH190aNHD0yaNAkHDx6scf9OTk4y0ylHjx4NGxsbzJ49G1euXJG2R0VFwcrKCtra2rh16xYAoH379tDV1UV0dDQ++eQTAJAWc/n5+WjVqpXSeVQQCoVVXjd79mzExcVh9+7d6NatW5WxRUVFKCoqkmkrLiqClrZ2tfMkIiIiIqK6wXcEK5Gamorc3Fx06NCh0hgtLS14enri8OHDKCwsrPU99fT04ODggF9++QVPnz4F8GKU7+jRo/j7779hY2MjPWxtbfHs2TPExMRAIpEAADp16gQA+O2332qdy8usWrUKX331FdauXVutdxJDQkIgEolkjojNofWeJxERERERyeOIYCX27NkDAHB1da0yrrCwEBKJBPn5+dDR0an1fSummRYUFKBFixY4fPgwnj9/js2bN8uN2t24cQPLli3DuXPn0L9/f4wYMQIhISGIiorCgAEDap1LZTZt2oSVK1ciICAAixcvrtY1gYGBmDdvnkxb0sOiSqKJiIiIiKg+sRBUQCwWY/Xq1bCysoKPjw8AIDMzU26aaE5ODg4dOgQLCwu5c8rIzs7G+fPnYWJiIu0vKioK1tbW0oVi/qmoqAhr165FdHQ0+vfvD0dHR7i5uWHHjh1wd3eHl5eXTHxxcTGWLFlSqy0k9u/fjzlz5sDHxwehodUf0dPW1ob2v6aBaj3OUzoPIiIiIuKG8qS8Jl8Injx5EikpKSgtLUVGRgbEYjESEhJgaWmJ2NhY6Xty7u7uMDc3h4ODA4yMjHDv3j3s2rULDx8+xP79+5W698GDB6GnpweJRIKHDx/iv//9L548eYItW7ZAIBDg4cOHOHXqFObMmaPwem1tbbi6uuLAgQMIDw9H8+bNsXv3bgwdOhQjR47EiBEjMGTIELRo0QJ//vkn9u3bh7S0NKULwcuXL2PChAl4/fXXMWTIEERHR8uc79u3L6ytrZXqm4iIiIiIGk6TLwSXL18O4MX7fgYGBrCzs0NYWBj8/f1lVt+cPHky9u3bhw0bNiAnJwevvfYa3nrrLcTExCg9DfODDz6Q/nOLFi3QtWtXfPzxxxgzZgwAYN++fSgvL8eIESMq7WPEiBE4dOgQTp48CU9PTxgaGuL8+fP46quvsH//fixduhTFxcWwtLSEp6cn5s6dq1SuAPDHH3+guLgYjx49wuTJk+XO79q1i4UgEREREdErQCCpWGmEqIH9/DenhipSVs7/JP+tez9baGU8RKGxKU4lXlN1Ompj8BB76GSk4amRCY6c+EnV6aiVRdsuqzoFtXT+42GqTkEtFZWWqzoFtVRUwu+lMl0t9FSdgtTBX9NUnUK9Gd2Ne1TXpyY/IkhERERE9KriFgCkLBaCdSg3N/el20iYmJg0UDZERERERESKsRCsQ3PnzkVkZGSVMZyJS0REREREqsZCsA4tWrQIvr6+qk6DiIiIiIioSiwE65CtrS1sbW1VnQYRERERNRHcR5CUxfdLiYiIiIiImhgWgkRERERERE0MC0EiIiIiIqImhu8IksqUlnGjWkWelZapOgW1U47/X21XAnDh3X/4/++imUADHUTqs7mxOjAyFak6BbXUQsj/7SsiLOdvLIrkF5aqOgUiqkf8PwIRERER0SuKS8WQsjg1lIiIiIiIqIlhIUhERERERNTEsBAkIiIiIiJqYviOIBERERHRK4r7yZOyOCJYCxERERAIBNJDKBTCzMwMrq6uCA8PR35+vkz8mTNn4OnpCQsLCwiFQpiYmMDNzQ3nzp2r8b1XrlwJgUCArKysSmNOnz4tk9+/j3379klj27VrV2nc8+fPAQAFBQVYsWIF3NzcYGBgAIFAgIiIiBrnTkREREREqsURwToQHBwMKysrlJSUID09HadPn0ZAQABCQ0MRGxuLrl27AgBu3rwJDQ0NzJgxAyYmJnjy5AmioqLg5OSE48ePw83NrV7ymzNnDnr37i3X7ujoKPPZ3t4e8+fPl4vT0tICAGRlZSE4OBht27ZFt27dcPr06XrJl4iIiIiI6hcLwTrg7u6OXr16ST8HBgZCLBbDw8MDnp6eSE5Oho6ODqZOnYqpU6fKXDtz5kxYW1sjLCys3grBAQMGYPTo0S+Na9OmDXx9fSs9b2pqirS0NJiYmODnn39WWFwSEREREZH649TQeuLs7IygoCDcvXsXUVFRlcbp6urC0NAQOTk5DZeckrS1tWFiYqLqNIiIiIjo/2lA0GgPql8sBOuRn58fACA+Pl6mPS8vD1lZWUhJScGSJUuQlJSEIUOG1Fse+fn5yMrKkjskEolMXElJiVzMs2fP6i0vIiIiIiJSDU4NrUfm5uYQiUS4ffu2TLu3tze+++47AC/ev5s+fTqCgoLqLY/JkycrbK+Y5lkhPj4ehoaGMjErVqzAypUr6y03IiIiIiJqeCwE65menp7c6qFr167F/Pnzcf/+fURGRqK4uBilpaX1lsPy5csxYMAAuXYDAwOZzw4ODlizZo1Mm7W1db3lRUREREREqsFCsJ4VFBTAyMhIps3e3l76z76+vujRowcmTZqEgwcP1ksOdnZ2cHFxeWlc69atqxWnjKKiIhQVFcm0FRcVQUtbu17uR0REREREleM7gvUoNTUVubm56NChQ6UxWlpa8PT0xOHDh1FYWNiA2TWskJAQiEQimWP31g2qTouIiIjolSYQNN6D6hdHBOvRnj17AACurq5VxhUWFkIikSA/Px86OjoNkVqDCwwMxLx582Tart1vvIUvEREREZE6YyFYT8RiMVavXg0rKyv4+PgAADIzM+Wmiebk5ODQoUOwsLCQO9eYaGtrQ/tf00C1tMtVlA0RERERUdPGQrAOnDx5EikpKSgtLUVGRgbEYjESEhJgaWmJ2NhYCIVCAC82njc3N4eDgwOMjIxw79497Nq1Cw8fPsT+/fuVundoaCh0dXVl2jQ0NLBkyRLp5x9//BHPnz+Xu7Zr167o2rVrje63ceNG5OTk4OHDhwCAo0ePIjU1FQAwe/ZsiESimj4CERERERE1MBaCdWD58uUAXrzvZ2BgADs7O4SFhcHf3x8tW7aUxk2ePBn79u3Dhg0bkJOTg9deew1vvfUWYmJiFK7qWR0hISFybZqamjKFYHh4uMJrV6xYUeNC8PPPP8fdu3elnw8fPozDhw8DeLHwDQtBIiIiooYj4MbrpCSB5N+7ihM1kIu3clSdglp6Vlqm6hTUTt9BdhBmpKHQyBTixGuqTkdtOA+xh05mGoqMzXD17O+qTketzIi6ouoU1FLCgoGqTkEtlZXzj0KK5BfW39ZWrzobY/VZ0+F4UqaqU6g3w7s03tem1AFXDSUiIiIiImpiODVUzeTm5r50GwkTE5MGyoaIiIiIiBojFoJqZu7cuYiMjKwyhrN5iYiIiIioNlgIqplFixbB19dX1WkQERER0SuAG6+TslgIqhlbW1vY2tqqOg0iIiIiImrEuFgMERERERFRE8NCkIiIiIiIqInh1FAiIiIioleUBjeUJyWxECSVMW4lVHUKaqm5Jn9D/7dmGv+bvCABV839t+aaArzZRl/VaaiVb2b3U3UKaqmFlqaqU1BLp24+UnUKasmmtZ6qUyCiesSpoURERERERE0MC0EiIiIiIqImhlNDiYiIiIheUdxHkJTFEUEiIiIiIqImhoUgERERERFRE8NCsIYiIiIgEAikh1AohJmZGVxdXREeHo78/HyZ+DNnzsDT0xMWFhYQCoUwMTGBm5sbzp07p3QO165dg6+vLywsLKCtrQ0DAwO4uLhg165dKCsrk4l9/vw5NmzYAAcHB4hEIgiFQnTs2BGzZs3CzZs3pXErV66EQCBAVlZWpfe9c+eOzLP/89i3b5/Sz0NERERERA2L7wgqKTg4GFZWVigpKUF6ejpOnz6NgIAAhIaGIjY2Fl27dgUA3Lx5ExoaGpgxYwZMTEzw5MkTREVFwcnJCcePH4ebm1uN7rtjxw7MmDEDxsbG8PPzg42NDfLz85GYmIgpU6YgLS0NS5YsAQBkZWXBzc0NV65cgYeHB9577z3o6enhxo0b2LdvH7Zt24bi4uIaP/v48eMxbNgwmTZHR8ca90NERERERKrBQlBJ7u7u6NWrl/RzYGAgxGIxPDw84OnpieTkZOjo6GDq1KmYOnWqzLUzZ86EtbU1wsLCalQIXrx4ETNmzICjoyNOnDiBli1bSs8FBATg559/RlJSkrRt0qRJuHr1Kg4ePIhRo0bJ9LV69WosXbq0po8NAOjRowd8fX2VupaIiIiI6g4XiyFlcWpoHXJ2dkZQUBDu3r2LqKioSuN0dXVhaGiInJycGvW/atUqCAQCREdHyxSBFXr16oVJkyYBAC5duoTjx49jypQpckUgAGhra+Pzzz+v0f3/6enTp0qNJhIRERERkeqxEKxjfn5+AID4+HiZ9ry8PGRlZSElJQVLlixBUlIShgwZUu1+nz17hsTERDg5OaFt27YvjY+NjZXJpy6tWrUKenp6EAqF6N27t9yzEhERERGReuPU0Dpmbm4OkUiE27dvy7R7e3vju+++AwBoaWlh+vTpCAoKqna/t27dQklJCezs7KoVn5ycDADVjq8ODQ0NDB06FO+++y7atGmDv/76C6GhoXB3d0dsbCyGDx9eZ/ciIiIiIqL6w0KwHujp6cmtHrp27VrMnz8f9+/fR2RkJIqLi1FaWlrtPvPy8gBA4ZTQuoivjrZt20qL2Qp+fn6wtbXF/PnzWQgSERERNTAB+JIgKYdTQ+tBQUGBXAFmb2+Pt99+G5MnT0ZCQgIuX74sfZ+vOvT19QFArsCsq3hlGRgYwN/fHzdu3EBqamqlcUVFRcjLy5M5ioqK6jU3IiIiIiJSjIVgHUtNTUVubi46dOhQaYyWlhY8PT1x+PBhFBYWVqvfDh06oFmzZvjtt9+qFd+pUycAqHZ8bVhYWAAAsrOzK40JCQmBSCSSOTZ/8Vm950ZERERERPJYCNaxPXv2AABcXV2rjCssLIREIqn2iJ2uri6cnZ1x5swZ3L9//6XxI0aMAIAqVy+tK3/99RcAwNDQsNKYwMBA5ObmyhwfzF1Y77kREREREZE8FoJ1SCwWY/Xq1bCysoKPjw8AIDMzUy4uJycHhw4dgoWFBYyMjKrd/4oVKyCRSODn54eCggK581euXEFkZCSAFxu8u7m5YceOHfjmm2/kYouLi7FgwYJq3xsAHj16JNf24MED7Ny5E127doWpqWml12pra0NfX1/m0NbWrtH9iYiIiEiWhqDxHlS/uFiMkk6ePImUlBSUlpYiIyMDYrEYCQkJsLS0RGxsLIRCIYAXG8+bm5vDwcEBRkZGuHfvHnbt2oWHDx9i//79Nbpn3759sWnTJsycOROdOnWCn58fbGxskJ+fj9OnTyM2NhZr1qyRxu/evRtDhw7FyJEjMWLECAwZMgQtWrTAn3/+iX379iEtLU1uL8HQ0FDo6urKtGloaGDJkiVYtGgRbt++jSFDhsDMzAx37tzB1q1b8fTpU3zxxRdKfpNERERERNTQWAgqafny5QBevO9nYGAAOzs7hIWFwd/fX2ahmMmTJ2Pfvn3YsGEDcnJy8Nprr+Gtt95CTEwMBgwYUOP7Tp8+Hb1798b69euxe/duPHr0CHp6eujRowd27doFX19faayhoSHOnz+Pr776Cvv378fSpUtRXFwMS0tLeHp6Yu7cuXL9h4SEyLVpampiyZIlGDp0KLZs2YJNmzbhyZMnaNWqFZycnLBs2TL06NGjxs9CRERERESqIZBIJBJVJ0FN099Zz1Wdglpqrsm5EP9m0rk9mqU9QKGRKRITr6o6HbUxZEh36GSmodysDfJv3VN1OmrlybNiVaegloxackq+Iqduyr/6QIBNaz1Vp6C2OprovjyogSSmZKk6hXozpFNrVafQqPEdQSIiIiIioiaGU0PVQG5u7ku3kTAxMWmgbIiIiIjoVcEN5UlZLATVwNy5c6WrfVaGM3iJiIiIiKiusBBUA4sWLZJZ5IWIiIiIiKg+sRBUA7a2trC1tVV1GkRERERE1ESwECQiIiIiekUJ+IogKYmrhhIRERERETUxHBEklSkv5wI4ipRr8Ed7VeG6SfIkAAqLy1Sdhlp5kF31SsxNlZE+9xFUxN68lapTUEv8/zRR48YRQSIiIiIioiaGI4JERERERK8o7iNIyuKIIBERERERURPDQpCIiIiIiKiJYSFIRERERETUxLAQVEJERAQEAoH0EAqFMDMzg6urK8LDw5Gfny8Tf+bMGXh6esLCwgJCoRAmJiZwc3PDuXPnlM7h+vXr8Pf3h5WVFYRCIfT09GBvb49Fixbhr7/+komdNGkSBAIBunbtComCJRcFAgFmzZol/Xznzh0IBAJ8/vnnMnFr166FQCDAd999pzCnYcOGQSQS4eHDh0o/FxERERER1T8uFlMLwcHBsLKyQklJCdLT03H69GkEBAQgNDQUsbGx6Nq1KwDg5s2b0NDQwIwZM2BiYoInT54gKioKTk5OOH78ONzc3Gp03+3bt+ODDz5A69at4ePjg06dOqG0tBRJSUnYvXs3wsLCUFhYCE1NTZnrfvvtNxw+fBijRo1S6nnnz5+PmJgYzJw5E0lJSdDR0ZGeO3DgAE6ePIlNmzbBzMxMqf6JiIiIqGa46xQpi4VgLbi7u6NXr17Sz4GBgRCLxfDw8ICnpyeSk5Oho6ODqVOnYurUqTLXzpw5E9bW1ggLC6tRIXj+/Hl88MEH6NevH44dO4aWLVvKnF+/fj0+/vhjuet0dHRgYWGB4OBgjBw5EgJBzX/XaN68ObZt24Z+/fph9erV+OSTTwAA+fn5CAgIwFtvvYUZM2bUuF8iIiIiImpYnBpax5ydnREUFIS7d+8iKiqq0jhdXV0YGhoiJyenRv2vWrUKAoEA0dHRckUgAAiFQqxevVpuNFBDQwPLli3D9evXceTIkRrd858qir3PP/8cf/zxBwBg2bJlyMzMxLZt26ChwV9SRERERETqjn9qrwd+fn4AgPj4eJn2vLw8ZGVlISUlBUuWLEFSUhKGDBlS7X6fPXsGsViMQYMGwdzcvMZ5vffee7CxsUFwcLDCdwWrKyQkBIaGhpg+fTquXLmCTZs2YcGCBbCzs1O6TyIiIiIiajgsBOuBubk5RCIRbt++LdPu7e0NQ0NDvPnmm1i/fj2mT5+OoKCgavd769YtlJaWokuXLnLnsrOzkZWVJT2Ki4vlYjQ1NbFs2TL8+uuv+Oabb2r8XBX09fURHh6Os2fPYujQobC0tMTy5cuV7o+IiIiIlCNoxH8pY9OmTWjXrh2EQiEcHBxw+fLlKuMPHDiATp06QSgUws7ODidOnJA5L5FIsHz5cpiamkJHRwcuLi74888/ZWKys7Ph4+MDfX19tGrVClOmTEFBQYHC+926dQstW7ZEq1atlHq+usRCsJ7o6enJrR66du1axMfH47///S/eeustFBcXo7S0tNp95uXlSfv+N2traxgaGkqP2NhYhX34+PjUyajgqFGjMGzYMGRnZ2PTpk0yC8cQERERETW0/fv3Y968eVixYgV++eUXdOvWDa6ursjMzFQYf/78eYwfPx5TpkzB1atX4eXlBS8vLyQlJUljPv30U4SHh2PLli24dOkSWrRoAVdXVzx//lwa4+Pjg99//x0JCQk4duwYzpw5g2nTpsndr6SkBOPHj8eAAQPq/uGVwEKwnhQUFMi9w2dvb4+3334bkydPRkJCAi5fvoxJkyZVu8+K/hT9hOHbb79FQkKC3JYP/1YxKnjt2rVajQoCQO/evQFAZsGcyhQVFSEvL0/mKCoqqtX9iYiIiIgqhIaG4v3334e/vz9sbW2xZcsW6OrqYufOnQrjv/jiC7i5uWHhwoV48803sXr1avTo0QMbN24E8GI0MCwsDMuWLcM777yDrl27Yvfu3Xj48KH0z9HJycmIi4vDjh074ODggP79++PLL7/Evn375LZUW7ZsGTp16gRvb+96/R6qi4VgPUhNTUVubi46dOhQaYyWlhY8PT1x+PBhFBYWVqvfDh06oFmzZjI/pagwcOBAuLi4oGfPni/tx8fHBx06dKj1qGBNhISEQCQSyRxbwj9rkHsTERERUeNWXFyMK1euwMXFRdqmoaEBFxcXXLhwQeE1Fy5ckIkHAFdXV2n833//jfT0dJkYkUgEBwcHacyFCxfQqlUrmYERFxcXaGho4NKlS9I2sViMAwcOYNOmTbV/2DrC7SPqwZ49ewC8+IVUlcLCQkgkEuTn51dramWLFi0waNAg/PDDD3jw4AHatGmjVH4Vo4KTJk3Ct99+q1QfNRUYGIh58+bJtKXmljfIvYmIiIgaKyV2BHtlFBUVyc0g09bWhra2tlxsVlYWysrKYGxsLNNubGyMlJQUhf2np6crjE9PT5eer2irKsbIyEjmfLNmzWBgYCCNefz4MSZNmoSoqCjo6+tX+cwNiSOCdUwsFmP16tWwsrKCj48PACicl5yTk4NDhw7BwsJC7hdPVZYvX46ysjL4+voqnCJa3RE+X19fdOjQAatWrar2vWtDW1sb+vr6Moei/4iJiIiIiADFM8pCQkJUnVaNvf/++3jvvffg5OSk6lRkcESwFk6ePImUlBSUlpYiIyMDYrEYCQkJsLS0RGxsLIRCIYAXG8+bm5vDwcEBRkZGuHfvHnbt2oWHDx9i//79NbrngAEDsHHjRsyePRs2Njbw8fFBp06dUFxcjJs3byI6OhpaWlowMTGpsh9NTU0sXboU/v7+lcYkJibKvAhbwcvLS+HKpUREREREdUXRjLLKBhJat24NTU1NZGRkyLRnZGRU+udiExOTKuMr/p6RkQFTU1OZGHt7e2nMvwd9SktLkZ2dLb1eLBYjNjZWupaHRCJBeXk5mjVrhm3btmHy5MmVfgf1iYVgLVRsmaClpQUDAwPY2dkhLCwM/v7+MgvFTJ48Gfv27cOGDRuQk5OD1157DW+99RZiYmKUWjXogw8+gKOjIzZs2IADBw4gPT0dzZs3R/v27TFx4kR88MEHaN++/Uv78fX1xZo1a+S2uagQFxeHuLg4ufZ27dqxECQiIiKielXZNFBFtLS00LNnTyQmJsLLywsAUF5ejsTERMyaNUvhNY6OjkhMTERAQIC0LSEhAY6OjgAAKysrmJiYIDExUVr45eXl4dKlS/jggw+kfeTk5ODKlSvStTrEYjHKy8vh4OAA4MV7hGVlZdJ7fPvtt1i3bh3Onz+v9KtedUEgaajVQoj+5XZm9RbJaWqaN+OM7X8z69IezdIeoNDIFN9/f1XV6agNF5fu0MlMQ5lZGzz6429Vp6NWbmcq3r+pqeverpWqU1BLT56WqDoFtVRezj8iVsbCQH1ebzn75xNVp1Bv+tu8VqP4/fv3Y+LEidi6dSv69OmDsLAwfP3110hJSYGxsTEmTJiANm3aSKeXnj9/HgMHDsTatWsxfPhw7Nu3D5988gl++eUX6aDHunXrsHbtWkRGRsLKygpBQUG4fv06/vjjD5nZfxkZGdiyZQtKSkrg7++PXr16ISYmRmGeERERCAgIQE5OjvJfTh3giCARERER0SuqEa8VU2Njx47Fo0ePsHz5cqSnp8Pe3h5xcXHSxV7u3bsHDY3//cC9b9++iImJwbJly7BkyRLY2Njgm2++kZn5tmjRIjx9+hTTpk1DTk4O+vfvj7i4OGkRCADR0dGYNWsWhgwZAg0NDYwaNQrh4eEN9+BK4oigmsjNzX3pNhIve+/vVcMRQcU4IiiPI4KKcUSwchwRVIwjgopxRFAxjghWTp1GBM814hHBfjUcEaSa4Yigmpg7dy4iIyOrjGHNTkREREREdYGFoJpYtGgRfH19VZ0GERERERE1ASwE1YStrS1sbW1VnQYRERERvUI0GvOO8lSv+DISERERERFRE8NCkIiIiIiIqInh1FAiNdJmaD80y8xUdRpqRzMjDQAgzEyDi0t3FWejPoRZGapOgYiIiF5RLARJZVIy81WdgtoxS0tHs8w0VaehtgQAdPj9yCkuK8cvqY13+XBlGAi1VJ2CWsouKFZ1CmrpMb8Xhb7/65GqU1Bb8wdaqzoFKb4hSMpiIUikhiQaGigyNFZ1GmpDOyMNArz4XsqMTVWdjlopKS9H8etGqk6DiIiIXjEsBInUUJGhMc6c+lXVaagNp8HdIMxIQ5mxKdKSbqs6HbXyW1quqlMgIiKiVxAXiyEiIiIiImpiWAgSERERERE1MZwaSkRERET0quJqMaQkjghWU0REBAQCgfQQCoUwMzODq6srwsPDkZ8vuwJmYmIiJk+ejI4dO0JXVxfW1taYOnUq0tKUX/Hw2rVr8PX1hYWFBbS1tWFgYAAXFxfs2rULZWVl0rh/5tmsWTMYGBigZ8+emDt3Lv7444+XPtu/j+jo6EpzevvttyEQCDBr1iyln4uIiIiIiBoWRwRrKDg4GFZWVigpKUF6ejpOnz6NgIAAhIaGIjY2Fl27dgUALF68GNnZ2RgzZgxsbGzw119/YePGjTh27BiuXbsGExOTGt13x44dmDFjBoyNjeHn5wcbGxvk5+cjMTERU6ZMQVpaGpYsWSKNf/vttzFhwgRIJBLk5ubi119/RWRkJL766iusW7cO8+bNk8Y6OTlhz549cvfcsGEDfv31VwwZMkRhTocPH8aFCxdq9BxERERERKR6LARryN3dHb169ZJ+DgwMhFgshoeHBzw9PZGcnAwdHR2Ehoaif//+0ND436Crm5sbBg4ciI0bN2LNmjXVvufFixcxY8YMODo64sSJE2jZsqX0XEBAAH7++WckJSXJXNOxY0f4+vrKtK1duxYjRozA/Pnz0alTJwwbNgwAYG1tDWtr2f1wCgsLMXPmTDg7OyssWp8/f4758+dj8eLFWL58ebWfhYiIiIiIVI9TQ+uAs7MzgoKCcPfuXURFRQF4Mcr2zyKwos3AwADJyck16n/VqlXSKZr/LAIr9OrVC5MmTXppP6+//jr27duHZs2a4eOPP64y9ujRo8jPz4ePj4/C859++inKy8uxYMGCaj0DEREREdU9QSP+i+oXC8E64ufnBwCIj4+vNKagoAAFBQVo3bp1tft99uwZEhMT4eTkhLZt29Y6z7Zt22LgwIG4ePEi8vLyKo2Ljo6Gjo4ORo4cKXfu3r17WLt2LdatWwcdHZ1a50RERERERA2LhWAdMTc3h0gkwu3blW92HRYWhuLiYowdO7ba/d66dQslJSWws7OrizQBAF26dEF5eTnu3Lmj8Hx2djbi4uIwYsQIhSOQ8+fPR/fu3TFu3Lg6y4mIiIiIiBoO3xGsQ3p6enKrh1Y4c+YMVq1aBW9vbzg7O1e7z4pRO0UFmbL09PQAoNJcDx48iOLiYoXTQk+dOoVDhw7h0qVLdZYPERERERE1LBaCdaigoABGRkZy7SkpKXj33XfRpUsX7Nixo0Z96uvrA6i8aFNGQUEBgMqLy+joaBgYGMDd3V2mvbS0FHPmzIGfnx969+5do3sWFRWhqKhIpq2kuAjNtbRr1A8RERER/Y+Ar9KRkjg1tI6kpqYiNzcXHTp0kGm/f/8+hg4dCpFIJLfiZ3V06NABzZo1w2+//VZnuSYlJUFTUxNWVlZy5+7du4cff/wRY8aMQfPmzWXO7d69Gzdu3MD06dNx584d6QG8KFTv3LmDZ8+eKbxnSEgIRCKRzPH1jvA6eyYiIiIiIqo+FoJ1pGIfPldXV2nb48ePMXToUBQVFeG7776DqalpjfvV1dWFs7Mzzpw5g/v379c6z3v37uGHH36Ao6OjwqJ07969kEgkCqeF3rt3DyUlJejXrx+srKykB/CiSLSysqp0sZzAwEDk5ubKHN5T59T6eYiIiIiIqOY4NbQOiMVirF69GlZWVtIC6unTpxg2bBgePHiAU6dOwcbGRun+V6xYgcTERPj5+eHYsWPSd/wqXLlyBUlJSZg4cWKV/WRnZ2P8+PEoKyvD0qVLFcbExMSgbdu26N+/v9y5cePGwd7eXq793XffxbBhw/D+++/DwcFBYb/a2trQ1padBtpc63mV+RIRERERUf1gIVhDJ0+eREpKCkpLS5GRkQGxWIyEhARYWloiNjYWQqEQAODj44PLly9j8uTJSE5Oltk7UE9PD15eXtW+Z9++fbFp0ybMnDkTnTp1gp+fH2xsbJCfn4/Tp08jNjZWboP6mzdvIioqChKJBHl5efj1119x4MABFBQUIDQ0FG5ubnL3SUpKwvXr1/HRRx9BoGDCeadOndCpUyeFOVpZWdXomYiIiIiISHVYCNbQ8uXLAQBaWlowMDCAnZ0dwsLC4O/vLzPV8tq1awCAnTt3YufOnTJ9WFpa1rhomj59Onr37o3169dj9+7dePToEfT09NCjRw/s2rULvr6+MvEJCQlISEiAhoYG9PX1YWVlhYkTJ2LatGmwtbVVeI/o6GgAwHvvvVej3IiIiIhINbhWDClLIJFIJKpOgpqm40mZqk5B7TgPsYdOZhqeG5vizKlfVZ2O2nAa3A3CjDSUmrZBWlLle3U2Rb+l5ao6BbVkINRSdQpqydxAR9UpqKXHBcWqTkEtff/XI1WnoLbmD7RWdQpSP/3VeP8/0NtapOoUGjUuFkNERERERNTEcGqoCuXm5qKwsLDKGBMTkwbKhoiIiIiImgoWgio0d+5cREZGVhnDmbtEREREVCm+JEhKYiGoQosWLZJb5IWIiIiIiKi+sRBUIVtb20pX8CQiIiIiIqovXCyGiIiIiIioieGIIBERERHRK0rAlwRJSRwRJCIiIiIiamI4IkgqU1xeruoU1BoXjJWnIQD0hPxt659sWrdUdQpqyWHBEVWnoJauh49WdQpqyVgkVHUKasnbro2qUyCiesQRQSIiIiIioiaGhSAREREREVETwzlWRERERESvKAHXiiElcUSQiIiIiIioiWEhSERERERE1MQ0ykIwIiICAoFAegiFQpiZmcHV1RXh4eHIz8+XiU9MTMTkyZPRsWNH6OrqwtraGlOnTkVaWlqN771y5UqZe+vq6sLW1hbLli1DXl6eXI4///xzpX3t2LEDAoEAkZGRcucuXLgADQ0NLFiwQNo2aNAg6X01NDSgr6+PN954A35+fkhISKj0PmVlZdi1axcGDRoEAwMDaGtro127dvD395fJ76effsKsWbPQuXNntGjRAm3btoW3tzdu3rxZ06+JiIiIiIhUqFG/IxgcHAwrKyuUlJQgPT0dp0+fRkBAAEJDQxEbG4uuXbsCABYvXozs7GyMGTMGNjY2+Ouvv7Bx40YcO3YM165dg4mJSY3vvXnzZujp6aGgoADx8fH4+OOPIRaLce7cOQiqOZl7ypQpiIyMxIIFC+Dh4YHXX38dAFBSUoJp06bBwsICq1atkrnG3NwcISEhAICnT5/i1q1bOHz4MKKiouDt7Y2oqCg0b95cGl9YWIiRI0ciLi4OTk5OWLJkCQwMDHDnzh18/fXXiIyMxL1792Bubo5169bh3LlzGDNmDLp27Yr09HRs3LgRPXr0wMWLF9GlS5caf09EREREpDy+IkjKatSFoLu7O3r16iX9HBgYCLFYDA8PD3h6eiI5ORk6OjoIDQ1F//79oaHxvwFSNzc3DBw4EBs3bsSaNWtqfO/Ro0ejdevWAIAZM2Zg1KhROHz4MC5evAhHR8dq9SEQCLB161bY29tjwYIF2LVrFwBg/fr1SEpKQmxsLFq0aCFzjUgkgq+vr0zb2rVrMWfOHHz11Vdo164d1q1bJz23cOFCxMXFYcOGDQgICJC5bsWKFdiwYYP087x58xATEwMtLS1p29ixY2FnZ4e1a9ciKiqqWs9FRERERESq1SinhlbF2dkZQUFBuHv3rrRwcXJykikCK9oMDAyQnJxcZ/cFgL///rtG19na2mLhwoWIiIjADz/8gL///hvBwcEYOXIkRowYUa0+NDU1ER4eDltbW2zcuBG5ubkAgNTUVGzduhVvv/22XBFYcd2CBQtgbm4OAOjbt69MEQgANjY26Ny5c519T0REREREVP+aXCEIAH5+fgCA+Pj4SmMKCgpQUFAgHdWrrdu3bwOAdHpnTSxbtgzW1taYPn06ZsyYgWbNmiE8PLxGfWhqamL8+PF49uwZzp49CwA4efIkSktLpd+HMiQSCTIyMurseyIiIiIiovrXqKeGVsbc3BwikUhanCkSFhaG4uJijB07Vql7ZGdnA4D0HcGvvvoKxsbGGDBgQI370tHRwebNm+Hq6oobN24gLCwMbdq0qXE/Fe/wVTx3xSienZ1djfuqEB0djQcPHiA4OFjpPoiIiIhISXxJkJTUJAtBANDT05NbPbTCmTNnsGrVKnh7e0undNbUG2+8IfO5c+fOiIyMhK6urlL9GRgYQENDA+Xl5Rg6dKhSfejp6QGA9LkrVjFt2bKlUv2lpKTgww8/hKOjIyZOnKhUH0RERERE1PCabCFYUFAAIyMjufaUlBS8++676NKlC3bs2KF0/4cOHYK+vj6aN28Oc3NztG/fXum+ysrKMG3aNJiZmaGgoABz5sypcjuIyhQUFAD4X+Gnr68PAJUWxFVJT0/H8OHDIRKJcPDgQWhqalYZX1RUhKKiIpm2kuIiNNfSrvG9iYiIiIiodprkO4KpqanIzc1Fhw4dZNrv37+PoUOHQiQS4cSJE0qPlAEvFptxcXHBwIEDa1UEAsAXX3yBq1evYuPGjfj444/x/fffIyYmpsb9JCUlAYD0uTt16gQA+O2332rUT25uLtzd3ZGTk4O4uDiYmZm99JqQkBCIRCKZ49B/v6zhExARERERUV1okoXgnj17AACurq7StsePH2Po0KEoKirCd999B1NTU1WlJ+P+/ftYsWIF3nnnHbzzzjuYMWMG+vTpg3nz5klX/6yOsrIyxMTEQFdXF/379wfwYnsNTU3NGm378Pz5c4wYMQI3b97EsWPHYGtrW63rAgMDkZubK3OMmjK72vclIiIiIqK60+QKQbFYjNWrV8PKygo+Pj4AXmy8PmzYMDx48AAnTpyAjY2NirP8n9mzZ0MikeDLL1+MnmloaGDLli3IysrCkiVLqtVHWVkZ5syZg+TkZMyZM0c6JdTCwgLvv/8+4uPjpf3/U3l5OdavX4/U1FRpP2PHjsWFCxdw4MCBau+HCADa2trQ19eXOTgtlIiIiKh2BI34L6pfjfodwZMnTyIlJQWlpaXIyMiAWCxGQkICLC0tERsbC6FQCADw8fHB5cuXMXnyZCQnJ8vsiaenpwcvL696y3Hnzp2Ii4uTa587dy6+//57fPvtt1i/fj0sLCyk57p3744PP/wQGzduxKRJk9C7d2/pudzcXOkI37Nnz3Dr1i0cPnwYt2/fxrhx47B69WqZ+6xfvx63b9/GnDlzcPjwYXh4eOC1117DvXv3cODAAaSkpGDcuHEAgPnz5yM2NhYjRoxAdna23EjivzeyJyIiIiIi9dSoC8Hly5cDALS0tGBgYAA7OzuEhYXB399f5v2/a9euAXhRlO3cuVOmD0tLy3otBDdv3qywfdKkSZgzZw7s7e0xd+5cufNr1qzBwYMHMWPGDFy+fFm6WEtqaqp0X0A9PT2YmprC0dERmzdvxttvvy3Xj66uLk6ePImIiAhERkZi9erVePbsGczMzODs7Izo6GjpVhUV39PRo0dx9OhRub5YCBIRERERvRoEEolEouokqGk6cj1d1SmoHbe3e0AnMw3PjU3xg/hXVaejNgY6d4MwIw3lZm2Q++ddVaejVrLyi1WdglpyWHBE1Smopevho1WdglpqpskpaIqUlJarOgW1ZWGgPq+3XL1b89XfXxXdLZVfuJFerlGPCBIRERERNWYC/hyDlMRCsJpyc3NRWFhYZYyJiUkDZUNERERERKQ8FoLVNHfuXERGRlYZw1m2RERERET0KmAhWE2LFi3iYihERERERNQosBCsJltb22pvnk5ERERE1BD4iiApq8ltKE9ERERERNTUsRAkIiIiIiJqYlgIEhERERERNTF8R5BURkuDP4eojEQCFJdxI98KFQvylkkk3ED9X+49earqFNRS1Ecuqk5BLZXw9xWFcp6WqjoFtXTyVqaqU1BbCwdZqzoFolpjIUhERERE9KriajGkJA7JEBERERERNTEsBImIiIiIiJoYFoJERERERERNDN8RJCIiIiJ6RQn4kiApiSOCVYiIiIBAIJAeQqEQZmZmcHV1RXh4OPLz82XiExMTMXnyZHTs2BG6urqwtrbG1KlTkZaWVqs8Fi1aBIFAgLFjx1YZ9/vvv8PX1xdt2rSBtrY2zMzM4OPjg99//73K67y9vSEQCLB48WKF50+fPg2BQICDBw8qPD9p0iTo6elV72GIiIiIiEjlWAhWQ3BwMPbs2YPNmzdj9uzZAICAgADY2dnh+vXr0rjFixfj9OnTePfddxEeHo5x48bh66+/Rvfu3ZGenq7UvSUSCfbu3Yt27drh6NGjcsVnhcOHD6NHjx5ITEyEv78/vvrqK0yZMgWnTp1Cjx49cOTIEYXX5eXl4ejRo2jXrh327t0LScU6/URERERE1Ghxamg1uLu7o1evXtLPgYGBEIvF8PDwgKenJ5KTk6Gjo4PQ0FD0798fGv/YH8/NzQ0DBw7Exo0bsWbNmhrf+/Tp00hNTYVYLIarqysOHz6MiRMnysTcvn0bfn5+sLa2xpkzZ2BoaCg9N3fuXAwYMAB+fn64fv06rK1l9705dOgQysrKsHPnTjg7O+PMmTMYOHBgjfMkIiIiIqJXB0cEleTs7IygoCDcvXsXUVFRAAAnJyeZIrCizcDAAMnJyUrdJzo6Gra2thg8eDBcXFwQHR0tF/PZZ5/h2bNn2LZtm0wRCACtW7fG1q1b8fTpU3z66acK+3/77bcxePBgvPnmmwr7JyIiIiL1JBA03oPqFwvBWvDz8wMAxMfHVxpTUFCAgoICtG7dusb9FxUV4dChQxg/fjwAYPz48RCLxXLTTCumdg4YMEBhP05OTmjXrh2OHz8u0/7w4UOcOnVKpv+DBw+iuLhYYT/5+fnIysqSO4qKimr8bEREREREpDosBGvB3NwcIpEIt2/frjQmLCwMxcXFL13oRZFjx44hJycH48aNAwB4eXmhefPm2LdvnzQmNzcXDx8+RLdu3arsq2vXrkhNTZV5x3Dv3r3Q1tbGO++8AwAYN24cnjx5ghMnTijsY/LkyTA0NJQ7/pkPERERERGpPxaCtaSnp1fpAi5nzpzBqlWr4O3tDWdn5xr3HR0djV69eqFDhw4AgJYtW2L48OEy0zcr7t2yZcsq+6o4n5eXJ9P/8OHDpedsbGzQs2fPSqeHLl++HAkJCXLH0KFDa/xsRERERESkOlwsppYKCgpgZGQk156SkoJ3330XXbp0wY4dO2rcb05ODk6cOIFZs2bh1q1b0vZ+/frh0KFDuHnzJjp27Cgt4iorRiv8u2BMTk7G1atXMWHCBJn+Bw0ahE2bNiEvLw/6+voyfdjZ2cHFxUWu74p3JKtSVFQkN4W0pLgIzbW0X3otERERERHVLY4I1kJqaipyc3OlI3YV7t+/j6FDh0IkEuHEiRMvHa1T5MCBAygqKsL69ethY2MjPebNmwcA0lE7kUgEU1NTmW0sFLl+/TratGkjLe4qirf//Oc/Mv2vX78ez58/x6FDh2qcc1VCQkIgEolkjq93hNfpPYiIiIiaGkEjPqh+cUSwFvbs2QMAcHV1lbY9fvwYQ4cORVFRERITE2FqaqpU39HR0ejSpQtWrFghd27r1q2IiYnBqlWrAAAeHh7Yvn07zp49i/79+8vF//jjj7hz5w6mT58O4MXehDExMRg8eDBmzpwpF7969WpER0fD399fqdwVCQwMlBaxFcS3cuusfyIiIiIiqj4WgkoSi8VYvXo1rKys4OPjAwB4+vQphg0bhgcPHuDUqVOwsbFRqu/79+9L3y8cPXq03Pni4mL4+Pjg0qVLcHBwwMKFCxEVFYXp06fjzJkzeP3116Wx2dnZmDFjBnR1dbFw4UIAwLlz53Dnzh0EBwcr7P/mzZsICgrCw4cPYWZmptQz/Ju2tja0tWWngTbXel4nfRMRERERUc2wEKyGkydPIiUlBaWlpcjIyIBYLEZCQgIsLS0RGxsLoVAIAPDx8cHly5cxefJkJCcny+wdqKenBy8vr2rdLyYmBhKJBJ6engrPDxs2DM2aNUN0dDQcHBxgY2ODyMhI+Pj4wM7ODlOmTIGVlRXu3LmD//73v8jKysLevXvRvn17AC9GGzU1NTF8+HCF/Xt6emLp0qXYt2+f3CgeERERERG9+lgIVsPy5csBAFpaWjAwMICdnR3CwsLg7+8v8/7ftWvXAAA7d+7Ezp07ZfqwtLSsdiEYHR2Ntm3bVrolRKtWrdC/f3/s378foaGhaNasGcaMGYNOnTohJCREWvy9/vrrGDx4MJYsWYIuXboAAEpKSnDgwAH07dsXBgYGCvvv0qULrKysEBUVxUKQiIiISJ3xZTpSkkAikUhUnQQ1TceTMlWdgtpxHmIPncw0FBqZ4vvvr6o6HbXh4tIdOplpKDE1w52rt15+QRNy78lTVaeglorKylWdglp601j/5UFN0NPnpapOQS2dvMX/T1dm4SBrVacglfSgQNUp1JsubfRUnUKjxlVDiYiIiIiImhhODW1gubm5KCwsrDLGxMSkgbIhIiIiIqKmiIVgA5s7dy4iIyOrjOFsXSIiIiKqDgFfEiQlsRBsYIsWLYKvr6+q0yAiIiIioiaMhWADs7W1ha2trarTICIiIiKiJoyLxRARERERETUxLASJiIiIiIiaGE4NJSIiIiJ6RQm4VgwpiYUgqUzW8yJVp6B2ysEVY6uiKRDgdT0tVaehVjQ1+CcARabE/KLqFNTSLt8eqk5BLZ1PfazqFNTS+G5tVJ0CEdUjTg0lIiIiIiJqYlgIEhERERERNTGcGkpERERE9IriCwKkLI4IEhERERERNTEsBBtQREQEBAKB9BAKhTAzM4OrqyvCw8ORn58vE5+YmIjJkyejY8eO0NXVhbW1NaZOnYq0tLQa33vlypUQCATIysqStpWUlMDOzg7t27dHYWGh3DV37tyBrq4uxowZAwAyuVd2rFy5ssa5ERERERFRw+LUUBUIDg6GlZUVSkpKkJ6ejtOnTyMgIAChoaGIjY1F165dAQCLFy9GdnY2xowZAxsbG/z111/YuHEjjh07hmvXrsHExKRWeTRv3hzbtm1Dv379sHr1anzyyScy52fNmgUtLS2Eh4cDAPbs2VNpXytXrsTt27fh4OBQq5yIiIiIiKj+sRBUAXd3d/Tq1Uv6OTAwEGKxGB4eHvD09ERycjJ0dHQQGhqK/v37Q0PjfwO3bm5uGDhwIDZu3Ig1a9bUOhdHR0fMmDEDn3/+OXx8fNC5c2cAwKFDh3D8+HF89dVXMDU1BQD4+voq7GPHjh24ffs2Zs+eDXd391rnRERERETVxJcESUmcGqomnJ2dERQUhLt37yIqKgoA4OTkJFMEVrQZGBggOTm5zu4dEhKC1q1bY8aMGZBIJCgoKEBAQIC0SKzK77//jjlz5qB79+747LPP6iwnIiIiIiKqPywE1Yifnx8AID4+vtKYgoICFBQUoHXr1nV2X5FIhPDwcJw9exY7duxAUFAQMjIysG3bNggElf+Y6dmzZ/D29oampib27dsHbW3tOsuJiIiIiIjqD6eGqhFzc3OIRCLcvn270piwsDAUFxdj7NixdXrv0aNHw8PDAwsXLkRBQQEWLVqELl26VHnN7Nmz8ccffyAyMhIdO3as03yIiIiIiKj+cERQzejp6cmtHlrhzJkzWLVqFby9veHs7Fzn9960aROKi4thYWGBoKCgKmNjYmKwc+dO+Pn5YcKECXWeCxERERER1R+OCKqZgoICGBkZybWnpKTg3XffRZcuXbBjx456uXfbtm1hZGSEzp07Q0dHp9K4P//8EzNmzEDHjh3x1VdfVavvoqIiFBUVybSVFBehuRankxIREREpS8DVYkhJHBFUI6mpqcjNzUWHDh1k2u/fv4+hQ4dCJBLhxIkTaNmypYoyfFHQjR07FsXFxdi3bx/09PSqdV1ISAhEIpHMcSxiUz1nS0REREREirAQVCMV+/S5urpK2x4/foyhQ4eiqKgI3333nXQrB1VZsGABrl69ik8//RTdu3ev9nWBgYHIzc2VOTwmfViPmRIRERERUWU4NVRNiMVirF69GlZWVvDx8QEAPH36FMOGDcODBw9w6tQp2NjYqDTHI0eOYOPGjfD09MScOXNqdK22trbcqqLNtXLrMj0iIiIiIqomFoIqcPLkSaSkpKC0tBQZGRkQi8VISEiApaUlYmNjIRQKAQA+Pj64fPkyJk+ejOTkZJm9A/X09ODl5VXje4eGhkJXV1emTUNDA0uWLKnyurS0NEyZMgWampoYMmSIdK/Df2vfvj0cHR1rnBcRERER1VwVO30RVYmFoAosX74cAKClpQUDAwPY2dkhLCwM/v7+Mu//Xbt2DQCwc+dO7Ny5U6YPS0tLpQrBkJAQuTZNTc2XFoI3btzAkydPAABz586tNG7ixIksBImIiIiI1BwLwQY0adIkTJo0qdrxd+7cqbN7r1y5EitXrlT6noMGDYJEIqmzfIiIiIiISHW4WAwREREREVETwxHBV1xubi4KCwurjDExMWmgbIiIiIioIfEVQVIWC8FX3Ny5cxEZGVllDKd0EhERERHRP7EQfMUtWrQIvr6+qk6DiIiIiIheISwEX3G2trawtbVVdRpERERERPQK4WIxRERERERETQxHBImIiIiIXlVcLYaUxBFBIiIiIiKiJoYjgqQyLZrzl9+/Cf7/x3rlEgkyC4tUnI36KP//lW9LyyW4nVGg4mzUS2t9bVWnoJZ0tPn7iyJl5VxFWpFx9haqTkEt5T4rUXUKRFSPOCJIRERERETUxPBHpkRERERErygBXxIkJXFEkIiIiIiIqIlhIUhERERERNTEsBAkIiIiIiJqYlgI1kBERAQEAoH0EAqFMDMzg6urK8LDw5Gfny8Tn5iYiMmTJ6Njx47Q1dWFtbU1pk6dirS0tBrd986dOzL3reyIiIioVvzatWulfQ8aNEjmnJaWFqysrDBt2jTcv39fJo+ffvoJs2bNQufOndGiRQu0bdsW3t7euHnzpnJfKBERERHVikDQeA+qX1wsRgnBwcGwsrJCSUkJ0tPTcfr0aQQEBCA0NBSxsbHo2rUrAGDx4sXIzs7GmDFjYGNjg7/++gsbN27EsWPHcO3aNZiYmFTrfoaGhtizZ4/Cc2VlZZg3bx4KCgrQvXt3mXPjx4/HsGHD5K75d5y5uTlCQkIAAMXFxfjjjz+wZcsWfPfdd0hOToauri4AYN26dTh37hzGjBmDrl27Ij09HRs3bkSPHj1w8eJFdOnSpVrPQ0REREREqsVCUAnu7u7o1auX9HNgYCDEYjE8PDzg6emJ5ORk6OjoIDQ0FP3794eGxv8GXt3c3DBw4EBs3LgRa9asqdb9WrRoAV9fX4Xnli1bhuzsbKxfvx7dunWTOdejR49Kr/snkUgkF2dlZYVZs2bh3LlzePvttwEA8+bNQ0xMDLS0tKRxY8eOhZ2dHdauXYuoqKhqPQ8REREREakWp4bWEWdnZwQFBeHu3bvSgsjJyUmmCKxoMzAwQHJycq3vmZiYiJCQEAwbNgz/+c9/at3fP1WMVjZr9r+fFfTt21emCAQAGxsbdO7cuU6eh4iIiIiIGgYLwTrk5+cHAIiPj680pqCgAAUFBWjdunWt7pWRkQEfHx+YmJggMjISAgUTqZ89e4asrCy5o7S0VCaurKxMei4tLQ1isRgrVqxAhw4d0K9fvyrzkEgkyMjIqPXzEBERERFRw2EhWIfMzc0hEolw+/btSmPCwsJQXFyMsWPHKn2f8vJy+Pn54dGjR4iOjq60CFuxYgUMDQ3ljp9//lkmLiUlRXrOzMwMQ4YMQXl5OeLj4+VGAP8tOjoaDx48qNXzEBEREZFyBI34oPrFdwTrmJ6entzqoRXOnDmDVatWwdvbG87OzkrfY+3atUhISEBQUBAGDRpUady0adMwZswYuXZbW1uZz+3atcP27dsBAKWlpbhx4wY+/fRTuLu748cff4ShoaHC/lNSUvDhhx/C0dEREydOVPp5iIiIiIioYbEQrGMFBQUwMjKSa09JScG7776LLl26YMeOHUr3f+7cOaxYsQIDBgzAihUrqoy1sbGBi4vLS/ts0aKFTJybmxv69++PXr16Ye3atVi/fr3cNenp6Rg+fDhEIhEOHjwITU3NKu9RVFSEoqIimbaS4iI019J+aX5ERERERFS3ODW0DqWmpiI3NxcdOnSQab9//z6GDh0KkUiEEydOoGXLlkr1n52djfHjx0NfXx8xMTEvLb5qo2fPnhCJRDhz5ozcudzcXLi7uyMnJwdxcXEwMzN7aX8hISEQiUQyx5H/flkfqRMRERER0UtwRLAOVez15+rqKm17/Pgxhg4diqKiIiQmJsLU1FTp/idNmoT79+/j22+/hbm5ea3zfZmysjIUFBTItD1//hwjRozAzZs38f3338tNM61MYGAg5s2bJ9N2/EZ2neVKRERE1CTxZTpSEgvBOiIWi7F69WpYWVnBx8cHAPD06VMMGzYMDx48wKlTp2BjY6N0/2FhYTh69Chmz54NT0/Pukq7UqdOnUJBQYHM3oRlZWUYO3YsLly4gG+//RaOjo7V7k9bWxva2rLTQJtrPa2zfImIiIiIqPpYCCrh5MmTSElJQWlpKTIyMiAWi5GQkABLS0vExsZCKBQCAHx8fHD58mVMnjwZycnJMnvt6enpwcvLq1r3u379OhYvXgw9PT1069at0o3bu3btiq5du0o///LLLwpj27dvL1PE5ebmSuMqFovZvHkzdHR08NFHH0nj5s+fj9jYWIwYMQLZ2dlyfVdn83oiIiIiIlI9FoJKWL58OQBAS0sLBgYGsLOzQ1hYGPz9/WXe/7t27RoAYOfOndi5c6dMH5aWltUuBH/55RcUFxejuLgYU6dOrTRuxYoVMoXg3r17sXfvXrm4iRMnyhSCqamp0j0QBQIBXnvtNQwcOBArVqyAvb293PMcPXoUR48eleuXhSARERER0auBhWANTJo0CZMmTap2/J07d1Ry33bt2kEikVQr9vTp09XutyaxRERERFT/BHxJkJTEVUOJiIiIiIiaGI4Iqlhubi4KCwurjDExMWmgbIiIiIiIqClgIahic+fORWRkZJUx1Z3mSUREREREVB0sBFVs0aJFXGSFiIiIiIgaFAtBFbO1ta32puxERERERP8k4FoxpCQuFkNERERERNTEsBAkIiIiIiJqYjg1lFRm2NCeqk5B7QizMgAAuo/SMc7DQcXZqA+dx5mqToGIiIioUWEhSCqjm5mm6hTUlgBAi0fpqk5D7WgIgNdbaqs6DbXyvKRM1SmoJXc7Q1WnoJYKnpeqOgW11EKbfxxSREODL5+9CvhviZTF3/lIpSQaGnje2ljVaagNYWYaBHjxvRQb8nv5J00NAcqMuKcmERERUV1gIUgq9by1Mb7//qqq01AbLi7doZOZhmJDY/z04++qTketmLXSUXUKREREpOY2bdqEzz77DOnp6ejWrRu+/PJL9OnTp9L4AwcOICgoCHfu3IGNjQ3WrVuHYcOGSc9LJBKsWLEC27dvR05ODvr164fNmzfDxsZGGpOdnY3Zs2fj6NGj0NDQwKhRo/DFF19AT08PAHD69Gls2LABly9fRl5eHmxsbLBw4UL4+PjU3xdRDVwshoiIiIiIXnn79+/HvHnzsGLFCvzyyy/o1q0bXF1dkZmpeK2B8+fPY/z48ZgyZQquXr0KLy8veHl5ISkpSRrz6aefIjw8HFu2bMGlS5fQokULuLq64vnz59IYHx8f/P7770hISMCxY8dw5swZTJs2TeY+Xbt2xaFDh3D9+nX4+/tjwoQJOHbsWP19GdUgkEgkEpVmQE2XQIBCI1OOCP5DxYhgkbEpRwT/hSOCihWV8h1BRb7/iwsMKeLUtrWqU1BLxiKhqlNQS2Xl/CNiZdq00lJ1ClJ3Hj9/edArqt3rNftv08HBAb1798bGjRsBAOXl5bCwsMDs2bPx0UcfycWPHTsWT58+lSnI3nrrLdjb22PLli2QSCQwMzPD/PnzsWDBAgBAbm4ujI2NERERgXHjxiE5ORm2trb46aef0KtXLwBAXFwchg0bhtTUVJiZmSnMdfjw4TA2NsbOnTtr9Ix1iSOCRERERET0SisuLsaVK1fg4uIibdPQ0ICLiwsuXLig8JoLFy7IxAOAq6urNP7vv/9Genq6TIxIJIKDg4M05sKFC2jVqpW0CAQAFxcXaGho4NKlS5Xmm5ubCwMDg5o/aB1iIdgAIiIiIBAIpIdQKISZmRlcXV0RHh6O/Px8mfi0tDR89NFHGDx4MFq2bAmBQIDTp08rde+VK1dCIBAgKytL2lZSUgI7Ozu0b98ehYWFctfcuXMHurq6GDNmjEz+QqEQDx48kIsfNGgQunTpolR+RERERESKFBUVIS8vT+YoKipSGJuVlYWysjIYG8sutmdsbIz0dMUrsaenp1cZX/H3l8UYGRnJnG/WrBkMDAwqve/XX3+Nn376Cf7+/grPNxQWgg0oODgYe/bswebNmzF79mwAQEBAAOzs7HD9+nVp3I0bN7Bu3To8ePAAdnZ2dZ5H8+bNsW3bNvz9999YvXq13PlZs2ZBS0sL4eHhMu1FRUVYu3ZtnedDRERERPRvISEhEIlEMkdISIiq06qVU6dOwd/fH9u3b0fnzp1VmgsLwQbk7u4OX19f+Pv7IzAwEN999x2+//57ZGZmwtPTUzo617NnTzx+/Bg3b97EvHnz6iUXR0dHzJgxA59//jl+//1/76IdOnQIx48fR0hICExNTWWusbe3x/bt2/Hw4cN6yYmIiIiIqEJgYCByc3NljsDAQIWxrVu3hqamJjIyMmTaMzIyYGKiePspExOTKuMr/v6ymH8vRlNaWors7Gy5+/7www8YMWIENmzYgAkTJlT16A2ChaCKOTs7IygoCHfv3kVUVBQAoGXLlg0yZzgkJAStW7fGjBkzIJFIUFBQgICAAGmR+G9LlixBWVkZRwWJiIiI1ISgEf+lra0NfX19mUNbW1vh96ClpYWePXsiMTFR2lZeXo7ExEQ4OjoqvMbR0VEmHgASEhKk8VZWVjAxMZGJycvLw6VLl6Qxjo6OyMnJwZUrV6QxYrEY5eXlcHBwkLadPn0aw4cPx7p162RWFFUlFoJqwM/PDwAQHx/foPcViUQIDw/H2bNnsWPHDgQFBSEjIwPbtm2DQCCQi7eyssKECRM4KkhEREREamfevHnYvn07IiMjkZycjA8++ABPnz6Vvos3YcIEmRHFuXPnIi4uDuvXr0dKSgpWrlyJn3/+GbNmzQIACAQCBAQEYM2aNYiNjcVvv/2GCRMmwMzMDF5eXgCAN998E25ubnj//fdx+fJlnDt3DrNmzcK4ceOkK4aeOnUKw4cPx5w5czBq1Cikp6cjPT0d2dnZDfsF/QsLQTVgbm4OkUiE27dvN/i9R48eDQ8PDyxcuBBffvklFixYUOXCL0uXLkVpaSnWrVvXgFkSEREREVVt7Nix+Pzzz7F8+XLY29vj2rVriIuLky72cu/ePaSlpUnj+/bti5iYGGzbtg3dunXDwYMH8c0338j8WXjRokWYPXs2pk2bht69e6OgoABxcXEQCv+3tUV0dDQ6deqEIUOGYNiwYejfvz+2bdsmPR8ZGYlnz55JX72qOEaOHNkA30rlmqn07iSlp6cnt3poQ9m0aRM6deoECwsLBAUFVRlrbW0NPz8/bNu2DR999JHce4RERERERKoya9Ys6YjevylahX/MmDHSlfIVEQgECA4ORnBwcKUxBgYGiImJqfR8REQEIiIiKj2vKhwRVBMFBQVo2bKlSu7dtm1bGBkZoXPnztDRefmm3cuWLUNpaWmN3hVUuPxvbZImIiIiIggEjfeg+sVCUA2kpqYiNzcXHTp0UHUq1WJtbQ1fX19s27ZNZni9KgqX/63nPImIiIiISDEWgmpgz549AABXV1cVZ1J9FaOC1X1XUOHyv/WcIxERERERKcZ3BFVMLBZj9erVsLKygo+Pj6rTqbb27dvD19cXW7duhaWlJZo1q/qXkra2tsLlfgvrK0EiIiIiIqoUC8EGdPLkSaSkpKC0tBQZGRkQi8VISEiApaUlYmNjZVYfWrNmDQBIN3vfs2cPzp49C+DFaFxNhYaGQldXV6ZNQ0MDS5YsUfZxsHTpUuzZswc3btxA586dle6HiIiIiJTDV+lIWSwEG9Dy5csBvNjw0sDAAHZ2dggLC4O/v7/cQjH/Xr1z586d0n9WphAMCZF/I09TU7NWhWCHDh3g6+uLyMhIpfsgIiIiIqKGJ5BIJBJVJ0FNlECAQiNTfP/9VVVnojZcXLpDJzMNRcam+OnH31Wdjloxa/XyFW2boqLSMlWnoJa+/ytT1SmoJae2rVWdgloyFglfHtQElZXzj4iVadNKS9UpSN3PbrzrsFsYyL9WRHWHi8UQERERERE1MZwa+orKzc1FYWHVS62YmJg0UDZERERERPQqYSH4ipo7d+5L383jrF8iIiKixo0br5OyWAi+ohYtWgRfX19Vp0FERERERK8gFoKvKFtbW9ja2qo6DSIiIiIiegVxsRgiIiIiIqImhiOCRERERESvLL4kSMrhiCAREREREVETwxFBUrlycHXTf5MAKORG4TJa6TZXdQpq6Vkxf56nyInr3FBekbetjVWdgloqeF6q6hTU0qO8xrtReW2p04byRMrinyCIiIiIiIiaGBaCRERERERETQynhhIRERERvaK4oTwpiyOCRERERERETQwLQSIiIiIioiamUReCEREREAgE0kMoFMLMzAyurq4IDw9Hfn6+THxaWho++ugjDB48GC1btoRAIMDp06eVvv/Zs2fh7u6ONm3aQCgUom3bthgxYgRiYmJk4gQCAWbNmlWtPo8cOQJ3d3e0bt0aWlpaMDMzg7e3N8RisdyzTJs2DVZWVtDR0UH79u0xb948PH78uNK++/TpA4FAgM2bNys8X1BQgBUrVsDNzQ0GBgYQCASIiIioVt5ERERERKQ+GnUhWCE4OBh79uzB5s2bMXv2bABAQEAA7OzscP36dWncjRs3sG7dOjx48AB2dna1uueBAwfg5OSEjIwMzJ07F19++SV8fX3x5MkTbN++vcb9SSQS+Pv7Y+TIkcjIyMC8efOwZcsWfPjhh/jrr78wZMgQnD9/HsCLgs3R0RFHjhzBhAkT8OWXX2LYsGHYuHEjXFxcUF5eLtf/n3/+iZ9++gnt2rVDdHS0whyysrIQHByM5ORkdOvWrcbPQERERER1S9CID6pfTWKxGHd3d/Tq1Uv6OTAwEGKxGB4eHvD09ERycjJ0dHTQs2dPPH78GAYGBjh48CDGjBmj9D1XrlwJW1tbXLx4EVpasnvNZGbWfH+r9evXIyIiAgEBAQgNDYXgH28GL126FHv27EGzZi/+dcbGxuLu3bs4duwYhg8fLo0zMDBAcHAwfv31V3Tv3l2m/6ioKBgZGWH9+vUYPXo07ty5g3bt2snEmJqaIi0tDSYmJvj555/Ru3fvGj8HERERERGpXpMYEVTE2dkZQUFBuHv3LqKiogAALVu2hIGBQZ30f/v2bfTu3VuuCAQAIyOjGvVVWFiIkJAQdOrUCZ9//rlMEVjBz88Pffr0AQDk5eUBAIyNZTcONjU1BQDo6OjIXR8TE4PRo0fDw8MDIpFIbvoqAGhra8PExKRGuRMRERERkfppsoUg8KJ4AoD4+Pg679vS0hKJiYlITU2tdV9nz55FdnY23nvvPWhqar403snJCRoaGpg7dy4uXryI1NRUnDhxAh9//DG8vLzQqVMnmfhLly7h1q1bGD9+PLS0tDBy5MhKp4cSEREREdGrr0kXgubm5hCJRLh9+3ad97148WLcv38f7du3h7OzM5YvX46zZ88qfD/vZZKTkwGg2u8t2traYtu2bfjjjz/g6OgICwsLDB8+HEOGDMGBAwfk4qOiomBhYYF+/foBAMaNG4c//vgD165dq3GuRERERNRwBILGe1D9atKFIADo6enJrR5aFyZPnoy4uDgMGjQIZ8+exerVqzFgwADY2NhIF3Wproqpni1btqz2NW3atEGfPn0QFhaGI0eOYN68eYiOjsZHH30kE1daWor9+/dj7Nix0imnzs7OMDIy4qggEREREVEj1SQWi6lKQUFBjd/Zqy5XV1e4urri2bNnuHLlCvbv348tW7bAw8MDKSkp1b6vvr4+AFS7YD137hw8PDxw8eJF6SI5Xl5e0NfXx6pVqzB58mTY2toCeDEt9tGjR+jTpw9u3bol7WPw4MHYu3cv1q1bBw2N2v+8oKioCEVFRTJt2rXulYiIiIiIlNGkRwRTU1ORm5uLDh061Ot9dHV1MWDAAGzcuBHLli3DkydPcPLkyWpfX/FO32+//Vat+K1bt8LY2FhmpVQA8PT0hEQikRmRrBj18/b2ho2NjfTYv38/Hjx4gB9++KHaeVYlJCQEIpFI5gipk56JiIiIiKimmvSI4J49ewC8GLlrKBXFWVpaWrWv6d+/P1577TXs3bsXS5YseemCMRkZGSgrK5NrLykpAfBiOigAPH36FN9++y3Gjh2L0aNHy8XPmTMH0dHRGDx4cLVzrUxgYCDmzZsn06YtEqHmb0wSEREREVFtNdkRQbFYjNWrV8PKygo+Pj513n9iYqLC9hMnTgAA3njjjWr3pauri8WLFyM5ORmLFy+GRCKRi4mKisLly5cBAB07dkRGRgZOnz4tE7N3714AkO4heOTIETx9+hQffvghRo8eLXd4eHjg0KFDclM6laGtrQ19fX2Zg1NDiYiIiGpH0Ij/ovrVJEYET548iZSUFJSWliIjIwNisRgJCQmwtLREbGwshEKhNHbNmjUAgN9//x3Ai1HDs2fPAgCWLVtW7Xu+8847sLKywogRI9C+fXs8ffoU33//PY4ePYrevXtjxIgRMvE///yz9N7/NGjQIPTv3x8LFy7E77//jvXr1+PUqVMYPXo0TExMkJ6ejm+++QaXL1+WTvmcNWsWdu3ahREjRmD27NmwtLTEDz/8gL179+Ltt9+Gg4MDgBfTQl9//XX07dtX4TN4enpi+/btOH78OEaOHAkA2LhxI3JycvDw4UMAwNGjR6VbZMyePRsikaja3xEREREREamGQKJoeKmRiIiIgL+/v/SzlpYWDAwMYGdnBw8PD/j7+8utxKlos/YKNfmq9u3bh2+//RY//fQTHj58CIlEAmtra7z77rtYvHixzH2ruufq1atlCtBDhw5h27Zt+Pnnn5GXlwdDQ0M4OTnhgw8+wMCBA6VxN27cwLJly3Dp0iWkp6fDzMwMY8aMwapVq6Crq4vMzEy0adMG48ePx+7duxXeu7CwEK1bt4arqysOHz4MAGjXrh3u3r2rMP7vv/9Gu3btqv0dQSBAoZEp4r//pfrXNHJDXXpAJzMNz41N8eOp66pOR630bPuaqlNQS8+K5aeBE/D+vquqTkEtbfCq3jZETU0zTY48KPIor/Yzghorxw6tVJ2CVHpuiapTqDcmouaqTqFRa9SFIKk5FoJyWAhWjoWgYiwEFWMhqBgLQcVYCCrGQrByLAQbBgvB+tUkpoYSERERETVK/DkGKYmFYA3l5uaisLCwyhgTE5MGyoaIiIiIiKjmWAjW0Ny5cxEZGVllDGfbEhERERGROmMhWEOLFi2Cr6+vqtMgIiIiIiJSGgvBGrK1tYWtra2q0yAiIiIi4iuCpLQmu6E8ERERERFRU8VCkIiIiIiIqIlhIUhERERERNTE8B1BUrlyLrKqUDlXn5Uh4EsQChVyQ3mFLp2/reoU1JLeuO6qTkEtcUN5xbSbcbyAqDFjIUhERERE9IriD0pJWfxRDxERERERURPDQpCIiIiIiKiJYSFIRERERETUxPAdQSIiIiKiV5SAW8qTkhrliGBERAQEAoH0EAqFMDMzg6urK8LDw5Gfny8Tn5aWho8++giDBw9Gy5YtIRAIcPr0aaXuvXLlSpl76+rqwtbWFsuWLUNeXp5cjj///HOlfe3YsQMCgQCRkZFy5y5cuAANDQ0sWLBA2jZo0CDpfTU0NKCvr4833ngDfn5+SEhIqPQ+ZWVl2LVrFwYNGgQDAwNoa2ujXbt28Pf3l8nvp59+wqxZs9C5c2e0aNECbdu2hbe3N27evFnTr4mIiIiIiFSoUY8IBgcHw8rKCiUlJUhPT8fp06cREBCA0NBQxMbGomvXrgCAGzduYN26dbCxsYGdnR0uXLhQ63tv3rwZenp6KCgoQHx8PD7++GOIxWKcO3cOgmou7zRlyhRERkZiwYIF8PDwwOuvvw4AKCkpwbRp02BhYYFVq1bJXGNubo6QkBAAwNOnT3Hr1i0cPnwYUVFR8Pb2RlRUFJo3by6NLywsxMiRIxEXFwcnJycsWbIEBgYGuHPnDr7++mtERkbi3r17MDc3x7p163Du3DmMGTMGXbt2RXp6OjZu3IgePXrg4sWL6NKlS62/NyIiIiIiqn+NuhB0d3dHr169pJ8DAwMhFovh4eEBT09PJCcnQ0dHBz179sTjx49hYGCAgwcPYsyYMbW+9+jRo9G6dWsAwIwZMzBq1CgcPnwYFy9ehKOjY7X6EAgE2Lp1K+zt7bFgwQLs2rULALB+/XokJSUhNjYWLVq0kLlGJBLB19dXpm3t2rWYM2cOvvrqK7Rr1w7r1q2Tnlu4cCHi4uKwYcMGBAQEyFy3YsUKbNiwQfp53rx5iImJgZaWlrRt7NixsLOzw9q1axEVFVWt5yIiIqL/a+/ug6K+7j2Of9ZNeBLYyVYJUoyhQJKBQOJD4yUmrTE1BAUz0xtDLBJ12lpmULDGGMlIoOIUTYtjiTEPzVgYI3EmJGnQXi20W2107G1zTa6tAWyZaMEBEqXudQuCLHv/cNxm5UHY7rKr+37N/Gbc3fM757uMOvPhnN85AOBbN+XS0JHMmzdPxcXFOnPmjDO4REREyGw2e31cSfrss8/GdF9SUpKee+45VVVV6fDhw/rss8+0adMmffvb31ZWVtao+jAajaqsrFRSUpJ27Nghq9UqSWpra9Prr7+u+fPnDwqBV+9bt26dYmNjJUkPPvigSwiUpMTERCUnJ6uxsXFM3wsAAAAeYLiJL3hVwAVBScrNzZUk1dfXj9uYLS0tkuRc3jkWGzdu1Ne+9jX94Ac/UF5enm655RZVVlaOqQ+j0aglS5aou7tbR44ckSQdOHBA/f39zp+HOxwOhzo7O52znwAAAAD83029NHQ4sbGxMplMznDmDV1dXZLkfEZw586duv322/Xwww+Pua/Q0FC9+uqrSk9PV3Nzs7Zv366vfvWrY+7n6jN8V7/31Vm8lJSUMfd11Z49e3T27Flt2rTJ7T4AAAAAjK+ADIKSFB4ePmj3UE+6++67XV4nJyerurpaYWFhbvVnNps1YcIEDQwM6LHHHnOrj/DwcElyfu+ru5hGRES41V9TU5Py8/OVlpamZcuWudUHAAAAgPEXsEHQZrMpKirKa/2/++67ioyM1K233qrY2FjFx8e73ZfdbtfKlSsVExMjm82mgoKCEY+DGI7NZpP0r+AXGRkpSW4F4o6ODi1cuFAmk0m1tbUyGo0jtu/t7VVvb6/Le8FjHhUAAACAJwRkEGxra5PValVCQoLXxvjGN77hsefmfvazn+njjz/WL3/5S509e1b5+fmqqanRd77znTH185e//EWSnN/7nnvukST9+c9/1v333z/qfqxWqzIyMnThwgV9+OGHiomJue495eXlg466KJH0/KhHBQAAwLXYUwXuCsjNYnbv3i1JSk9P93El19fa2qqSkhI98cQTeuKJJ5SXl6cHHnhAa9eude7+ORp2u101NTUKCwvTQw89JOnK8RpGo3FMxz5cunRJWVlZOnXqlPbv36+kpKRR3VdUVCSr1epyFY16VAAAAACeFHBB0GKxqKysTHFxccrJyfF1Ode1evVqORwOvfzyy5KkCRMm6LXXXtO5c+f0wgsvjKoPu92ugoICNTY2qqCgwLkkdOrUqfr+97+v+vp6Z/9fNjAwoIqKCrW1tTn7yc7O1rFjx/TOO++M+jxESQoODlZkZKTLxdJQAAAAwDdu6qWhBw4cUFNTk/r7+9XZ2SmLxaKGhgZNmzZNdXV1CgkJcbbdvHmzJOnkyZOSrswaXj1mYePGjV6rcdeuXTp48OCg9wsLC/Wb3/xGH3zwgSoqKjR16lTnZ9OnT1d+fr527Nih5cuX6+tf/7rzM6vV6pzh6+7u1t/+9je99957amlp0dNPP62ysjKXcSoqKtTS0qKCggK99957yszM1G233aa///3veuedd9TU1KSnn35akvTss8+qrq5OWVlZ6urqGjSTeO1B9gAAAAD8k8HhcDh8XYSnVVVVacWKFc7XQUFBMpvNSklJUWZmplasWDFop0yDYfgV1mP5EZWWlupHP/qRvvjiixGfEby2xmu1trYqLS1NkyZN0kcffTRoM5aLFy/qnnvuUXR0tP74xz/KaDRq7ty5Onz4sLNNeHi4pkyZotmzZ+uZZ57R/PnzhxzLbrerqqpK1dXVOnHihLq7uxUTE6N58+apoKDA+fzgtf1fa8x/lQwG9URN0cGG42O77yb2+PwZCv28XZdun6LDlv/1dTl+5YE7zb4uwS+du9jn6xL80ux17/u6BL90ovJJX5fgl24x8pTVUC73D/i6BL811ew/65rO/7Pf1yV4zVcm3tRzVj53UwZB3CAIgoMQBIdHEBwaQXBoBMGhEQSHRhAcGkFweATB8UEQ9K6Ae0YQAAAAAAIdMXuUrFarenp6RmwTHR09TtUAAAAAgPsIgqNUWFio6urqEduwyhYAAADjycBJgnATQXCU1q9fz66YAAAAAG4KBMFRSkpKGvXh6QAAAADgz9gsBgAAAAACDEEQAAAAAAIMS0PhUw5JdjbZcbr6k7APONRmG3mX2kBj7rD5ugS/NCnSf86y8if/MSfB1yX4pd5+u69L8EuRofw7GoqVcwRvCAb2ioGbmBEEAAAAgABDEAQAAACAAEMQBAAAAIAAQxAEAAAAgABDEAQAAACAAEMQHIOqqioZDAbnFRISopiYGKWnp6uyslIXL150ad/e3q4NGzbokUceUUREhAwGgw4dOjTmcU+fPu0y7nBXVVXVqNpv2bLF2ffcuXNdPgsKClJcXJxWrlyp1tZWlzr+9Kc/adWqVUpOTtbEiRN1xx136KmnntKpU6fG/J0AAAAA+A7HR7hh06ZNiouL0+XLl9XR0aFDhw5pzZo12rZtm+rq6pSamipJam5u1tatW5WYmKiUlBQdO3bMrfEmT56s3bt3D/mZ3W7X2rVrZbPZNH36dJfPlixZogULFgy659p2sbGxKi8vlyT19fXp008/1WuvvaZf//rXamxsVFhYmCRp69atOnr0qBYvXqzU1FR1dHRox44dmjFjhv7whz/o3nvvdev7AQAAABhfBEE3ZGRkaNasWc7XRUVFslgsyszM1KJFi9TY2KjQ0FDNnDlT58+fl9lsVm1trRYvXuzWeBMnTtTSpUuH/Gzjxo3q6upSRUWF7rvvPpfPZsyYMex9X2YymQa1i4uL06pVq3T06FHNnz9fkrR27VrV1NQoKCjI2S47O1spKSnasmWL3nrrrbF+NQAAAPwbOEcQ7mJpqIfMmzdPxcXFOnPmjDMQRUREyGw2e23M3/72tyovL9eCBQv0wx/+0KN9R0dHS5JuueVfvyt48MEHXUKgJCUmJio5OVmNjY0eHR8AAACA9xAEPSg3N1eSVF9f7/WxOjs7lZOTo+joaFVXV8swxK+Duru7de7cuUFXf3+/Szu73e78rL29XRaLRSUlJUpISNCcOXNGrMPhcKizs1OTJk3y6PcDAAAA4D0EQQ+KjY2VyWRSS0uLV8cZGBhQbm6uvvjiC+3Zs2fYEFZSUqLJkycPuj766COXdk1NTc7PYmJi9Oijj2pgYED19fWDZgCvtWfPHp09e1bZ2dke+34AAAAAvItnBD0sPDx80O6hnrZlyxY1NDSouLhYc+fOHbbdypUrh3wuMSkpyeX1nXfeqZ///OeSpP7+fjU3N+ull15SRkaGPvzwQ02ePHnI/puampSfn6+0tDQtW7bM/S8EAAAAYFwRBD3MZrMpKirKa/0fPXpUJSUlevjhh1VSUjJi28TERH3rW9+6bp8TJ050aff444/roYce0qxZs7RlyxZVVFQMuqejo0MLFy6UyWRSbW2tjEbjiGP09vaqt7fX5b3g61YGAACAkRjEbjFwD0tDPaitrU1Wq1UJCQle6b+rq0tLlixRZGSkampqrhu+/h0zZ86UyWTS73//+0GfWa1WZWRk6MKFCzp48KBiYmKu2195eblMJpPLVe6NwgEAAABcF0HQg66e9Zeenu6V/pcvX67W1lb94he/UGxsrFfG+DK73S6bzeby3qVLl5SVlaVTp05p//79g5aZDqeoqEhWq9XlKvJG0QAAAACui6WhHmKxWFRWVqa4uDjl5OR4vP/t27dr3759Wr16tRYtWuTx/q/1u9/9TjabzeVsQrvdruzsbB07dkwffPCB0tLSRt1fcHCwgoMHLwbt9ki1AAAAAMaCIOiGAwcOqKmpSf39/ers7JTFYlFDQ4OmTZumuro6hYSEONtu3rxZknTy5ElJV2YNjxw5IunKYfCjceLECT3//PMKDw/XfffdN+zB7ampqUpNTXW+Pn78+JBt4+PjXUKc1Wp1tru6Wcyrr76q0NBQbdiwwdnu2WefVV1dnbKystTV1TWo79EcXg8AAADP4UB5uIsg6IYXX3xRkhQUFCSz2ayUlBRt375dK1asUEREhEvb4uJil9e7du1y/nm0QfD48ePq6+tTX1+fvve97w3brqSkxCUIvv3223r77bcHtVu2bJlLEGxra3OegWgwGHTbbbfpm9/8pkpKSnT//fc7233yySeSpH379mnfvn2D+iUIAgAAADcGg8PhcPi6CAQog0HdUVP0X/X/4+tK/MaCx2Yq7PN2/XNytPbu/29fl+NXUieZfF2CX5oUyf67Q8mvPeHrEvzSy/+Z4usS/NLkCP4dDcXafdnXJfitqWb/+Tvzf5cGfF2C10SGsJ2JN/HTBQAAAIAAw9JQH7Narerp6RmxTXR09DhVAwAAgBsJjwjCXQRBHyssLFR1dfWIbVi9CwAAAMCTCII+tn79ejZZAQAAADCuCII+lpSUNOpD2QEAAADAE9gsBgAAAAACDDOCAAAAwI2K3WLgJmYEAQAAACDAMCMIn3LIoX9e7vd1GX7DoSs7xE4wGBQV6j+H1fqD+NvDfV2CX7rAgc9D6unl/5WhGCcwdTCUvZ+0+roEv7Tgbo6vAm5mzAgCAAAAQIBhRhAAAAC4QRl4SBBuYkYQAAAAAAIMQRAAAAAAAgxBEAAAAAACDEFwHFVVVclgMDivkJAQxcTEKD09XZWVlbp48aJL+/b2dm3YsEGPPPKIIiIiZDAYdOjQIbfGLi0tlcFg0Llz55zvXb58WSkpKYqPj1dPT8+ge06fPq2wsDAtXrxYklxqH+4qLS11qz4AAACMncFw817wLjaL8YFNmzYpLi5Oly9fVkdHhw4dOqQ1a9Zo27ZtqqurU2pqqiSpublZW7duVWJiolJSUnTs2DGP1nHrrbfqjTfe0Jw5c1RWVqYf//jHLp+vWrVKQUFBqqyslCTt3r172L5KS0vV0tKi2bNne7RGAAAAAJ5HEPSBjIwMzZo1y/m6qKhIFotFmZmZWrRokRobGxUaGqqZM2fq/PnzMpvNqq2tdc7MeVJaWpry8vL005/+VDk5OUpOTpYkvfvuu/rVr36lnTt3asqUKZKkpUuXDtnHm2++qZaWFq1evVoZGRkerxEAAACAZ7E01E/MmzdPxcXFOnPmjN566y1JUkREhMxms9fHLi8v16RJk5SXlyeHwyGbzaY1a9Y4Q+JITp48qYKCAk2fPl0/+clPvF4rAAAAgH8fQdCP5ObmSpLq6+vHdVyTyaTKykodOXJEb775poqLi9XZ2ak33nhDhhEWaHd3d+upp56S0WjU3r17FRwcPI5VAwAAAHAXS0P9SGxsrEwmk1paWsZ97CeffFKZmZl67rnnZLPZtH79et17770j3rN69Wp9+umnqq6u1l133TVOlQIAAOAq9lSBu5gR9DPh4eGDdg8dL6+88or6+vo0depUFRcXj9i2pqZGu3btUm5urp555plxqhAAAACAJzAj6GdsNpuioqJ8MvYdd9yhqKgoJScnKzQ0dNh2f/3rX5WXl6e77rpLO3fuHFXfvb296u3tdXmPhaQAAACAbzAj6Efa2tpktVqVkJDg61KG1dvbq+zsbPX19Wnv3r0KDw8f1X3l5eUymUwuV7mXawUAAAAwNIKgH7l6Tl96erqPKxneunXr9PHHH+ull17S9OnTR31fUVGRrFary1XkxToBAAACguEmvuBVLA31ExaLRWVlZYqLi1NOTo6vyxnS+++/rx07dmjRokUqKCgY073BwcFD7ira76niAAAAAIwaQdAHDhw4oKamJvX396uzs1MWi0UNDQ2aNm2a6urqFBIS4my7efNmSVfO65OuzBoeOXJEkrRx48Yxj71t2zaFhYW5vDdhwgS98MILI97X3t6u7373uzIajXr00UedZx1eKz4+XmlpaWOuCwAAAMD4IQj6wIsvvihJCgoKktlsVkpKirZv364VK1YoIiLCpe21u3fu2rXL+Wd3gmB5+eAn84xG43WDYHNzs/7xj39IkgoLC4dtt2zZMoIgAAAA4OcIguNo+fLlWr58+ZjucTgcHhm7tLRUpaWl1213+vTpId+fO3eux2oBAACAZxh4mA5uYrMYAAAAAAgwzAje4KxWq3p6ekZsEx0dPU7VAAAAALgREARvcIWFhaqurh6xDUs6AQAAAHwZQfAGt379ei1dutTXZQAAAAC4gRAEb3BJSUlKSkrydRkAAADwAQN7xcBNbBYDAAAAAAGGIAgAAAAAAYalofCp0HOf68mFD/i6DL8Reu5zX5cAAACAAGBwsKUkAAAAAAQUloYCAAAAQIAhCAIAAABAgCEIAgAAAECAIQgCAAAAQIAhCAIAAABAgCEIAgAAAECAIQgCAAAAQIAhCAIAAABAgCEIAgAAAECA+X8RrN/ulpjJQwAAAABJRU5ErkJggg==\n"
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "For the RNA modality, we perform pathway/GO enrichment analysis on the top genes for each latent dimension by running run_pathway_analysis_top_genes(top_genes_by_dim), which queries g:Profiler and returns per-dimension enrichment results (GO/KEGG/Reactome)."
      ],
      "metadata": {
        "id": "Rw3-L89dHayE"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "enrichment_results_by_dim = run_pathway_analysis_top_genes(top_genes_by_dim)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZR3kWcXVHBg7",
        "outputId": "e454ad65-be1e-400d-8b90-40e0361c1908"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "--- Pathway Analysis for Dimension 0 ---\n",
            "  source      native                                            name  \\\n",
            "0  GO:BP  GO:0002682             regulation of immune system process   \n",
            "1  GO:BP  GO:0002376                           immune system process   \n",
            "2  GO:BP  GO:0006955                                 immune response   \n",
            "3  GO:BP  GO:0002684    positive regulation of immune system process   \n",
            "4  GO:BP  GO:0051239  regulation of multicellular organismal process   \n",
            "\n",
            "        p_value  significant  \\\n",
            "0  1.835380e-21         True   \n",
            "1  1.037405e-20         True   \n",
            "2  1.334316e-17         True   \n",
            "3  2.626855e-17         True   \n",
            "4  3.723644e-17         True   \n",
            "\n",
            "                                         description  term_size  query_size  \\\n",
            "0  \"Any process that modulates the frequency, rat...       1581         180   \n",
            "1  \"Any process involved in the development or fu...       2871         180   \n",
            "2  \"Any immune system process that functions in t...       2070         180   \n",
            "3  \"Any process that activates or increases the f...       1099         180   \n",
            "4  \"Any process that modulates the frequency, rat...       2956         180   \n",
            "\n",
            "   intersection_size  effective_domain_size  precision    recall    query  \\\n",
            "0                 61                  21026   0.338889  0.038583  query_1   \n",
            "1                 80                  21026   0.444444  0.027865  query_1   \n",
            "2                 64                  21026   0.355556  0.030918  query_1   \n",
            "3                 47                  21026   0.261111  0.042766  query_1   \n",
            "4                 76                  21026   0.422222  0.025710  query_1   \n",
            "\n",
            "                                parents  \\\n",
            "0              [GO:0002376, GO:0050789]   \n",
            "1                          [GO:0008150]   \n",
            "2              [GO:0002376, GO:0050896]   \n",
            "3  [GO:0002376, GO:0002682, GO:0048518]   \n",
            "4              [GO:0032501, GO:0050789]   \n",
            "\n",
            "                                       intersections  \\\n",
            "0  [LYN, CD74, HLA-DRA, HLA-DRB1, CAMK4, FCN1, AN...   \n",
            "1  [LYZ, LYN, CD74, HLA-DRA, BCL11B, HLA-DRB1, CA...   \n",
            "2  [LYZ, LYN, CD74, HLA-DRA, HLA-DRB1, CAMK4, FCN...   \n",
            "3  [LYN, CD74, HLA-DRA, HLA-DRB1, FCN1, ANXA1, TH...   \n",
            "4  [SLC8A1, LYN, VCAN, CD74, HLA-DRA, HLA-DRB1, C...   \n",
            "\n",
            "                                           evidences  \n",
            "0  [[IMP, ISS, IBA, TAS, IEA], [IDA, IMP, ISS, IB...  \n",
            "1  [[TAS], [IMP, ISS, IBA, TAS, IEA], [IDA, IMP, ...  \n",
            "2  [[TAS], [ISS, IBA, TAS, IEA], [IDA, ISS, IBA, ...  \n",
            "3  [[ISS, IBA, TAS, IEA], [IDA, IMP, ISS, IBA, IE...  \n",
            "4  [[IMP, ISS, TAS, IC], [ISS, TAS, IEA], [IBA], ...  \n",
            "\n",
            "--- Pathway Analysis for Dimension 1 ---\n",
            "  source      native                                 name       p_value  \\\n",
            "0  GO:BP  GO:0002682  regulation of immune system process  8.066030e-25   \n",
            "1  GO:BP  GO:0001775                      cell activation  1.325153e-24   \n",
            "2  GO:BP  GO:0002376                immune system process  2.041598e-24   \n",
            "3  GO:BP  GO:0050865        regulation of cell activation  2.497794e-22   \n",
            "4  GO:BP  GO:0045321                 leukocyte activation  4.568174e-22   \n",
            "\n",
            "   significant                                        description  term_size  \\\n",
            "0         True  \"Any process that modulates the frequency, rat...       1581   \n",
            "1         True  \"A multicellular organismal process by which e...       1132   \n",
            "2         True  \"Any process involved in the development or fu...       2871   \n",
            "3         True  \"Any process that modulates the frequency, rat...        632   \n",
            "4         True  \"A change in morphology and behavior of a leuk...        980   \n",
            "\n",
            "   query_size  intersection_size  effective_domain_size  precision    recall  \\\n",
            "0         180                 65                  21026   0.361111  0.041113   \n",
            "1         180                 56                  21026   0.311111  0.049470   \n",
            "2         180                 85                  21026   0.472222  0.029606   \n",
            "3         180                 42                  21026   0.233333  0.066456   \n",
            "4         180                 50                  21026   0.277778  0.051020   \n",
            "\n",
            "     query                               parents  \\\n",
            "0  query_1              [GO:0002376, GO:0050789]   \n",
            "1  query_1              [GO:0009987, GO:0032501]   \n",
            "2  query_1                          [GO:0008150]   \n",
            "3  query_1  [GO:0001775, GO:0050794, GO:0051239]   \n",
            "4  query_1              [GO:0001775, GO:0002376]   \n",
            "\n",
            "                                       intersections  \\\n",
            "0  [CD74, LEF1, HLA-DRA, LYN, CAMK4, ANXA1, FCN1,...   \n",
            "1  [ZEB2, CD74, LEF1, HLA-DRA, BCL11B, LYN, CAMK4...   \n",
            "2  [LYZ, CD74, LEF1, HLA-DRA, BCL11B, LYN, CAMK4,...   \n",
            "3  [CD74, LEF1, HLA-DRA, LYN, CAMK4, ANXA1, JAK2,...   \n",
            "4  [CD74, LEF1, HLA-DRA, BCL11B, LYN, CAMK4, ANXA...   \n",
            "\n",
            "                                           evidences  \n",
            "0  [[IDA, IMP, ISS, IBA, NAS, IEA], [IDA, IMP, IS...  \n",
            "1  [[IEA], [IMP, IBA, NAS, IEA], [ISS, IEA], [IDA...  \n",
            "2  [[TAS], [IDA, IMP, ISS, IBA, NAS, IEA], [IDA, ...  \n",
            "3  [[IMP, NAS, IEA], [ISS, IEA], [IDA, IBA], [ISS...  \n",
            "4  [[IMP, IBA, NAS, IEA], [ISS, IEA], [IDA, IBA],...  \n",
            "\n",
            "--- Pathway Analysis for Dimension 2 ---\n",
            "  source      native                   name       p_value  significant  \\\n",
            "0  GO:BP  GO:0002376  immune system process  5.233409e-29         True   \n",
            "1  GO:BP  GO:0006955        immune response  7.663880e-28         True   \n",
            "2  GO:BP  GO:0001775        cell activation  7.281455e-26         True   \n",
            "3  GO:BP  GO:0046649  lymphocyte activation  3.142806e-25         True   \n",
            "4  GO:BP  GO:0045321   leukocyte activation  2.496772e-24         True   \n",
            "\n",
            "                                         description  term_size  query_size  \\\n",
            "0  \"Any process involved in the development or fu...       2871         184   \n",
            "1  \"Any immune system process that functions in t...       2070         184   \n",
            "2  \"A multicellular organismal process by which e...       1132         184   \n",
            "3  \"A change in morphology and behavior of a lymp...        814         184   \n",
            "4  \"A change in morphology and behavior of a leuk...        980         184   \n",
            "\n",
            "   intersection_size  effective_domain_size  precision    recall    query  \\\n",
            "0                 92                  21026   0.500000  0.032045  query_1   \n",
            "1                 78                  21026   0.423913  0.037681  query_1   \n",
            "2                 58                  21026   0.315217  0.051237  query_1   \n",
            "3                 50                  21026   0.271739  0.061425  query_1   \n",
            "4                 53                  21026   0.288043  0.054082  query_1   \n",
            "\n",
            "                    parents  \\\n",
            "0              [GO:0008150]   \n",
            "1  [GO:0002376, GO:0050896]   \n",
            "2  [GO:0009987, GO:0032501]   \n",
            "3              [GO:0045321]   \n",
            "4  [GO:0001775, GO:0002376]   \n",
            "\n",
            "                                       intersections  \\\n",
            "0  [GNLY, CCL5, LEF1, BACH2, NKG7, CAMK4, SERINC5...   \n",
            "1  [GNLY, CCL5, LEF1, BACH2, NKG7, CAMK4, SERINC5...   \n",
            "2  [CCL5, LEF1, NKG7, CAMK4, ZEB2, PRKCA, CCR7, I...   \n",
            "3  [CCL5, LEF1, NKG7, CAMK4, CCR7, IL7R, PRF1, TC...   \n",
            "4  [CCL5, LEF1, NKG7, CAMK4, CCR7, IL7R, PRF1, TC...   \n",
            "\n",
            "                                           evidences  \n",
            "0  [[IDA, IBA], [IDA, IEP, IBA, TAS, IC, IEA], [I...  \n",
            "1  [[IDA, IBA], [IEP, IBA, TAS, IEA], [ISS, IEA],...  \n",
            "2  [[IDA], [ISS, IEA], [ISS], [IMP, TAS], [IEA], ...  \n",
            "3  [[IDA], [ISS, IEA], [ISS], [TAS], [ISS, IC, IE...  \n",
            "4  [[IDA], [ISS, IEA], [ISS], [IMP, TAS], [ISS, I...  \n",
            "\n",
            "--- Pathway Analysis for Dimension 3 ---\n",
            "  source      native                                 name       p_value  \\\n",
            "0  GO:BP  GO:0006955                      immune response  1.394454e-23   \n",
            "1  GO:BP  GO:0002376                immune system process  2.375714e-23   \n",
            "2  GO:BP  GO:0002682  regulation of immune system process  1.564387e-21   \n",
            "3  GO:BP  GO:0006952                     defense response  2.088741e-19   \n",
            "4  GO:BP  GO:0050776        regulation of immune response  1.112364e-17   \n",
            "\n",
            "   significant                                        description  term_size  \\\n",
            "0         True  \"Any immune system process that functions in t...       2070   \n",
            "1         True  \"Any process involved in the development or fu...       2871   \n",
            "2         True  \"Any process that modulates the frequency, rat...       1581   \n",
            "3         True  \"Reactions, triggered in response to the prese...       1853   \n",
            "4         True  \"Any process that modulates the frequency, rat...        947   \n",
            "\n",
            "   query_size  intersection_size  effective_domain_size  precision    recall  \\\n",
            "0         185                 73                  21026   0.394595  0.035266   \n",
            "1         185                 85                  21026   0.459459  0.029606   \n",
            "2         185                 62                  21026   0.335135  0.039216   \n",
            "3         185                 64                  21026   0.345946  0.034539   \n",
            "4         185                 45                  21026   0.243243  0.047518   \n",
            "\n",
            "     query                               parents  \\\n",
            "0  query_1              [GO:0002376, GO:0050896]   \n",
            "1  query_1                          [GO:0008150]   \n",
            "2  query_1              [GO:0002376, GO:0050789]   \n",
            "3  query_1                          [GO:0006950]   \n",
            "4  query_1  [GO:0002682, GO:0006955, GO:0048583]   \n",
            "\n",
            "                                       intersections  \\\n",
            "0  [HLA-DPA1, IFITM3, BACH2, IFITM2, FCGR3A, S100...   \n",
            "1  [HLA-DPA1, IFITM3, ITGB1, BACH2, IFITM2, FCGR3...   \n",
            "2  [HLA-DPA1, FCGR3A, PLCB1, S100A8, HLA-DRB1, LS...   \n",
            "3  [HLA-DPA1, IFITM3, ITGB1, COTL1, IFITM2, FCGR3...   \n",
            "4  [HLA-DPA1, FCGR3A, S100A8, HLA-DRB1, CD74, HLA...   \n",
            "\n",
            "                                           evidences  \n",
            "0  [[IDA, IBA, NAS, IEA], [IDA, IMP, IBA, TAS, IE...  \n",
            "1  [[IDA, IMP, IBA, NAS, IEA], [IDA, IMP, IBA, TA...  \n",
            "2  [[IMP, IBA], [IDA], [ISS, IEA], [NAS], [IDA, I...  \n",
            "3  [[IDA], [IDA, IMP, IBA, IEA], [ISS, TAS, IEA],...  \n",
            "4  [[IBA], [IDA], [NAS], [IDA, IBA], [IBA, IEA], ...  \n",
            "\n",
            "--- Pathway Analysis for Dimension 4 ---\n",
            "  source      native                                 name       p_value  \\\n",
            "0  GO:BP  GO:0002682  regulation of immune system process  1.088617e-14   \n",
            "1  GO:BP  GO:0002376                immune system process  3.302929e-14   \n",
            "2  GO:BP  GO:0032501     multicellular organismal process  4.272460e-13   \n",
            "3  GO:CC  GO:0005737                            cytoplasm  7.556269e-13   \n",
            "4  GO:BP  GO:0001775                      cell activation  3.048345e-12   \n",
            "\n",
            "   significant                                        description  term_size  \\\n",
            "0         True  \"Any process that modulates the frequency, rat...       1581   \n",
            "1         True  \"Any process involved in the development or fu...       2871   \n",
            "2         True  \"Any biological process, occurring at the leve...       7322   \n",
            "3         True  \"The contents of a cell excluding the plasma m...      12527   \n",
            "4         True  \"A multicellular organismal process by which e...       1132   \n",
            "\n",
            "   query_size  intersection_size  effective_domain_size  precision    recall  \\\n",
            "0         179                 52                  21026   0.290503  0.032891   \n",
            "1         179                 70                  21026   0.391061  0.024382   \n",
            "2         179                116                  21026   0.648045  0.015843   \n",
            "3         188                157                  22149   0.835106  0.012533   \n",
            "4         179                 41                  21026   0.229050  0.036219   \n",
            "\n",
            "     query                   parents  \\\n",
            "0  query_1  [GO:0002376, GO:0050789]   \n",
            "1  query_1              [GO:0008150]   \n",
            "2  query_1              [GO:0008150]   \n",
            "3  query_1  [GO:0005622, GO:0110165]   \n",
            "4  query_1  [GO:0009987, GO:0032501]   \n",
            "\n",
            "                                       intersections  \\\n",
            "0  [ANXA1, PLCB1, LEF1, CD36, FCN1, TXK, CSF3R, R...   \n",
            "1  [ANXA1, BACH2, PLCB1, LEF1, CD36, LYZ, FCN1, T...   \n",
            "2  [VCAN, ANXA1, PLCB1, LEF1, CD36, FCN1, SLC8A1,...   \n",
            "3  [VCAN, ANXA1, BACH2, PLCB1, LEF1, DPYD, CD36, ...   \n",
            "4  [ANXA1, PLCB1, LEF1, ZEB2, BCL11B, CD247, JAK2...   \n",
            "\n",
            "                                           evidences  \n",
            "0  [[IDA, ISS, IEA], [ISS, IEA], [IDA, IMP, ISS, ...  \n",
            "1  [[IDA, IMP, ISS, IEA], [IMP], [ISS, IEA], [IDA...  \n",
            "2  [[IDA, HDA, IBA], [IDA, IMP, ISS, IEA], [ISS, ...  \n",
            "3  [[TAS], [IDA, ISS, IBA, TAS, IEA], [IDA, IEA],...  \n",
            "4  [[IDA, ISS, IEA], [ISS, IEA], [ISS, IEA], [IEA...  \n",
            "\n",
            "--- Pathway Analysis for Dimension 5 ---\n",
            "  source      native                           name       p_value  \\\n",
            "0  GO:BP  GO:0002376          immune system process  7.411843e-25   \n",
            "1  GO:BP  GO:0006955                immune response  5.103731e-22   \n",
            "2  GO:BP  GO:0050896           response to stimulus  5.214937e-22   \n",
            "3  GO:BP  GO:0006952               defense response  1.022680e-21   \n",
            "4  GO:BP  GO:0009605  response to external stimulus  1.104125e-21   \n",
            "\n",
            "   significant                                        description  term_size  \\\n",
            "0         True  \"Any process involved in the development or fu...       2871   \n",
            "1         True  \"Any immune system process that functions in t...       2070   \n",
            "2         True  \"Any process that results in a change in state...       8999   \n",
            "3         True  \"Reactions, triggered in response to the prese...       1853   \n",
            "4         True  \"Any process that results in a change in state...       2415   \n",
            "\n",
            "   query_size  intersection_size  effective_domain_size  precision    recall  \\\n",
            "0         185                 87                  21026   0.470270  0.030303   \n",
            "1         185                 71                  21026   0.383784  0.034300   \n",
            "2         185                148                  21026   0.800000  0.016446   \n",
            "3         185                 67                  21026   0.362162  0.036158   \n",
            "4         185                 76                  21026   0.410811  0.031470   \n",
            "\n",
            "     query                   parents  \\\n",
            "0  query_1              [GO:0008150]   \n",
            "1  query_1  [GO:0002376, GO:0050896]   \n",
            "2  query_1              [GO:0008150]   \n",
            "3  query_1              [GO:0006950]   \n",
            "4  query_1              [GO:0050896]   \n",
            "\n",
            "                                       intersections  \\\n",
            "0  [FCGR3A, IFITM3, AIF1, IFITM2, LST1, TNFSF10, ...   \n",
            "1  [FCGR3A, IFITM3, AIF1, IFITM2, LST1, TNFSF10, ...   \n",
            "2  [TCF7L2, FCGR3A, MTSS1, CDKN1C, IFITM3, AFF3, ...   \n",
            "3  [FCGR3A, IFITM3, AIF1, SERPINA1, IFITM2, GBP2,...   \n",
            "4  [FCGR3A, IFITM3, AIF1, IFITM2, GBP2, CTSS, COT...   \n",
            "\n",
            "                                           evidences  \n",
            "0  [[IDA, IMP, IBA, TAS, IEA], [IDA, IMP, IBA, TA...  \n",
            "1  [[IDA, IBA, TAS], [IDA, IMP, IBA, TAS, IEA], [...  \n",
            "2  [[IMP, ISS, IBA, IC, IEA], [IDA, IBA, TAS], [I...  \n",
            "3  [[IDA, IBA], [IDA, IMP, IBA, IEA], [IEP, ISS, ...  \n",
            "4  [[IDA], [IDA, IMP, IBA, IEA], [IDA, IEP, ISS],...  \n",
            "\n",
            "--- Pathway Analysis for Dimension 6 ---\n",
            "  source      native                   name       p_value  significant  \\\n",
            "0  GO:BP  GO:0042110      T cell activation  1.992777e-21         True   \n",
            "1  GO:BP  GO:0006955        immune response  3.775498e-21         True   \n",
            "2  GO:BP  GO:0046649  lymphocyte activation  6.955784e-20         True   \n",
            "3  GO:BP  GO:0001775        cell activation  2.040475e-19         True   \n",
            "4  GO:BP  GO:0002376  immune system process  5.129033e-18         True   \n",
            "\n",
            "                                         description  term_size  query_size  \\\n",
            "0  \"The change in morphology and behavior of a ma...        582         172   \n",
            "1  \"Any immune system process that functions in t...       2070         172   \n",
            "2  \"A change in morphology and behavior of a lymp...        814         172   \n",
            "3  \"A multicellular organismal process by which e...       1132         172   \n",
            "4  \"Any process involved in the development or fu...       2871         172   \n",
            "\n",
            "   intersection_size  effective_domain_size  precision    recall    query  \\\n",
            "0                 39                  21026   0.226744  0.067010  query_1   \n",
            "1                 67                  21026   0.389535  0.032367  query_1   \n",
            "2                 43                  21026   0.250000  0.052826  query_1   \n",
            "3                 49                  21026   0.284884  0.043286  query_1   \n",
            "4                 74                  21026   0.430233  0.025775  query_1   \n",
            "\n",
            "                    parents  \\\n",
            "0              [GO:0046649]   \n",
            "1  [GO:0002376, GO:0050896]   \n",
            "2              [GO:0045321]   \n",
            "3  [GO:0009987, GO:0032501]   \n",
            "4              [GO:0008150]   \n",
            "\n",
            "                                       intersections  \\\n",
            "0  [THEMIS, CD8B, CD8A, KLRK1, HLA-DRA, CCL5, STA...   \n",
            "1  [THEMIS, CD8B, CD8A, LYZ, IL32, KLRK1, HLA-DRA...   \n",
            "2  [THEMIS, CD8B, CD8A, KLRK1, HLA-DRA, CCL5, STA...   \n",
            "3  [THEMIS, CD8B, CD8A, KLRK1, HLA-DRA, CCL5, STA...   \n",
            "4  [THEMIS, CD8B, CD8A, LYZ, IL32, KLRK1, HLA-DRA...   \n",
            "\n",
            "                                           evidences  \n",
            "0  [[ISS, IEA], [IDA, NAS], [IDA, ISS, IBA, NAS],...  \n",
            "1  [[ISS, IBA, IEA], [IDA, NAS, IEA], [IDA, ISS, ...  \n",
            "2  [[ISS, IEA], [IDA, NAS], [IDA, ISS, IBA, NAS],...  \n",
            "3  [[ISS, IEA], [IDA, NAS], [IDA, ISS, IBA, NAS],...  \n",
            "4  [[ISS, IBA, IEA], [IDA, NAS, IEA], [IDA, ISS, ...  \n",
            "\n",
            "--- Pathway Analysis for Dimension 7 ---\n",
            "  source      native                                          name  \\\n",
            "0  GO:BP  GO:0006955                               immune response   \n",
            "1  GO:BP  GO:0002376                         immune system process   \n",
            "2  GO:BP  GO:0001775                               cell activation   \n",
            "3  GO:BP  GO:0002682           regulation of immune system process   \n",
            "4  GO:BP  GO:0002684  positive regulation of immune system process   \n",
            "\n",
            "        p_value  significant  \\\n",
            "0  2.752551e-22         True   \n",
            "1  1.203099e-21         True   \n",
            "2  2.818021e-20         True   \n",
            "3  3.269748e-19         True   \n",
            "4  4.044589e-19         True   \n",
            "\n",
            "                                         description  term_size  query_size  \\\n",
            "0  \"Any immune system process that functions in t...       2070         179   \n",
            "1  \"Any process involved in the development or fu...       2871         179   \n",
            "2  \"A multicellular organismal process by which e...       1132         179   \n",
            "3  \"Any process that modulates the frequency, rat...       1581         179   \n",
            "4  \"Any process that activates or increases the f...       1099         179   \n",
            "\n",
            "   intersection_size  effective_domain_size  precision    recall    query  \\\n",
            "0                 70                  21026   0.391061  0.033816  query_1   \n",
            "1                 81                  21026   0.452514  0.028213  query_1   \n",
            "2                 51                  21026   0.284916  0.045053  query_1   \n",
            "3                 58                  21026   0.324022  0.036686  query_1   \n",
            "4                 49                  21026   0.273743  0.044586  query_1   \n",
            "\n",
            "                                parents  \\\n",
            "0              [GO:0002376, GO:0050896]   \n",
            "1                          [GO:0008150]   \n",
            "2              [GO:0009987, GO:0032501]   \n",
            "3              [GO:0002376, GO:0050789]   \n",
            "4  [GO:0002376, GO:0002682, GO:0048518]   \n",
            "\n",
            "                                       intersections  \\\n",
            "0  [S100A9, S100A8, HLA-DRA, PDE4D, CD83, CD74, H...   \n",
            "1  [S100A9, S100A8, HLA-DRA, PDE4D, CD83, CD74, H...   \n",
            "2  [HLA-DRA, CD83, CD74, HLA-DRB1, HLA-DPB1, HLA-...   \n",
            "3  [S100A9, S100A8, HLA-DRA, PDE4D, CD83, CD74, H...   \n",
            "4  [S100A9, S100A8, HLA-DRA, PDE4D, CD83, CD74, H...   \n",
            "\n",
            "                                           evidences  \n",
            "0  [[IDA, IBA, NAS, IEA], [NAS, IEA], [IDA, IBA, ...  \n",
            "1  [[IDA, IBA, NAS, IEA], [IDA, IBA, NAS, IEA], [...  \n",
            "2  [[IDA, IBA], [IEA], [IMP, IBA, NAS, IEA], [IDA...  \n",
            "3  [[NAS], [NAS], [IDA, IBA], [IMP], [IEA], [IDA,...  \n",
            "4  [[NAS], [NAS], [IDA, IBA], [IMP], [IEA], [IDA,...  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Similarly, we apply the same analysis to the ATAC modality, using the appropriate libraries for chromatin accessibility data."
      ],
      "metadata": {
        "id": "mjwB99jCI88b"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "avg_abs_sens_peaks, top_peaks_by_dim = interpret_embedding_top_genes(\n",
        "    adata=mdata[\"atac\"],\n",
        "    alpha=beta,\n",
        "    K_centered=kernel_y_center,\n",
        "    gamma=gamma_y,\n",
        "    pca_key=\"X_pca\",\n",
        "    pcs_key=\"PCs\",\n",
        "    top_k=1000,\n",
        "    print_top=False  # set True to print\n",
        ")\n",
        "\n",
        "print(\"Top genes per dimension (names only):\")\n",
        "for d, genes in top_peaks_by_dim.items():\n",
        "    print(f\"Dim {d}: {genes[:10]} ...\")  # preview\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7WjJkBLXIauD",
        "outputId": "9ca091ca-e19f-4e5b-b07c-22bcb171da5d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Top genes per dimension (names only):\n",
            "Dim 0: ['chr6:44058491-44059286', 'chr2:1822751-1823661', 'chr2:86785399-86785901', 'chr12:10554267-10555223', 'chr2:86825765-86826652', 'chr20:50274666-50275585', 'chr6:44064154-44065076', 'chr19:47388776-47389731', 'chr9:134371498-134372350', 'chr19:28632185-28632982'] ...\n",
            "Dim 1: ['chr14:105817767-105818662', 'chr14:64235492-64236400', 'chr6:44058491-44059286', 'chr16:88069188-88070082', 'chr10:48671241-48672145', 'chr22:41935938-41936849', 'chr1:178605679-178606546', 'chr6:88141509-88142457', 'chr3:13110453-13111334', 'chr20:43681796-43682628'] ...\n",
            "Dim 2: ['chr12:52566610-52567512', 'chr17:36066303-36067216', 'chr4:8409189-8410117', 'chr12:121557431-121558344', 'chr14:24633897-24634800', 'chr7:121530228-121531217', 'chr1:202197397-202198309', 'chr20:20215859-20216748', 'chr2:231425721-231426641', 'chr1:25065672-25066545'] ...\n",
            "Dim 3: ['chr16:83744246-83745164', 'chr5:54774499-54775391', 'chr10:113169759-113170685', 'chr1:241964511-241965409', 'chr2:152499603-152500505', 'chr17:4759632-4760546', 'chr1:212632970-212633894', 'chr18:13568766-13569621', 'chr5:54772389-54773316', 'chr1:18160238-18161146'] ...\n",
            "Dim 4: ['chr16:83744246-83745164', 'chr10:113169759-113170685', 'chr17:4759632-4760546', 'chr5:54774499-54775391', 'chr1:241964511-241965409', 'chr18:13568766-13569621', 'chr6:108841157-108842083', 'chr21:45224239-45225152', 'chr2:152499603-152500505', 'chr21:43352469-43353375'] ...\n",
            "Dim 5: ['chr5:54774499-54775391', 'chr16:83744246-83745164', 'chr1:241964511-241965409', 'chr10:113169759-113170685', 'chr1:212632970-212633894', 'chr20:33724178-33725064', 'chr4:141332301-141333204', 'chr7:98641556-98642549', 'chr11:2836884-2837790', 'chr2:152499603-152500505'] ...\n",
            "Dim 6: ['chr2:86785399-86785901', 'chr12:10554267-10555223', 'chr2:86825765-86826652', 'chr2:86786009-86786618', 'chr9:126396796-126397681', 'chr2:86790654-86791467', 'chr5:172918590-172919495', 'chr2:86793463-86794314', 'chr11:122838071-122838960', 'chr22:26596749-26597659'] ...\n",
            "Dim 7: ['chr7:12584725-12585624', 'chr20:5657202-5658071', 'chr19:13831037-13831692', 'chr11:64497882-64498767', 'chr8:47526045-47526835', 'chr17:7094030-7094901', 'chr3:46404934-46405701', 'chr1:15261003-15261948', 'chr9:105154576-105155489', 'chr2:113124204-113124889'] ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!python --version\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ly1FXTdJcHjH",
        "outputId": "2cb3cda4-6d31-4bd6-8c90-1e79512d19e0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Python 3.11.13\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install gimmemotifs gprofiler\n",
        "!apt-get install bedtools\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "si4i1GVKcN2d",
        "outputId": "8e285876-c4f2-4a0a-f986-a40b15cdc210"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting gimmemotifs\n",
            "  Downloading gimmemotifs-0.18.0.tar.gz (12.2 MB)\n",
            "\u001b[?25l     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m0.0/12.2 MB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[91mâ”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m\u001b[91mâ•¸\u001b[0m\u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m4.2/12.2 MB\u001b[0m \u001b[31m133.8 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\r\u001b[2K     \u001b[91mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m\u001b[91mâ•¸\u001b[0m\u001b[90mâ”â”â”â”â”\u001b[0m \u001b[32m10.5/12.2 MB\u001b[0m \u001b[31m192.0 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\r\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m12.2/12.2 MB\u001b[0m \u001b[31m143.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting gprofiler\n",
            "  Downloading gprofiler-1.2.2-py2.py3-none-any.whl.metadata (340 bytes)\n",
            "Requirement already satisfied: setuptools>=0.7 in /usr/local/lib/python3.11/dist-packages (from gimmemotifs) (75.2.0)\n",
            "Collecting biofluff>=3.0.4 (from gimmemotifs)\n",
            "  Downloading biofluff-3.0.4.tar.gz (28 kB)\n",
            "  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting configparser (from gimmemotifs)\n",
            "  Downloading configparser-7.2.0-py3-none-any.whl.metadata (5.5 kB)\n",
            "Collecting diskcache (from gimmemotifs)\n",
            "  Downloading diskcache-5.6.3-py3-none-any.whl.metadata (20 kB)\n",
            "Collecting feather-format (from gimmemotifs)\n",
            "  Downloading feather-format-0.4.1.tar.gz (3.2 kB)\n",
            "  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting genomepy>=0.13.0 (from gimmemotifs)\n",
            "  Downloading genomepy-0.16.3-py3-none-any.whl.metadata (26 kB)\n",
            "Collecting iteround (from gimmemotifs)\n",
            "  Downloading iteround-1.0.4-py3-none-any.whl.metadata (4.2 kB)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from gimmemotifs) (3.1.6)\n",
            "Collecting logomaker (from gimmemotifs)\n",
            "  Downloading logomaker-0.8.7-py3-none-any.whl.metadata (2.6 kB)\n",
            "Collecting loguru (from gimmemotifs)\n",
            "  Downloading loguru-0.7.3-py3-none-any.whl.metadata (22 kB)\n",
            "Requirement already satisfied: matplotlib>=3.3 in /usr/local/lib/python3.11/dist-packages (from gimmemotifs) (3.10.0)\n",
            "Requirement already satisfied: numpy>=1.18 in /usr/local/lib/python3.11/dist-packages (from gimmemotifs) (1.26.4)\n",
            "Requirement already satisfied: pandas>=1.3.0 in /usr/local/lib/python3.11/dist-packages (from gimmemotifs) (2.2.2)\n",
            "Collecting pybedtools>=0.9.0 (from gimmemotifs)\n",
            "  Downloading pybedtools-0.12.0.tar.gz (12.6 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m12.6/12.6 MB\u001b[0m \u001b[31m24.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting pysam>=0.16 (from gimmemotifs)\n",
            "  Downloading pysam-0.23.3-cp311-cp311-manylinux_2_28_x86_64.whl.metadata (1.7 kB)\n",
            "Collecting xxhash (from gimmemotifs)\n",
            "  Downloading xxhash-3.6.0-cp311-cp311-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl.metadata (13 kB)\n",
            "Collecting qnorm>=0.8.1 (from gimmemotifs)\n",
            "  Downloading qnorm-0.9.0-py3-none-any.whl.metadata (12 kB)\n",
            "Requirement already satisfied: scikit-learn>=0.23.2 in /usr/local/lib/python3.11/dist-packages (from gimmemotifs) (1.6.1)\n",
            "Requirement already satisfied: scipy>=1.5 in /usr/local/lib/python3.11/dist-packages (from gimmemotifs) (1.15.3)\n",
            "Requirement already satisfied: seaborn>=0.10.1 in /usr/local/lib/python3.11/dist-packages (from gimmemotifs) (0.13.2)\n",
            "Requirement already satisfied: statsmodels in /usr/local/lib/python3.11/dist-packages (from gimmemotifs) (0.14.5)\n",
            "Requirement already satisfied: tqdm>=4.46.1 in /usr/local/lib/python3.11/dist-packages (from gimmemotifs) (4.67.1)\n",
            "Collecting xdg (from gimmemotifs)\n",
            "  Downloading xdg-6.0.0-py3-none-any.whl.metadata (1.3 kB)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from gprofiler) (2.32.3)\n",
            "Collecting HTSeq (from biofluff>=3.0.4->gimmemotifs)\n",
            "  Downloading HTSeq-2.0.9-cp311-cp311-manylinux_2_24_x86_64.manylinux_2_28_x86_64.whl.metadata (3.8 kB)\n",
            "Collecting palettable (from biofluff>=3.0.4->gimmemotifs)\n",
            "  Downloading palettable-3.3.3-py2.py3-none-any.whl.metadata (3.3 kB)\n",
            "Collecting pyBigWig (from biofluff>=3.0.4->gimmemotifs)\n",
            "  Downloading pyBigWig-0.3.24-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (19 kB)\n",
            "Collecting appdirs (from genomepy>=0.13.0->gimmemotifs)\n",
            "  Downloading appdirs-1.4.4-py2.py3-none-any.whl.metadata (9.0 kB)\n",
            "Collecting biopython>=1.73 (from genomepy>=0.13.0->gimmemotifs)\n",
            "  Downloading biopython-1.85-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (13 kB)\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.11/dist-packages (from genomepy>=0.13.0->gimmemotifs) (8.2.1)\n",
            "Collecting colorama (from genomepy>=0.13.0->gimmemotifs)\n",
            "  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)\n",
            "Requirement already satisfied: filelock>=3.5 in /usr/local/lib/python3.11/dist-packages (from genomepy>=0.13.0->gimmemotifs) (3.18.0)\n",
            "Collecting mygene (from genomepy>=0.13.0->gimmemotifs)\n",
            "  Downloading mygene-3.2.2-py2.py3-none-any.whl.metadata (10 kB)\n",
            "Collecting mysql-connector-python<=8.4 (from genomepy>=0.13.0->gimmemotifs)\n",
            "  Downloading mysql_connector_python-8.4.0-cp311-cp311-manylinux_2_17_x86_64.whl.metadata (1.9 kB)\n",
            "Collecting pyfaidx>=0.7.2.1 (from genomepy>=0.13.0->gimmemotifs)\n",
            "  Downloading pyfaidx-0.9.0.3-py3-none-any.whl.metadata (25 kB)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.11/dist-packages (from genomepy>=0.13.0->gimmemotifs) (6.0.2)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.3->gimmemotifs) (1.3.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.3->gimmemotifs) (0.12.1)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.3->gimmemotifs) (4.58.5)\n",
            "Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.3->gimmemotifs) (1.4.8)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.3->gimmemotifs) (25.0)\n",
            "Requirement already satisfied: pillow>=8 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.3->gimmemotifs) (11.3.0)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.3->gimmemotifs) (3.2.3)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.3->gimmemotifs) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas>=1.3.0->gimmemotifs) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas>=1.3.0->gimmemotifs) (2025.2)\n",
            "Requirement already satisfied: numba in /usr/local/lib/python3.11/dist-packages (from qnorm>=0.8.1->gimmemotifs) (0.61.2)\n",
            "Requirement already satisfied: joblib>=1.2.0 in /usr/local/lib/python3.11/dist-packages (from scikit-learn>=0.23.2->gimmemotifs) (1.5.1)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.11/dist-packages (from scikit-learn>=0.23.2->gimmemotifs) (3.6.0)\n",
            "Requirement already satisfied: pyarrow>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from feather-format->gimmemotifs) (20.0.0)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from jinja2->gimmemotifs) (3.0.2)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->gprofiler) (3.4.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->gprofiler) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->gprofiler) (2.4.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->gprofiler) (2025.7.14)\n",
            "Requirement already satisfied: patsy>=0.5.6 in /usr/local/lib/python3.11/dist-packages (from statsmodels->gimmemotifs) (1.0.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.7->matplotlib>=3.3->gimmemotifs) (1.17.0)\n",
            "Collecting biothings-client>=0.2.6 (from mygene->genomepy>=0.13.0->gimmemotifs)\n",
            "  Downloading biothings_client-0.4.1-py3-none-any.whl.metadata (10 kB)\n",
            "Requirement already satisfied: llvmlite<0.45,>=0.44.0dev0 in /usr/local/lib/python3.11/dist-packages (from numba->qnorm>=0.8.1->gimmemotifs) (0.44.0)\n",
            "Requirement already satisfied: httpx>=0.22.0 in /usr/local/lib/python3.11/dist-packages (from biothings-client>=0.2.6->mygene->genomepy>=0.13.0->gimmemotifs) (0.28.1)\n",
            "Requirement already satisfied: anyio in /usr/local/lib/python3.11/dist-packages (from httpx>=0.22.0->biothings-client>=0.2.6->mygene->genomepy>=0.13.0->gimmemotifs) (4.9.0)\n",
            "Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.11/dist-packages (from httpx>=0.22.0->biothings-client>=0.2.6->mygene->genomepy>=0.13.0->gimmemotifs) (1.0.9)\n",
            "Requirement already satisfied: h11>=0.16 in /usr/local/lib/python3.11/dist-packages (from httpcore==1.*->httpx>=0.22.0->biothings-client>=0.2.6->mygene->genomepy>=0.13.0->gimmemotifs) (0.16.0)\n",
            "Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.11/dist-packages (from anyio->httpx>=0.22.0->biothings-client>=0.2.6->mygene->genomepy>=0.13.0->gimmemotifs) (1.3.1)\n",
            "Requirement already satisfied: typing_extensions>=4.5 in /usr/local/lib/python3.11/dist-packages (from anyio->httpx>=0.22.0->biothings-client>=0.2.6->mygene->genomepy>=0.13.0->gimmemotifs) (4.14.1)\n",
            "Downloading gprofiler-1.2.2-py2.py3-none-any.whl (3.1 kB)\n",
            "Downloading genomepy-0.16.3-py3-none-any.whl (83 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m83.3/83.3 kB\u001b[0m \u001b[31m11.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pysam-0.23.3-cp311-cp311-manylinux_2_28_x86_64.whl (26.5 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m26.5/26.5 MB\u001b[0m \u001b[31m59.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading qnorm-0.9.0-py3-none-any.whl (14 kB)\n",
            "Downloading configparser-7.2.0-py3-none-any.whl (17 kB)\n",
            "Downloading diskcache-5.6.3-py3-none-any.whl (45 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m45.5/45.5 kB\u001b[0m \u001b[31m8.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading iteround-1.0.4-py3-none-any.whl (7.3 kB)\n",
            "Downloading logomaker-0.8.7-py3-none-any.whl (13.2 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m13.2/13.2 MB\u001b[0m \u001b[31m87.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading loguru-0.7.3-py3-none-any.whl (61 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m61.6/61.6 kB\u001b[0m \u001b[31m10.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading xdg-6.0.0-py3-none-any.whl (3.9 kB)\n",
            "Downloading xxhash-3.6.0-cp311-cp311-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl (193 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m193.9/193.9 kB\u001b[0m \u001b[31m28.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading biopython-1.85-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.3 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m3.3/3.3 MB\u001b[0m \u001b[31m143.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading mysql_connector_python-8.4.0-cp311-cp311-manylinux_2_17_x86_64.whl (19.5 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m19.5/19.5 MB\u001b[0m \u001b[31m141.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pyfaidx-0.9.0.3-py3-none-any.whl (29 kB)\n",
            "Downloading appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)\n",
            "Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)\n",
            "Downloading HTSeq-2.0.9-cp311-cp311-manylinux_2_24_x86_64.manylinux_2_28_x86_64.whl (2.2 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m2.2/2.2 MB\u001b[0m \u001b[31m145.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading mygene-3.2.2-py2.py3-none-any.whl (5.4 kB)\n",
            "Downloading palettable-3.3.3-py2.py3-none-any.whl (332 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m332.3/332.3 kB\u001b[0m \u001b[31m54.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pyBigWig-0.3.24-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (185 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m185.8/185.8 kB\u001b[0m \u001b[31m25.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading biothings_client-0.4.1-py3-none-any.whl (46 kB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m46.7/46.7 kB\u001b[0m \u001b[31m8.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hBuilding wheels for collected packages: gimmemotifs, biofluff, pybedtools, feather-format\n",
            "  Building wheel for gimmemotifs (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for gimmemotifs: filename=gimmemotifs-0.18.0-cp311-cp311-linux_x86_64.whl size=11234400 sha256=ae2bf4cb07ae412341ba55f72c045d411f0621ec500b6cef830c88c267ec5506\n",
            "  Stored in directory: /root/.cache/pip/wheels/d2/e0/bd/236585a8aaad86d1d25432e4ff6eaf69f5225fbdcb6be1352f\n",
            "  Building wheel for biofluff (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for biofluff: filename=biofluff-3.0.4-py3-none-any.whl size=31296 sha256=68f9790bd16eb2cb4bd4b0e89b3dda365b30413cffb931204c3c76128265a76a\n",
            "  Stored in directory: /root/.cache/pip/wheels/6f/a9/4b/fb42de9cae2c8805900bc90d33a6b42a027d0687d91c5e757b\n",
            "  Building wheel for pybedtools (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.12.0-cp311-cp311-linux_x86_64.whl size=14261449 sha256=fbcdba19c4041e2486da9d562eb3497635b75d8a83bd9b91fd6c9220243f7d6e\n",
            "  Stored in directory: /root/.cache/pip/wheels/95/62/57/e8c47c2a43884227b6644cf1d2dabf3162a0f40b9c1493c1f5\n",
            "  Building wheel for feather-format (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for feather-format: filename=feather_format-0.4.1-py3-none-any.whl size=2490 sha256=ce72c44185e7c401f8eec410807d9cd9990baaf7eceb4c7033a58267c0d20a82\n",
            "  Stored in directory: /root/.cache/pip/wheels/77/5b/0e/0e63d10b6353208a085a321ea2eed2578f220a77bb8a4bd7ab\n",
            "Successfully built gimmemotifs biofluff pybedtools feather-format\n",
            "Installing collected packages: iteround, appdirs, xxhash, xdg, pysam, pyfaidx, pyBigWig, palettable, mysql-connector-python, loguru, feather-format, diskcache, configparser, colorama, biopython, qnorm, HTSeq, pybedtools, logomaker, gprofiler, biothings-client, mygene, biofluff, genomepy, gimmemotifs\n",
            "Successfully installed HTSeq-2.0.9 appdirs-1.4.4 biofluff-3.0.4 biopython-1.85 biothings-client-0.4.1 colorama-0.4.6 configparser-7.2.0 diskcache-5.6.3 feather-format-0.4.1 genomepy-0.16.3 gimmemotifs-0.18.0 gprofiler-1.2.2 iteround-1.0.4 logomaker-0.8.7 loguru-0.7.3 mygene-3.2.2 mysql-connector-python-8.4.0 palettable-3.3.3 pyBigWig-0.3.24 pybedtools-0.12.0 pyfaidx-0.9.0.3 pysam-0.23.3 qnorm-0.9.0 xdg-6.0.0 xxhash-3.6.0\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "backports"
                ]
              },
              "id": "41be3b836ea34a739e679f16edee3bc5"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 2 not upgraded.\n",
            "Need to get 563 kB of archives.\n",
            "After this operation, 1,548 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 bedtools amd64 2.30.0+dfsg-2ubuntu0.1 [563 kB]\n",
            "Fetched 563 kB in 0s (3,134 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 122124 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.30.0+dfsg-2ubuntu0.1_amd64.deb ...\n",
            "Unpacking bedtools (2.30.0+dfsg-2ubuntu0.1) ...\n",
            "Setting up bedtools (2.30.0+dfsg-2ubuntu0.1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install \"pandas<2.0\"  # e.g., 1.5.3\n",
        "# then rerun your command\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 439
        },
        "id": "KX8bK7BGDFOi",
        "outputId": "287989c8-cc6d-4187-9508-3161808187b9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting pandas<2.0\n",
            "  Downloading pandas-1.5.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (11 kB)\n",
            "Requirement already satisfied: python-dateutil>=2.8.1 in /usr/local/lib/python3.11/dist-packages (from pandas<2.0) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas<2.0) (2025.2)\n",
            "Requirement already satisfied: numpy>=1.21.0 in /usr/local/lib/python3.11/dist-packages (from pandas<2.0) (1.26.4)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.8.1->pandas<2.0) (1.17.0)\n",
            "Downloading pandas-1.5.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.0 MB)\n",
            "\u001b[?25l   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m0.0/12.0 MB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K   \u001b[91mâ”â”â”â”â”â”â”\u001b[0m\u001b[90mâ•º\u001b[0m\u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m2.2/12.0 MB\u001b[0m \u001b[31m66.1 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\r\u001b[2K   \u001b[91mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m\u001b[91mâ•¸\u001b[0m \u001b[32m12.0/12.0 MB\u001b[0m \u001b[31m329.6 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\r\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m12.0/12.0 MB\u001b[0m \u001b[31m218.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: pandas\n",
            "  Attempting uninstall: pandas\n",
            "    Found existing installation: pandas 2.2.2\n",
            "    Uninstalling pandas-2.2.2:\n",
            "      Successfully uninstalled pandas-2.2.2\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "anndata 0.12.3 requires pandas!=2.1.2,>=2.1.0, but you have pandas 1.5.3 which is incompatible.\n",
            "google-colab 1.0.0 requires pandas==2.2.2, but you have pandas 1.5.3 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed pandas-1.5.3\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "pandas"
                ]
              },
              "id": "402182af489642b4b5b72a335d073f90"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "pip install pyranges"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "h8LJv0ioJOb3",
        "outputId": "84ed9253-9725-4d33-f312-0330429fcc9a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: pyranges in /usr/local/lib/python3.11/dist-packages (0.1.4)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (from pyranges) (1.5.3)\n",
            "Requirement already satisfied: ncls>=0.0.63 in /usr/local/lib/python3.11/dist-packages (from pyranges) (0.0.70)\n",
            "Requirement already satisfied: tabulate in /usr/local/lib/python3.11/dist-packages (from pyranges) (0.9.0)\n",
            "Requirement already satisfied: sorted_nearest>=0.0.33 in /usr/local/lib/python3.11/dist-packages (from pyranges) (0.0.41)\n",
            "Requirement already satisfied: natsort in /usr/local/lib/python3.11/dist-packages (from pyranges) (8.4.0)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from ncls>=0.0.63->pyranges) (1.26.4)\n",
            "Requirement already satisfied: python-dateutil>=2.8.1 in /usr/local/lib/python3.11/dist-packages (from pandas->pyranges) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas->pyranges) (2025.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.8.1->pandas->pyranges) (1.17.0)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We download the hg38 reference genome FASTA (hg38.fa.gz) from UCSC because ATAC-seq peaks are defined by genomic coordinates, and downstream analyses such as peak annotation, motif enrichment, and peak-to-gene mapping require access to the underlying DNA sequence."
      ],
      "metadata": {
        "id": "A2AHe1RJLJBg"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir -p genome_data\n"
      ],
      "metadata": {
        "id": "Ek6uBQp6ffp6"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "cd genome_data\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WpWBBxWn1UYC",
        "outputId": "61322fe5-86c9-4373-f618-21404e44a9fa"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/genome_data\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "cI1z50ci1mBY",
        "outputId": "594a791b-2668-4198-c1f0-4ec382f9e648"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-10-21 12:55:06--  http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz\n",
            "Resolving hgdownload.cse.ucsc.edu (hgdownload.cse.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.cse.ucsc.edu (hgdownload.cse.ucsc.edu)|128.114.119.163|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 983659424 (938M) [application/x-gzip]\n",
            "Saving to: â€˜hg38.fa.gzâ€™\n",
            "\n",
            "hg38.fa.gz          100%[===================>] 938.09M  17.0MB/s    in 56s     \n",
            "\n",
            "2025-10-21 12:56:03 (16.8 MB/s) - â€˜hg38.fa.gzâ€™ saved [983659424/983659424]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gunzip hg38.fa.gz"
      ],
      "metadata": {
        "id": "IX6B6mST1mjV"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "cd .."
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "A70-cfWm2Ms3",
        "outputId": "2626a0da-5268-4559-e4bc-3ff178362f0d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We export all ATAC peaks to a BED file to provide a standard coordinate input (and a dataset-specific background universe) for downstream analyses such as motif enrichment and peak-based annotation."
      ],
      "metadata": {
        "id": "r3uxkwHhLkA9"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "# Usage:\n",
        "write_all_peaks_to_bed(mdata['atac'], \"all_atac_peaks.bed\")\n",
        "print(\"Wrote all ATAC peaks to all_atac_peaks.bed.\")\n",
        "# Hypothetical: You have a file \"all_atac_peaks.bed\" in your environment.\n",
        "# If not, you can create it similarly.\n",
        "BACKGROUND_BED = \"all_atac_peaks.bed\"\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sFRqnrBp2NH3",
        "outputId": "ddd94bcc-ffe4-4110-fa3d-433c41963f97"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Wrote all ATAC peaks to all_atac_peaks.bed.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We use bedtools getfasta to extract the DNA sequences corresponding to ATAC peak coordinates from the hg38 reference genome, generating a FASTA file for downstream sequence-based analyses."
      ],
      "metadata": {
        "id": "yz2hnjzmL3Yh"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta \\\n",
        "    -fi genome_data/hg38.fa \\\n",
        "    -bed all_atac_peaks.bed \\\n",
        "    -fo all_atac_peaks.fa\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JXIRfI5e2NoB",
        "outputId": "03eb3c86-1231-43a5-af61-cc999b59b57d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "index file genome_data/hg38.fa.fai not found, generating...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "For each latent dimension, we take the corresponding set of top ATAC peaks, write their genomic intervals to a dimension-specific BED file, extract the underlying hg38 DNA sequences into a FASTA file using bedtools getfasta, and then run gimmeMotifs on that FASTA to identify motifs enriched in that dimensionâ€™s peaks, using the full genome (-g hg38.fa) to enable GC-aware background normalization."
      ],
      "metadata": {
        "id": "-8WZmOnjSyJT"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "\n",
        "# ----------------------------------------------\n",
        "# MAIN WORKFLOW: For each dimension, do BED->FA->Gimme\n",
        "# ----------------------------------------------\n",
        "\n",
        "for dim, peaks in top_peaks_by_dim.items():\n",
        "    # 1) Write a BED file for this dimension\n",
        "    bed_file = f\"dim_{dim}_peaks.bed\"\n",
        "    write_peaks_to_bed(peaks, bed_file)\n",
        "\n",
        "    # 2) Convert BED -> FASTA\n",
        "    fa_file = f\"dim_{dim}_peaks.fa\"\n",
        "    bed_to_fasta(bed_file, fa_file, REFERENCE_GENOME)\n",
        "\n",
        "    # 3) Run gimme motifs with -g to enable GC z-scoring\n",
        "    #    If you want original peak lengths, pass keep_original_size=True\n",
        "    out_dir = f\"gimme_dim_{dim}_fa_withGenome_1000_peak\"\n",
        "    run_gimme_motifs_with_genome(fa_file, out_dir, REFERENCE_GENOME, keep_original_size=True)\n",
        "\n",
        "    print(f\"Finished dimension {dim}\\n\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lsOmZrDICudi",
        "outputId": "69156208-e74d-4f00-ce35-02d27206410b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Running: /usr/bin/bedtools getfasta -fi genome_data/hg38.fa -bed dim_0_peaks.bed -fo dim_0_peaks.fa\n",
            "Success! Created dim_0_peaks.fa from dim_0_peaks.bed.\n",
            "Running: gimme motifs dim_0_peaks.fa gimme_dim_0_fa_withGenome_1000_peak -g genome_data/hg38.fa --size 0\n",
            "Success! gimme motifs results are in: gimme_dim_0_fa_withGenome_1000_peak/report.html\n",
            "Finished dimension 0\n",
            "\n",
            "Running: /usr/bin/bedtools getfasta -fi genome_data/hg38.fa -bed dim_1_peaks.bed -fo dim_1_peaks.fa\n",
            "Success! Created dim_1_peaks.fa from dim_1_peaks.bed.\n",
            "Running: gimme motifs dim_1_peaks.fa gimme_dim_1_fa_withGenome_1000_peak -g genome_data/hg38.fa --size 0\n",
            "Success! gimme motifs results are in: gimme_dim_1_fa_withGenome_1000_peak/report.html\n",
            "Finished dimension 1\n",
            "\n",
            "Running: /usr/bin/bedtools getfasta -fi genome_data/hg38.fa -bed dim_2_peaks.bed -fo dim_2_peaks.fa\n",
            "Success! Created dim_2_peaks.fa from dim_2_peaks.bed.\n",
            "Running: gimme motifs dim_2_peaks.fa gimme_dim_2_fa_withGenome_1000_peak -g genome_data/hg38.fa --size 0\n",
            "Success! gimme motifs results are in: gimme_dim_2_fa_withGenome_1000_peak/report.html\n",
            "Finished dimension 2\n",
            "\n",
            "Running: /usr/bin/bedtools getfasta -fi genome_data/hg38.fa -bed dim_3_peaks.bed -fo dim_3_peaks.fa\n",
            "Success! Created dim_3_peaks.fa from dim_3_peaks.bed.\n",
            "Running: gimme motifs dim_3_peaks.fa gimme_dim_3_fa_withGenome_1000_peak -g genome_data/hg38.fa --size 0\n",
            "Success! gimme motifs results are in: gimme_dim_3_fa_withGenome_1000_peak/report.html\n",
            "Finished dimension 3\n",
            "\n",
            "Running: /usr/bin/bedtools getfasta -fi genome_data/hg38.fa -bed dim_4_peaks.bed -fo dim_4_peaks.fa\n",
            "Success! Created dim_4_peaks.fa from dim_4_peaks.bed.\n",
            "Running: gimme motifs dim_4_peaks.fa gimme_dim_4_fa_withGenome_1000_peak -g genome_data/hg38.fa --size 0\n",
            "Success! gimme motifs results are in: gimme_dim_4_fa_withGenome_1000_peak/report.html\n",
            "Finished dimension 4\n",
            "\n",
            "Running: /usr/bin/bedtools getfasta -fi genome_data/hg38.fa -bed dim_5_peaks.bed -fo dim_5_peaks.fa\n",
            "Success! Created dim_5_peaks.fa from dim_5_peaks.bed.\n",
            "Running: gimme motifs dim_5_peaks.fa gimme_dim_5_fa_withGenome_1000_peak -g genome_data/hg38.fa --size 0\n",
            "Success! gimme motifs results are in: gimme_dim_5_fa_withGenome_1000_peak/report.html\n",
            "Finished dimension 5\n",
            "\n",
            "Running: /usr/bin/bedtools getfasta -fi genome_data/hg38.fa -bed dim_6_peaks.bed -fo dim_6_peaks.fa\n",
            "Success! Created dim_6_peaks.fa from dim_6_peaks.bed.\n",
            "Running: gimme motifs dim_6_peaks.fa gimme_dim_6_fa_withGenome_1000_peak -g genome_data/hg38.fa --size 0\n",
            "Success! gimme motifs results are in: gimme_dim_6_fa_withGenome_1000_peak/report.html\n",
            "Finished dimension 6\n",
            "\n",
            "Running: /usr/bin/bedtools getfasta -fi genome_data/hg38.fa -bed dim_7_peaks.bed -fo dim_7_peaks.fa\n",
            "Success! Created dim_7_peaks.fa from dim_7_peaks.bed.\n",
            "Running: gimme motifs dim_7_peaks.fa gimme_dim_7_fa_withGenome_1000_peak -g genome_data/hg38.fa --size 0\n",
            "Success! gimme motifs results are in: gimme_dim_7_fa_withGenome_1000_peak/report.html\n",
            "Finished dimension 7\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We then convert ATAC peak intervals and RNA gene transcription start sites (TSSs) into PyRanges genomic objects by parsing their chr:start-end annotations (and strand-aware TSS positions for genes), providing a standardized coordinate framework for downstream peakâ€“gene association analyses."
      ],
      "metadata": {
        "id": "R0lsGx5Wd9-Y"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "atac_adata = mdata['atac'].copy()\n",
        "\n",
        "import scanpy as sc\n",
        "import pandas as pd\n",
        "import pyranges as pr\n",
        "\n",
        "# 1) Load your scATAC AnnData\n",
        "\n",
        "# We expect atac_adata.var['interval'] to have something like 'chrN:start-end'\n",
        "# Parse into separate columns: Chromosome, Start, End\n",
        "interval_split = atac_adata.var[\"interval\"].str.split(\":\", expand=True)\n",
        "atac_adata.var[\"chr\"] = interval_split[0]\n",
        "\n",
        "start_end = interval_split[1].str.split(\"-\", expand=True)\n",
        "atac_adata.var[\"start\"] = start_end[0].astype(int)\n",
        "atac_adata.var[\"end\"]   = start_end[1].astype(int)\n",
        "\n",
        "# Create a BED-like DataFrame for peaks\n",
        "peaks_df = pd.DataFrame({\n",
        "    \"Chromosome\": atac_adata.var[\"chr\"],\n",
        "    \"Start\": atac_adata.var[\"start\"],\n",
        "    \"End\": atac_adata.var[\"end\"],\n",
        "    \"peak_id\": atac_adata.var_names  # or any identifier for these peaks\n",
        "}).reset_index(drop=True)\n",
        "\n",
        "# Drop missing/NaN if necessary\n",
        "peaks_df = peaks_df.dropna()\n",
        "\n",
        "# Convert to PyRanges\n",
        "peaks_pr = pr.PyRanges(peaks_df)\n",
        "peaks_pr\n",
        "\n",
        "\n",
        "rna_adata =  mdata['rna'].copy()\n",
        "# Split out chr, start, end from rna_adata.var[\"interval\"]\n",
        "interval_split = rna_adata.var[\"interval\"].str.split(\":\", expand=True)\n",
        "rna_adata.var[\"chr\"] = interval_split[0]\n",
        "\n",
        "start_end = interval_split[1].str.split(\"-\", expand=True)\n",
        "rna_adata.var[\"start_raw\"] = start_end[0].astype(int)\n",
        "rna_adata.var[\"end_raw\"]   = start_end[1].astype(int)\n",
        "\n",
        "def get_tss(row):\n",
        "    # If there's a \"strand\" column, use it:\n",
        "    if row[\"strand\"] == \"+\":\n",
        "        return row[\"start_raw\"]\n",
        "    else:\n",
        "        return row[\"end_raw\"]\n",
        "\n",
        "rna_adata.var[\"tss\"] = rna_adata.var.apply(get_tss, axis=1)\n",
        "\n",
        "genes_df = pd.DataFrame({\n",
        "    \"Chromosome\": rna_adata.var[\"chr\"],\n",
        "    \"Start\": rna_adata.var[\"tss\"],\n",
        "    \"End\": rna_adata.var[\"tss\"] + 1,  # TSS as a 1-bp region\n",
        "    \"gene_id\": rna_adata.var_names,   # or rna_adata.var[\"gene_ids\"]\n",
        "    \"strand\": rna_adata.var[\"strand\"]\n",
        "}).dropna()\n",
        "\n",
        "genes_df[\"Start\"] = genes_df[\"Start\"].astype(int)\n",
        "genes_df[\"End\"]   = genes_df[\"End\"].astype(int)\n",
        "\n",
        "genes_pr = pr.PyRanges(genes_df)\n",
        "genes_pr\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 424
        },
        "id": "AUfgjzKzN1y2",
        "outputId": "7643f76f-03aa-4495-d7fb-075093311e04"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "+--------------+-----------+-----------+------------+--------------+\n",
              "| Chromosome   | Start     | End       | gene_id    | strand       |\n",
              "| (category)   | (int64)   | (int64)   | (object)   | (category)   |\n",
              "|--------------+-----------+-----------+------------+--------------|\n",
              "| chr1         | 1000172   | 1000173   | HES4       | -            |\n",
              "| chr1         | 1001137   | 1001138   | ISG15      | +            |\n",
              "| chr1         | 1206592   | 1206593   | TNFRSF18   | -            |\n",
              "| chr1         | 1214153   | 1214154   | TNFRSF4    | -            |\n",
              "| ...          | ...       | ...       | ...        | ...          |\n",
              "| chrX         | 153830548 | 153830549 | PDZD4      | -            |\n",
              "| chrX         | 155022723 | 155022724 | F8         | -            |\n",
              "| chrX         | 155334614 | 155334615 | CLIC2      | -            |\n",
              "| chrX         | 155466539 | 155466540 | TMLHE-AS1  | +            |\n",
              "+--------------+-----------+-----------+------------+--------------+\n",
              "Unstranded PyRanges object has 3,943 rows and 5 columns from 23 chromosomes.\n",
              "For printing, the PyRanges was sorted on Chromosome."
            ],
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "      <th>gene_id</th>\n",
              "      <th>strand</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>1000172</td>\n",
              "      <td>1000173</td>\n",
              "      <td>HES4</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr1</td>\n",
              "      <td>1001137</td>\n",
              "      <td>1001138</td>\n",
              "      <td>ISG15</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr1</td>\n",
              "      <td>1206592</td>\n",
              "      <td>1206593</td>\n",
              "      <td>TNFRSF18</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr1</td>\n",
              "      <td>1214153</td>\n",
              "      <td>1214154</td>\n",
              "      <td>TNFRSF4</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr1</td>\n",
              "      <td>1449688</td>\n",
              "      <td>1449689</td>\n",
              "      <td>ATAD3C</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3938</th>\n",
              "      <td>chrX</td>\n",
              "      <td>148500616</td>\n",
              "      <td>148500617</td>\n",
              "      <td>AFF2</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3939</th>\n",
              "      <td>chrX</td>\n",
              "      <td>153830548</td>\n",
              "      <td>153830549</td>\n",
              "      <td>PDZD4</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3940</th>\n",
              "      <td>chrX</td>\n",
              "      <td>155022723</td>\n",
              "      <td>155022724</td>\n",
              "      <td>F8</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3941</th>\n",
              "      <td>chrX</td>\n",
              "      <td>155334614</td>\n",
              "      <td>155334615</td>\n",
              "      <td>CLIC2</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3942</th>\n",
              "      <td>chrX</td>\n",
              "      <td>155466539</td>\n",
              "      <td>155466540</td>\n",
              "      <td>TMLHE-AS1</td>\n",
              "      <td>+</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3943 rows Ã— 5 columns</p>\n",
              "</div>"
            ]
          },
          "metadata": {},
          "execution_count": 46
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "ATAC side:\n",
        "gimmeMotifs motif BED files give genomic motif hits\n",
        "(motif hit â†’ ATAC peak).\n",
        "\n",
        "RNA side:\n",
        "genes are represented by strand-aware TSS coordinates\n",
        "(gene â†’ TSS).\n",
        "\n",
        "Integration:\n",
        "we map motif-containing peaks to the nearest gene TSS within Â±20 kb, and\n",
        "intersect with top RNA genes for the same dimension, yielding\n",
        "motif â†’ peak â†’ gene (supported by RNA).\n",
        "\n"
      ],
      "metadata": {
        "id": "6SRf6GIFf8rw"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "import glob\n",
        "import pyranges as pr\n",
        "\n",
        "base_dir = \"/content/\"\n",
        "max_dist = 20_000  # Adjust as needed\n",
        "\n",
        "# Suppose you already have these PyRanges loaded:\n",
        "# peaks_pr = ...\n",
        "# genes_pr = ...\n",
        "\n",
        "# Suppose you also have a dictionary/list of top genes: top_genes_by_dim[i]\n",
        "# for dimension i. If not, adjust as needed.\n",
        "# top_genes_by_dim = { 0: [...], 1: [...], ... }\n",
        "\n",
        "results_dict = {}\n",
        "\n",
        "# 1) Find all subdirectories that look like \"gimme_dim_XXX_fa_withGenome_1000_peak\"\n",
        "dim_dirs = [\n",
        "    d\n",
        "    for d in os.listdir(base_dir)\n",
        "    if d.startswith(\"gimme_dim_\")\n",
        "       and os.path.isdir(os.path.join(base_dir, d))\n",
        "]\n",
        "\n",
        "for dim_dir in dim_dirs:\n",
        "    # Extract the dimension number from the directory name\n",
        "    split_parts = dim_dir.split(\"_\")\n",
        "    dim_str = split_parts[2]  # e.g. \"0\"\n",
        "    try:\n",
        "        dim = int(dim_str)\n",
        "    except ValueError:\n",
        "        print(f\"Warning: Could not parse dimension from directory {dim_dir}, skipping.\")\n",
        "        continue\n",
        "\n",
        "    print(f\"\\n=== Processing directory '{dim_dir}' (dimension {dim}) ===\")\n",
        "    motif_dir = os.path.join(base_dir, dim_dir, \"motif_scan_results\")\n",
        "\n",
        "    if not os.path.isdir(motif_dir):\n",
        "        print(f\"No motif_scan_results found in {dim_dir}. Skipping.\")\n",
        "        continue\n",
        "\n",
        "    # 2) Find all .bed files in motif_scan_results\n",
        "    bed_files = glob.glob(os.path.join(motif_dir, \"*.bed\"))\n",
        "    if not bed_files:\n",
        "        print(f\"No .bed files found in {motif_dir}.\")\n",
        "        continue\n",
        "\n",
        "    results_dict[dim] = {}\n",
        "\n",
        "    # 3) Loop over each BED file\n",
        "    for bed_file in bed_files:\n",
        "        bed_basename = os.path.basename(bed_file)\n",
        "\n",
        "        # 3a) We want to read the *first data line* to get the motif name from column 4\n",
        "        motif_name = None\n",
        "        cleaned_bed_path = bed_file.replace(\".bed\", \".cleaned.bed\")\n",
        "\n",
        "        with open(bed_file, \"r\") as infile, open(cleaned_bed_path, \"w\") as outfile:\n",
        "            for line in infile:\n",
        "                if line.startswith(\"#\"):\n",
        "                    # Skip writing commented lines\n",
        "                    continue\n",
        "                # If this is the first data line, extract motif name from 4th column\n",
        "                if motif_name is None:\n",
        "                    fields = line.strip().split(\"\\t\")\n",
        "                    if len(fields) >= 4:\n",
        "                        motif_name = fields[3]  # 4th column\n",
        "                    else:\n",
        "                        motif_name = \"UnknownMotif\"\n",
        "                # Write the line to the \"cleaned\" file\n",
        "                outfile.write(line)\n",
        "\n",
        "        if motif_name is None:\n",
        "            print(f\"  -> {bed_basename} has no data lines or no motif name found!\")\n",
        "            continue\n",
        "\n",
        "        print(f\"  -> Processing motif: {motif_name}\")\n",
        "\n",
        "        # 3b) Load the cleaned bed with PyRanges\n",
        "        motif_hits_pr = pr.read_bed(cleaned_bed_path)\n",
        "\n",
        "        # 3c) Join motif hits with peaks\n",
        "        peak_motif_joined = peaks_pr.join(motif_hits_pr)\n",
        "        peak_motif_df = peak_motif_joined.as_df()\n",
        "\n",
        "        # 3d) Convert to PyRanges\n",
        "        peak_motif_pr = pr.PyRanges(\n",
        "            peak_motif_df.rename(\n",
        "                columns={\n",
        "                    \"peak_chrom\": \"Chromosome\",\n",
        "                    \"peak_start\": \"Start\",\n",
        "                    \"peak_end\":   \"End\"\n",
        "                }\n",
        "            )\n",
        "        )\n",
        "\n",
        "        # 3e) Nearest gene TSS\n",
        "        nearest_result = peak_motif_pr.nearest(genes_pr)\n",
        "        nearest_df = nearest_result.as_df()\n",
        "\n",
        "        # 3f) Filter by distance\n",
        "        filtered_df = nearest_df[nearest_df[\"Distance\"] <= max_dist]\n",
        "        print(f\"     Found {len(filtered_df)} motif-peak-gene combos within Â±{max_dist} bp.\")\n",
        "\n",
        "        # 3g) Grab gene IDs\n",
        "        if \"gene_id_b\" in filtered_df.columns:\n",
        "            motif_genes = set(filtered_df[\"gene_id_b\"])\n",
        "        else:\n",
        "            motif_genes = set(filtered_df[\"gene_id\"])\n",
        "\n",
        "        # 3h) Intersect with top genes for this dimension\n",
        "        if dim in top_genes_by_dim:\n",
        "            top_genes_dim_set = set(top_genes_by_dim[dim])\n",
        "            common_genes = motif_genes.intersection(top_genes_dim_set)\n",
        "            overlap_count = len(common_genes)\n",
        "        else:\n",
        "            common_genes = set()\n",
        "            overlap_count = 0\n",
        "\n",
        "        print(f\"     Overlap with top genes (dim {dim}): {overlap_count}\")\n",
        "\n",
        "        # 3i) Store by the *motif name* rather than the file name\n",
        "        results_dict[dim][motif_name] = {\n",
        "            \"motif_genes\": motif_genes,\n",
        "            \"common_genes\": common_genes,\n",
        "            \"motif_genes_count\": len(motif_genes),\n",
        "            \"common_genes_count\": overlap_count\n",
        "        }\n",
        "\n",
        "# After processing all directories:\n",
        "print(\"\\n=== Summary ===\")\n",
        "for dim, motif_dict in results_dict.items():\n",
        "    print(f\"\\nDimension {dim} results:\")\n",
        "    for motif_name, info in motif_dict.items():\n",
        "        print(f\"  Motif: {motif_name}\")\n",
        "        print(f\"    # motif genes: {info['motif_genes_count']}\")\n",
        "        print(f\"    # overlapping with top genes: {info['common_genes_count']}\")\n",
        "        # Optionally, print the actual overlapping gene IDs\n",
        "        # print(f\"    Overlapping genes: {info['common_genes']}\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZO0djRSyPJoZ",
        "outputId": "3d2a7637-9de0-4477-9147-5723d1637480"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "=== Processing directory 'gimme_dim_7_fa_withGenome_1000_peak' (dimension 7) ===\n",
            "  -> Processing motif: GM.5.0.Ets.0037\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.STAT.0006\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 11 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0010\n",
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0017\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 18 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 3\n",
            "  -> Processing motif: GM.5.0.Rel.0015\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0037\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0013\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 11 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n",
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0019\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.STAT.0006\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 11 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0019\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0017\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0021\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Rel.0003\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0052\n",
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Rel.0003\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.STAT.0006\n",
            "     Found 11 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.Unknown.0108\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0052\n",
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 11 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0017\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0013\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0033\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0033\n",
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0021\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0106\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0014\n",
            "     Found 11 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0021\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0013\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 11 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 11 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 1\n",
            "  -> Processing motif: GM.5.0.IRF.0019\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0106\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0106\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0052\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Rel.0003\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0049\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0010\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n",
            "     Found 18 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 3\n",
            "  -> Processing motif: GM.5.0.Ets.0010\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0019\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0037\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0002\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GimmeMotifs_1\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 2\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0033\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 7): 1\n",
            "\n",
            "=== Processing directory 'gimme_dim_6_fa_withGenome_1000_peak' (dimension 6) ===\n",
            "  -> Processing motif: GM.5.0.bZIP.0016\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.bHLH.0119\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0006\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.IRF.0002\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.AT_hook.0006\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n",
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0119\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0108\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.Ets.0017\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0032\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.Sox.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0017\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0119\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.IRF.0002\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0197\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0002\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0014\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.Paired_box.0017\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.AT_hook.0006\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0197\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Sox.0013\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0108\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0057\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0017\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0119\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.C2H2_ZF.0014\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0032\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.Unknown.0197\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.Paired_box.0017\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0108\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 15 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 4\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0057\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0108\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0017\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n",
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.Ets.0017\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0013\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n",
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0014\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0013\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.bHLH.0119\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.bHLH.0119\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0006\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.RFX.0010\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Sox.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0014\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0119\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0016\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0014\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0057\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0032\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0006\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.Paired_box.0017\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.Paired_box.0017\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.RFX.0010\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.Runt.0002\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0016\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0002\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.AT_hook.0006\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0119\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Paired_box.0017\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.AT_hook.0006\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0108\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.AT_hook.0006\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.RFX.0010\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0002\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 6): 1\n",
            "\n",
            "=== Processing directory 'gimme_dim_5_fa_withGenome_1000_peak' (dimension 5) ===\n",
            "  -> Processing motif: GM.5.0.bZIP.0039\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0168\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0009\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0010\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0005\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0039\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0066\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GimmeMotifs_2\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0168\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0040\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0040\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GimmeMotifs_2\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0168\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0052\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.IRF.0009\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GimmeMotifs_2\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0005\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0066\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0009\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GimmeMotifs_2\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0005\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0005\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.Ets.0009\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bHLH.0106\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0066\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0013\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.bHLH.0106\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GimmeMotifs_2\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Rel.0004\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0010\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0168\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0010\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.Rel.0004\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0039\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0168\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.bHLH.0106\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0005\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Rel.0004\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0005\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0033\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0040\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 5): 1\n",
            "\n",
            "=== Processing directory 'gimme_dim_1_fa_withGenome_1000_peak' (dimension 1) ===\n",
            "  -> Processing motif: GM.5.0.bZIP.0002\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Unknown.0052\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0002\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 15 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.bHLH.0046\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 2\n",
            "  -> Processing motif: GM.5.0.Sox.0013\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.RFX.0006\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0002\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0096\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 15 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0039\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Rel.0006\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.Sox.0013\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Sox.0013\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0004\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0004\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.bHLH.0046\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 2\n",
            "  -> Processing motif: GM.5.0.Unknown.0140\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0002\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.RFX.0018\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Rel.0006\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.Unknown.0140\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Sox.0013\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.C2H2_ZF.0300\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0300\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0001\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0300\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.IRF.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Sox.0013\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.bHLH.0046\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0016\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0002\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Unknown.0029\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.Rel.0006\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0140\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0006\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0004\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 15 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0096\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.MADS_box.0021\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 17 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 2\n",
            "  -> Processing motif: GM.5.0.RFX.0006\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0011\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0096\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0002\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 1): 1\n",
            "\n",
            "=== Processing directory 'gimme_dim_2_fa_withGenome_1000_peak' (dimension 2) ===\n",
            "  -> Processing motif: GM.5.0.T-box.0001\n",
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0012\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0018\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0013\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.T-box.0015\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0006\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0120\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n",
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0314\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0018\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.T-box.0015\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0012\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.T-box.0016\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.SAND.0005\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.T-box.0001\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.T-box.0016\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.T-box.0001\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.IRF.0015\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0002\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Paired_box.0017\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.IRF.0004\n",
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n",
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.IRF.0004\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.T-box.0007\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.SAND.0005\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.C2H2_ZF.0314\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.SAND.0005\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.T-box.0023\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.T-box.0001\n",
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0120\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0120\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0012\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.T-box.0016\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0018\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0314\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0314\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.bHLH.0106\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.T-box.0001\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Homeodomain.0001\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0314\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.bHLH.0106\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0013\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0006\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.T-box.0023\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.Runt.0006\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.Paired_box.0017\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Paired_box.0017\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0013\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0002\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.IRF.0004\n",
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.T-box.0023\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n",
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0018\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.bHLH.0106\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0018\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0002\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0051\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "  -> Processing motif: GM.5.0.T-box.0015\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 2): 1\n",
            "\n",
            "=== Processing directory 'gimme_dim_0_fa_withGenome_1000_peak' (dimension 0) ===\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.bZIP.0016\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0049\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 15 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0035\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0035\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0011\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0002\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0036\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n",
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0061\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0011\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.bZIP.0049\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0032\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0007\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 11 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0002\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0069\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0006\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0035\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0036\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0008\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0104\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0007\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n",
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> GM_5_0_bZIP_0063.matches.cleaned.cleaned.cleaned.cleaned.bed has no data lines or no motif name found!\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 15 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0069\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0008\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0061\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0032\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n",
            "     Found 15 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0058\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0052\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GimmeMotifs_1\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0027\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 16 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 2\n",
            "  -> Processing motif: GM.5.0.Ets.0049\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0052\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0036\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0058\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0007\n",
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.Mixed.0027\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 16 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 2\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0300\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0104\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0052\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0032\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0007\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.Mixed.0104\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0104\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0052\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0013\n",
            "     Found 11 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> GM_5_0_bZIP_0063.matches.cleaned.cleaned.cleaned.bed has no data lines or no motif name found!\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0036\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0005\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0049\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 2\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0035\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0027\n",
            "     Found 16 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 2\n",
            "  -> Processing motif: GM.5.0.RFX.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 2\n",
            "  -> GM_5_0_bZIP_0063.matches.bed has no data lines or no motif name found!\n",
            "  -> Processing motif: GM.5.0.Mixed.0027\n",
            "     Found 16 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0016\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 14 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0032\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0052\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0011\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0016\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0016\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0002\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0058\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 10 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0049\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0006\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0104\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0069\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0006\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0049\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0061\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> GM_5_0_bZIP_0063.matches.cleaned.cleaned.bed has no data lines or no motif name found!\n",
            "  -> Processing motif: GM.5.0.bZIP.0016\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0032\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0011\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0035\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0027\n",
            "     Found 16 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0008\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0034\n",
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 3\n",
            "  -> GM_5_0_bZIP_0063.matches.cleaned.bed has no data lines or no motif name found!\n",
            "  -> Processing motif: GimmeMotifs_1\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.MADS_box.0011\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GimmeMotifs_1\n",
            "     Found 16 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0036\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0005\n",
            "     Found 18 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 3\n",
            "  -> Processing motif: GM.5.0.RFX.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 2\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n",
            "     Found 15 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0009\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "\n",
            "=== Processing directory 'gimme_dim_4_fa_withGenome_1000_peak' (dimension 4) ===\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0120\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0036\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0314\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0007\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GimmeMotifs_2\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0014\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Paired_box.0017\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0110\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0029\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0026\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0004\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GimmeMotifs_2\n",
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0004\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0314\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 1\n",
            "  -> Processing motif: GM.5.0.Mixed.0027\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0059\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bHLH.0024\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0005\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0036\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0120\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0059\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0120\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0120\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0042\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0003\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0120\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0042\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GimmeMotifs_2\n",
            "     Found 9 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0014\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 1\n",
            "  -> Processing motif: GM.5.0.RFX.0005\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0110\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0314\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0013\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0013\n",
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0014\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n",
            "     Found 7 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0010\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0027\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0059\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Paired_box.0017\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0003\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Paired_box.0017\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0042\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0086\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0049\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0004\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bHLH.0024\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Nuclear_receptor.0110\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0027\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bHLH.0024\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0005\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0036\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 4): 0\n",
            "\n",
            "=== Processing directory 'gimme_dim_3_fa_withGenome_1000_peak' (dimension 3) ===\n",
            "  -> Processing motif: GM.5.0.bZIP.0044\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0035\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0011\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0036\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0043\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0049\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0091\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0043\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0069\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0006\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0020\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0090\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0011\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0008\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.IRF.0020\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0069\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0090\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0003\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0021\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0052\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0044\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0049\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0091\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0042\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0091\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.MADS_box.0021\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0101\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0091\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.IRF.0009\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0090\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0042\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0101\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0049\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0003\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0101\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0021\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0035\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0042\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0044\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0016\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0011\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0091\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.bZIP.0090\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0020\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 2\n",
            "  -> Processing motif: GM.5.0.bZIP.0042\n",
            "     Found 8 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.MADS_box.0021\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0043\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.Ets.0049\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0006\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0062\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0044\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0069\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 1\n",
            "  -> Processing motif: GM.5.0.MADS_box.0021\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.RFX.0006\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0039\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0035\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0049\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0011\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0090\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GimmeMotifs_1\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0011\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0044\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "  -> Processing motif: GM.5.0.Unknown.0154\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 3): 0\n",
            "\n",
            "=== Processing directory 'gimme_dim_0_randomBG' (dimension 0) ===\n",
            "  -> Processing motif: GM.5.0.IRF.0014\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0003\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0003\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GimmeMotifs_3\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Paired_box.0003\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0036\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0014\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0036\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0019\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.Paired_box.0003\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Paired_box.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Homeodomain.0125\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Homeodomain.0125\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0003\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0019\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GimmeMotifs_3\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0048\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.IRF.0014\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GimmeMotifs_3\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Homeodomain.0125\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0019\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.IRF.0014\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0019\n",
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.Runt.0003\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.MADS_box.0019\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 6 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 1\n",
            "  -> Processing motif: GM.5.0.IRF.0014\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n",
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Paired_box.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0036\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Paired_box.0003\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GimmeMotifs_3\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Homeodomain.0125\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GimmeMotifs_3\n",
            "     Found 2 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Homeodomain.0125\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0036\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Mixed.0003\n",
            "     Found 3 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0036\n",
            "     Found 0 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.bZIP.0063\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     Found 5 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.C2H2_ZF.0203\n",
            "     Found 1 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "  -> Processing motif: GM.5.0.Ets.0015\n",
            "     Found 4 motif-peak-gene combos within Â±20000 bp.\n",
            "     Overlap with top genes (dim 0): 0\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Dimension 7 results:\n",
            "  Motif: GM.5.0.Ets.0037\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.STAT.0006\n",
            "    # motif genes: 11\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0010\n",
            "    # motif genes: 10\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.C2H2_ZF.0062\n",
            "    # motif genes: 18\n",
            "    # overlapping with top genes: 3\n",
            "  Motif: GM.5.0.Ets.0017\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Rel.0015\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0013\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0042\n",
            "    # motif genes: 11\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0086\n",
            "    # motif genes: 9\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.IRF.0019\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.MADS_box.0019\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0021\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Rel.0003\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0052\n",
            "    # motif genes: 10\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Runt.0003\n",
            "    # motif genes: 7\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Unknown.0108\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0009\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0033\n",
            "    # motif genes: 7\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Mixed.0106\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.IRF.0014\n",
            "    # motif genes: 10\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.IRF.0013\n",
            "    # motif genes: 11\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0063\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0049\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Ets.0002\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GimmeMotifs_1\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 2\n",
            "\n",
            "Dimension 6 results:\n",
            "  Motif: GM.5.0.bZIP.0016\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bHLH.0119\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Runt.0006\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.IRF.0002\n",
            "    # motif genes: 8\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.AT_hook.0006\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Mixed.0051\n",
            "    # motif genes: 8\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.C2H2_ZF.0119\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Unknown.0154\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Unknown.0108\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.Ets.0017\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.C2H2_ZF.0032\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Sox.0013\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0013\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.bZIP.0086\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.Unknown.0197\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Runt.0003\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Runt.0002\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.C2H2_ZF.0014\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.Paired_box.0017\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.C2H2_ZF.0057\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0063\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Ets.0009\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.RFX.0010\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 1\n",
            "\n",
            "Dimension 5 results:\n",
            "  Motif: GM.5.0.bZIP.0039\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.C2H2_ZF.0168\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.IRF.0009\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0029\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0010\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.MADS_box.0005\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.C2H2_ZF.0203\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0042\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0066\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GimmeMotifs_2\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0040\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0052\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Runt.0003\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0039\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.RFX.0005\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0009\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bHLH.0106\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.IRF.0013\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Rel.0004\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0033\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 1\n",
            "\n",
            "Dimension 1 results:\n",
            "  Motif: GM.5.0.bZIP.0002\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0011\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Unknown.0052\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Mixed.0051\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.C2H2_ZF.0062\n",
            "    # motif genes: 14\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bHLH.0046\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.Sox.0013\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0086\n",
            "    # motif genes: 8\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.RFX.0006\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Runt.0002\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0096\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Unknown.0039\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Rel.0006\n",
            "    # motif genes: 8\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.IRF.0004\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Unknown.0140\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.RFX.0018\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Runt.0003\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.C2H2_ZF.0300\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.C2H2_ZF.0001\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.IRF.0013\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0016\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Unknown.0029\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.MADS_box.0021\n",
            "    # motif genes: 17\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.MADS_box.0011\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "\n",
            "Dimension 2 results:\n",
            "  Motif: GM.5.0.T-box.0001\n",
            "    # motif genes: 9\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0012\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0018\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0013\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.T-box.0015\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Runt.0006\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Nuclear_receptor.0120\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Mixed.0051\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0029\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.Unknown.0154\n",
            "    # motif genes: 7\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.C2H2_ZF.0314\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.T-box.0016\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.SAND.0005\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.IRF.0015\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Runt.0003\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Runt.0002\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Paired_box.0017\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.IRF.0004\n",
            "    # motif genes: 7\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.T-box.0007\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.T-box.0023\n",
            "    # motif genes: 8\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.bHLH.0106\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Homeodomain.0001\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0063\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 1\n",
            "\n",
            "Dimension 0 results:\n",
            "  Motif: GM.5.0.IRF.0014\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0015\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Mixed.0003\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.C2H2_ZF.0203\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GimmeMotifs_3\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0048\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Paired_box.0003\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0036\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Runt.0003\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.MADS_box.0019\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0063\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Homeodomain.0125\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "\n",
            "Dimension 4 results:\n",
            "  Motif: GM.5.0.Ets.0015\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Nuclear_receptor.0120\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0036\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0029\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.C2H2_ZF.0314\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.C2H2_ZF.0062\n",
            "    # motif genes: 7\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0007\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0042\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0086\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GimmeMotifs_2\n",
            "    # motif genes: 9\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.IRF.0014\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Paired_box.0017\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Nuclear_receptor.0110\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0026\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.IRF.0004\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0039\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Runt.0003\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Mixed.0027\n",
            "    # motif genes: 8\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Nuclear_receptor.0059\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bHLH.0024\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.RFX.0005\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0042\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.MADS_box.0003\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0013\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.IRF.0013\n",
            "    # motif genes: 7\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.RFX.0010\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0063\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0049\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "\n",
            "Dimension 3 results:\n",
            "  Motif: GM.5.0.bZIP.0044\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0035\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.MADS_box.0011\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0036\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0011\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0043\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0048\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Unknown.0154\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0049\n",
            "    # motif genes: 0\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.C2H2_ZF.0062\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0091\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Ets.0042\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0069\n",
            "    # motif genes: 6\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.RFX.0006\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.IRF.0020\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 2\n",
            "  Motif: GM.5.0.bZIP.0090\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0011\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0008\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.Ets.0003\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.MADS_box.0021\n",
            "    # motif genes: 1\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0052\n",
            "    # motif genes: 8\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.Ets.0039\n",
            "    # motif genes: 5\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0042\n",
            "    # motif genes: 8\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0101\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.IRF.0009\n",
            "    # motif genes: 4\n",
            "    # overlapping with top genes: 1\n",
            "  Motif: GM.5.0.bZIP.0063\n",
            "    # motif genes: 2\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GM.5.0.bZIP.0016\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n",
            "  Motif: GimmeMotifs_1\n",
            "    # motif genes: 3\n",
            "    # overlapping with top genes: 0\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n",
            "join: Strand data from other will be added as strand data to self.\n",
            "If this is undesired use the flag apply_strand_suffix=False.\n",
            "To turn off the warning set apply_strand_suffix to True or False.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "gimmeMotifs provides motif-to-factor annotations via `combined.motif2factors.txt`\n",
        "(motif â†’ TF/factor).\n",
        "\n",
        "Integration:\n",
        "we combine (TF/factor â† motif) with previous script (motif â†’ peak â†’ gene) to obtain\n",
        "putative TF/factor â†’ motif â†’ peak â†’ gene links (reported as TF/factorâ€“gene pairs).\n"
      ],
      "metadata": {
        "id": "EcF2ajMlhgJh"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "import pandas as pd\n",
        "\n",
        "base_dir = \"/content/\"\n",
        "all_rows = []  # accumulate (dim, motif, factor, gene) rows here\n",
        "\n",
        "for dim in sorted(results_dict.keys()):\n",
        "    # Path to the dimension folder: e.g. gimme_dim_0_fa_withGenome_1000_peak\n",
        "    dim_folder = f\"gimme_dim_{dim}_fa_withGenome_1000_peak\"\n",
        "    dim_path = os.path.join(base_dir, dim_folder)\n",
        "\n",
        "    # combined.motif2factors file for this dimension\n",
        "    m2f_file = os.path.join(dim_path, \"combined.motif2factors.txt\")\n",
        "    if not os.path.exists(m2f_file):\n",
        "        print(f\"Warning: {m2f_file} not found, skipping dimension {dim}.\")\n",
        "        continue\n",
        "\n",
        "    # Read the motif2factor table (columns typically: \"Motif\", \"Factor\", \"Evidence\", \"Curated\")\n",
        "    m2f_df = pd.read_csv(m2f_file, sep=\"\\t\")\n",
        "\n",
        "    # Instead of looping over bed_filenames, we directly have motif names in results_dict[dim]\n",
        "    for motif_name, info in results_dict[dim].items():\n",
        "\n",
        "        # Find rows in the motif2factor table for this motif\n",
        "        sub_df = m2f_df[m2f_df[\"Motif\"] == motif_name]\n",
        "        if sub_df.empty:\n",
        "            print(f\"No factor info for motif '{motif_name}' in dimension {dim}.\")\n",
        "            continue\n",
        "\n",
        "        # Collect the unique \"Factor\" values from these rows\n",
        "        factors = sub_df[\"Factor\"].unique()\n",
        "\n",
        "        # The set of common genes for this motif\n",
        "        common_genes = info[\"common_genes\"]  # already a set\n",
        "\n",
        "        # For each factor, for each gene, add a row to all_rows\n",
        "        for factor in factors:\n",
        "            for gene in common_genes:\n",
        "                row_dict = {\n",
        "                    \"Dimension\": dim,\n",
        "                    \"Motif\": motif_name,\n",
        "                    \"Factor\": factor,\n",
        "                    \"Gene\": gene\n",
        "                }\n",
        "                all_rows.append(row_dict)\n",
        "\n",
        "# Convert the accumulated rows to a DataFrame\n",
        "results_df = pd.DataFrame(all_rows, columns=[\"Dimension\", \"Motif\", \"Factor\", \"Gene\"])\n",
        "\n",
        "print(results_df.shape)\n",
        "results_df\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 584
        },
        "id": "oynPGynfq-2m",
        "outputId": "768b6bad-d4a7-453c-8d47-9d4cdea403ce"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "No factor info for motif 'GimmeMotifs_3' in dimension 0.\n",
            "No factor info for motif 'GM.5.0.Unknown.0052' in dimension 1.\n",
            "No factor info for motif 'GM.5.0.Unknown.0039' in dimension 1.\n",
            "No factor info for motif 'GM.5.0.Unknown.0140' in dimension 1.\n",
            "No factor info for motif 'GimmeMotifs_2' in dimension 4.\n",
            "No factor info for motif 'GimmeMotifs_2' in dimension 5.\n",
            "No factor info for motif 'GM.5.0.Unknown.0108' in dimension 6.\n",
            "No factor info for motif 'GM.5.0.Unknown.0108' in dimension 7.\n",
            "(372, 4)\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     Dimension                 Motif   Factor    Gene\n",
              "0            0  GM.5.0.MADS_box.0019    MEF2C     VIM\n",
              "1            1     GM.5.0.Mixed.0051     FLI1   SKAP1\n",
              "2            1     GM.5.0.Mixed.0051   ZNF143   SKAP1\n",
              "3            1   GM.5.0.C2H2_ZF.0062     IRF4    CST3\n",
              "4            1   GM.5.0.C2H2_ZF.0062     Irf4    CST3\n",
              "..         ...                   ...      ...     ...\n",
              "367          7       GM.5.0.Ets.0049     ELK1   NAMPT\n",
              "368          7         GimmeMotifs_1  de novo   NAMPT\n",
              "369          7         GimmeMotifs_1  de novo  FAM49A\n",
              "370          7         GimmeMotifs_1     PU.1   NAMPT\n",
              "371          7         GimmeMotifs_1     PU.1  FAM49A\n",
              "\n",
              "[372 rows x 4 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-fa01fd0e-cc79-4b64-94f3-b4c00ad74021\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Dimension</th>\n",
              "      <th>Motif</th>\n",
              "      <th>Factor</th>\n",
              "      <th>Gene</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>0</td>\n",
              "      <td>GM.5.0.MADS_box.0019</td>\n",
              "      <td>MEF2C</td>\n",
              "      <td>VIM</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>GM.5.0.Mixed.0051</td>\n",
              "      <td>FLI1</td>\n",
              "      <td>SKAP1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>GM.5.0.Mixed.0051</td>\n",
              "      <td>ZNF143</td>\n",
              "      <td>SKAP1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>GM.5.0.C2H2_ZF.0062</td>\n",
              "      <td>IRF4</td>\n",
              "      <td>CST3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>GM.5.0.C2H2_ZF.0062</td>\n",
              "      <td>Irf4</td>\n",
              "      <td>CST3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>367</th>\n",
              "      <td>7</td>\n",
              "      <td>GM.5.0.Ets.0049</td>\n",
              "      <td>ELK1</td>\n",
              "      <td>NAMPT</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>368</th>\n",
              "      <td>7</td>\n",
              "      <td>GimmeMotifs_1</td>\n",
              "      <td>de novo</td>\n",
              "      <td>NAMPT</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>369</th>\n",
              "      <td>7</td>\n",
              "      <td>GimmeMotifs_1</td>\n",
              "      <td>de novo</td>\n",
              "      <td>FAM49A</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>370</th>\n",
              "      <td>7</td>\n",
              "      <td>GimmeMotifs_1</td>\n",
              "      <td>PU.1</td>\n",
              "      <td>NAMPT</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>371</th>\n",
              "      <td>7</td>\n",
              "      <td>GimmeMotifs_1</td>\n",
              "      <td>PU.1</td>\n",
              "      <td>FAM49A</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>372 rows Ã— 4 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-fa01fd0e-cc79-4b64-94f3-b4c00ad74021')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-fa01fd0e-cc79-4b64-94f3-b4c00ad74021 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-fa01fd0e-cc79-4b64-94f3-b4c00ad74021');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-06e43609-4849-4c66-97b3-e4383760731f\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-06e43609-4849-4c66-97b3-e4383760731f')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-06e43609-4849-4c66-97b3-e4383760731f button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "  <div id=\"id_4d5c2474-0b02-4741-8cfd-ef390eeb6319\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('results_df')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_4d5c2474-0b02-4741-8cfd-ef390eeb6319 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('results_df');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "results_df",
              "summary": "{\n  \"name\": \"results_df\",\n  \"rows\": 372,\n  \"fields\": [\n    {\n      \"column\": \"Dimension\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 0,\n        \"max\": 7,\n        \"num_unique_values\": 8,\n        \"samples\": [\n          1,\n          5,\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Motif\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 55,\n        \"samples\": [\n          \"GM.5.0.bZIP.0042\",\n          \"GM.5.0.bZIP.0096\",\n          \"GM.5.0.IRF.0009\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Factor\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 215,\n        \"samples\": [\n          \"Sp6\",\n          \"Rfx8\",\n          \"Runx1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 43,\n        \"samples\": [\n          \"CCL5\",\n          \"IFITM3\",\n          \"C5AR1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 49
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "H2RYt-7IYoPG"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}